{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import string\n",
    "import os\n",
    "from collections import Counter\n",
    "\n",
    "import nltk\n",
    "from nltk.tokenize import sent_tokenize, word_tokenize\n",
    "from segtok.segmenter import split_single, split_multi #seems to be the better choice than nltk despite lower speed\n",
    "\n",
    "from nltk.corpus import stopwords\n",
    "from nltk.stem import WordNetLemmatizer\n",
    "pd.options.display.max_colwidth = 50\n",
    "import re\n",
    "\n",
    "from sklearn.feature_extraction.text import TfidfVectorizer\n",
    "from sklearn.decomposition import TruncatedSVD"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Creating base"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "train = pd.read_csv('..//bases/training_variants')\n",
    "test = pd.read_csv('..//bases/test_variants')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "train_texts = pd.read_csv('..//bases/training_text', sep=\"\\|\\|\", engine='python', header=None, skiprows=1, names=[\"ID\",\"Text\"], encoding = \"utf-8\")\n",
    "test_texts = pd.read_csv('..//bases/test_text', sep=\"\\|\\|\", engine='python', header=None, skiprows=1, names=[\"ID\",\"Text\"], encoding = \"utf-8\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "train = pd.merge(train, train_texts, how='left', on='ID')\n",
    "test = pd.merge(test, test_texts, how='left', on='ID')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Transform Amino Acid (AA) Letter to their three-letter abbreviation in order to find them in the text when they appear\n",
    "One_to_Three_AA = {'C': 'Cys', 'D': 'Asp', 'S': 'Ser', 'Q': 'Gln', 'K': 'Lys',\n",
    "         'I': 'Ile', 'P': 'Pro', 'T': 'Thr', 'F': 'Phe', 'N': 'Asn', \n",
    "         'G': 'Gly', 'H': 'His', 'L': 'Leu', 'R': 'Arg', 'W': 'Trp', \n",
    "         'A': 'Ala', 'V': 'Val', 'E': 'Glu', 'Y': 'Tyr', 'M': 'Met'}\n",
    "pattern = re.compile('|'.join(One_to_Three_AA.keys()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 351,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#### process the train and test set together\n",
    "data_all = pd.concat((train, test), axis=0, ignore_index=True)\n",
    "data_all_backup = data_all[:] ## We keep backup in case we need to use again\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Substitutions (subs)\n",
    "## Functions for pre-processing of subs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# find_sub return the substituions that are in text and those that are not\n",
    "def find_sub(data):    \n",
    "    Boolean = [data.Variation[i][:-1] in data.Text[i] or #case 1.\n",
    "               pattern.sub(lambda x: One_to_Three_AA[x.group()], data.Variation[i][:-1]) # case2\n",
    "               in data.Text[i]  for i in data.index] ## because new indexing we use \n",
    "    \n",
    "    sub_in_text = data[Boolean]\n",
    "    not_Boolean = [not i for i in Boolean]  \n",
    "    sub_not_in_text = data[not_Boolean]\n",
    "    \n",
    "    return sub_in_text, sub_not_in_text\n",
    "\n",
    "# find_sub_numberChange searches for other number of a substitution i.e. G12V -> G_V because sometimes mistake in entry\n",
    "# Is currently without One_to_three substitution or Variation[:-1] only the full variation\n",
    "def find_sub_numberChange(data):\n",
    "    Booleans = [] #will contain the different Booleans if found in text\n",
    "    for i in data.index:\n",
    "        split_variation = re.split('(\\d+)', data.Variation[i]) # split based on a number\n",
    "        first_Amino = re.escape(split_variation[0]) #re.escpae uses variable as regex\n",
    "        last_Amino = re.escape(split_variation[-1])\n",
    "        new_regex  = first_Amino + r\"\\d+\" + last_Amino\n",
    "        Boolean = bool(re.search(new_regex, data.Text[i]))\n",
    "        Booleans.append(Boolean)\n",
    "        \n",
    "    sub_number_in_text = data[Booleans]\n",
    "    not_Boolean = [not i for i in Booleans]  \n",
    "    sub_number_no_text = data[not_Boolean]\n",
    "    \n",
    "    return sub_number_in_text, sub_number_no_text\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "##### get_sentences_sub use a window to extract sentences where the subs appear. \n",
    "# If window_left & window_right = 0 => just taking the sentences with subs\n",
    "\n",
    "def get_sentences_sub(data, splitted_sentences, window_left, window_right):\n",
    "    data.index = range(len(data)) #makes sure that index is right\n",
    "    sentences_with_sub = [[] for _ in range(len(data))]\n",
    "    for i in range(len(splitted_sentences)):\n",
    "        sentences = splitted_sentences[i]\n",
    "        one_to_three_variation = pattern.sub(lambda x: One_to_Three_AA[x.group()], data.Variation[i][:-1])\n",
    "        Variation = data.Variation[i][:-1]        \n",
    "        for j in range(len(sentences)):                              \n",
    "            if (Variation in sentences[j]) or (one_to_three_variation in sentences[j]):\n",
    "                new_regex = re.escape(Variation) + r\"[\\S]*\" ###  r\"[\\S]*\" because we look for Variation[:-1] not just Variation\n",
    "                sentences[j] = re.sub(new_regex, ' placeholderMutation', sentences[j]) #case 1\n",
    "                ### We add the space to ' placeholderMutation' because sometimes there are letters in front of it\n",
    "                new_regex = re.escape(one_to_three_variation) + r\"[\\S]*\"\n",
    "                sentences[j] = re.sub(new_regex, ' placeholderMutation', sentences[j]) #case 2\n",
    "                sentences_with_sub[i].extend(sentences[max(j-window_left,0) : min(j+1+window_right, len(sentences)-1)])               \n",
    "    \n",
    "    return sentences_with_sub\n",
    "\n",
    "##### get_sentences_sub_number use a window to extract sentences where the subs appear that have different number i.e. G12V -> G_V\n",
    "\n",
    "def get_sentences_sub_number(data, splitted_sentences, window_left, window_right):\n",
    "    #position_sentences = [[] for _ in range(len(data))]  #### currently not used\n",
    "    data.index = range(len(data))\n",
    "    sentences_with_sub_number = [[] for _ in range(len(data))]\n",
    "    for i in range(len(splitted_sentences)):\n",
    "        sentences = splitted_sentences[i] \n",
    "        for j in range(len(sentences)):\n",
    "            split_variation = re.split('(\\d+)', data.Variation[i]) # split based on a number\n",
    "            first_Amino = re.escape(split_variation[0]) #re.escpae uses variable as regex\n",
    "            last_Amino = re.escape(split_variation[-1])\n",
    "            new_regex  = first_Amino + r\"\\d+\" + last_Amino\n",
    "            Boolean = bool(re.search(new_regex, sentences[j]))            \n",
    "            if Boolean:\n",
    "                sentences[j] = re.sub(new_regex, ' placeholderMutation', sentences[j]) # Again replacing the sentence with placeholder\n",
    "                sentences_with_sub_number[i].extend(sentences[max(j-window_left,0) : min(j+1+window_right,len(sentences)-1)])\n",
    "    \n",
    "    return sentences_with_sub_number\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#### Converts list of sentences into one string of sentences for each document => to use for tfidf etc.\n",
    "def sentences_to_string(sentences_list):\n",
    "    sentence_strings = []\n",
    "    for sentences in sentences_list:\n",
    "        sentence_string =  ' '.join(str(sentence) for sentence in sentences)\n",
    "        sentence_strings.append(sentence_string)\n",
    "    \n",
    "    return sentence_strings \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Subs processing of the data set "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 352,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#### First find those that have the format of being a substitution in data\n",
    "data_all['Substitutions_var'] = data_all.Variation.apply(lambda x: bool(re.search('^[A-Z]\\\\d+[A-Z*]$', x))*1) #multiplying by 1 converts True to 1, False to 0 => Maybe modify this later?\n",
    "data_all['Stop_codon_var'] = data_all.Variation.apply(lambda x: bool(re.search('[*]', x))*1) #multiplying by 1 converts True to 1, False to 0\n",
    "data_sub = data_all[data_all['Substitutions_var']==1] ### Now we know the index of where a substitution occurs - the data_sub"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "sub_text, sub_no_text = find_sub(data_sub) \n",
    "sub_text_backup = sub_text[:] ## index gets changed by text_processing if we don't make a copy\n",
    "\n",
    "sub_number_text, sub_number_no_text = find_sub_numberChange(sub_no_text) # 108 such cases out of 411 = nice improvement\n",
    "sub_number_backup = sub_number_text[:]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "## use tokenizer to split into sentences of all the subs in text + extract sentences in window\n",
    "NLTK_sub = [sent_tokenize(sub_text.Text[i]) for i in sub_text.index] \n",
    "sub_sentences = get_sentences_sub(sub_text, NLTK_sub, window_left = 0, window_right = 0) # choosing for window 0 as default now"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "## use tokenizer to split into sentences of subs that have different number in text + extract sentences in window\n",
    "NLTK_sub_number = [split_single(sub_number_text.Text[i]) for i in sub_number_text.index]\n",
    "sub_number_sentences = get_sentences_sub_number(sub_number_text, NLTK_sub_number, window_left = 0, window_right = 0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Length of subs with text: 7794\n",
      "Length of subs with other number in text: 108\n",
      "Length of subs still not found: 304\n"
     ]
    }
   ],
   "source": [
    "print(\"Length of subs with text: %i\" %len(sub_sentences))\n",
    "print(\"Length of subs with other number in text: %i\" %len(sub_number_sentences))\n",
    "print(\"Length of subs still not found: %i\" %len(sub_number_no_text)) # other ways to process it like finding the word 'mutation'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 296,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# sub_number_no_text.Text[sub_number_no_text.index[0]] #prints out one of the texts"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Non-substitutions (no subs)\n",
    "## Functions for pre-processing of no_subs like fusions, deletions, truncations, etc."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "## Gets different non-subs variation types by using regex\n",
    "def variation_regex(data, pattern): \n",
    "    Boolean = [bool(re.search(pattern, i, re.IGNORECASE)) for i in data.Variation]\n",
    "    data_regex = data[Boolean] \n",
    "    not_Boolean = [not i for i in Boolean]  \n",
    "    data_no_regex = data[not_Boolean]\n",
    "    \n",
    "    return (data_regex, data_no_regex)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 431,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# find the nosub needed based on pattern\n",
    "def find_nosub(data, pattern):    \n",
    "    Boolean = [pattern in data.Text[i] for i in data.index] ## because new indexing we use \n",
    "    \n",
    "    nosub_in_text = data[Boolean]\n",
    "    not_Boolean = [not i for i in Boolean]  \n",
    "    nosub_not_in_text = data[not_Boolean]\n",
    "    \n",
    "    return nosub_in_text, nosub_not_in_text\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 434,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# find the fusion needed based on Gene and pattern\n",
    "def find_nosub_gene(data, pattern):    \n",
    "    Boolean = [(pattern in data.Text[i]) & (data.Gene[i] in data.Text[i]) for i in data.index] \n",
    "    \n",
    "    nosub_gene_in_text = data[Boolean]\n",
    "    not_Boolean = [not i for i in Boolean]  \n",
    "    nosub_gene_not_in_text = data[not_Boolean]\n",
    "    \n",
    "    return nosub_gene_in_text, nosub_gene_not_in_text"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 492,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# find the fusion needed based on Gene and pattern\n",
    "def find_gene_fusion(data, pattern):\n",
    "    Booleans = []\n",
    "    for i in data.index:\n",
    "        gene_fusion = data.Variation[i].split(' ')[0] # this forget about the word 'Fusion' inside the variation\n",
    "        # split on '-' to search for the two genes seperately now\n",
    "        gene_split = gene_fusion.split('-')\n",
    "        first_gene = gene_split[0]\n",
    "        if len(gene_split)>1:\n",
    "            second_gene = gene_fusion.split('-')[1]\n",
    "        else:\n",
    "            second_gene = '' # will always find if there is no second gene\n",
    "        \n",
    "        Boolean = (pattern in data.Text[i]) & ((first_gene in data.Text[i]) & (second_gene in data.Text[i]))\n",
    "        Booleans.append(Boolean)\n",
    "    \n",
    "    gene_fusion_in_text = data[Booleans]\n",
    "    not_Boolean = [not i for i in Booleans]  \n",
    "    gene_fusion_not_in_text = data[not_Boolean]\n",
    "        \n",
    "    return gene_fusion_in_text, gene_fusion_not_in_text"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 175,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def get_sentences_nosub(data, splitted_sentences, pattern, window_left, window_right):\n",
    "    data.index = range(len(data))\n",
    "    sentences_nosub = [[] for _ in range(len(data))]\n",
    "    \n",
    "    for i in range(len(splitted_sentences)):\n",
    "        sentences = splitted_sentences[i]   \n",
    "        for j in range(len(sentences)):    \n",
    "                if pattern in sentences[j]:\n",
    "                    sentences_nosub[i].extend(sentences[max(j-window_left,0) : min(j+1+window_right, len(sentences)-1)])\n",
    "                    \n",
    "    return sentences_nosub  ### This might take a while because it's looping through all sentences\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 333,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def get_sentences_fusion(data, splitted_sentences, pattern, window_left, window_right):\n",
    "    data.index = range(len(data))\n",
    "    sentences_fusion = [[] for _ in range(len(data))]\n",
    "    \n",
    "    for i in range(len(splitted_sentences)):\n",
    "        sentences = splitted_sentences[i]   \n",
    "        for j in range(len(sentences)):    \n",
    "                if (pattern in sentences[j]) & (data.Gene[i] in sentences[j]):\n",
    "                    sentences_fusion[i].extend(sentences[max(j-window_left,0) : min(j+1+window_right, len(sentences)-1)])\n",
    "                    \n",
    "    return sentences_fusion   ### This might take a while because it's looping through all sentences\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 399,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def get_sentences_gene_fusion(data, splitted_sentences, pattern, window_left, window_right):\n",
    "    data.index = range(len(data))\n",
    "    sentences_gene_fusion = [[] for _ in range(len(data))]\n",
    "    \n",
    "    for i in range(len(splitted_sentences)):\n",
    "        sentences = splitted_sentences[i]\n",
    "        gene_fusion = data.Variation[i].split(' ')[0] # this forget about the word 'Fusion' inside the variation\n",
    "        # split on '-' to have both genes\n",
    "        gene_split = gene_fusion.split('-')\n",
    "        first_gene = gene_split[0]\n",
    "        if len(gene_split)>1:\n",
    "            second_gene = gene_fusion.split('-')[1]\n",
    "        else:\n",
    "            second_gene = '' # will always find if there is no second gene\n",
    "        for j in range(len(sentences)):    \n",
    "                if (pattern in sentences[j]) & ((first_gene in sentences[j]) & (second_gene in sentences[j])):\n",
    "                    sentences_gene_fusion[i].extend(sentences[max(j-window_left,0) : min(j+1+window_right, len(sentences)-1)])\n",
    "                    \n",
    "    return sentences_gene_fusion   ### This might take a while because it's looping through all sentences\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## No subs processing of the data set "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 371,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\rafiz\\Anaconda3\\lib\\site-packages\\pandas\\core\\indexing.py:179: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  self._setitem_with_indexer(indexer, value)\n"
     ]
    }
   ],
   "source": [
    "#### First find those that have the format of being a non-substitutions in the data\n",
    "\n",
    "# Initialize some of the variables already (because we splice them, could be filled with NA's otherwise)\n",
    "data_all['gene_fusion_var'] = 0\n",
    "data_all['Deletion_var'] = 0\n",
    "data_all['del_or_ins_var'] = 0\n",
    "\n",
    "####### Fusions : 'Fusions' ############\n",
    "data_all['Fusion_var'] = data_all.Variation.apply(lambda x: bool(re.search('^fusion', x, re.IGNORECASE))*1) #multiplying by 1 converts True to 1, False to 0\n",
    "_ , new_data_all = variation_regex(data_all, '^fusion') \n",
    "\n",
    "###### Fusions: 'Gene-Gene fusion' ########\n",
    "data_all['gene_fusion_var'].loc[new_data_all.index] = new_data_all.Variation.apply(lambda x: bool(re.search('fusion', x, re.IGNORECASE))*1)\n",
    "_ , new_data_all = variation_regex(new_data_all, 'fusion') \n",
    "\n",
    "####### Deletions: 'Deletions' ############\n",
    "data_all['Deletion_var'].loc[new_data_all.index] = new_data_all.Variation.apply(lambda x: bool(re.search('^del', x, re.IGNORECASE))*1)\n",
    "_, new_data_all = variation_regex(new_data_all, '^del') \n",
    "\n",
    "####### Deletions & Insertions wheteher together or seperately (doesn't make a big difference IMO)\n",
    "data_all['del_or_ins_var'].loc[new_data_all.index] = new_data_all.Variation.apply(lambda x: bool(re.search('del|ins', x, re.IGNORECASE))*1)\n",
    "\n",
    "###### Amplifications #########\n",
    "data_all['Amplification_var'] = data_all.Variation.apply(lambda x: bool(re.search('ampl', x, re.IGNORECASE))*1) \n",
    "\n",
    "###### Truncations ########### Don't forget there are 'Truncating mutations' = 95 and '_trunc' = 4\n",
    "data_all['Truncation_var'] = data_all.Variation.apply(lambda x: bool(re.search('trunc', x, re.IGNORECASE))*1) \n",
    "\n",
    "####### Exons #########\n",
    "data_all['exon_var'] = data_all.Variation.apply(lambda x: bool(re.search('exon', x, re.IGNORECASE))*1) \n",
    "\n",
    "####### Frameshift mutations ########\n",
    "data_all['frameshift_var'] = data_all.Variation.apply(lambda x: bool(re.search('fs', x, re.IGNORECASE))*1) \n",
    "\n",
    "####### Duplications ##############\n",
    "data_all['dup_var'] = data_all.Variation.apply(lambda x: bool(re.search('dup', x, re.IGNORECASE))*1) \n",
    "\n",
    "\n",
    "all_variations = data_all.loc[:, \"Substitutions_var\":\"dup_var\"]\n",
    "# all_variations[all_variations.T == 0] # deal with those 61 cases that are not assigned any label "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Fusions: the normal variations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 435,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# Reduce text for all the fusions using the word 'fusion' in text\n",
    "Fusions = data_all[(data_all['Fusion_var']==1)]\n",
    "fusion_in_text, fusion_not_in_text = find_nosub_gene(Fusions, 'fusion') "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 436,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Class</th>\n",
       "      <th>Gene</th>\n",
       "      <th>ID</th>\n",
       "      <th>Text</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Substitutions_var</th>\n",
       "      <th>Stop_codon_var</th>\n",
       "      <th>gene_fusion_var</th>\n",
       "      <th>Deletion_var</th>\n",
       "      <th>del_or_ins_var</th>\n",
       "      <th>Fusion_var</th>\n",
       "      <th>Amplification_var</th>\n",
       "      <th>Truncation_var</th>\n",
       "      <th>exon_var</th>\n",
       "      <th>frameshift_var</th>\n",
       "      <th>dup_var</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [Class, Gene, ID, Text, Variation, Substitutions_var, Stop_codon_var, gene_fusion_var, Deletion_var, del_or_ins_var, Fusion_var, Amplification_var, Truncation_var, exon_var, frameshift_var, dup_var]\n",
       "Index: []"
      ]
     },
     "execution_count": 436,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fusion_not_in_text # for 'Fusion' variation we find that gene is always in text"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 437,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "NLTK_fusion = [sent_tokenize(fusion_in_text.Text[i]) for i in fusion_in_text.index] \n",
    "sentences_fusion = get_sentences_fusion(fusion_in_text, NLTK_fusion, 'fusion', window_left=0, window_right=0)\n",
    "# sentences_fusion did a good job reducing our text. only 3 cases left where sentences are 0, 1 and 2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Fusions: the gene-gene variations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 489,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "gene_fusions = data_all[data_all['gene_fusion_var']==1]\n",
    "gene_fusion_in_text, gene_fusion_not_in_text = find_gene_fusion(gene_fusions, 'fusion') "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 490,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "6"
      ]
     },
     "execution_count": 490,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(gene_fusion_not_in_text) # 6 gene_fusions don't include 'fusion' and one of the gene names => deal later with them if we want"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 491,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Class</th>\n",
       "      <th>Gene</th>\n",
       "      <th>ID</th>\n",
       "      <th>Text</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Substitutions_var</th>\n",
       "      <th>Stop_codon_var</th>\n",
       "      <th>gene_fusion_var</th>\n",
       "      <th>Deletion_var</th>\n",
       "      <th>del_or_ins_var</th>\n",
       "      <th>Fusion_var</th>\n",
       "      <th>Amplification_var</th>\n",
       "      <th>Truncation_var</th>\n",
       "      <th>exon_var</th>\n",
       "      <th>frameshift_var</th>\n",
       "      <th>dup_var</th>\n",
       "      <th>Text_words</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>941</th>\n",
       "      <td>2.0</td>\n",
       "      <td>PDGFRB</td>\n",
       "      <td>941</td>\n",
       "      <td>Chronic myelomonocytic leukemia (CMML) is a my...</td>\n",
       "      <td>ATF7IP-PDGFRB Fusion</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>116</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1057</th>\n",
       "      <td>2.0</td>\n",
       "      <td>EWSR1</td>\n",
       "      <td>1057</td>\n",
       "      <td>As a result of chromosome translocations, the ...</td>\n",
       "      <td>EWSR1-FEV Fusion</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>207</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1060</th>\n",
       "      <td>7.0</td>\n",
       "      <td>EWSR1</td>\n",
       "      <td>1060</td>\n",
       "      <td>Fibroblast growth factors (FGFs) expressed in...</td>\n",
       "      <td>EWSR1-ETV1 Fusion</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6237</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1065</th>\n",
       "      <td>2.0</td>\n",
       "      <td>EWSR1</td>\n",
       "      <td>1065</td>\n",
       "      <td>Ewing’s sarcoma/PNET, a tumor of the bone and ...</td>\n",
       "      <td>EWSR1-ETV4 Fusion</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1531</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3223</th>\n",
       "      <td>7.0</td>\n",
       "      <td>NTRK1</td>\n",
       "      <td>3223</td>\n",
       "      <td>In this study, we utilized retroviral transfer...</td>\n",
       "      <td>Delta-NTRK1 Fusion</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6524</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6183</th>\n",
       "      <td>NaN</td>\n",
       "      <td>PDGFRB</td>\n",
       "      <td>2862</td>\n",
       "      <td>Chromosomal translocations involving the plate...</td>\n",
       "      <td>RABEP1-PDGFRB Fusion</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5245</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Class    Gene    ID                                               Text  \\\n",
       "941     2.0  PDGFRB   941  Chronic myelomonocytic leukemia (CMML) is a my...   \n",
       "1057    2.0   EWSR1  1057  As a result of chromosome translocations, the ...   \n",
       "1060    7.0   EWSR1  1060   Fibroblast growth factors (FGFs) expressed in...   \n",
       "1065    2.0   EWSR1  1065  Ewing’s sarcoma/PNET, a tumor of the bone and ...   \n",
       "3223    7.0   NTRK1  3223  In this study, we utilized retroviral transfer...   \n",
       "6183    NaN  PDGFRB  2862  Chromosomal translocations involving the plate...   \n",
       "\n",
       "                 Variation  Substitutions_var  Stop_codon_var  \\\n",
       "941   ATF7IP-PDGFRB Fusion                  0               0   \n",
       "1057      EWSR1-FEV Fusion                  0               0   \n",
       "1060     EWSR1-ETV1 Fusion                  0               0   \n",
       "1065     EWSR1-ETV4 Fusion                  0               0   \n",
       "3223    Delta-NTRK1 Fusion                  0               0   \n",
       "6183  RABEP1-PDGFRB Fusion                  0               0   \n",
       "\n",
       "      gene_fusion_var  Deletion_var  del_or_ins_var  Fusion_var  \\\n",
       "941                 1             0               0           0   \n",
       "1057                1             0               0           0   \n",
       "1060                1             0               0           0   \n",
       "1065                1             0               0           0   \n",
       "3223                1             0               0           0   \n",
       "6183                1             0               0           0   \n",
       "\n",
       "      Amplification_var  Truncation_var  exon_var  frameshift_var  dup_var  \\\n",
       "941                   0               0         0               0        0   \n",
       "1057                  0               0         0               0        0   \n",
       "1060                  0               0         0               0        0   \n",
       "1065                  0               0         0               0        0   \n",
       "3223                  0               0         0               0        0   \n",
       "6183                  0               0         0               0        0   \n",
       "\n",
       "      Text_words  \n",
       "941          116  \n",
       "1057         207  \n",
       "1060        6237  \n",
       "1065        1531  \n",
       "3223        6524  \n",
       "6183        5245  "
      ]
     },
     "execution_count": 491,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_fusion_not_in_text"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 428,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "NLTK_gene_fusion = [sent_tokenize(gene_fusion_in_text.Text[i]) for i in gene_fusion_in_text.index] \n",
    "sentences_gene_fusion = get_sentences_gene_fusion(gene_fusion_in_text, NLTK_gene_fusion, 'fusion', window_left=0, window_right=0)\n",
    "# sentences_gene_fusion from 154 sentences still has some low ones (14 smaller than 5)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Deletions: the normal variations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 448,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# Reduce text for all the Deletion Variation using the word 'deletion' in text\n",
    "deletions = data_all[(data_all['Deletion_var']==1)]\n",
    "deletion_in_text, deletion_not_in_text = find_nosub_gene(deletions, 'del') # only 5 left in deletion_not_in_text"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 449,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Class</th>\n",
       "      <th>Gene</th>\n",
       "      <th>ID</th>\n",
       "      <th>Text</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Substitutions_var</th>\n",
       "      <th>Stop_codon_var</th>\n",
       "      <th>gene_fusion_var</th>\n",
       "      <th>Deletion_var</th>\n",
       "      <th>del_or_ins_var</th>\n",
       "      <th>Fusion_var</th>\n",
       "      <th>Amplification_var</th>\n",
       "      <th>Truncation_var</th>\n",
       "      <th>exon_var</th>\n",
       "      <th>frameshift_var</th>\n",
       "      <th>dup_var</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>539</th>\n",
       "      <td>4.0</td>\n",
       "      <td>SMAD2</td>\n",
       "      <td>539</td>\n",
       "      <td>Transforming growth factor- (TGF-) plays a com...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1157</th>\n",
       "      <td>1.0</td>\n",
       "      <td>KMT2C</td>\n",
       "      <td>1157</td>\n",
       "      <td>Abstract Monomethylation of histone H3 on Lys...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3201</th>\n",
       "      <td>1.0</td>\n",
       "      <td>RASA1</td>\n",
       "      <td>3201</td>\n",
       "      <td>Phospholipase C gamma 1 (PLC gamma 1) and p21r...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3253</th>\n",
       "      <td>1.0</td>\n",
       "      <td>CASP8</td>\n",
       "      <td>3253</td>\n",
       "      <td>Evidence exists that alterations of the genes ...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7153</th>\n",
       "      <td>NaN</td>\n",
       "      <td>ASXL2</td>\n",
       "      <td>3832</td>\n",
       "      <td>Acute myeloid leukemia (AML) with t(8;21) (q22...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Class   Gene    ID                                               Text  \\\n",
       "539     4.0  SMAD2   539  Transforming growth factor- (TGF-) plays a com...   \n",
       "1157    1.0  KMT2C  1157   Abstract Monomethylation of histone H3 on Lys...   \n",
       "3201    1.0  RASA1  3201  Phospholipase C gamma 1 (PLC gamma 1) and p21r...   \n",
       "3253    1.0  CASP8  3253  Evidence exists that alterations of the genes ...   \n",
       "7153    NaN  ASXL2  3832  Acute myeloid leukemia (AML) with t(8;21) (q22...   \n",
       "\n",
       "     Variation  Substitutions_var  Stop_codon_var  gene_fusion_var  \\\n",
       "539   Deletion                  0               0                0   \n",
       "1157  Deletion                  0               0                0   \n",
       "3201  Deletion                  0               0                0   \n",
       "3253  Deletion                  0               0                0   \n",
       "7153  Deletion                  0               0                0   \n",
       "\n",
       "      Deletion_var  del_or_ins_var  Fusion_var  Amplification_var  \\\n",
       "539              1               0           0                  0   \n",
       "1157             1               0           0                  0   \n",
       "3201             1               0           0                  0   \n",
       "3253             1               0           0                  0   \n",
       "7153             1               0           0                  0   \n",
       "\n",
       "      Truncation_var  exon_var  frameshift_var  dup_var  \n",
       "539                0         0               0        0  \n",
       "1157               0         0               0        0  \n",
       "3201               0         0               0        0  \n",
       "3253               0         0               0        0  \n",
       "7153               0         0               0        0  "
      ]
     },
     "execution_count": 449,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "deletion_not_in_text"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 438,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "NLTK_deletion = [sent_tokenize(deletion_in_text.Text[i]) for i in deletion_in_text.index] \n",
    "sentences_deletion = get_sentences_fusion(deletion_in_text, NLTK_deletion, 'del', window_left=0, window_right=0)\n",
    "# still work to do"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Deletions: the del_or_ins variations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 462,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "L747_T751delinsP\n",
      "S752_I759del\n",
      "E746_T751insIP\n",
      "D770_N771insD\n",
      "K745_A750del\n",
      "D770_N771insNPG\n",
      "E746_A750del\n",
      "A859_L883delinsV\n",
      "A750_E758del\n",
      "V769_D770insGVV\n",
      "A750_E758delinsP\n",
      "L747_P753delinsS\n",
      "H773insLGNP\n",
      "H773_V774insH\n",
      "L747_A750delinsP\n",
      "L747_E749del\n",
      "L747_T751del\n",
      "L747_A750del\n",
      "E746_T751delinsA\n",
      "E709_T710delinsD\n",
      "A763_Y764insFQEA\n",
      "I744_K745delinsKIPVAI\n",
      "E746_A750delinsQ\n",
      "L747_P753del\n",
      "S768_V769insVAS\n",
      "E746_T751delinsVA\n",
      "A767_V769del\n",
      "T34_A289del\n",
      "D770_N771insVDSVDNP\n",
      "E746_S752delinsA\n",
      "E746_S752delinsI\n",
      "E161del\n",
      "3' Deletion\n",
      "534_536del\n",
      "533_534del\n",
      "DNA binding domain deletions\n",
      "DNA binding domain insertions\n",
      "G776delinsVC\n",
      "M774_A775insAYVM\n",
      "G776delinsLC\n",
      "G776_V777insYVMA\n",
      "C450_K451insMIEWMI\n",
      "C456_N468del\n",
      "I843_D846del\n",
      "V561_I562insER\n",
      "Y375_K455del\n",
      "D842_M844del\n",
      "D842_H845del\n",
      "S566_E571delinsR\n",
      "E311_K312del\n",
      "I843del\n",
      "C456_R481del\n",
      "H845_N848delinsP\n",
      "560_561insER\n",
      "V544_L545insAVLVLLVIVIISLI\n",
      "M1_E165DEL\n",
      "V422del\n",
      "385_418del\n",
      "N198_F199delinsI\n",
      "V128del\n",
      "E1552del\n",
      "P449_L455del\n",
      "G106_R108del\n",
      "T576del\n",
      "I559_D560insDKRMNS\n",
      "W237_Y242del\n",
      "Q579_L581del\n",
      "K459_S460delinsN\n",
      "E439del\n",
      "W290_I291delinsC\n",
      "Y599_D600insGLYVDFREYEY\n",
      "D835del\n",
      "I836del\n",
      "Y599_D600insSTDNEYFYVDFREYEY\n",
      "E598_Y599insDVDFREYE\n",
      "W603_E604insDREYEYDLKW\n",
      "S840_N841insGS\n",
      "L601_K602insREYEYDL\n",
      "F594_R595insSDNEYFYVDF\n",
      "L611_E612insCSSDNEYFYVDFREYEYDLKWEFPRENL\n",
      "E612_F613insGYVDFREYEYDLKWEFRPRENLEF\n",
      "Y599_D600insPAPQIMSTSTLISENMNIA\n",
      "E598_Y599insGLVQVTGSSDNEYFYVDFREYE\n",
      "Y599_D600insEYEYEYEY\n",
      "D600_L601insFREYEYD\n",
      "S459del\n",
      "Q58_Q59insL\n",
      "L57del\n",
      "Q347_A348del\n",
      "S45del\n",
      "Q56_V60del\n",
      "E102_I103del\n",
      "K59del\n",
      "Q58_E62del\n",
      "T244_I245insCPT\n",
      "D289del\n",
      "D289_D292del\n",
      "L283_D294del\n",
      "I32del\n",
      "M199del\n",
      "I33del\n",
      "T319del\n",
      "F537_K539delinsL\n",
      "N542_E543del\n",
      "S1297del\n",
      "V1688del\n",
      "T599_V600insV\n",
      "L485_P490del\n",
      "L485_P490delinsF\n",
      "P490_Q494del\n",
      "R506_K507insVLR\n",
      "L485_P490delinsY\n",
      "T488_P492del\n",
      "V487_P492delinsA\n",
      "L485_Q494del\n",
      "T599_V600insEAT\n",
      "G469del\n",
      "N486_P490del\n",
      "V600D_K601insFGLAT\n",
      "L576del\n",
      "I563_L576del\n",
      "K550_K558del\n",
      "P577_D579del\n",
      "Y553_K558del\n",
      "P551_E554del\n",
      "T417_D419delinsRG\n",
      "T417_D419delinsI\n",
      "K558delinsNP\n",
      "Y553_Q556del\n",
      "P577_W582delinsPYD\n",
      "E554_K558del\n",
      "P551_V555del\n",
      "D419del\n",
      "T574_R588delinsL\n",
      "V559_V560del\n",
      "M552_K558del\n",
      "A504_Y505ins\n",
      "V559del\n",
      "Q556_K558del\n",
      "P551_W557delinsL\n",
      "N564_Y578del\n",
      "D579del\n",
      "K558_V559del\n",
      "V569_L576del\n",
      "E554_V559del\n",
      "550_592del\n",
      "W557_V559delinsC\n",
      "P573_D579del\n",
      "W557_K558del\n",
      "E554_I571del\n",
      "K558_E562del\n",
      "W557_V560del\n",
      "Y418_D419del\n",
      "K550_W557del\n",
      "V555_V559del\n",
      "K550_V555delinsI\n",
      "V555_L576del\n",
      "M552_W557del\n",
      "T574insTQLPYD\n",
      "V1578del\n",
      "V1605del\n",
      "P2415del\n",
      "A11_G12insGA\n",
      "N480del\n",
      "E632_L633del\n",
      "P551_M552del\n",
      "W559_R560del\n",
      "A775_G776insYVMA\n",
      "D842_I843delinsIM\n",
      "K1110del\n",
      "V600delinsYM\n",
      "P447_L455del\n",
      "V241del\n",
      "L747_S752del\n",
      "D600_L601insDFREYEYD\n",
      "V600_K601delinsE\n",
      "V769_D770insASV\n",
      "Y568_L576delinsVN\n",
      "V559_G565del\n",
      "D770_N771insSVD\n",
      "P772_H773insYNP\n",
      "E746_S752delinsV\n",
      "T599_V600insETT\n",
      "T599insTT\n",
      "E746_T751delinsL\n",
      "L755_T759del\n",
      "K602_W603insYEYDLK\n",
      "V560del\n"
     ]
    }
   ],
   "source": [
    "del_or_ins = data_all[(data_all['del_or_ins_var']==1) & (data_all['exon_var']==0)]\n",
    "for Variation in del_or_ins.Variation:\n",
    "    print(Variation)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 460,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Class</th>\n",
       "      <th>Gene</th>\n",
       "      <th>ID</th>\n",
       "      <th>Text</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Substitutions_var</th>\n",
       "      <th>Stop_codon_var</th>\n",
       "      <th>gene_fusion_var</th>\n",
       "      <th>Deletion_var</th>\n",
       "      <th>del_or_ins_var</th>\n",
       "      <th>Fusion_var</th>\n",
       "      <th>Amplification_var</th>\n",
       "      <th>Truncation_var</th>\n",
       "      <th>exon_var</th>\n",
       "      <th>frameshift_var</th>\n",
       "      <th>dup_var</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>138</th>\n",
       "      <td>7.0</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>138</td>\n",
       "      <td>Non–small-cell lung cancer is the leading caus...</td>\n",
       "      <td>L747_T751delinsP</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139</th>\n",
       "      <td>2.0</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>139</td>\n",
       "      <td>In contrast to other primary epidermal growth...</td>\n",
       "      <td>S752_I759del</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>146</th>\n",
       "      <td>7.0</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>146</td>\n",
       "      <td>Somatic mutations introduced into the epiderma...</td>\n",
       "      <td>E746_T751insIP</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>147</th>\n",
       "      <td>7.0</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>147</td>\n",
       "      <td>In contrast to other primary epidermal growth...</td>\n",
       "      <td>D770_N771insD</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>149</th>\n",
       "      <td>7.0</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>149</td>\n",
       "      <td>Non–small-cell lung cancer is the leading caus...</td>\n",
       "      <td>K745_A750del</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165</th>\n",
       "      <td>7.0</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>165</td>\n",
       "      <td>In contrast to other primary epidermal growth...</td>\n",
       "      <td>D770_N771insNPG</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>166</th>\n",
       "      <td>7.0</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>166</td>\n",
       "      <td>Abstract  Malignant pleural effusions (MPEs) ...</td>\n",
       "      <td>E746_A750del</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>169</th>\n",
       "      <td>7.0</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>169</td>\n",
       "      <td>Somatic mutations in the tyrosine kinase (TK) ...</td>\n",
       "      <td>Exon 19 deletion/insertion</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>171</th>\n",
       "      <td>2.0</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>171</td>\n",
       "      <td>Somatic mutations introduced into the epiderma...</td>\n",
       "      <td>A859_L883delinsV</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>174</th>\n",
       "      <td>7.0</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>174</td>\n",
       "      <td>Somatic mutations introduced into the epiderma...</td>\n",
       "      <td>A750_E758del</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>175</th>\n",
       "      <td>7.0</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>175</td>\n",
       "      <td>In contrast to other primary epidermal growth...</td>\n",
       "      <td>V769_D770insGVV</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>184</th>\n",
       "      <td>7.0</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>184</td>\n",
       "      <td>Non–small-cell lung cancer is the leading caus...</td>\n",
       "      <td>A750_E758delinsP</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>187</th>\n",
       "      <td>7.0</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>187</td>\n",
       "      <td>Abstract  Malignant pleural effusions (MPEs) ...</td>\n",
       "      <td>L747_P753delinsS</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>188</th>\n",
       "      <td>7.0</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>188</td>\n",
       "      <td>Somatic mutations in the tyrosine kinase (TK) ...</td>\n",
       "      <td>Exon 19 deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193</th>\n",
       "      <td>7.0</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>193</td>\n",
       "      <td>In contrast to other primary epidermal growth...</td>\n",
       "      <td>H773insLGNP</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>194</th>\n",
       "      <td>7.0</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>194</td>\n",
       "      <td>NTRODUCTION Non–small cell lung cancers (NSCLC...</td>\n",
       "      <td>H773_V774insH</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>195</th>\n",
       "      <td>7.0</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>195</td>\n",
       "      <td>Abstract  Malignant pleural effusions (MPEs) ...</td>\n",
       "      <td>L747_A750delinsP</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197</th>\n",
       "      <td>7.0</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>197</td>\n",
       "      <td>In contrast to other primary epidermal growth...</td>\n",
       "      <td>Exon 20 insertion</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>198</th>\n",
       "      <td>7.0</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>198</td>\n",
       "      <td>The discovery of Rous sarcoma virus (RSV) led ...</td>\n",
       "      <td>Exon 19 insertion</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>203</th>\n",
       "      <td>7.0</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>203</td>\n",
       "      <td>Mutations in the epidermal growth factor recep...</td>\n",
       "      <td>L747_E749del</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>206</th>\n",
       "      <td>7.0</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>206</td>\n",
       "      <td>Somatic mutations introduced into the epiderma...</td>\n",
       "      <td>L747_T751del</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>212</th>\n",
       "      <td>7.0</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>212</td>\n",
       "      <td>Mutations in the epidermal growth factor recep...</td>\n",
       "      <td>L747_A750del</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>218</th>\n",
       "      <td>7.0</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>218</td>\n",
       "      <td>Non–small-cell lung cancer is the leading caus...</td>\n",
       "      <td>E746_T751delinsA</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>225</th>\n",
       "      <td>7.0</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>225</td>\n",
       "      <td>Purpose: Clinical features of epidermal growth...</td>\n",
       "      <td>E709_T710delinsD</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>226</th>\n",
       "      <td>7.0</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>226</td>\n",
       "      <td>NTRODUCTION Non–small cell lung cancers (NSCLC...</td>\n",
       "      <td>A763_Y764insFQEA</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>229</th>\n",
       "      <td>7.0</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>229</td>\n",
       "      <td>Non–small-cell lung cancer is the leading caus...</td>\n",
       "      <td>I744_K745delinsKIPVAI</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>231</th>\n",
       "      <td>7.0</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>231</td>\n",
       "      <td>Non–small-cell lung cancer is the leading caus...</td>\n",
       "      <td>E746_A750delinsQ</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>232</th>\n",
       "      <td>7.0</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>232</td>\n",
       "      <td>Non–small-cell lung cancer is the leading caus...</td>\n",
       "      <td>L747_P753del</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>239</th>\n",
       "      <td>7.0</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>239</td>\n",
       "      <td>In contrast to other primary epidermal growth...</td>\n",
       "      <td>S768_V769insVAS</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>240</th>\n",
       "      <td>7.0</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>240</td>\n",
       "      <td>Abstract  Malignant pleural effusions (MPEs) ...</td>\n",
       "      <td>E746_T751delinsVA</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3066</th>\n",
       "      <td>2.0</td>\n",
       "      <td>NOTCH1</td>\n",
       "      <td>3066</td>\n",
       "      <td>The majority of human pancreatic cancers have ...</td>\n",
       "      <td>V1578del</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3069</th>\n",
       "      <td>2.0</td>\n",
       "      <td>NOTCH1</td>\n",
       "      <td>3069</td>\n",
       "      <td>The majority of human pancreatic cancers have ...</td>\n",
       "      <td>V1605del</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3086</th>\n",
       "      <td>2.0</td>\n",
       "      <td>NOTCH1</td>\n",
       "      <td>3086</td>\n",
       "      <td>Since Notch phenotypes in Drosophila melanogas...</td>\n",
       "      <td>P2415del</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3119</th>\n",
       "      <td>7.0</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>3119</td>\n",
       "      <td>KRAS mutational status has been shown to be a ...</td>\n",
       "      <td>A11_G12insGA</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3209</th>\n",
       "      <td>5.0</td>\n",
       "      <td>RB1</td>\n",
       "      <td>3209</td>\n",
       "      <td>Abstract Certain kindreds with low-penetrant ...</td>\n",
       "      <td>N480del</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3263</th>\n",
       "      <td>7.0</td>\n",
       "      <td>RET</td>\n",
       "      <td>3263</td>\n",
       "      <td>Somatic RET mutations have been identi®ed in a...</td>\n",
       "      <td>E632_L633del</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3440</th>\n",
       "      <td>NaN</td>\n",
       "      <td>KIT</td>\n",
       "      <td>119</td>\n",
       "      <td>The pathogenesis of most gastrointestinal stro...</td>\n",
       "      <td>P551_M552del</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3665</th>\n",
       "      <td>NaN</td>\n",
       "      <td>PDGFRA</td>\n",
       "      <td>344</td>\n",
       "      <td>Gastrointestinal stromal tumors (GISTs) are co...</td>\n",
       "      <td>W559_R560del</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4251</th>\n",
       "      <td>NaN</td>\n",
       "      <td>ERBB2</td>\n",
       "      <td>930</td>\n",
       "      <td>The epidermal growth factor receptor (EGFR) ha...</td>\n",
       "      <td>A775_G776insYVMA</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4282</th>\n",
       "      <td>NaN</td>\n",
       "      <td>PDGFRA</td>\n",
       "      <td>961</td>\n",
       "      <td>Gastrointestinal stromal tumors (GISTs) common...</td>\n",
       "      <td>D842_I843delinsIM</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4700</th>\n",
       "      <td>NaN</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>1379</td>\n",
       "      <td>Mutations in BRCA1 and BRCA2 account for the m...</td>\n",
       "      <td>K1110del</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4949</th>\n",
       "      <td>NaN</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>1628</td>\n",
       "      <td>Papillary thyroid carcinoma (PTC) represents t...</td>\n",
       "      <td>V600delinsYM</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5222</th>\n",
       "      <td>NaN</td>\n",
       "      <td>PIK3CA</td>\n",
       "      <td>1901</td>\n",
       "      <td>ABSTRACT Germline mutations in the BRCAJ tumo...</td>\n",
       "      <td>P447_L455del</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5223</th>\n",
       "      <td>NaN</td>\n",
       "      <td>TSC2</td>\n",
       "      <td>1902</td>\n",
       "      <td>Tuberous sclerosis complex (TSC) is an autosom...</td>\n",
       "      <td>V241del</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5472</th>\n",
       "      <td>NaN</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>2151</td>\n",
       "      <td>Mutations in the epidermal growth factor recep...</td>\n",
       "      <td>L747_S752del</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5551</th>\n",
       "      <td>NaN</td>\n",
       "      <td>FLT3</td>\n",
       "      <td>2230</td>\n",
       "      <td>An internal tandem duplication (ITD) of the FL...</td>\n",
       "      <td>D600_L601insDFREYEYD</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5582</th>\n",
       "      <td>NaN</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>2261</td>\n",
       "      <td>Vemurafenib has shown significant activity in ...</td>\n",
       "      <td>V600_K601delinsE</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5602</th>\n",
       "      <td>NaN</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>2281</td>\n",
       "      <td>In contrast to other primary epidermal growth...</td>\n",
       "      <td>V769_D770insASV</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5969</th>\n",
       "      <td>NaN</td>\n",
       "      <td>KIT</td>\n",
       "      <td>2648</td>\n",
       "      <td>The pathogenesis of most gastrointestinal stro...</td>\n",
       "      <td>Y568_L576delinsVN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6830</th>\n",
       "      <td>NaN</td>\n",
       "      <td>KIT</td>\n",
       "      <td>3509</td>\n",
       "      <td>Gastrointestinal stromal tumors (GIST) are cha...</td>\n",
       "      <td>V559_G565del</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7357</th>\n",
       "      <td>NaN</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>4036</td>\n",
       "      <td>In contrast to other primary epidermal growth...</td>\n",
       "      <td>D770_N771insSVD</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7822</th>\n",
       "      <td>NaN</td>\n",
       "      <td>KIT</td>\n",
       "      <td>4501</td>\n",
       "      <td>The outcome of patients diagnosed with advance...</td>\n",
       "      <td>Exon 11 deletions</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7849</th>\n",
       "      <td>NaN</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>4528</td>\n",
       "      <td>In contrast to other primary epidermal growth...</td>\n",
       "      <td>P772_H773insYNP</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7886</th>\n",
       "      <td>NaN</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>4565</td>\n",
       "      <td>Non–small-cell lung cancer is the leading caus...</td>\n",
       "      <td>E746_S752delinsV</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7928</th>\n",
       "      <td>NaN</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>4607</td>\n",
       "      <td>Abstract Activating mutations of the BRAF gen...</td>\n",
       "      <td>T599_V600insETT</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7952</th>\n",
       "      <td>NaN</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>4631</td>\n",
       "      <td>Pilocytic astrocytoma (PA) is emerging as a tu...</td>\n",
       "      <td>T599insTT</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7976</th>\n",
       "      <td>NaN</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>4655</td>\n",
       "      <td>Non–small-cell lung cancer is the leading caus...</td>\n",
       "      <td>E746_T751delinsL</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8159</th>\n",
       "      <td>NaN</td>\n",
       "      <td>ERBB2</td>\n",
       "      <td>4838</td>\n",
       "      <td>Functional characterization of cancer-associat...</td>\n",
       "      <td>L755_T759del</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8747</th>\n",
       "      <td>NaN</td>\n",
       "      <td>FLT3</td>\n",
       "      <td>5426</td>\n",
       "      <td>An internal tandem duplication (ITD) of the FL...</td>\n",
       "      <td>K602_W603insYEYDLK</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8785</th>\n",
       "      <td>NaN</td>\n",
       "      <td>KIT</td>\n",
       "      <td>5464</td>\n",
       "      <td>To analyze a multi-institutional series of typ...</td>\n",
       "      <td>V560del</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>196 rows × 16 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Class    Gene    ID                                               Text  \\\n",
       "138     7.0    EGFR   138  Non–small-cell lung cancer is the leading caus...   \n",
       "139     2.0    EGFR   139   In contrast to other primary epidermal growth...   \n",
       "146     7.0    EGFR   146  Somatic mutations introduced into the epiderma...   \n",
       "147     7.0    EGFR   147   In contrast to other primary epidermal growth...   \n",
       "149     7.0    EGFR   149  Non–small-cell lung cancer is the leading caus...   \n",
       "165     7.0    EGFR   165   In contrast to other primary epidermal growth...   \n",
       "166     7.0    EGFR   166   Abstract  Malignant pleural effusions (MPEs) ...   \n",
       "169     7.0    EGFR   169  Somatic mutations in the tyrosine kinase (TK) ...   \n",
       "171     2.0    EGFR   171  Somatic mutations introduced into the epiderma...   \n",
       "174     7.0    EGFR   174  Somatic mutations introduced into the epiderma...   \n",
       "175     7.0    EGFR   175   In contrast to other primary epidermal growth...   \n",
       "184     7.0    EGFR   184  Non–small-cell lung cancer is the leading caus...   \n",
       "187     7.0    EGFR   187   Abstract  Malignant pleural effusions (MPEs) ...   \n",
       "188     7.0    EGFR   188  Somatic mutations in the tyrosine kinase (TK) ...   \n",
       "193     7.0    EGFR   193   In contrast to other primary epidermal growth...   \n",
       "194     7.0    EGFR   194  NTRODUCTION Non–small cell lung cancers (NSCLC...   \n",
       "195     7.0    EGFR   195   Abstract  Malignant pleural effusions (MPEs) ...   \n",
       "197     7.0    EGFR   197   In contrast to other primary epidermal growth...   \n",
       "198     7.0    EGFR   198  The discovery of Rous sarcoma virus (RSV) led ...   \n",
       "203     7.0    EGFR   203  Mutations in the epidermal growth factor recep...   \n",
       "206     7.0    EGFR   206  Somatic mutations introduced into the epiderma...   \n",
       "212     7.0    EGFR   212  Mutations in the epidermal growth factor recep...   \n",
       "218     7.0    EGFR   218  Non–small-cell lung cancer is the leading caus...   \n",
       "225     7.0    EGFR   225  Purpose: Clinical features of epidermal growth...   \n",
       "226     7.0    EGFR   226  NTRODUCTION Non–small cell lung cancers (NSCLC...   \n",
       "229     7.0    EGFR   229  Non–small-cell lung cancer is the leading caus...   \n",
       "231     7.0    EGFR   231  Non–small-cell lung cancer is the leading caus...   \n",
       "232     7.0    EGFR   232  Non–small-cell lung cancer is the leading caus...   \n",
       "239     7.0    EGFR   239   In contrast to other primary epidermal growth...   \n",
       "240     7.0    EGFR   240   Abstract  Malignant pleural effusions (MPEs) ...   \n",
       "...     ...     ...   ...                                                ...   \n",
       "3066    2.0  NOTCH1  3066  The majority of human pancreatic cancers have ...   \n",
       "3069    2.0  NOTCH1  3069  The majority of human pancreatic cancers have ...   \n",
       "3086    2.0  NOTCH1  3086  Since Notch phenotypes in Drosophila melanogas...   \n",
       "3119    7.0    KRAS  3119  KRAS mutational status has been shown to be a ...   \n",
       "3209    5.0     RB1  3209   Abstract Certain kindreds with low-penetrant ...   \n",
       "3263    7.0     RET  3263  Somatic RET mutations have been identi®ed in a...   \n",
       "3440    NaN     KIT   119  The pathogenesis of most gastrointestinal stro...   \n",
       "3665    NaN  PDGFRA   344  Gastrointestinal stromal tumors (GISTs) are co...   \n",
       "4251    NaN   ERBB2   930  The epidermal growth factor receptor (EGFR) ha...   \n",
       "4282    NaN  PDGFRA   961  Gastrointestinal stromal tumors (GISTs) common...   \n",
       "4700    NaN   BRCA1  1379  Mutations in BRCA1 and BRCA2 account for the m...   \n",
       "4949    NaN    BRAF  1628  Papillary thyroid carcinoma (PTC) represents t...   \n",
       "5222    NaN  PIK3CA  1901   ABSTRACT Germline mutations in the BRCAJ tumo...   \n",
       "5223    NaN    TSC2  1902  Tuberous sclerosis complex (TSC) is an autosom...   \n",
       "5472    NaN    EGFR  2151  Mutations in the epidermal growth factor recep...   \n",
       "5551    NaN    FLT3  2230  An internal tandem duplication (ITD) of the FL...   \n",
       "5582    NaN    BRAF  2261  Vemurafenib has shown significant activity in ...   \n",
       "5602    NaN    EGFR  2281   In contrast to other primary epidermal growth...   \n",
       "5969    NaN     KIT  2648  The pathogenesis of most gastrointestinal stro...   \n",
       "6830    NaN     KIT  3509  Gastrointestinal stromal tumors (GIST) are cha...   \n",
       "7357    NaN    EGFR  4036   In contrast to other primary epidermal growth...   \n",
       "7822    NaN     KIT  4501  The outcome of patients diagnosed with advance...   \n",
       "7849    NaN    EGFR  4528   In contrast to other primary epidermal growth...   \n",
       "7886    NaN    EGFR  4565  Non–small-cell lung cancer is the leading caus...   \n",
       "7928    NaN    BRAF  4607   Abstract Activating mutations of the BRAF gen...   \n",
       "7952    NaN    BRAF  4631  Pilocytic astrocytoma (PA) is emerging as a tu...   \n",
       "7976    NaN    EGFR  4655  Non–small-cell lung cancer is the leading caus...   \n",
       "8159    NaN   ERBB2  4838  Functional characterization of cancer-associat...   \n",
       "8747    NaN    FLT3  5426  An internal tandem duplication (ITD) of the FL...   \n",
       "8785    NaN     KIT  5464  To analyze a multi-institutional series of typ...   \n",
       "\n",
       "                       Variation  Substitutions_var  Stop_codon_var  \\\n",
       "138             L747_T751delinsP                  0               0   \n",
       "139                 S752_I759del                  0               0   \n",
       "146               E746_T751insIP                  0               0   \n",
       "147                D770_N771insD                  0               0   \n",
       "149                 K745_A750del                  0               0   \n",
       "165              D770_N771insNPG                  0               0   \n",
       "166                 E746_A750del                  0               0   \n",
       "169   Exon 19 deletion/insertion                  0               0   \n",
       "171             A859_L883delinsV                  0               0   \n",
       "174                 A750_E758del                  0               0   \n",
       "175              V769_D770insGVV                  0               0   \n",
       "184             A750_E758delinsP                  0               0   \n",
       "187             L747_P753delinsS                  0               0   \n",
       "188             Exon 19 deletion                  0               0   \n",
       "193                  H773insLGNP                  0               0   \n",
       "194                H773_V774insH                  0               0   \n",
       "195             L747_A750delinsP                  0               0   \n",
       "197            Exon 20 insertion                  0               0   \n",
       "198            Exon 19 insertion                  0               0   \n",
       "203                 L747_E749del                  0               0   \n",
       "206                 L747_T751del                  0               0   \n",
       "212                 L747_A750del                  0               0   \n",
       "218             E746_T751delinsA                  0               0   \n",
       "225             E709_T710delinsD                  0               0   \n",
       "226             A763_Y764insFQEA                  0               0   \n",
       "229        I744_K745delinsKIPVAI                  0               0   \n",
       "231             E746_A750delinsQ                  0               0   \n",
       "232                 L747_P753del                  0               0   \n",
       "239              S768_V769insVAS                  0               0   \n",
       "240            E746_T751delinsVA                  0               0   \n",
       "...                          ...                ...             ...   \n",
       "3066                    V1578del                  0               0   \n",
       "3069                    V1605del                  0               0   \n",
       "3086                    P2415del                  0               0   \n",
       "3119                A11_G12insGA                  0               0   \n",
       "3209                     N480del                  0               0   \n",
       "3263                E632_L633del                  0               0   \n",
       "3440                P551_M552del                  0               0   \n",
       "3665                W559_R560del                  0               0   \n",
       "4251            A775_G776insYVMA                  0               0   \n",
       "4282           D842_I843delinsIM                  0               0   \n",
       "4700                    K1110del                  0               0   \n",
       "4949                V600delinsYM                  0               0   \n",
       "5222                P447_L455del                  0               0   \n",
       "5223                     V241del                  0               0   \n",
       "5472                L747_S752del                  0               0   \n",
       "5551        D600_L601insDFREYEYD                  0               0   \n",
       "5582            V600_K601delinsE                  0               0   \n",
       "5602             V769_D770insASV                  0               0   \n",
       "5969           Y568_L576delinsVN                  0               0   \n",
       "6830                V559_G565del                  0               0   \n",
       "7357             D770_N771insSVD                  0               0   \n",
       "7822           Exon 11 deletions                  0               0   \n",
       "7849             P772_H773insYNP                  0               0   \n",
       "7886            E746_S752delinsV                  0               0   \n",
       "7928             T599_V600insETT                  0               0   \n",
       "7952                   T599insTT                  0               0   \n",
       "7976            E746_T751delinsL                  0               0   \n",
       "8159                L755_T759del                  0               0   \n",
       "8747          K602_W603insYEYDLK                  0               0   \n",
       "8785                     V560del                  0               0   \n",
       "\n",
       "      gene_fusion_var  Deletion_var  del_or_ins_var  Fusion_var  \\\n",
       "138                 0             0               1           0   \n",
       "139                 0             0               1           0   \n",
       "146                 0             0               1           0   \n",
       "147                 0             0               1           0   \n",
       "149                 0             0               1           0   \n",
       "165                 0             0               1           0   \n",
       "166                 0             0               1           0   \n",
       "169                 0             0               1           0   \n",
       "171                 0             0               1           0   \n",
       "174                 0             0               1           0   \n",
       "175                 0             0               1           0   \n",
       "184                 0             0               1           0   \n",
       "187                 0             0               1           0   \n",
       "188                 0             0               1           0   \n",
       "193                 0             0               1           0   \n",
       "194                 0             0               1           0   \n",
       "195                 0             0               1           0   \n",
       "197                 0             0               1           0   \n",
       "198                 0             0               1           0   \n",
       "203                 0             0               1           0   \n",
       "206                 0             0               1           0   \n",
       "212                 0             0               1           0   \n",
       "218                 0             0               1           0   \n",
       "225                 0             0               1           0   \n",
       "226                 0             0               1           0   \n",
       "229                 0             0               1           0   \n",
       "231                 0             0               1           0   \n",
       "232                 0             0               1           0   \n",
       "239                 0             0               1           0   \n",
       "240                 0             0               1           0   \n",
       "...               ...           ...             ...         ...   \n",
       "3066                0             0               1           0   \n",
       "3069                0             0               1           0   \n",
       "3086                0             0               1           0   \n",
       "3119                0             0               1           0   \n",
       "3209                0             0               1           0   \n",
       "3263                0             0               1           0   \n",
       "3440                0             0               1           0   \n",
       "3665                0             0               1           0   \n",
       "4251                0             0               1           0   \n",
       "4282                0             0               1           0   \n",
       "4700                0             0               1           0   \n",
       "4949                0             0               1           0   \n",
       "5222                0             0               1           0   \n",
       "5223                0             0               1           0   \n",
       "5472                0             0               1           0   \n",
       "5551                0             0               1           0   \n",
       "5582                0             0               1           0   \n",
       "5602                0             0               1           0   \n",
       "5969                0             0               1           0   \n",
       "6830                0             0               1           0   \n",
       "7357                0             0               1           0   \n",
       "7822                0             0               1           0   \n",
       "7849                0             0               1           0   \n",
       "7886                0             0               1           0   \n",
       "7928                0             0               1           0   \n",
       "7952                0             0               1           0   \n",
       "7976                0             0               1           0   \n",
       "8159                0             0               1           0   \n",
       "8747                0             0               1           0   \n",
       "8785                0             0               1           0   \n",
       "\n",
       "      Amplification_var  Truncation_var  exon_var  frameshift_var  dup_var  \n",
       "138                   0               0         0               0        0  \n",
       "139                   0               0         0               0        0  \n",
       "146                   0               0         0               0        0  \n",
       "147                   0               0         0               0        0  \n",
       "149                   0               0         0               0        0  \n",
       "165                   0               0         0               0        0  \n",
       "166                   0               0         0               0        0  \n",
       "169                   0               0         1               0        0  \n",
       "171                   0               0         0               0        0  \n",
       "174                   0               0         0               0        0  \n",
       "175                   0               0         0               0        0  \n",
       "184                   0               0         0               0        0  \n",
       "187                   0               0         0               0        0  \n",
       "188                   0               0         1               0        0  \n",
       "193                   0               0         0               0        0  \n",
       "194                   0               0         0               0        0  \n",
       "195                   0               0         0               0        0  \n",
       "197                   0               0         1               0        0  \n",
       "198                   0               0         1               0        0  \n",
       "203                   0               0         0               0        0  \n",
       "206                   0               0         0               0        0  \n",
       "212                   0               0         0               0        0  \n",
       "218                   0               0         0               0        0  \n",
       "225                   0               0         0               0        0  \n",
       "226                   0               0         0               0        0  \n",
       "229                   0               0         0               0        0  \n",
       "231                   0               0         0               0        0  \n",
       "232                   0               0         0               0        0  \n",
       "239                   0               0         0               0        0  \n",
       "240                   0               0         0               0        0  \n",
       "...                 ...             ...       ...             ...      ...  \n",
       "3066                  0               0         0               0        0  \n",
       "3069                  0               0         0               0        0  \n",
       "3086                  0               0         0               0        0  \n",
       "3119                  0               0         0               0        0  \n",
       "3209                  0               0         0               0        0  \n",
       "3263                  0               0         0               0        0  \n",
       "3440                  0               0         0               0        0  \n",
       "3665                  0               0         0               0        0  \n",
       "4251                  0               0         0               0        0  \n",
       "4282                  0               0         0               0        0  \n",
       "4700                  0               0         0               0        0  \n",
       "4949                  0               0         0               0        0  \n",
       "5222                  0               0         0               0        0  \n",
       "5223                  0               0         0               0        0  \n",
       "5472                  0               0         0               0        0  \n",
       "5551                  0               0         0               0        0  \n",
       "5582                  0               0         0               0        0  \n",
       "5602                  0               0         0               0        0  \n",
       "5969                  0               0         0               0        0  \n",
       "6830                  0               0         0               0        0  \n",
       "7357                  0               0         0               0        0  \n",
       "7822                  0               0         1               0        0  \n",
       "7849                  0               0         0               0        0  \n",
       "7886                  0               0         0               0        0  \n",
       "7928                  0               0         0               0        0  \n",
       "7952                  0               0         0               0        0  \n",
       "7976                  0               0         0               0        0  \n",
       "8159                  0               0         0               0        0  \n",
       "8747                  0               0         0               0        0  \n",
       "8785                  0               0         0               0        0  \n",
       "\n",
       "[196 rows x 16 columns]"
      ]
     },
     "execution_count": 460,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "del_or_ins"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Truncations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 455,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "truncations = data_all[(data_all['Truncation_var']==1)]\n",
    "truncation_in_text, truncation_not_in_text = find_nosub_gene(truncations, 'trunc') # only 5 left in deletion_not_in_text"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 456,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Class</th>\n",
       "      <th>Gene</th>\n",
       "      <th>ID</th>\n",
       "      <th>Text</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Substitutions_var</th>\n",
       "      <th>Stop_codon_var</th>\n",
       "      <th>gene_fusion_var</th>\n",
       "      <th>Deletion_var</th>\n",
       "      <th>del_or_ins_var</th>\n",
       "      <th>Fusion_var</th>\n",
       "      <th>Amplification_var</th>\n",
       "      <th>Truncation_var</th>\n",
       "      <th>exon_var</th>\n",
       "      <th>frameshift_var</th>\n",
       "      <th>dup_var</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>1.0</td>\n",
       "      <td>RYBP</td>\n",
       "      <td>88</td>\n",
       "      <td>The mouse double minute 2 (MDM2)–p53 interacti...</td>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>313</th>\n",
       "      <td>1.0</td>\n",
       "      <td>ELF3</td>\n",
       "      <td>313</td>\n",
       "      <td>Previous studies have shown that the promoter ...</td>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>532</th>\n",
       "      <td>1.0</td>\n",
       "      <td>TP53BP1</td>\n",
       "      <td>532</td>\n",
       "      <td>Introduction  Inhibition of PARP1 induces synt...</td>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1015</th>\n",
       "      <td>1.0</td>\n",
       "      <td>TSC2</td>\n",
       "      <td>1015</td>\n",
       "      <td>SUMMARY mTORC1 promotes cell growth in respon...</td>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1080</th>\n",
       "      <td>1.0</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>1080</td>\n",
       "      <td>ARID1A (BAF250A) promotes the formation of SWI...</td>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1277</th>\n",
       "      <td>1.0</td>\n",
       "      <td>ARID5B</td>\n",
       "      <td>1277</td>\n",
       "      <td>null</td>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1826</th>\n",
       "      <td>1.0</td>\n",
       "      <td>NCOR1</td>\n",
       "      <td>1826</td>\n",
       "      <td>Estrogen receptors (ERs) are normally expresse...</td>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1902</th>\n",
       "      <td>1.0</td>\n",
       "      <td>TET1</td>\n",
       "      <td>1902</td>\n",
       "      <td>TET2 is a close relative of TET1, an enzyme th...</td>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1911</th>\n",
       "      <td>1.0</td>\n",
       "      <td>FOXP1</td>\n",
       "      <td>1911</td>\n",
       "      <td>The transcriptional network of the androgen re...</td>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2146</th>\n",
       "      <td>1.0</td>\n",
       "      <td>PTCH1</td>\n",
       "      <td>2146</td>\n",
       "      <td>Basal cell carcinoma (BCC) is the most common ...</td>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2285</th>\n",
       "      <td>1.0</td>\n",
       "      <td>INPP4B</td>\n",
       "      <td>2285</td>\n",
       "      <td>Although the gastrointestinal (GI) tract posse...</td>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3160</th>\n",
       "      <td>1.0</td>\n",
       "      <td>NPM1</td>\n",
       "      <td>3160</td>\n",
       "      <td>Hemizygous deletions are common molecular abno...</td>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3202</th>\n",
       "      <td>1.0</td>\n",
       "      <td>BCL2L11</td>\n",
       "      <td>3202</td>\n",
       "      <td>Lung cancer is the leading cause of major canc...</td>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3252</th>\n",
       "      <td>1.0</td>\n",
       "      <td>CASP8</td>\n",
       "      <td>3252</td>\n",
       "      <td>Evidence exists that alterations of the genes ...</td>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7150</th>\n",
       "      <td>NaN</td>\n",
       "      <td>SMAD3</td>\n",
       "      <td>3829</td>\n",
       "      <td>Smad proteins mediate transforming growth fact...</td>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8043</th>\n",
       "      <td>NaN</td>\n",
       "      <td>ERF</td>\n",
       "      <td>4722</td>\n",
       "      <td>ETS proteins are a group of evolutionarily rel...</td>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8758</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CCND3</td>\n",
       "      <td>5437</td>\n",
       "      <td>The activities of cyclin D-dependent kinases s...</td>\n",
       "      <td>D286_L292trunc</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Class     Gene    ID                                               Text  \\\n",
       "88      1.0     RYBP    88  The mouse double minute 2 (MDM2)–p53 interacti...   \n",
       "313     1.0     ELF3   313  Previous studies have shown that the promoter ...   \n",
       "532     1.0  TP53BP1   532  Introduction  Inhibition of PARP1 induces synt...   \n",
       "1015    1.0     TSC2  1015   SUMMARY mTORC1 promotes cell growth in respon...   \n",
       "1080    1.0   ARID1A  1080  ARID1A (BAF250A) promotes the formation of SWI...   \n",
       "1277    1.0   ARID5B  1277                                               null   \n",
       "1826    1.0    NCOR1  1826  Estrogen receptors (ERs) are normally expresse...   \n",
       "1902    1.0     TET1  1902  TET2 is a close relative of TET1, an enzyme th...   \n",
       "1911    1.0    FOXP1  1911  The transcriptional network of the androgen re...   \n",
       "2146    1.0    PTCH1  2146  Basal cell carcinoma (BCC) is the most common ...   \n",
       "2285    1.0   INPP4B  2285  Although the gastrointestinal (GI) tract posse...   \n",
       "3160    1.0     NPM1  3160  Hemizygous deletions are common molecular abno...   \n",
       "3202    1.0  BCL2L11  3202  Lung cancer is the leading cause of major canc...   \n",
       "3252    1.0    CASP8  3252  Evidence exists that alterations of the genes ...   \n",
       "7150    NaN    SMAD3  3829  Smad proteins mediate transforming growth fact...   \n",
       "8043    NaN      ERF  4722  ETS proteins are a group of evolutionarily rel...   \n",
       "8758    NaN    CCND3  5437  The activities of cyclin D-dependent kinases s...   \n",
       "\n",
       "                 Variation  Substitutions_var  Stop_codon_var  \\\n",
       "88    Truncating Mutations                  0               0   \n",
       "313   Truncating Mutations                  0               0   \n",
       "532   Truncating Mutations                  0               0   \n",
       "1015  Truncating Mutations                  0               0   \n",
       "1080  Truncating Mutations                  0               0   \n",
       "1277  Truncating Mutations                  0               0   \n",
       "1826  Truncating Mutations                  0               0   \n",
       "1902  Truncating Mutations                  0               0   \n",
       "1911  Truncating Mutations                  0               0   \n",
       "2146  Truncating Mutations                  0               0   \n",
       "2285  Truncating Mutations                  0               0   \n",
       "3160  Truncating Mutations                  0               0   \n",
       "3202  Truncating Mutations                  0               0   \n",
       "3252  Truncating Mutations                  0               0   \n",
       "7150  Truncating Mutations                  0               0   \n",
       "8043  Truncating Mutations                  0               0   \n",
       "8758        D286_L292trunc                  0               0   \n",
       "\n",
       "      gene_fusion_var  Deletion_var  del_or_ins_var  Fusion_var  \\\n",
       "88                  0             0               0           0   \n",
       "313                 0             0               0           0   \n",
       "532                 0             0               0           0   \n",
       "1015                0             0               0           0   \n",
       "1080                0             0               0           0   \n",
       "1277                0             0               0           0   \n",
       "1826                0             0               0           0   \n",
       "1902                0             0               0           0   \n",
       "1911                0             0               0           0   \n",
       "2146                0             0               0           0   \n",
       "2285                0             0               0           0   \n",
       "3160                0             0               0           0   \n",
       "3202                0             0               0           0   \n",
       "3252                0             0               0           0   \n",
       "7150                0             0               0           0   \n",
       "8043                0             0               0           0   \n",
       "8758                0             0               0           0   \n",
       "\n",
       "      Amplification_var  Truncation_var  exon_var  frameshift_var  dup_var  \n",
       "88                    0               1         0               0        0  \n",
       "313                   0               1         0               0        0  \n",
       "532                   0               1         0               0        0  \n",
       "1015                  0               1         0               0        0  \n",
       "1080                  0               1         0               0        0  \n",
       "1277                  0               1         0               0        0  \n",
       "1826                  0               1         0               0        0  \n",
       "1902                  0               1         0               0        0  \n",
       "1911                  0               1         0               0        0  \n",
       "2146                  0               1         0               0        0  \n",
       "2285                  0               1         0               0        0  \n",
       "3160                  0               1         0               0        0  \n",
       "3202                  0               1         0               0        0  \n",
       "3252                  0               1         0               0        0  \n",
       "7150                  0               1         0               0        0  \n",
       "8043                  0               1         0               0        0  \n",
       "8758                  0               1         0               0        0  "
      ]
     },
     "execution_count": 456,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "truncation_not_in_text"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 256,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Mre11, Rad50, and Nbs1 function in a protein complex that is central to the metabolism of chromosome breaks. Null mutants of each are inviable. We demonstrate here that hypomorphic Rad50 mutant mice (Rad50S/S mice) exhibited growth defects and cancer predisposition. Rad50S/S mice died with complete bone marrow depletion as a result of progressive hematopoietic stem cell failure. Similar attrition occurred in spermatogenic cells. In both contexts, attrition was substantially mitigated by p53 deficiency, whereas the tumor latency of p53−/− and p53+/− animals was reduced by Rad50S/S. Indices of genotoxic stress and chromosomal rearrangements were evident in Rad50S/S cultured cells, as well as in Rad50S/S and p53−/−Rad50S/S lymphomas, suggesting that the Rad50S/S phenotype was attributable to chromosomal instability. These outcomes were not associated with overt defects in the Mre11 complex\\'s previously established double strand break repair and cell cycle checkpoint regulation functions. The data indicate that even subtle perturbation of Mre11 complex functions results in severe genotoxic stress, and that the complex is critically important for homeostasis of proliferative tissues.  Keywords: Mre11 complex, Rad50S, double strand breaks, hematopoiesis The Mre11 complex, consisting of the highly conserved Mre11 and Rad50 proteins and Nbs1 (or Xrs2 in yeast), mediates diverse functions in double strand break (DSB) metabolism. In both mammals and Saccharomyces cerevisiae, the complex functions as a sensor of DSBs and influences the induction of DNA damage-dependent cell cycle checkpoints (Petrini 2000a; D\\'Amours and Jackson 2001; Grenon et al. 2001; Mirzoeva and Petrini 2001; Usui et al. 2001). Mre11 complex deficiency in S. cerevisiae profoundly impairs recombinational DNA repair as well as telomere maintenance and meiotic recombination (Haber 1998). In vertebrates, null mutants in each member of the Mre11 complex are inviable (Xiao and Weaver 1997; Luo et al. 1999; Yamaguchi-Iwai et al. 1999; Zhu et al. 2001).  The Mre11 complex specifies ssDNA endonuclease, 3′-to-5′ exonuclease, and DNA unwinding activities in vitro. Accordingly, it has been implicated in DNA end processing events in vivo. Spo11, the topoisomerase II-like enzyme that forms DSBs to initiate meiotic recombination, remains bound to the DSB termini in a group of hypomorphic RAD50 and MRE11 mutants, called S mutants (Alani et al. 1990; Usui et al. 1998). This suggests that the Spo11-DNA intermediate is cleaved by the Mre11 complex in wild-type cells.  In addition, rad50S and nuclease-deficient mre11 mutants are inviable in the absence of Rad27, a nuclease required for the completion of lagging strand DNA synthesis, raising the possibility that the Mre11 complex participates in the nucleolytic processing of DNA replication intermediates (Harrington and Lieber 1994; Moreau et al. 1999; Debrauwere et al. 2001). It appears that Mre11 complex nuclease activities are not critical to its DSB repair functions because rad50S, mre11S, and nuclease-deficient mre11 mutants exhibit only minor sensitivity to clastogens, and resection of DSBs generated by the HO endonuclease is normal (Alani et al. 1990; Nairz and Klein 1997; Bressan et al. 1998, 1999; Tsubouchi and Ogawa 1998; Usui et al. 1998; Moreau et al. 1999; Lee et al. 2002).  As suggested by the functional interaction with Rad27, the Mre11 complex appears to be particularly important during DNA replication (Petrini 2000b). γ-irradiation of synchronous cultures revealed that the extreme clastogen sensitivity of S. cerevisiae mre11Δ mutants is largely attributable to the inability of those cells to utilize sister chromatids in DNA repair, and that this effect is most pronounced in S phase (Ivanov et al. 1992; Bressan et al. 1999). Further, SbcCD, the bacterial Mre11 complex ortholog, appears to be required for processing aberrant DNA structures that arise at replication forks (Leach 1994; Connelly et al. 1998). In bacteriophage T4, recombination-dependent DNA replication and restart of stalled replication forks requires gp46/47, the T4 orthologs of Mre11/Rad50 (George et al. 2001; Stohr and Kreuzer 2001).  Several lines of evidence support a similar role for the Mre11 complex in vertebrates. First, Mre11-depleted DT40 chicken cells die with chromosomal damage suggestive of failure to resolve DSBs arising during DNA replication (Yamaguchi-Iwai et al. 1999). Second, DNA replication in Mre11-depleted Xenopus laevis extracts is associated with the accumulation of chromosome breaks (Costanzo et al. 2001). Finally, the human Mre11 complex becomes avidly associated with chromatin during S phase (Maser et al. 2001). These studies underscore the role of the Mre11 complex in DNA replication, and define a likely basis for the inviability of Mre11 complex-deficient vertebrate cells.  Mre11 complex functions at telomeres may also account for the inviability of Mre11 complex-deficient mammalian cells. The complex functions at telomeres in S. cerevisiae as well as in mammals, although the extent to which Mre11 complex telomere functions are conserved in yeast and mammalian cells is unknown. The human Mre11 complex is situated at telomeres via physical association with the telomere protection protein, TRF2 (Zhu et al. 2000). Given its association with TRF2, the Mre11 complex may function in telomere protection, and would be required for cellular viability via a role in preventing telomere failure (de Lange and Petrini 2000; Lombard and Guarente 2000; Zhu et al. 2000).  In this study, mice harboring a Rad50 allele analogous to the S. cerevisiae rad50S allele rad50-R20M (Alani et al. 1990) were derived. Rad50S/S mice were viable, but exhibited partial embryonic lethality, growth defects, and cancer predisposition. These pathologic outcomes were associated with progressive failure of Rad50S/S hematopoietic and germline cells which was partially mitigated by p53 deficiency. Primary and transformed cell cultures and tumor lines established from Rad50S/S mice were characterized by increased spontaneous apoptosis and chromosomal instability. We conclude that this effect of the Rad50K22M allele constitutes the mechanistic basis of hematopoietic and spermatogenic depletion in Rad50S/S mice. These findings underscore the importance of the Mre11 complex for maintenance of chromosome stability and in homeostasis of proliferative tissues.  Go to: Results Derivation of Rad50S/S\\u2009mice  Null Rad50 mutations were lethal at the cellular and organismal levels, precluding conventional gene inactivation as a means to analyze the Mre11 complex in vivo (Luo et al. 1999). Three hypomorphic Rad50 mutants, Rad50K6E, Rad50K22M, and Rad50R83I, corresponding to S. cerevisiae rad50S alleles were tested for their ability to support cell viability. mRad50Brdm1/mRad50Brdc1 embryonic stem (ES) cells harbor a null Rad50 allele (mRad50Brdm1) and a Rad50 allele that is inactivated upon expression of Cre recombinase (mRad50Brdc1). Because Rad50 is required for viability, Cre expression in mRad50Brdm1/mRad50Brdc1 cells is lethal unless a functional Rad50 cDNA is ectopically expressed (Luo et al. 1999). These cells were cotransfected with a Cre expression vector (pPGKhygro-Cre) and a wild-type or Rad50S cDNA expression construct. Wild-type and pRad50K22M, but not pRad50R83I or pRad50K6E, were sufficient for viability. On this basis, the Rad50K22M mutation, which corresponds to the rad50S allele rad50-R20M in yeast (Alani et al. 1990), was chosen for the derivation of mutant mice.  pkin-24ploxp contains exons 1, 2, and 3 of the Rad50 locus with an AAG to ATG change in exon 1, resulting in K→M at amino acid residue 22 (Fig. (Fig.1A;1A; Dolganov et al. 1996). Homologous integration of pkin-24ploxp in ES cells was assessed by Southern blotting (Fig. (Fig.1B),1B), and the ensuing targeted ES cells were used to derive mutant animals. Rad50S/S mice were susceptible to partial embryonic lethality at embryonic day 14–16 (E14–E16; P\\u2009=\\u20090.002), and were significantly underrepresented among 3-wk-old animals (P\\u2009=\\u20090.00001). These data indicated that the Rad50K22Mallele is hypomorphic.  Figure 1Figure 1 Figure 1 Derivation of Rad50K22M mutants. (A) Schematic of Rad50 locus. 5′ region and exons 1–3 (black boxes) are shown. Homologous integration of the targeting vector pk-in-24ploxp with the K22M mutation encoded in exon 1 (*) inserts a loxP-flanked ... Rad50S/S MEFs were not sensitive to ionizing radiation (IR), mitomycin C, hydroxyurea, or etoposide, nor did they exhibit growth defects. The formation of Mre11 IR-induced foci (IRIF) and the intra-S phase checkpoint were also unaffected (data not shown). In addition, wild-type levels of Mre11, Nbs1, and Rad50SK22M protein were present in extracts of Rad50S/S murine embryonic fibroblasts (MEFs), and their ability to assemble in complex was not overtly affected (Fig. (Fig.11C).  Although the Rad50K22M allele exerted a minimal cellular phenotype, it had a profound impact at the organismal level. Rad50S/S mice weighed 60% of wild-type littermates, and by 4–8 wk of age, most Rad50S/S animals had a gray, wasted appearance suggestive of anemia. The mean age of death of Rad50S/S mice was 2.6\\u2009±\\u20091.8 mo (N\\u2009=\\u2009166). Whereas most Rad50S/S mice died tumor-free by 4 mo of age, 20% survived 4–7 mo. In this group of long-lived Rad50S/S mice, 20% died with metastatic thymic lymphomas, four exhibited splenic hyperplasia, and one mouse developed myeloid leukemia.  Meiotic progression in Rad50S/S\\u2009mice  In S. cerevisiae rad50S mutants, meiotic DSBs are not processed due to the covalent attachment of Spo11 (Alani et al. 1990; Keeney and Kleckner 1995; Keeney et al. 1997). Because the function of Spo11 appears to be conserved in mice (Baudat et al. 2000; Romanienko and Camerini-Otero 2000), we analyzed Rad50S/S ovaries and testes to determine whether murine Rad50K22M led to an analogous block in meiotic progression.  Rad50S/S females were fertile and had normal ovaries (not shown). Testicular development was normal in prepubescent Rad50S/S males. Despite decreased cellularity in the seminiferous tubules of 4-wk-old Rad50S/S mice (Fig. (Fig.2A),2A), older mice were also fertile.  Figure 2Figure 2Figure 2 Figure 2 Progressive degeneration in Rad50S/S testes is not associated with meiotic recombination defects. (A) Hematoxylin and eosin (H&E) staining of testes sections from 2- and 4-wk-old Rad50+/+ and Rad50S/S mice. Bar, 100 μm; ... Meiotic recombination defects in Spo11−/− mice result in depletion of germ cell precursors (Baudat et al. 2000; Romanienko and Camerini-Otero 2000). We reasoned that the cellular attrition in Rad50S/S testes might be the result of a similar, though less severe meiotic defect. TUNEL analysis of 4-wk-old Rad50S/S testes, unlike those from 2-wk-old Rad50S/S mice, revealed a 3–5-fold increase in the number of apoptotic tubules (Fig. (Fig.2B).2B). However, cytologic analysis of meiotic progression did not indicate a stage-specific arrest, nor were aberrant chromosomal structures noted (Fig. (Fig.2C).2C). These data suggest that the Rad50K22M allele did not impair meiotic recombination, and that cell death in Rad50S/S testes is attributable to a different source of genotoxic stress.  Hematopoietic stem cell failure in Rad50S/S\\u2009mice  The premature death of Rad50S/S mice was associated with severe anemia. Two-wk-old Rad50S/S mice did not exhibit differences in the abundance of hematopoietic cells. However, by 4–8 wk of age, the bone marrow of most Rad50S/S mice was virtually devoid of hematopoietic cells and was largely composed of adipocytes (Fig. (Fig.3A).3A). Spleen and thymus architecture was also grossly abnormal, with few lymphocytes present (data not shown). Long-lived Rad50S/S mice that did not develop thymic lymphomas died with bone marrow attrition identical to that seen in 4–8-wk-old mutants.  Figure 3Figure 3Figure 3 Figure 3 Rad50S/S progressive hematopoietic depletion is intrinsic to stem cells. (A) H&E staining of bone marrow sections from Rad50+/+ and Rad50S/S. Bar, 100 μm; magnification, 400×. (B) FACS analysis of hematopoietic ... Fluorescence-activated cell sorting (FACS) analysis was used to correlate abundance of hematopoietic components with age. Double negative (CD4−,CD8−, DN) T cells, pro B cells (B220+,CD43+), and macrophages (B220−,CD43+) were present at wild-type levels in E16 Rad50S/S embryos (Fig. (Fig.3B).3B). By 1 wk of age, pre B (B220+,CD43−) and immature B (B220+,IgM+) cells in Rad50S/S mice were 1.5–3-fold reduced, whereas pro B cells and macrophages were 2–3-fold increased. Thymocytes were unaffected at this age, but splenic B and T cells were decreased to 4.8%–12% and 7%–49% of wild-type, respectively (Fig. (Fig.3B;3B; data not shown). By 4 wk of age, lymphocytes, macrophages, red blood cells and platelets were severely depleted (0.1%–18% wild-type, Fig. Fig.3B,3B, data not shown). These results demonstrated that progressive hematopoietic failure in Rad50S/S mice affected all hematopoietic lineages.  Cell transfer experiments were performed to determine whether Rad50S/S pathology was intrinsic to hematopoietic cells. Irradiated Rad50S/S mice were injected with wild-type fetal liver cells to assess the ability of Rad50S/S bone marrow stroma to support hematopoietic differentiation. Reconstitution of hematopoietic cells in Rad50S/S recipients of wild-type fetal liver was comparable to that in control recipients (Fig. (Fig.3C).3C). PCR-based genotyping confirmed that wild-type cells were responsible for reconstitution of Rad50S/S mice (data not shown). In contrast, wild-type recipients of Rad50S/S fetal liver had 10–1000-fold fewer hematopoietic cells than controls at 6 wk after transfer (Fig. (Fig.3C).3C). These data suggested that Rad50S/S bone marrow failure was attributable to hematopoietic failure independent of stromal defects.  To confirm that interpretation, we mixed wild-type and Rad50S/S fetal liver cells in a 1:1 ratio and transferred them into wild-type recipients. In this setting, Rad50S/S cells would be supported by the wild-type hematopoietic milieu during differentiation. Full reconstitution was evident 4 wk after transfer, and genotypic analysis showed that wild-type, but not Rad50S/S cells, were present in hematopoietic tissues of reconstituted animals (Fig. (Fig.3C;3C; data not shown). Thus, the presence of wild-type hematopoietic cells and stroma did not increase the ability of Rad50S/S stem cells to reconstitute hematopoietic compartments. These experiments indicated that progressive bone marrow depletion in Rad50S/S mice was primarily attributable to hematopoietic stem cell failure.  Normal V(D)J recombination in Rad50S/S\\u2009mice  DNA hairpins are intermediates in the site-specific recombination process required for antigen receptor gene assembly, V(D)J recombination (Roth et al. 1992; McBlane et al. 1995). Recent in vitro experiments suggest that the artemis/DNAPK complex may open V(D)J hairpins (Ma et al. 2002). However, in light of in vitro and in vivo data, the possibility remains that the Mre11 complex may also influence this process (Paull and Gellert 1998, 1999; Lobachev et al. 2002). We used ligation-mediated PCR to assess whether hairpin opening was defective in Rad50S/S mice (Zhu and Roth 1995). TCR Dδ2 hairpins were readily detected in thymus DNA from 1-wk-old scid mice; however none were detected in Rad50S/S samples (Fig. (Fig.4).4).  Figure 4 Figure 4 Normal hairpin resolution in Rad50S/S thymocytes. Hairpin coding ends (CE), oligonucleotides were ligated to the D2/J1 blunt signal ends (SE) and the open CEs. Ends were PCR amplified using Dδ2– and oligospecific primers (arrows), and ... Quantitative PCR showed that in vivo V to DJ and D to J rearrangement frequencies, as well as the structure of the joints formed, were normal in E16 and in 4-wk-old Rad50S/S mice (data not shown). Transient V(D)J recombination assays indicated that coding and signal joints were formed at comparable frequencies in wild-type and Rad50S/S MEFs (recombination frequency for signal joints: 0.12% in wild-type, 0.29% in Rad50S/S; coding joints: 0.5% in wild-type, 1.2% in Rad50S/S), and like Rad50S/S in vivo joints, did not exhibit abnormal structures (data not shown). These data indicate that V(D)J recombination was normal in Rad50S/S mice, and suggest that hairpin opening during V(D)J recombination is not impaired by the Rad50K22M mutation.  p53 deficiency mitigates the Rad50S/S\\u2009phenotype  The correlation of age with malignancy, together with the observation that testes and bone marrow showed age-dependent cellular attrition, suggested that the Rad50K22M mutation caused chronic genotoxic stress. Because p53 profoundly influences genotoxic stress responses (Prives and Hall 1999), we hypothesized that p53 deficiency would mitigate the Rad50S/S phenotype. Conversely, because potentially oncogenic genetic alterations would accumulate without inducing apoptosis in p53-deficient Rad50S/S mice, we expected Rad50S to reduce the tumor latency of p53−/− animals.  Indeed, reduced p53 dosage partially rescued Rad50S-dependent depletion in the hematopoietic compartment, and the Rad50K22M mutation accelerated p53+/− and p53−/− tumorigenesis (Fig. (Fig.5A,B).5A,B). B cell numbers were increased 3–10-fold and 5–20-fold above Rad50S/S levels in p53+/− Rad50S/S and p53−/− Rad50S/S mice, respectively. Macrophages were threefold increased and T cells were 3–20-fold increased in both p53+/− Rad50S/S and p53−/− Rad50S/S mice (cf. Figs. Figs.3B3B and and5A).5A). The partial rescue of hematopoietic cell survival in p53+/− Rad50S/S and p53−/− Rad50S/S mice was associated with increased lifespan (Fig. (Fig.5B).5B). Rad50K22M significantly altered p53+/− and p53−/− tumor susceptibility. p53+/− control mice in our colony developed osteosarcomas after 12 mo of age (N\\u2009=\\u200934), whereas p53+/− Rad50S/S mice died from thymic lymphomas at an average of 5 mo (N\\u2009=\\u200917; P\\u2009=\\u20090.002). p53−/− Rad50S/S animals (N\\u2009=\\u20097, average age of death 3.3 mo; P\\u2009=\\u20090.05) died earlier than p53−/− mice (N\\u2009=\\u200917, average age of death 4.5 mo) without changes in p53−/− tumor spectrum (Jacks et al. 1994). In contrast to Rad50S/S animals, the double mutants succumbed to lymphoma and did not show overt signs of anemia.  Figure 5Figure 5Figure 5 Figure 5 p53 deficiency partially rescues the Rad50S/S phenotype. (A) FACS analysis of 4-wk-old 129/SvEv Rad50+/+ (○), p53+/− Rad50S/S (▴), and p53−/−Rad50S/S (▪) mice; depicted as in Fig. ... p53 deficiency also suppressed the attrition of cells in the seminiferous tubules, correlated with abrogation of testicular apoptosis in both p53+/− Rad50S/S and p53−/− Rad50S/S mice (Fig. (Fig.5C).5C). The suppression of testicular apoptosis raised the possibility that defects in meiotic recombination, obscured by apoptotic attrition in p53-proficient cells, would be evident in double mutants. However, p53+/− Rad50S/S and p53−/− Rad50S/S males did not exhibit an arrest in meiotic progression, nor were aberrant chromosomal structures detected (data not shown). These data indicate that apoptosis in Rad50S/S testes was not associated with meiotic recombination defects.  Chronic genotoxic stress in cells from Rad50S/S\\u2009mice  The rescue of Rad50S/S hematopoietic and testicular attrition and the reduced tumor latency in p53+/− Rad50S/S and p53−/− Rad50S/S mice supported the interpretation that chronic genotoxic stress was the underlying cause of the Rad50S/S phenotype. Indices of genotoxic stress were assessed in primary cell cultures. First, the apoptotic indices of wild-type and Rad50S/S MEFs were analyzed by FACS. Compared to wild-type controls, unirradiated Rad50S/S MEFs exhibited a 1.4–3.4-fold increase in the fraction of cells with sub-G1 DNA content. A similar increase in annexinV staining was also found in unirradiated Rad50S/S MEFs. The increase in apoptotic cells after irradiation (IR) was similar in Rad50S/S and control cells, indicating that the apoptotic response to IR was not compromised (Table (Table1).1).  Table 1 Table 1 Genotoxic stress in Rad50S/S\\u2009cells Second, the presence of phosphorylated histone H2AX (γ-H2AX), which is correlated with the presence of DNA damage (Rogakou et al. 1999; Paull et al. 2000; Ward and Chen 2001), was examined in MEFs by immunofluorescence and flow cytometry with γ-H2AX antiserum. The percentage of γ-H2AX-positive cells was increased in unirradiated Rad50S/S cells to levels comparable to that in wild-type irradiated cells (Table (Table11).  Finally, since the mammalian Mre11 complex localizes to sites of DNA replication (Maser et al. 2001), sites of DNA damage (Nelms et al. 1998; Mirzoeva and Petrini 2001), and to human telomeres (Zhu et al. 2000), we reasoned that attenuation of Mre11 complex function at any one of these venues could induce chromosome aberrations in Rad50S/S cells. To address this possibility, spectral karyotype (SKY) and conventional karyotypic analyses were performed. Increased levels of chromosome breaks were evident in ear fibroblast cultures established from Rad50S/S and p53−/− Rad50S/S mice (Table (Table2;2; Fig. Fig.6A).6A). Increases in chromosome breaks and rearrangements were also detected in SV40-transformed Rad50S/S ear fibroblasts by conventional analysis of Giemsa-stained metaphase cells (Table (Table2).2). These data were consistent with the interpretation that the Rad50K22M mutation exerted an impact on chromosome stability.  Table 2 Table 2 Karyotypic analysis of ear fibroblast\\u2009cultures Figure 6Figure 6Figure 6 Figure 6 Chromosome instability in Rad50S/S cells. (A) Images of Giemsa-stained metaphase cells. (Left) a normal metaphase spread from a Rad50S/S ear fibroblast. (Right) Rad50S/S ear fibroblast metaphase with one broken chromosome (arrow). Magnification, 1000×. ... Because the events that lead to depletion of hematopoietic cells also predispose malignancy, the mechanistic basis of Rad50S/S pathology should be evident in tumors from these mice. Rad50S/S and p53−/− Rad50S/S thymic lymphoma cells exhibited increased chromosomal instability in SKY analysis (Table (Table3;3; Fig. Fig.6B).6B). Compared to p53−/− tumors, most Rad50S/S and p53−/− Rad50S/S tumors had more chromosome rearrangements and some exhibited an increased frequency of nonclonal karyotypes, suggesting that events leading to instability were ongoing (Table (Table3).3).  Table 3 Table 3 Spectral karyotype analysis of thymic\\u2009lymphomas Giemsa-stained chromosomes from tumors revealed similar karyotypic instability. Notable was a significant increase in short-arm chromosome fusions (Table (Table4).4). Fluorescent in situ hybridization (FISH) with a CCCTAA probe was performed to determine whether fusions in Rad50S/S cells resulted from joining of telomeric or interstitial DNA. The majority of short-arm fusions in Rad50S/S and p53−/− Rad50S/S tumors contained two intense telomeric signals at the join, indicating that telomeric DNA was retained in these fusions (Fig. (Fig.6C).6C). Fusions containing telomeric DNA were detected by FISH in all Rad50S/S and p53−/− Rad50S/S tumors, but not in p53−/− tumors (Table (Table4).4). These data raise the possibility that telomere failure was among the chromosomal phenotypes associated with the Rad50K22M mutation, and collectively support the interpretation that Rad50S/S cells are subject to chronic genotoxic stress.  Table 4 Table 4 Chromosome aberrations in thymic\\u2009lymphomasa Go to: Discussion The Mre11 complex is integral to a diverse spectrum of functions that bear upon chromosome stability. Genetic analyses in S. cerevisiae establish that the complex\\'s influence can be generally attributed to either structural roles, which primarily influence the DNA recombination functions of the complex, or enzymatic (nucleolytic) roles, which appear to influence DSB end processing or degradation of certain DNA structures. The complex\\'s structural role has been invoked to account for the inability of null Mre11 complex mutants to utilize sister chromatids during recombinational DNA repair, whereas nuclease-deficient mutants do not exhibit this property (Ivanov et al. 1992; Moore and Haber 1996; Bressan et al. 1999). A molecular basis for such a role is suggested by electron microscopic studies of the human Mre11 complex and structural analysis of the Pyrococcus furiosis Mre11 complex. Those data indicate that the complex may bridge DNA ends or sister chromatids via the coiled coil regions of Rad50 (de Jager et al. 2001; Hopfner et al. 2002). The in vivo role of the complex\\'s nuclease functions have been inferred from the S. cerevisiae S alleles, in which the initial processing of meiotic DSBs is impaired, and from the synthetic lethality of nuclease-deficient mutants with Rad27 deficiency (Symington 1998; Debrauwere et al. 2001).  We determined previously that a null Rad50 mouse mutant, in which the structural role of the Mre11 complex is presumably abrogated, is inviable (Luo et al. 1999). Therefore, we established a Rad50S/S mouse mutant in which the Mre11 complex is intact, but functionally hypomorphic. In contrast to Rad50-deficient mice, Rad50S/S mice exhibit only partial embryonic lethality. Rad50S/S mice have shortened lifespan associated with progressive loss of cells in the male germ line and in hematopoietic lineages, and die with hematopoietic failure by 3 mo of age. Sporadically appearing longer-lived Rad50S/S animals are highly predisposed to malignancy. Underlying these phenotypic outcomes is the accumulation of chromosome aberrations. In p53-proficient mice, these events precipitate the cellular attrition observed, whereas in the context of p53 deficiency, abrogation of apoptosis diminishes cellular attrition but dramatically enhances tumorigenesis.  Progressive attrition of hematopoietic and spermatogenic lineages in Rad50S/S\\u2009mice  Rad50S/S pathology is largely restricted to hematopoietic and germ cell lineages. Fetal liver cell transfer experiments demonstrated that the progressive hematopoietic failure observed is correlated with depletion of stem cells. An analogous progenitor depletion may underlie the progressive attrition of cells in the seminiferous tubules, because meiotic progression is not impaired by Rad50S. In this regard, the Rad50S/S mice are similar to late-generation telomerase-deficient (mTR−/−) mice, which experience telomere failure as a sequela to telomere shortening, and exhibit relatively selective depletion of hematopoietic and spermatogenic cells (Lee et al. 1998; Herrera et al. 1999).  Chronic genotoxic stress associated with the Rad50K22M allele would differentially affect hematopoietic and spermatogenic tissues for two reasons. First, in contrast to other proliferative tissues such as intestine or skin, they arise from extensive proliferation of a relatively small number of stem cells and committed precursors (Weissman 2000). Therefore, for each stem or precursor cell depleted, a relatively large proportion of the tissue—arising from geometric expansion from the depleted stem cell—would be lost. Second, decreased cellularity, as a primary outcome of stem cell depletion and the ensuing reduction in differentiated cells, imposes increased proliferative demand on the residual population. Whereas mutations that result in gross chromosomal instability would have broad pathological outcomes, less severe mutations would more specifically affect tissues that rely upon limited precursor populations.  The selective effect of Rad50K22M on hematopoietic and spermatogenic tissue is thus consistent with the observation that the magnitude of chromosomal instability in Rad50S/S mice was relatively low. The chromosomal aberrations seen in Rad50S/S and p53−/− Rad50S/S lymphomas were not seen in p53−/− tumors, and were largely clonal, indicating that these chromosomal aberrations are unique to the Rad50K22M allele, and further, that they accumulate at a relatively low rate. Were Rad50S/S mice longer-lived, a broader spectrum of pathology would likely emerge. By analogy, whereas hematopoietic and spermatogenic depletion are seen first in late-generation mTR−/− mice, an aged cohort of these mice exhibited much broader pathology, including skin and intestine defects (Herrera et al. 1999; Rudolph et al. 1999).  Murine Rad50S alleles  The inability of Rad50K6E and Rad50R83I to support cell viability was not predicted from the behavior of the corresponding S. cerevisiae mutants. Given the conservation of Mre11 and Rad50, it is unlikely that the enzymatic functions impaired by the murine S mutations differ from those in S. cerevisiae. Assuming that the three S mutations examined impair the same functions, the inviability of Rad50K6E and Rad50R83I could simply reflect a more severe form of the Rad50K22M phenotype, in which cell death ensues from greater increase in the levels of chromosome instability. It is also conceivable that increased severity could result from defects in processing a more diverse spectrum of substrates in Rad50K6E- and Rad50R83I-expressing cells.  The lack of meiotic recombination and V(D)J recombination defects in Rad50K22M mutants, which was also unexpected, could be explained by the same argument. S mutations that exert a profound effect on meiotic progression would be inviable. Although a meiotic arrest was not clearly evident in the Rad50K22M mice, a subtle defect might not have been detected by our analysis.  The Mre11 complex: Diverse mechanisms in maintaining genomic\\u2009integrity  Mutations in MRE11 and NBS1 are associated with cancer predisposition, chromosome instability, and cell cycle checkpoint defects in A-TLD and NBS (Carney et al. 1998; Stewart et al. 1999). Despite their severe clinical and cellular phenotypes, those mutations do not confer overt recombinational DNA repair defects, as was true in Rad50S/S mice. In contrast, null Mre11 complex mutations, which cause profound DNA recombination defects in S. cerevisiae, are lethal in vertebrates. These outcomes support the possibility that essential and nonessential Mre11 complex functions are mechanistically distinct. The P. furiosis Mre11/Rad50 complex suggests a molecular basis for distinct mechanisms. This complex contains a bifunctional DNA binding domain in which the active site of the Mre11 nuclease is capable of binding ssDNA, whereas an adjacent site accommodates double-stranded DNA. Occupancy of the sites is mutually exclusive, suggesting that a given interaction with DNA facilitates enzymatic (in the ssDNA binding domain) or structural (in the dsDNA binding domain) functions (Hopfner et al. 2001). The data presented here argue that the essential functions of the complex cannot be clearly divided according to its enzymatic or structural roles.  Finally, the data clearly indicate that the Mre11 complex exerts a profound influence on homeostasis in mammalian tissues even when its checkpoint and DNA recombination functions are not overtly impaired. The Rad50S/S mouse therefore establishes that subtle hypomorphism in a DNA damage response pathway is sufficient to induce severe pathology. This observation has important implications for understanding the mechanistic basis of disease related to chromosomal metabolism.  The RAD50 gene of Saccharomyces cerevisiae is required for chromosome synapsis and recombination during meiosis and for repair of DNA damage during vegetative growth. The precise role of the RAD50 gene product in these processes is not known. Most rad50 mutant phenotypes can be explained by the proposal that the RAD50 gene product is involved in the search for homology between interacting DNA molecules or chromosomes, but there is no direct evidence for this model. We present here the nucleotide sequence of the RAD50 locus and an analysis of the predicted 153-kD RAD50 protein. The amino terminal region of the predicted protein contains residues suggestive of a purine nucleotide binding domain, most likely for adenine. The remaining 1 170 amino acids consist of two 250 amino acid segments of heptad repeat sequence separated by 320 amino acids, plus a short hydrophobic carboxy-terminal tail. Heptad repeats occur in proteins such as myosin and intermediate filaments that form a-helical coiled coils. One of the two heptad regions in RAD50 shows similarity to the S-2 domain of rabbit myosin beyond that expected for two random coiled coil proteins. T HE RAD50 gene of Saccharomyces cerevisiae plays an important role in both production of viable spores in meiosis and in the repair of DNA damage during vegetative growth (HAYNES and KUNZ 1981; GAME et al. 1980). During meiosis, RAD50 function is required for two interrelated events of meiotic prophase, chromosome synapsis and recombination, both of which are required for proper segregation of chromosomes at the first meiotic division (BAKER et al. 1976). In rad50 mutants, the defect in chromosome synapsis is manifested cytologically by the absence of mature synaptonemal complexes (SC) (BYERS, as cited in MALONE, JORDAN and WARDMAN 1985). More specifically, axial elements form between each pair of sister chromatids but never come together to form mature tripartite SCs (asynapsis) (FARNET et al. 1988; R. PADMORE and N. KLECKNER, unpublished data). The rad50 defect in meiotic recombination is absolute and involves a block at a very early step in the recombination process as revealed by both phenotypic and epistasis analysis (MALONE and ESPOSITO 198 1 ; MALONE 1983; MONTELONE, HOEKSTRA and MALONE 1988). RAD50 appears to act at a very early time in meiosis; it is required for the appearance of a specific meiotic signal which occurs just at the conclusion of premeiotic DNA replication, and which manifests itself as a site specific double strand break or pair of double strand ends (L. CAO, E. ALAN and N. KLECKNER, unpublished data). The rad50 defects in meiotic recombination and synaptonemal complex formation are sufficient to account for the fact that fewer than (knrtics 122: 47-57 (May, 1080) one percent of the spores from a rad50 meiosis are viable (GAME et al. 1980). During vegetative growth, RAD50 function is required for repair of some types of DNA damage, including double strand breaks induced by ionizing radiation or the radiomimetic alkylating agent rnethylmethane sulfonate (MMS) (GAME and MORTIMER 1974; STRIKE, cited in HAYNES and KUNZ 198 1; J. NITISS and M. RESNICK, personal communication). rad50 mutants also exhibit a delay in the spindle elongation step of normal vegetative growth, a phenotype which suggests that they are defective in repair of spontaneous DNA damage as well (FARNET et al. 1988; MONTELONE, HOEKSTRA and MALONE 1988). RAD50-dependent DNA repair is thought to occur by a recombinational pathway (HAYNES and KUNZ 198 1). Both chromosome synapsis (SC formation) and meiotic recombination require that homologous chromosomes or DNA molecules find one another. Some models propose that synapsis is required for recombination; other models propose more complex relationships. An economical hypothesis would be that a single homology search is used for both processes. We have proposed that RAD50 is required for such a homology search and that the RAD50 dependent meiotic signal is a manifestation of this search (FARNET et al. 1988; L. CAO, E. ALANI and N. KLECKNER, unpublished data). The relationship between meiotic and mitotic phenotypes of rad50 mutants is also not established. However, the involvement of RAD50 in DNA repair makes it seem more likely that it is directly involved in  48 E. Alani, S. Subbiah and N. Kleckner meiotic chromosome metabolism rather than playing an indirect and/or regulatory role. The role proposed for RAD50 in meiosis could also account for its role in DNA repair, since a recombinational repair pathway would also require recognition and interaction of homologous molecules. Arguments that RAD50 is required for recognition of homologous DNA segments have also been put forward by J. NITISS and M. RESNICK (personal communication). To further probe the nature of the RAD50 gene product, we have carried out DNA sequence analysis of a previously cloned RAD50 gene. This analysis, described below, has identified a 1312 amino acid open reading frame which corresponds to the genetically defined RAD50 gene. The predicted 153-kD protein has three interesting features: (1) The amino terminal region of the predicted protein contains residues diagnostic of a purine nucleotide binding domain (ROSSMAN, MORAS and OLSEN 1974), probably for adenine (FRY, KUBY and MILDVAN 1986). (2) Two regions, one at amino acids 177-421 and another at amino acids 743-995, are segments of heptad repeat sequence characteristic of alpha helical coiled coil proteins such as myosin and intermediate filaments (MCLACHLAN and KARN 1982; STEINERT, STEVENS and Roop 1985). (3) Heptad 743-995 also shows a significant similarity to the S2 domain of rabbit myosin (CAPONY and ELZINGA 198 1). MATERIALS AND METHODS Strains: Escherichia coli stain \"294 (F-, endA, hsdR, supE44, thiA) was used for all plasmid manipulations (GUARENTE et al. 1980). JMlOl (Alac, thi, supE/F, traD36, proA, proB, laciq, lacZA, M15) was used as a host to prepare single stranded DNA for dideoxy sequencing (MESSING 1983). Saccharomyces cerevisiae strain NKY551 is a diploid that contains a complete coding region deletion of RAD50 (HindIII-Sal1 deletion, Figure 1). NKY551 (a/a, rad50A:: hisG/rad50A::hisG, ura3lura3, lys2/lys2, ho::LYSB/ho::LysP) was constructed by transforming NKY278 (a/a, ura3/ura3, lys2/lys2, ho::LYS2/ho::LYS2 to UraC with BglII and EcoRI digested pNKY83 (described below). The integration of this fragment resulted in a complete coding region deletion of RAD50 (blot not shown). One such transformant, NKY492, was plated onto 5-fluoro-orotic (5-FOA) acid plates and a Ura- strain, NKY546, was selected (ALANI, CAO and KLECKNER 1987). Tetrad dissection of NKY546 yielded spores which segregated 2:2, MMS\\', wild-type growth: MMS\", slow growth (FARNET et al. 1988). Two rad50A haploid segregants from NKY546 were mated to form the homozygous rad50A strain NKY551. This strain sporulated at high frequency (50%) but yielded inviable spores. Media: E. coli was grown in LB broth (MILLER 1972). Ampicillin was supplemented at 25 pg/ml. Yeast was grown in either YPD or minimal medium (SHERMAN, FINK and HICKS 1983). Minimal medium contained 0.7%; yeast nitrogen base, 2% agar, 2% glucose and 0.004% uracil. 5-FOA was purchased from SCM specialty chemical, Gainesville, Florida. 5-FOA plates were prepared as described previously (BOEKE, LACROUTE and FINK 1984). Yeast transformations: All transformations were performed using standard methods (ITO et al. 1983). Nucleic acid techniques: All restriction enzymes, T4 DNA ligase, T4 DNA polymerase and BAL31 were purchased from New England Biolabs and used according to the manufacturer\\'s specifications. Plasmid DNA was isolated by a cleared lysate protocol (MANIATIS, FRITSCH and SAMBROOK 1982). DNA manipulations including isolation of single stranded DNA from \"13 origin plasmids was described previously. Chromosomal yeast DNA isolation and Southern blotting were described previously (HOLM et al. 1986; SOUTHERN 1975; MELTON et al. 1984). Plasmid constructions: All plasmids were derived from the following vectors (relevant genotype and/or description): YEP24 (URA3,2p; ROSE, GRISAFI and BOTSTEIN 1984; New England Biolabs Catalog 1986), YCP50 (URA3, ARSI, CEN4; STINCHCOMB, MANN and DAVIS 1982) pSG205 (7-kb RAD50 HindIII fragment inserted into YCP50, from S. GOTTLIEB and R. ESPOSITO), pOL13 (pSG205 derivative with RAD50 in opposite orientation), and \"13 sequencing vectors pGC-1, pGC-2, POL10 (pGC-I with the HindIII site destroyed) and pNKY47 (POL10 with a HindIII linker at BamHI site) (MYERS and MANIATIS 1985). pNKYlOO is a vector that was used to clone chromosomal sequences upstream of the 5\\' HindIII site in RAD50. A 4.0- kb HindIII-Sal1 subclone from pSG205 was inserted into the HindIII-Sal1 backbone of pBR322 to make pNKY 19. A HindIII URA3 subclone isolated from YEP24 was inserted into the HindIII site of pNKY19 to form pNKY100. In pNKYlOO URA3 transcription is in the same direction as RAD50 and EcoRI cuts only once in the plasmid, between amp\\' and URA3. pNKYlOl is a vector bearing the RAD50 gene and its upstream sequences. pNKYlO0 was linearized within the RAD50 gene by an NsiI restriction digest and then integrated into the RAD50 locus of NKY501 (a, his#-7 13, lys2- 20, sufl0, ura3, Ura- derivative of RE82 1, kindly supplied by R. ESPOSITO) by lithium acetate transformation (blot not shown). Chromosomal DNA from the integrant was isolated, digested with EcoRI, ligated and transformed into \"294. An amp\\' plasmid bearing the HindIII-Sal1 RAD50 fragment plus 2.8 kb of additional upstream RAD50 sequence was isolated and designated pNKY 10 1. pNKY74 is a pGC derived vector (MYERS and MANIATIS 1985) that contains an 8.8-kb RAD50 clone. RAD50 DNA upstream of the coding region was isolated on a 2 kb BglIIHindIII fragment from pNKYlOl and inserted into the BglIl and HindIII backbone of pNKY47 to form pNKY73. The RAD50 locus, containing the coding region plus both 5\\' and 3\\' flanking sequence, was reconstructed by inserting the 7-kb HindIII RAD50 fragment from pSG205 into pNKY73 to form the integrating vector pNKY74. pNKY1070 is an ARSI, CEN4 vector that contains 6.6 kb of RAD50 sequence. pNKY74 was digested with EcoRI and a 6.6-kb RAD50 fragment was inserted into the EcoRI site of YCP50. pNKY83 was used to delete the Hind-Sal1 RAD50 sequences from the chromosome in order to create complete chromosomal coding region deletions of RAD50. The 6.6- kb RAD50 EcoRI fragment from pNKY74 was inserted into the EcoRI site of a pBR322 derived vector (PvuII-EcoRI deletion) to form pNKY79. pNKY79 was digested with HindIII and Sal1 (complete RAD50 coding region deletion, filled in with T4 DNA polymerase and religated in the presence of BamHI 8-bp linkers. Finally, a 3.8-kb BglIIBamHI hi&-URA3-hisG fragment (ALANI, CAO and KLECKNER 1987) was inserted into the BamHI linker site to form pNKY83. pNKY5 and pNKY6 are RAD50 vectors that were used  Analysis of Yeast RAD50 Gene 49 to sequence both strands of RAD50 (strands A and B, respectively). POL13 was digested with XhoI and EcoRI. A 9.7-kb RAD50 fragment was inserted into the XhoI and EcoRI sites of POL10 to form pNKY5. The 7-kb HindIII fragment from pSG205 was blunt ended with T4 DNA polymerase and inserted into the blunt ended Sal1 site of pGC-2 to form pNKY6 (HindIII site is regenerated). Dideoxy sequencing: A 4.0-kb HindIII-Sal1 RAD50 subclone that was shown to complement the rad50-1 mutation was sequenced by the dideoxy approach (MESSING 1983). Two M13 origin vectors, pNKY5 and 6, which bear the HindIIISalI fragment in opposite orientations, were used as substrates for BAL31 deletion. The end points of 24 deletions in pNKY5 and 25 deletions in pNKY6 were sequenced (Figure 3). DNA sequences that were not uncovered by BAL31 deletion were sequenced by cloning restriction fragments into appropriate MI 3 vectors. Using this approach the entire HindIII-Sal1 fragment was sequenced on both strands. Sequences downstream of Sal1 were sequenced on only one strand (pNKY5). Transcription studies indicated that the RAD50 gene is transcribed from HindIII to SalI (FARNET et al. 1988; W. RAYMOND and N. KLECKNER, unpublished data). Because aberrant RAD50 transcripts are observed on ARS CEN vectors that contain the HindIII-Sal1 clone, sequences upstream of the HindIII site were cloned (pNKYlO1). Upstream from the HindIII site 600 bp of DNA were sequenced on both strands by cloning restriction fragments into pGC-1 and pGC-2. Computer analysis: All computer analysis was performed on a Digital VAX 8700 computer. RAD50 amino acid sequence was compared to the NBRF protein data bank, release #I 5 (493 1 sequences, 1,264,388 residues) using the FASTP algorithm described by LIPMAN and PEARSON (1985). Fourier analysis was performed by a method described by MCLACHLAN and KARN (1983). Heptad shuffle analysis using NEEDLEMAN-WUNSCH matrix analysis was described by s. SUBBIAH (unpublished data). The SIM matrix (similarity cutoff = 5 SD, gap penalties = 18 or 24 (McLACHLAN 1971)) and UP matrix (similarity cutoff = 3 SD, gap penalties = 3 or 5 (DAYHOFF 1978)) were used in this analysis and the following 2 values (LIPMAN and PEARSON 1985) were calculated using these matrices: 2 regular = (similarity score - mean similarity score of scrambled sequences)/(standard deviation of scrambled similarity scores) (NEEDLEMAN and WUNSCH 1970). 2 heptad = (similarity score - mean similarity score of heptad shuffled sequences)/ (standard deviation of heptad shuffled similarity scores) (S. SUBBIAH, unpublished data). 2 control = 2 heptad analysis of aligned protein sequences scrambled once in register (S. SUBBIAH, unpublished data). RESULTS AND DISCUSSION A 9.5-kb region which includes the RAD50 gene is diagrammed in Figure 1. RAD50 function is contained within a four kilobase region defined by HindIII and SalI restriction sites; this segment complements the mitotic and meiotic defects of both point and deletion mutants in RAD50 when present on an autonomously replicating yeast plasmid (ARSZ, CEN4) (KUPIEC and SIMCHEN 1984; S. GOTTLIEB and R. E. ESPOSITO, personal communication; see MATERIALS AND METHODS). Essentially all of the sequences in this 4-kb segment are required for complementation; TnlO-LUK insertion mutations that abolish such comSEQUENCED REGION E Bg P PH Bg Pv SE P B? H I I PB / \\\\ / \\\\ lkb I1 \\\\ PREDICTED 1312 AMINO ACID POLYPEPTIDE FIGURE I.-The 9.5-kb RAD50 locus, 4.775 kb of which was sequenced. The point of initiation and the direction of transcription is indicated by the arrow. E = EcoRI, P = PstI, Bg = BglII, B = BamHI, H = HindIII, Pv = PvulI, S = SalI. P = purine nucleotide binding domain similarity; I = heptad repeat I (amino acids 177- 421); I1 = heptad repeat I1 (amino acids (743-995). plementation span the entire region (HUISMAN et al. 1987). Additional information outside of this minimal segment is required for normal expression of the RAD50 gene. When expressed as a HindIII to SalI fragment on an ARS CEN vector, both the size and quantity of the RAD50 transcripts differ from that observed for a chromosomal RAD50 gene (W. RAYMOND and N. KLECKNER, unpublished data). However, a normal chromosomal transcription pattern is obtained with a larger 6.6-kb BgZII to EcoRI RAD50 subclone (W. RAYMOND and N. KLECKNER, unpublished data). To define the RAD50 coding region in more detail, the DNA sequence was determined for a 4775-bp region that includes the HindIII-Sal1 fragment (Figure 2). This analysis, described in Figure 3 and MATERIALS AND METHODS, reveals a continuous 1312 amino acid open reading frame (ORF) that is fully contained in the minimal RAD50 HindIII-Sal1 segment (Figure 2). The corresponding protein begins with a methionine encoded by bp 558-560 and terminates with a stop codon at bp 4492-4494 and has a predicted M,. of 153 kD. Immunoblot analysis of yeast proteins has subsequently identified a single polypeptide of the expected M,. which has been correlated genetically with RAD50 function (W. RAYMOND and N. KLECKNER unpublished data). The RAD50 protein predicted from the DNA sequence is composed of 17.5% acidic, 15.9% basic, 30.9% hydrophobic and 36.0% polar residues; nine glycosylation consensus sequences (NXS, X # P) are present. KYTE-DOOLITTLE (1 983) analysis suggests that the protein is predominantly hydrophilic; the longest hydrophobic region is a 35 amino acid segment near the carboxy terminus of the protein (amino acids 1205-1232, corresponding to bp 4170-4256 in Figure 2) which contains 20 hydrophobic residues. Nucleotide binding domain: Initial inspection of the RAD50 amino acid sequence reveals a short glycine-rich sequence that has previously been found in the sequences of many adenine and guanine nucleotide binding proteins (Figure 4; FRY, KUBY and MILD-  50 E. Alani, S. Subbiah and N. Kleckner 12 gszqmmmm mmmmmmm m \"\"N:RC::\"-mmWw rlmfnmf-rlm \\\\o OI pl\"\"b w w w w w w w w \\\\o om4 ,,ma\\'; - 5 a, o N - w m o W $$ON* wwwww N\"lnwf-oa 2!Ld3S2 ?l IIII  Analysis of Yeast RAD50 Gene 51 ~g;$;,,rlm mmmm ,I- mm44 rl 16 2 rl ~~ 1 Ill ~~ 3 il f ip 9 %I H r4 ko 3 ji ![ H e rl $ \"5 rl c:b4 Ai 11 2 8 i 1 t rl J d El 1 rl 8 i i iL I 4 d 1 3 rl H 3 z 4 $ ! 2 [I !i 3 $1 $I g iI q 1 id ! I! 1 1 H I $ 8 i 4 5 8 \\'61 rl E g 3 8 3 3 wwww OD B2SrnWP stggse W WNNNNNNR3 mmmmmm 3zzsi3 $ E 2 a WOD$$SSPONL; ww :2*mgz: SG wwo w z\" .Y z 0 rn E gA; 8 t F2 3 2 .- q22 r ;* p; q+pg @@B r-: n 2 \\'-4 2sSZZ g200 - ;;g sy; e2 rn 6 - 333 pi &E ::g * Ebk3 \\' \\'Z 0 mwm u %u d 4 x Yj E nEe! z ,Y QU .- y1 m rn\" t3 g.s d5 3 b 8 w& u \\'G .E 2 2 $p...Z& :-E $$$ m fis 09 gj.5 rq 3 % z52 .+q* <;sj z 2 2.2 y g m\\'E2 r: e! 5 B z.g6 y~3m \\'GZ -? 9 * c Ab &A 8 I $..u $? gz; m6,n Vi pg2< F\\'= rnm \\'2 5 \\'C z 2 b;%- A,? 5 5 z \"5 ; 5 0 g : .4 w sa gEn 3- 0 = +,z 2 2 \"05 5 i 0 rn\" 8 .L. Y E ..$ T 2 T Tee; 0 m4: kZZ sgz I g 22552 ;,;2 I e e\" 21 g 2 ;;; .o P u c.2zF 5 z ajm .5 E 522 u 20 5022E rnr=ern x \" .- Luo W€ ,- w QUT rn CS5b er - Un 0 x L \\'S .>, \"3 L. iQi mpu .-.s e ED\">: rn &.E y 5 .I2 2 pcm  = p:; .c . \\'e -hero - 2 \\'7 u .\\'\" c dm .- 3 \" m CI v) LmUS JZC II) m - I U m 0 0 0 3 W If: G I 0 0 3 t. m w I n a d 0 k i 0 m m I 0. n \\'2 t s n W er 3 k ,P .- 5 5 3 - - 2 b 2 e! B f z u c h - I .- a 3 K ? Y d si U m *2 2 g, e, .m 0 Qi * .* I M 3 3-G 22 rtU  52 E. Alani, S. Subbiah and N. Kleckner Xhol BarnHl / 1. PvuII or Hind111 digest 1. PvuII or Sal1 digest 2. BAL31 digestion 2. BAL31 digestion 3. XhoI digest 3. BamHI digest 4. T4 DNA Polymerase 4. T4 DNA Polymerase 5. T4 DNA Ligase 5. T4 DNA Ligase \\\\/ 1. order deletions by restriction mapping 2. isolate single stranded DNA far dideoxy Sequencing hydrophobic loop a-helix P-strand consensus: v>XXgsGKTI a I RADSO 129-50) M N G s G X T IT I I E c L K Y .J Adenylate Kinase (10-30) EV>G 0 P G S G K G TlQ C E K I V H 4 FIGURE 4.-RAD50 protein shows similarity to purine nucleotide binding proteins. Structural analysis of adenylate kinase and sequence comparisons between ATP and GTP binding proteins has led to ;I consensus sequence for purine nucleotide binding pockets (FRY, KURY and MILDVAN 1986; ROSSMAN, MoRAsand OLSEN 1974; BRADLEY et al. 1987). This pocket is composed of a hydrophobic 0- strand and an a-helix which flank a glycine rich flexible loop (top line). The corresponding structures in adenylate kinase, based on crystal structure and RAD50, based on CHOW-FASMAN analysis, are shown (FRY, KURY and MILDVAN 1986; CHOU and FASMAN 1978; RALPH, WEBSTER and SMITH 1987). In the consensus sequence, upper case and lower case letters indicate strongly and less strongly conserved residues, respectively, and X indicates nonconserved residues. VAN, 1986; WALKER et al. 1982; DEVER, GLYNIAS and MERRICK 1987). In adenylate kinase, this sequence specifies a flexible loop whose terminal lysine residue appears to interact with the a-phosphoryl group of Mg-ATP (FRY, KUBY and MILDVAN 1986; DREUSICKE and SCHULZ 1986). In a number of ATP and GTP binding proteins, this loop is suggested to mediate conformational changes in the protein brought about in some cases by interactions with additional proteins (FRY, KUBY and MILDVAN 1986; REINSTEIN, BRUNE and WITTINCHOFER 1988). FIGURE 3.-RAD50 sequencing strategy. pNKY5 and pNKY6 contain ColEl and M13 origins of replication, an ampicillin resistance determinant, a site used for binding of an oligonucleotide primer for DNA sequencing, and the RAD50 gene, inserted in opposite orientations in the two plasmids. Sequence of both strands of the RAD50 HindIII-Sal1 segment was obtained as described in this figure and in the MATERIALS AND METHODS. In addition, RAD50 protein is predicted by ARIADNE, a computer matching program that utilizes a descriptor pattern for the ATP binding motif (LATHROP, WEBSTER and SMITH 1987; RALPH, WEBSTER and SMITH 1987), and CHOU-FASMAN (1978) analysis to have specific secondary structure features common to the ATP binding pockets of many other nucleotide binding proteins. These include a hydrophobic Pstrand and an a-helix which flank the flexible loop (ROSSMAN, MORAS and OLSEN 1974; BRADLEY et al. 1987; Figure 4). Preliminary analysis reveals several reasonable candidates for a magnesium coordination site (BRADLEY et al. 1987; T. WEBSTER and M. BRADLEY, personal communication). RAD50 protein probably binds an adenine nucleotide. It lacks two consensus sequence features that are specific to the known set of closely related guanine nucleotide binding proteins (DEVER, GLYNIAS and MERRICK 1987). ATP binding is a common feature of DNA repair and recombination proteins from E. coli (RecA, RecBCD, RecN, UvrABC, UvrD), S. typhimurium (MutS) and S. cerevisiae (RAD3) (SMITH 1988 for review, BACKENDORF et al. 1986; HABER et al. 1988; NAUMOVSKI and FRIEDBERG 1986). However, in the event that RAD50 is involved structurally rather than or in addition to having an enzymatic role in recombination, the role of ATP in RAD50 function may be  Analysis of Yeast RAD50 Gene 53 f b e CORE INNER OUTER a, d e,q b,c,f FIGURE 5,\"Cross section of a-helical coiled coil when viewed from the amino end of parallel helices (N). Filled circles indicate core hydrophobic residues. (Adapted from MCLACHLAN and KARN 1983.) different from its role in these other proteins (see below). Heptad repeat sequences and similarity to myosin S2 domain: Two regions within the RAD50 amino acid sequence, designated heptad I (177-421) and heptad I1 (743-995) display a heptad repeat pattern found in proteins such as myosin, keratin and tropomyosin (MCLACHLAN and KARN 1982, 1983; COHEN and PARRY 1986). The heptad repeat is a seven amino acid sequence with primarily hydrophobic amino acids at the first and fourth positions and a high percentage of charged amino acids at the other positions. When present in a repeated motif, this unit forms an alpha helical structure (Figure 5). Two heptad-containing a-helices can coil around one another in a regular helical structure that is stabilized by repeated interactions between hydrophobic residues and specific, oppositely charged residues on the two chains (Figure 5) (MCLACHLAN and KARN 1982). The RAD50 heptad repeat regions were identified and analyzed by three separate methods, discussed below. These approaches also reveal a significant similarity between heptad I1 and the S-2 fragment of rabbit myosin heavy chain. The RAD50 protein sequence was compared to other protein sequences in the NBRF database using the FASTP computer algorithm of LIPMAN and PEARSON (1985). This analysis provides a similarity score for each such comparison. A significant match is indicated if the difference between the similarity score of a match of interest and the average similarity score is more than 10 SD (Z 2 10). When comparisons are made under conditions were significant gaps in matching sequences were allowed, eight protein sequences show significant homology to RAD50 protein. These include rabbit skeletal muscle myosin (heavy chain S- 2 fragment), rabbit cardiac muscle myosin (a and P heavy chains), and several muscle tropomyosins from chicken, rabbit and rat. When stricter penalties are assigned for gaps, only the rabbit S-2 myosin fragment continues to show strong similarity to RAD50 (Z = 12.5). RAD50 protein and the rabbit S-2 myosin fragment are 22% identical over the entire 259 amino acid rabbit myosin fragment (CAPONY and ELZINGA 198 1, Figure 6). The five proteins initially identified by FASTP analysis are all known to contain extensive regions that form a-helical coiled coils. Therefore, a systematic search for heptad repeats in the RAD50 protein was carried out using Fourier analysis. A computer program based on the analysis of MCLACHLAN and KARN (1 983) was written to search for repeats of seven residue sequences containing hydrophobic amino acids at the first and fourth positions (MCLACHLAN and KARN 1983). The RAD50 protein sequence was divided into eleven blocks of 120 amino acids each; each block was analyzed individually. This approach eliminates the possibility that small heptad containing regions might be obscured by large non-heptad regions; it is sensitive enough to detect heptad-containing sequences only twenty to thirty amino acids long (three or four heptads). The Fourier analysis reveals that two large regions of the protein contain significant amounts of heptad repeat sequence: three consecutive blocks of 120 amino acids in the amino-terminal third of the protein and three consecutive 120 amino acid blocks in the carboxy-terminal third of the protein. Visual inspection of these two regions identified by Fourier analysis reveals that each contain a single, virtually uninterrupted series of heptad repeats; both the end points of each region and rare one or two amino acid deletions in the heptad pattern (skip residues) could be unambiguously identified. The amino-terminal region heptad I spans amino acids 177-421; it consists of thirty-five heptad repeats, one of which contains a skip residue. Heptad I1 spans amino acids 743-995; it consists of thirty-seven heptads, three of which contain one skip residue and one contains twp residues. Given the rise of an alpha helix at 1.485 A per amino acid (FRASER and MACRAE 1973), the length of alpha helical coiled coil structures involving heptad I and heptad 11, if present as fully extended rods, would be 364 A and 385 A, respectively. However, other coiled coil interactions exist which do not result in extended rod structures (BANNER, KOKKINIDIS and TSERNOCLOU 1987). Interestingly, heptad I1 and the entire S-2 fragment of rabbit myosin are exactly co-extensive, with 37 heptads each. The amino acid compositions of the two RAD50 heptad regions and the S-2 domain of rabbit myosin are very similar to those of previously analyzed CYhelical coiled coil regions. These three regions and the relevant portions of nematode myosin, keratin and tropomyosin are compared in Table 1 (MCLACHLAN and KARN 1983; PARRY et al. 1977; STONE and SMILLIE 1978; KARN, BRENNER and BARNETT  54 E. Alani, S. Subbiah and N. Kieckner RADS0 (743) LNSINEKIDNSQKCLEKAKEETKTSKSKLDELEVDSTKiRPLIEKFTYLEKELKDLENSSKTISEEL-SIYNTSED ........ I .. .II.II. .... I .Ill .. )..I[. ......... I I .I1 ........... .I1 RABBIT S2 (n) SAETEKEMANMKEEFEKTKESLAKAEAKRKELEEKMVALMQEKNDLQLQVQAEADS~AE-ERQDLIKTKIQLEAKIKEVTERAED GIQTVDELRoQQRKMNDSLRLRKTISDLQ--MEKDEKVRENS-RMINLIKEKELTVSEIESSLT-QKQNIDDSIRSKRENINDIDS ... .II ... II..I. .II.I.I.Il. .I 1 ........ ..I. ... I. .I1 .I...... ......... EEEINAELTAKKRKLEDECSELKKDIDDLELTLAKVEKEKHATENKVXNLTEEMAGLDETIAKLTKEKKALQEAHQQTLDDLQAEED RVKELEARIISLKNKKDEAQSVLDKVKNERDIQVRNKQK~ADINRLIDRFQTIYNEWDFEAKGFDELQTTIKELELNKAQMLE RAD50 (996) .[..I. .. I... I .. I.. I. I 1.l.I .I.. . .I I. .... I . I ..... I I K~TLTKAKTKLEQQVDDLEGSLEQEKKIRMDLERAKRKLEGDLKLAQETSMDIENDKQQLDEKLKKLEFMTNLQSKIEDEQALM RABBIT S2 (nt258 I = Identity . = PAM250 Matrix conservative substitution FIGURE 6.-Amino acid alignment of rabbit S2 fragment of myosin (CAPONY and ELZINGA 198 1) and amino acids 743-996 of the RAD50 protein. I = amino acid identity, = Conservative substitution according to PAM 250 matrix (DAYHOFF 1978). TABLE 1 Amino acid distributions in RAD50 heptad I and heptad I1 compared to distributions in coiled coil regions of other proteins Protein (No. of heptads) RAD50 heptad I RAD50 heptad I1 Rabbit myosin (S2) Nematode myosin a-Keratin Tropomyosin Distribution (35) (37) (37) (152) (29) (40) Core (a, d) % hydrophobic 61 63 61 56 65 88 % charged 25 16 23 18 17 8 % hydrophobic 16 19 8 7 17 17 % charged 34 49 55 53 45 63 Inner, outer (b, c, e, f, g) Hydrophobic amino acids were: methionine, tyrosine, leucine, tryptophan, phenylalanine, valine and isoleucine; charged amino acids were: lysine, arginine, histidine, aspartic acid and glutamic acid. References for the protein sequences were as follows: rabbit myosin S2 domain (CAPONY and ELZINGA 1981), nematode myosin (KARN, BRENNER and BARNETT 1983), a-keratin (PARRY et al. 1977), and tropomyosin (STONE and SMILLIE 1978). 1983). All five regions show the distribution of hydrophobic and charged amino acids characteristic of coiled coil proteins: high percentages of hydrophobic residues at the first and fourth positions of the heptad (56-88%), and at the other positions, low percentages of hydrophobic residues (8-1 9%) and a high percentage of charged amino acid residues (34-63%) (Figure 5, Table 1). Significance of similarity with rabbit myosin S-2 domain: The similarity between RAD50 heptad I1 and the rabbit S-2 myosin fragment (Figure 6) is suggestive. Further analysis indicates the existence of a deeper relationship between the two proteins than expected simply on the basis of their heptad nature. The RAD50 and the rabbit S-2 fragment protein sequences were examined by a computer comparison program designed to eliminate the contributions of the repeating heptad pattern itself. This program, a modified NEEDLEMAN-WUNSCH analysis that incorporates a “heptad shuffle,” is described below (Table 2; S. SUBBIAH, unpublished data). Standard NEEDLEMAN-WUNSCH (1 970) analysis allows the optimal alignment of two sequences to yield a similarity score. To determine the significance of that optimal alignment, both sequences are scrambled at the same time and a NEEDLEMAN-WUNSCH similarity score is determined after each scrambling. From this process an average random score and its standard deviation is determined. Then the difference between the similarity score of the optimal alignment and the average random score of the distribution is assessed in terms of standard deviation units. This calculation yields a value called 2 regular. In this type of analysis, two heptad-containing proteins should always show greater similarity to one another than to a scrambled sequence in which the constraints of the heptad sequence are removed. This was verified by creating and then comparing two nonidentical amino acid sequences each consisting of 30 “perfect” heptads; 2 regular for these two sequences is close to (UP matrix, DAYHOFF 1978) or above (SIM matrix, MCLACHLAN 1971) the cutoff for significant  Analysis of Yeast RAD50 Gene 55 TABLE 2 Detailed comparison of RAD50 heptad I (HEPI), RAD50 heptad I1 (HEPII), nematode myosin S-2 domain (NEM), and rabbit myosin S-2 domain (RAB) protein sequences UP matrix analysis SIM matrix analysis CornDdrison 2-hemad 2-regular Z-heotad Z-regular 1. HEPII-RAB 4.2 ~ 1 5.25 4.82 - 8.50 2. HEPII-NEM 2.19 __ 2.93 2.72 4.91 1.04 1.37 3.07 4.06 - - - 3. RAB-NEM 21.13 25.44 20.78 25.94 4. HEPI-HEPII \"\" 5. HEPI-RAB 1.31 1.77 0.49 2.20 6. HEPI-NEM -0.83 -0.21 0.99 3.49 7. R1-R2 -1.61 2.76 -1.24 11.36 8. R1-R3 \"\" 19.54 37.82 13.28 35.45 Z-regular and Z-heptad calculations based on the UP and SIM matrices are described in the MATERIALS AND METHODS. Underlined sequences indicate similarity at or above the cutoff values for the UP (cutoff = 3) or SIM (cutoff = 5) comparisons. R1 and R2 are unrelated synthetic 30 heptad segments, each 210 amino acids long. R3 is a 30 heptad segment, 210 amino acids long that is 33% identical to R1. References are as follows: Rabbit myosin S2 domain (CAPONY and ELZINCA 1981) and nematode myosin (KARN, BRENNER and BARNETT 1983). homologies (Table 2). This defect was eliminated by performing the usual significance analysis on a set of control sequences scrambled in such a way that the heptad nature was always retained. Specifically, amino acids at the hydrophobic core positions of the reference sequence were scrambled separately from amino acids at other positions (MATERIALS AND METHODS, the heptad shuffle). The comparison of a sequence of interest with this control set gives a value called 2 heptad. Additional calculations show that the heptad shuffle does not dramatically change the similarity scores obtained either for two non-heptad proteins of similar amino acid composition or for two heptadcontaining proteins with a significant percentage of amino acid identities (S. SUBBIAH, unpublished data). For each pairwise combination of RAD50 heptads I and I1 and the S-2 domains of rabbit and nematode myosins, alignments were made by the standard NEEDLEMAN-WUNSCH procedure. For each pair, alignments were made using two different matrices to define amino acid relatedness (UP and SIM, MATERIALS AND METHODS). The significance of each alignment was evaluated both with and without the heptad shuffle (Table 2). Heptad I exhibits no significant similarity to any of the other sequences regardless of the relatedness matrix and significance analysis used. In contrast, heptad I1 and the rabbit S-2 domain shows significant relatedness under all conditions, although in one case the level of relatedness was slightly less than the number of standard deviations usually taken as a cutoff point (MCLACHLAN 1971; DAYHOFF 1978). Biological importance of coiled coil regions: Proteins that contain coiled coil regions participate in a variety of biological processes including muscle movement, anti-sense RNA pairing, nuclear envelope formation and viral pathogenesis (STEINERT, STEVENS and Roop 1985; MCLACHLAN and KARN 1982; BANNER, KOKKINIDIS and TSERN~CLOU 1987; AEBI et al. 1986; MCKEON, KIRSCHNER and CAPUT 1986). Some of these proteins show an additional feature that is also observed in the RAD50 protein sequence; the nuclear and cytoplasmic intermediate filament protein sequences contain long heptad repeat regions that are interspersed with non-heptad spacers (STEINERT, STEVENS and Roop 1985). The RAD50 protein also shows a 320 amino acid non-heptad spacer region between heptad I and heptad 11. However, the length of the RAD50 spacer is much larger than those present in intermediate filament proteins (STEINERT, STEVENS and Roop 1985). In the absence of any specific information about the role of RAD50 function, we can only speculate as to the role of its coiled coil regions. Three intriguing and somewhat unusual possibilities have arisen thus far. One possibility would be that the RAD50 protein stretches across the synaptonemal complex. The length of each coiled coil region is approximately the distance between an axial element and the central element, so the protein might extend from one axial element to the other, with the two heptad containing regions spanning the two regions that separate the axial and central elements. A second possibility, suggested by the similarity of heptad I1 to the S2 domain of myosin would be that RAD50 is part of a contractile system which actively brings homologous chromosomes together once homology is sensed. A third possibility, raised by the apparent similarity in overall structure between RAD50 protein and kinesin (J. GELLES, personal communication), would be that RAD50 is involved in translocation of DNA molecules along microtubules, or vice versa. Kinesin is an ATP dependent translocator that travels along microtubules. Electron microscopy suggests that each kinesin molecule consists of a globular domain at one end plus two rod-like elements connected by a flexible joint (AMOS 1987). The globular domain is in contact with microtubules and presumably encodes the ATP binding activity; the arrangement of the rod-like elements is suggestive of two a-helical coiled coils interrupted by a spacer (AMOS 1987). This organization is similar to that of RAD50, with its amino terminal purine nucleotide binding domain and two separated heptad repeat regions. However, computer comparisons between RAD50 and a partial sequence of Drosophila kinesin do not reveal any significant similarity beyond that expected from their common coiled coil and purine nucleotide binding features (L. GOLDSTEIN, personal communication). Biochemical purification of the RAD50 protein and genetic analysis of rad50 mutants should be useful in understanding the  56 E. Alani, S. Subbiah and N. Kleckner role of RAD50 protein in mitotic DNA repair and in meiotic recombination.  Approximately 5–10% of all breast and ovarian cancers are due to genetic predisposition.1 Germline mutations in two major susceptibility genes, BRCA1 and BRCA2, explain most of the hereditary ovarian cancer cases, but worldwide have been observed only in a fraction of the breast cancer families; in Finland, about 20%.2–4 A small portion of the remaining cases, perhaps 1–2%, appears to be caused by germline mutations in other cancer susceptibility genes such as TP53, PTEN, AR, and ATM. 5 Recently, moderate or low penetrance germline mutations such as 1100delC of CHK2 have also been implicated,6 suggesting that in many cases the phenotypic expression of the disease is accounted for by the combined effect of multiple but more subtle mutations in different target genes.5 7 Despite all these discoveries, many breast cancer families, especially those with fewer than four cases and/or with later disease onset, have cancers due to mutations in still unknown genes. As the protein products of the genes so far indicated are central players in the pathways involved in cell cycle checkpoint functions and in the sensing, transduction, and repair of DNA lesions,5 other similarly acting genes may represent new potential candidates for breast and/or ovarian cancer susceptibility. The Mre11 complex, composed of the proteins MRE11, NBS1, and RAD50, is a vital component in the DNA damage response and is at least partially responsible for connecting DNA damage detection to DNA repair and cell cycle checkpoint functions. The early association of the complex with damaged DNA suggests its involvement in sensing DNA double strand breaks (DSBs). In eukaryotes, DSB repair is handled by two different pathways, homologous recombination (HR) and non-homologous end joining (NHEJ), and the initial step in both is the recognition and signalling of the damage by the Mre11 complex.8–10 In addition to DNA damage detection and repair, the complex has an important role in co-ordinating checkpoint responses and DNA replication,11–14 and is also required for the maintenance of telomeres.15 The sites important for the composition and various functional activities of the Mre11 complex have been characterised (summarised in fig 1). Because of its independent interaction with both NBS1 and RAD50, MRE11 has been viewed as the core of the complex, whereas the interaction between NBS1 and RAD50 is indirect and mediated through MRE11.9 16 17 The sites in MRE11 responsible for binding to NBS1 and RAD50 have both been assigned to the aminoterminal half of the protein.16 18 In NBS1, the MRE11 binding domain has been located to the extreme carboxyterminus of the protein,16 and the corresponding site in RAD50 has been located to a 40 residue coiled coil region adjacent to the ABC domain.19 20 Additionally, several other important domains for the complex function have been identified. The DNA nuclease activity is restricted to MRE11 and specified by phosphodiesterase motifs in the aminoterminal half of the molecule.18 The aminoterminus of NBS1 exhibits a forkhead associated (FHA) domain, followed by a BRCA1 carboxyterminal (BRCT) domain.21 Both of these highly conserved domains are involved in protein–protein interactions and are therefore crucial for the functions of the Mre11 complex, including its ............................................................... . Abbreviations: AT, ataxia telangiectasia; ATLD, ataxia telangiectasialike disorde; BASC, BRCA1 associated genome surveillance complex; BRCT, BRCA1 carboxyterminus; CSGE, conformation sensitive gel electrophoresis; DSB, double strand break; FHA, forkhead associated; HR, homologous recombination; NBS, Nijmegen breakage syndrome; NHEJ, non-homologous end joining Key points N The Mre11 complex has an essential role in maintaining genomic integrity and preventing cells from malignancy. The complex components interact with several proteins (for example BRCA1, ATM and CHK2) already connected to hereditary breast and ovarian cancer susceptibility, which makes MRE11, NBS1, and RAD50 suitable candidate genes for predisposition to these cancers. N We screened the affected index cases of 151 families with signs of hereditary susceptibility to breast and/or ovarian cancer for germline mutations in the Mre11 complex genes. Seven novel heterozygous alterations were observed, three of which were considered potentially disease-related: MRE11 913CRT (Arg305Trp), NBS1 448CRT (Leu150Phe) and RAD50 687delT (stop codon at 234). N RAD50 687delT is expected to cause major truncation in both known protein isoforms and occurs in multiple individuals of two families. Curiously, RAD50 687delT was also present in some of the controls, but at a lower frequency, probably referring to incomplete penetrance. The MRE11 and NBS1 alterations affected highly conserved amino acids and were seen in single cancer cases, but they were both absent in 1000 healthy controls. N Our results suggest that these three mutations in the Mre11 complex genes associate with hereditary susceptibility to breast and/or ovarian cancer, and may be important in other types of cancer too. This is the first time to our knowledge that germline mutations in RAD50 have been linked to genetic disease in humans. 1 www.jmedgenet.com subcellular localisation after DNA damage.17 22 RAD50 in turn consists of bipartite N- and C-terminal ATPase segments, which assemble into a single ABC-type cassette at the end of the predicted antiparallel coiled coil.19 A functionally important zinc hook has been identified in the central portion of the coiled coil domain.23 This zinc hook is thought to mediate RAD50 dimerisation, which allows the required molecular flexibility of the Mre11 complex and keeps it functionally assembled during the DNA recombination and repair process.14 The Mre11 complex proteins interact with several important protein partners, many of which have been related to cancer predisposition. For instance, following ionising radiation, NBS1 is phosphorylated in an ATM dependent manner to ensure proper response to DNA damage.11 In turn, MRE11 phosphorylation relies on NBS1.24 Co-operation between a functional Mre11 complex and ATM is also required for activation of the CHK2 kinase in response to DNA damage.12 In addition, MRE11, NBS1 and RAD50 are linked through protein–protein interactions with BRCA1. After exposure to ionising radiation, BRCA1 forms nuclear foci that co-localise with the Mre11 complex.25 Along with BRCA1, ATM, and other proteins, MRE11, NBS1, and RAD50 are also part of a multi-subunit protein complex referred to as the BRCA1 associated genome surveillance complex (BASC).26 Failure in the DNA damage response and repair processes causes chromosomal instability that may result in cancer predisposition. For two of the Mre11 complex genes, namely MRE11 and NBS1, the connection to cancer has already been established: germline mutations in MRE11 are associated with the ataxia telangiectasia-like disorder (ATLD),27 whereas defects in NBS1 cause Nijmegen breakage syndrome (NBS).9 21 Cells derived from ATLD and NBS patients display similar characteristics as those from ataxia telangiectasia (AT) patients, and AT due to germline mutations in the ATM gene has been associated with an increased risk of developing various malignancies; breast cancer, lymphoma, and leukaemia in particular.28 As yet, germline mutations in RAD50 have not been connected to genetic diseases in humans. However, in mice certain germline RAD50 mutations result in growth defects and cancer predisposition.29 Because mutations in ATM, CHK2, and BRCA1 have all been previously linked to hereditary predisposition to breast and/or ovarian cancer5 it is possible that germline mutations in the individual Mre11 complex genes may also be directly involved in predisposition to these cancers. The suggested connection has been supported by a recent study, where the NBS1 657del5 founder mutation was found to be responsible for the occurrence of a small but significant portion of familial breast cancer cases.30 On these grounds, we tested 151 index patients from Finnish families with signs of hereditary susceptibility to breast and/or ovarian cancer for possible germline mutations in the MRE11, NBS1, and RAD50 genes. The results of our study indicate novel disease associated alterations. MATERIALS AND METHODS Subjects and unaffected controls Index cases diagnosed with breast or ovarian cancer of 151 families originating from northern Finland were selected for the screening of germline MRE11, NBS1, and RAD50 mutations. Of the studied families, 95 were associated with breast, 29 with breast–ovarian, and 4 with ovarian cancer. These families met the criteria for either high (76 families) or moderate (52 families) genetic susceptibility to breast and/or ovarian cancer. Inclusion criteria for the families were one or more of the following: (a) two to three or more cases of breast and/or ovarian cancer in first or second degree relatives; (b) early disease onset ((35 years); (c) bilateral breast cancer; or (d) multiple primary tumours including breast or ovarian cancer in the same individual. The high risk families mainly included three or more cancer cases. All of the high risk families had been screened for germline mutations in BRCA1, BRCA2, CHK2, and TP53, and eleven of them were found to be BRCA1 or BRCA2 mutation positive.4 31 32 Besides high or moderate risk families, 23 families showing single cases of breast or ovarian cancer along with multiple cases (two or more) of other kinds of cancer in first or second degree relatives were included in the study. Additionally, 1000 anonymous cancer free individuals were used as controls, and the occurrence of novel, potentially pathogenic alterations was also investigated in 192 breast cancer patients without family history of the disease. Both groups derived from the same geographical area as the studied families. All patients had given their informed consent for obtaining pedigree data and blood specimens for a study on cancer susceptibility gene mutations. Approval to perform the study was obtained from the ethics board of the Northern Ostrobotnia Health Care District and the Finnish Ministry of Social Affairs and Health. Screening for germline mutations DNA from blood lymphocytes was extracted using the standard phenol–chloroform method or the Puregene D- 50K purification kit (Gentra, Minneapolis, MN, USA). The coding regions and exon–intron boundaries of the MRE11, NBS1, and RAD50 genes were screened for germline mutations by conformation sensitive gel electrophoresis (CSGE), which has been shown to be a cost efficient way to scan genes with high sensitivity and specifity.4 33 Samples with band shift were reamplified and the sequencing analysis was performed with the Li-cor IR2 4200-S DNA Analysis system (Li-cor Inc., Lincoln, NE, USA) using the SequiTherm EXELTM II DNA Sequencing kit LC (Epicentre Technologies, Madison, WI, USA). Oligonucleotides for CSGE and sequencing were designed by using Primer3 software based on sequence information obtained from publicly available databases. Oligonucleotide sequences and PCR conditions are available upon request. LOH analysis Loss of heterozygosity (LOH) analysis was performed to assess whether the tumours of the patients displaying the novel MRE11 and RAD50 germline mutations had lost the wildtype allele. The flanking microsatellite DNA markers used for MRE11 were D11S4176 and D11S1757; and for RAD50 D5S2110, D5S2057, D5S1984, D5S2002, and D5S2117. The PCR products were analysed with the Li-cor DNA Analysis system using the IRD800 labelled forward primer. Figure 1 Components of the Mre11 complex. The functional domains of each protein are marked above the corresponding diagram (see text for more details). In addition, the putative DNA binding domain20 (NCBI; LocusLink) of MRE11 and the RAD50 hook construct containing the zinc binding Cys-X-X-Cys-motif23 are shown. FHA, forkhead associated; BRCT, BRCA1 carboxyterminal; MBS, MRE11 binding sites. 2 Electronic letter www.jmedgenet.com Evaluation of RAD50 687delT expression at the mRNA level Expression of the mutated allele at the mRNA level was evaluated for RAD50 687delT. Using the FastTrackH 2.0 Kit (Invitrogen, Carlsbad, CA, USA), mRNA from lymphoblast cell lines was isolated, following which cDNA synthesis was carried out with the GeneRacerTM kit (Invitrogen). The use of the GeneRacerTM Kit ensured that only transcripts containing full length cDNA ends were used in PCR analysis. Primers used for the cDNA specific amplification of RAD50 exon 5 were: forward 59-TTTGAAGCAAAAGTTTGATGAGA-39 (located in exon 4) and reverse 59- TCATTGTCAAGTTTCATTATTTTAGAG-39 (located in exon 6). RESULTS Novel germline alterations The mutation analysis of the MRE11, NBS1, and RAD50 genes revealed several germline alterations. Eleven heterozygous changes in protein encoding regions were observed (summarised in tables 1 and 2). Seven of the alterations were novel and three of these were considered potentially disease related. In addition, a large number of different apparently harmless alterations occurring in intron regions was seen (data not shown). MRE11 and NBS1 In the MRE11 gene, a single mutation in exon 10 was found in one index case with ovarian cancer (case 322, fig 2A). This novel single base change, a transition CRT at position 913, resulted in substitution of arginine to tryptophan at codon 305. The position of the involved arginine is extremely conserved between several species, even those distantly related (fig 4) (yeast and flowering plant: data not shown). Additionally, the flanking regions on both sides of this residue are well conserved, suggesting functional importance. In contrast to MRE11, multiple alterations within the protein encoding region of NBS1 were detected, but only one was considered potentially pathogenic. This novel change, a CRT transition at nucleotide 448, results in substitution of leucine to phenylalanine at codon 150 and was displayed by one index case diagnosed with breast cancer (case 277, fig 2B). Interestingly, this alteration occurs in the middle of the highly conserved BRCT domain, where the leucine residue at position 150 is conserved also in the mouse, rat, and chicken orthologues (fig 4). Both the MRE11 and NBS1 mutations were observed in single multi-cancer families including one ovarian or one breast cancer case, respectively, but co-segregation with the cancer phenotype could not be confirmed owing to lack of DNA specimens from additional family members. However, both of the mutations were absent from a large cohort of population controls (n = 1000), indicating that these mutations may indeed relate to cancer predisposition (table 1). RAD50 In RAD50, four novel germline changes affecting exon regions were discovered (tables 1 and 2). Of these changes, His68His appears to be a harmless polymorphism, but the roles of Ile94Leu and Arg224His alterations remain more difficult to determine. Both of these substitutions affect conserved residues, the former in the remarkably well conserved aminoterminal ATPase domain and the latter in the coiled coil domain,19 20 but they also occurred in the healthy controls, although at lower frequency. Interestingly, one of the breast cancer families displaying the RAD50 Ile94Leu variant also showed partial co-segregation with the previously identified CHK2 Ile157Thr unknown variant.31 Until further clarifying studies have been carried out, the cancer predisposing role of these two RAD50 variants remains unknown. However, at least one of the identified RAD50 alterations, 687delT, is likely to be disease related. The RAD50 687delT truncation mutation The RAD50 687delT mutation in exon 5 leads to a premature stop at codon 234 and affects both isoforms of the protein. Table 2 Observed sequence variants occurring in the exon regions of the Mre11 complex genes Gene Exon Nt change Effect on protein Allele frequency* Hereditary cases Controls NBS1 2 102GRA Leu34Leu  39.1% (59/151) ND 5 553GRC Glu185Gln  43.7% (66/151) ND 10 1197TRC Asp399Asp  50.3% (76/151) ND 11 1720ART Leu574Ile 0.7% (1/151) 1.3% (4/307)` 13 2016GRA Pro672Pro  37.1% (56/151) ND RAD50 2 204CRT His68His 0.7% (1/151) ND 3 280ARC Ile94Leu 1.3% (2/151) 0.3% (3/1000) 5 671GRA Arg224His 2.0% (3/151) 0.9% (9/1000) *Heterozygotes.  Reported by Varon et al (1998) or the SNP database. `Only 307 controls were tested for this alteration. ND, not done. Table 1 Novel MRE11, NBS1 and RAD50 mutations Gene Exon Nt change Effect on protein Allele frequency* Hereditary cases Sporadic cases Controls MRE11 10 913CRT Arg305Trp 0.7% (1/151) – (0/192) – (0/1000) NBS1 4 448CRT Leu150Phe 0.7% (1/151) – (0/192) – (0/1000) RAD50 5 687delT Frameshift R stop codon at 234 1.3% (2/151) – (0/192) 0.6% (6/1000) *Heterozygotes. Electronic letter 3 www.jmedgenet.com The deletion was found in two separate families (fig 3). In family 5, the index case and her sister (cases 828 and 474), both diagnosed with breast cancer, were carriers of the mutation. In family 6, the index (case 505) with ovarian cancer carried RAD50 687delT, but had also tested positive for the BRCA1 3745delT mutation in one of our previous studies.4 In addition, the deceased mother with ovarian cancer was an obligatory carrier of at least the RAD50 687delT mutation. In all the other cases tested, both the RAD50 and the BRCA1 truncation mutation segregated incompletely with the disease. The observation of unaffected RAD50 truncation mutation carriers in family 6, and also the surprisingly high prevalence at the population level (0.6%) suggested not only that RAD50 687delT, if truly pathogenic, exhibits low penetrance, but also that it might predispose carriers to cancer without clustering of the disease in the pedigrees. In order to test this hypothesis, we screened 192 breast cancer cases without family history of the disease for the RAD50 truncation mutation. In parallel, we also screened the same cases for the MRE11 and NBS1 mutations, as the disease segregation of these two alterations could not be confirmed from the pedigrees. As none of the tested alterations was observed in the studied cohort, it may suggest that these changes are mainly associated with patients with a family history of cancer. As RAD50 687delT is expected to cause truncation of most of the protein product, we evaluated whether these mutant transcripts are eliminated by nonsense mediated decay. Lymphoblast cell lines generated from blood specimens of three different mutation positive individuals (fig 3: family 5, case 828; family 6, cases 505 and 061) was used to produce mRNA for the cDNA synthesis. The results from sequencing analysis unambiguously demonstrated that in all studied cases both the wild type and the truncation allele were still present in the tested mRNA pool (data not shown). MRE11 and RAD50 LOH analysis In order to test the nature of the observed mutations, we investigated whether the tumorigenesis thought to be Figure 2 Families exhibiting the novel (A) MRE11 Arg305Trp and (B) NBS1 Leu150Phe germline mutations. Black circles (females) and boxes (males) indicate individuals with cancer. The grey filling indicates a pre-malignant lesion. The age at diagnosis, when known, is marked after the disease. The case number of individuals tested for a mutation is shown. Figure 3 Families with the novel RAD50 687delT germline mutation. Black circles (females) and boxes (males) indicate individuals with cancer. Age at diagnosis, when known, is indicated after the malignancy. The case number of individuals analysed and with parallel genetic changes is shown. For a disease free mutation carrier, the age at monitoring is shown after the case number. 4 Electronic letter www.jmedgenet.com associated with the novel germline mutations involved loss of the corresponding wild type allele. LOH analysis was performed on the available tumour specimens from four different patients (cases 322, 828, 474, and 505; figs 2 and 3) displaying either the MRE11 or RAD50 mutation, but no LOH was observed (data not shown). Owing to the lack of suitable tumour material the corresponding LOH analysis for the NBS1 mutation could not be performed. DISCUSSION The Mre11 complex is an integral part of the proper DNA damage response and repair of double strand breaks, essential for the wellbeing of any living organism. Failure in these highly conserved processes leads to the loss of genomic integrity and consequently to an elevated risk of cancer.14 The cellular functions of the Mre11 complex are indispensable, as seen in mice, where major deficiencies in any of the mre11, nbs1, or rad50 genes cause early embryonic lethality and only hypomorphic mutations appear to be viable.34–36 In humans, only a small number of hypomorphic germline mutations affecting the MRE11 and NBS1 genes are currently known, mainly associating with the severe and cancer prone phenotypes of ATLD and NBS, respectively.21 27 37 Disease related germline mutations in RAD50 have not to our knowledge previously been reported. In the current study, we have examined the prevalence of NBS1, MRE11, and RAD50 germline mutations in index cases with breast or ovarian cancer from 151 families. Of the novel heterozygous sequence alterations observed, Arg305Trp (MRE11), Leu150Phe (NBS1), and 687delT (RAD50) potentially relate to cancer susceptibility (table 1). As the LOH analysis of tumours from MRE11 or RAD50 mutation carriers showed retention of the wild type alleles, it may indicate that these germline mutations are sufficient to cause the observed adverse phenotypic effect. However, silencing of the wild type allele by epigenetic mechanisms (for example by promoter region hypermethylation) cannot be excluded. The association of the novel MRE11 and NBS1 missense mutations with hereditary cancer susceptibility is supported by their absence in both the controls and the unselected breast cancer cases. In addition, both amino acid substitutions occur in highly conserved protein domains. In MRE11, the Arg305Trp substitution takes place at the very end of the first of two major protein domains,20 replacing the conserved positively charged hydrophilic arginine with the neutral and highly hydrophobic tryptophan. Because of its position in the protein, it is possible that the Arg305Trp substitution may affect the structural organisation of the functionally active L shaped groove located between domains 1 and 2,38 and perhaps lower the affinity of MRE11 for DNA binding. The situation could be similar to that indicated for the Asn117Ser mutation in ATLD;27 despite the fact that the change occurs relatively remote from the active site, it has been suggested to disturb the macromolecular interaction of MRE11.20 In NBS1, the Leu150Phe substitution involves the highly conserved BRCT domain (fig 4), where the conserved leucine is replaced with the biochemically non-similar phenylalanine. Interestingly, another amino acid substitution affecting a conserved residue in the BRCT domain, Ile171Val, has previously been associated with cancer (acute lymphoblastic leukaemia).39 Additionally, recent site specific mutagenesis experiments by Cerosaletti & Concannon40 indicate that changes at the conserved residues GlyGly136–137GluGlu or Tyr176Ala are capable of disrupting Mre11 complex nuclear focus formation and blocking NBS1 phosphorylation after irradiation, emphasising the functional importance of this region and especially the role of the conserved amino acids in the BRCT domain. However, in order to obtain proof of the proposed deleterious nature of the MRE11 and NBS1 missense mutations discovered in the current study, the biochemical effects of these alterations has to be demonstrated by functional assays. Furthermore, additional case–control series are needed to confirm their possible role of as cancer susceptibility alleles. Of the novel germline alterations observed in this study, RAD50 687delT was probably the most dramatic as it is predicted to cause about 80% shortening of both isoforms of the protein. The truncated protein would have retained only the N-terminal portion of the sequence necessary for binding of the RAD50 dimer to MRE11, but would have lost the corresponding and equally important C-terminal site.23 Based on studies of lymphoblast cell lines we know that the deletion allele is expressed at the mRNA level, but it is currently unknown if it actually gives rise to a truncated protein, how stable this product would be, and whether it has effects on the function of the wild type protein, perhaps by disturbing the functionally important dimerisation of RAD50. If a shortened protein actually is produced, but is unstable, the effect is likely to be that of a null mutation. This dosage variation due to haploinsufficiency may have a direct Figure 4 Novel amino acid substitutions in MRE11 and NBS1, and comparison of sequence homology between Homo sapiens (hs), Rattus norvegicus (rn), Mus musculus (mm) and Gallus gallus (gg). The sites showing changed residues are boxed. The amino acids conserved in all four species are drawn in blue, biochemically similar residues in green, and dissimilar ones in black. The sequence shown for MRE11 extends to its last two phosphodiesterase motifs (PDE IV and V)18 38 and for NBS1 corresponds to its BRCT domain.22 In NBS1 the two sites (136–137 and 176) marked with an asterisk (*) have been found to be critical for Mre11 complex nuclear focus formation and phosphorylation of NBS1 after irradiation,40 and the one marked with an arrowhead (171.) has been associated with acute lymphoid leukaemia.39 Electronic letter 5 www.jmedgenet.com influence on the activities in DNA damage recognition and response pathways, and could modify the cancer risk by increasing genetic instability or by altering the effect of the normal DNA damage response.41 However, the possibility that RAD50 687delT is a hypomorphic mutation, analogous to the common Slavic NBS founder mutation NBS1 657del5,42 also cannot be excluded. Interestingly, according to the recent study by Gorski et al, the frequency of the NBS1 657del5 mutation in the population controls they used is comparable with that of RAD50 687delT in our study (both 0.6%). The RAD50 687delT mutation was observed altogether in seven individuals, two of whom showed breast and two ovarian cancers, all at relatively young age (41–50 years old). However, in family 6 (fig 3 right) the situation is less straightforward, as at least one of the affected individuals (case 505) showed parallel involvement of the previously identified BRCA1 3745delT truncation mutation.4 In addition, there were several unaffected carriers of both mutations, but the three daughters of case 505 found positive for either the RAD50 or the BRCA1 mutation, or both, may not yet contribute to the understanding of the disease penetrance owing to their relatively young age. Although at present it is difficult to determine the exact role of these two germline mutations in this family, there are several scenarios worth exploring. It is possible that both RAD50 and BRCA1 act independently on the resultant phenotype. On the other hand, co-occurrence of RAD50 687delT could also strengthen the effect of the BRCA1 mutation. Interestingly, a recent comparative genome hybridisation targeted linkage analysis has revealed a possible BRCA1 penetrance modifier locus on 5q in a location close to RAD50. 43 44 Besides penetrance modification, the RAD50 mutation may also in some way influence the tissue specificity of a BRCA1 associated malignancy. However, it is also possible that in the genetic background provided by this particular family the mutation in RAD50 does not significantly increase the risk of cancer in BRCA1 3745delT carriers. The surprisingly high frequency of RAD50 687delT mutation carriers among the healthy population controls and its absence from the breast cancer cases without familial disease background suggests that RAD50 687delT may be a new low penetrance susceptibility allele associated particularly with hereditary predisposition to cancer. At present it is unknown whether the RAD50 687delT allele exists outside Finland, as it is a well known fact that many disease related gene defects can be either unique or highly enriched in the Finnish population, owing to founder effects, isolation, and genetic drift.45 However, the situation could also be similar to that observed for the cancer predisposing CHK2 1100delC allele, which is enriched in southern Finland,31 46 but also occurs at significant frequencies in many other European populations (for example, the UK6 ). In order to obtain more information about the geographical occurrence and the overall relevance of the RAD50 687delT mutation, the intention is to study additional breast, ovarian, and other hereditary cancer materials along with suitable control specimens, initially from the Nordic countries. In addition, functional studies are needed to reveal the exact details about the molecular consequences of the RAD50 687delT mutation. In conclusion, this is the first time that germline mutations in RAD50 have been linked to genetic diseases in humans. The RAD50 deletion allele reported in this study provides evidence that deleterious germline mutations in the third member of the Mre11 complex exist, are viable, and appear to associate with predisposition to cancer. Additional investigations will be needed to confirm and follow up these preliminary leads on the role of Mre11 complex gene mutations in hereditary susceptibility to cancer.  DNA double-strand breaks (DSBs) are among the most cytotoxic form of DNA damage and can lead to chromosomal aberrations, disruption of genomic integrity and cancer (Carney et al. 1998; Game 1993; Luo et al. 1999; Stewart et al. 1999; Zdzienicka 1996). Although DSBs can arise from many sources, including genotoxic agents, ionizing radiation and enzymatic activities, the predominant cause of DSBs in proliferating cells is errors in DNA replication (Costanzo et al. 2001; Gellert 2002; Haber 1998; Keeney 2001; Kuzminov 2001; Ward 1988). Double-strand breaks can be repaired by two major DSB repair pathways, homologous recombination (HR) and non-homologous end joining (NHEJ). In HR, the DSBs are resected into 3′ single-strand (ss) DNA tails. These tails pair with the homologous DNA segment of the sister chromatid by formation of a D-loop structure and repair proceeds by DNA synthesis, essentially without loss of genetic information [see, e.g., Haber (2000), Thompson and Schild (2002) and references therein]. In NHEJ, the two broken ends are aligned, may be processed, and directly re-ligated. This process is potentially mutagenic [see, e.g., Lewis and Resnick (2000), Lieber et al. (2003) and references therein].  The multisubunit MRE11/RAD50/NBS1 complex (termed Mre11 complex) is a key player in the cellular response to DSBs and is involved in virtually all aspects of DNA end metabolism, including DSB detection (Mirzoeva and Petrini 2001), DSB processing (Connelly and Leach 1996; Lewis et al. 2004; Mickelson and Wiberg 1981; Moreau et al. 1999; Trujillo et al. 1998), homologous recombination and meiosis (Alani et al. 1990; Bressan et al. 1999; Tauchi et al. 2002a,b; Yamaguchi-Iwai et al. 1999), NHEJ (Milne et al. 1996; Moore and Haber 1996; Tsukamoto et al. 1996; Udayakumar et al. 2003), telomere maintenance (Boulton and Jackson 1998; Chamankhah et al. 2000; Kironmai and Muniyappa 1997; Lundblad 2002; Teng et al. 2000; Zhu et al. 2000) and the DSB-activated cell cycle checkpoint response (Carney et al. 1998; D’Amours and Jackson 2001; Zhao et al. 2000). This kaleidoscope of functions is explained by findings that the Mre11 complex combines several enzymatic and non-enzymatic activities, including DNA end processing, DNA cohesion and, recently identified, activation of the damage checkpoint kinase ATM (Table 1). Table 1 Activities of the Mre11 complex Activity Remarks References DNA processing  3′-5′ Exonuclease ATP dependent Connelly et al. (1999), Hopfner et al. (2001), Paull and Gellert (1998), Trujillo et al. (1998)  5′-3′ Exonuclease Only bacteriophage T4 gene products 46/47 Bleuit et al. (2001), Kreuzer (2000)  ssDNA endonuclease ATP independent Connelly et al. (1997), Furuse et al. (1998), Hopfner et al. (2000a,b), Trujillo et al. (1998)  Hairpin opening ATP dependent Paull and Gellert (1998), Trujillo and Sung (2001), Connelly et al. (1998), Trujillo and Sung (2001)  Protein bound DNA end Escherichia coli SbcCD, ATP dependent Connelly et al. (2003)  DNA unwinding ATP dependent Paull and Gellert (1999)  DNA end or sister chromatid joining Probably ATP dependent Bressan et al. (1999), Chen et al. (2001), de Jager et al. (2001), Gonzalez-Barrera et al. (2003), Hartsuiker et al. (2003), Hopfner et al. (2002) Coactivator function  DNA ligase IV  \\t Chen et al. (2001), Paull and Gellert (2000)  Ataxia telangiectasia mutated (ATM)  \\t Lee and Paull (2004)  Werner syndrome helicase (WRN)  \\t Cheng et al. (2004) The Mre11 complex The Mre11 complex consists of the large coiled-coil ATP-binding cassette (ABC) ATPase Rad50, the nuclease Mre11 and the checkpoint mediator NBS1 (Fig. 1a). Genetic studies have indicated that the Mre11 complex is required for genomic stability (Alani et al. 1990; Luo et al. 1999; Yamaguchi-Iwai et al. 1999). In higher eukaryotes, null mutations in components of the Mre11 complex are lethal (Gorski et al. 2004; Luo et al. 1999; Yamaguchi-Iwai et al. 1999; Zhu et al. 2001), while hypomorphic mutations in NBS1 and MRE11 cause the genome instability/cancer syndromes Nijmegen breakage syndrome (NBS) and Ataxia telangiectasia-like disease (ATLD), respectively (Carney et al. 1998; Matsuura et al. 1998; Stewart et al. 1999; Varon et al. 1998). A hypomorphic mutation in RAD50, originally identified as a separation of function mutation (Rad50S) in yeast (Alani et al. 1990), results in partial embryonic lethality and cancer susceptibility in mice (Bender et al. 2002). Thus even minor disturbances of Mre11 complex activity have profound effects on the integrity of the genome, underlining the central role of this complex in the maintenance of the genetic information. Fig. 1a–c Functional domains and architecture of the Mre11 complex proteins. a The domain structure and architecture of MRE11 and Rad50 are highly conserved in different organisms. RAD50 contains a bipartite ATP binding cassette (ABC) ATPase domain (orange) consisting of an N-terminal (ABC-N) and a C-terminal (ABC-C) segment. ABC-N harbors the Walker A motif (A); ABC-C harbors the Walker B (B) and signature motif (S). The ABC segments are the ends of a long heptad repeat sequence that folds into an antiparallel coiled-coil domain. Adjacent to the ABC segments are MRE11 binding sites. The center of the heptad repeats contains a highly conserved Cys–X–X–Cys (CXXC, green) motif that forms a metal-binding coiled-coil dimerization motif. MRE11 possesses a conserved phosphoesterase domain at the N-terminus (blue) that harbors the nuclease active site and a potential interaction loop with NBS1 (in eukaryotes). The central and C-terminal regions contain additional DNA binding sites (grey) as well as a binding site for RAD50 and for MRE11–MRE11 dimerization (indicated). Ataxia telangiectasia-like disease (ATLD) mutation sites are indicated (asterisks). NBS1 contains a forkhead associated (FHA) and breast cancer C-terminus (BRCT) domain at its N-terminus (magenta). These domains mediate interaction with phosphoproteins. The C-terminal region of NBS1 contains a putative interaction site with MRE11. The central region of NBS1/Xrs2 (mutation domain) contains the mutations that are found in Nijmegen breakage syndrome (including non-sense mutations). The checkpoint phosphorylation site Ser343 is indicated. b Structural model of the MRE112/RAD502/NBS12 complex architecture, using the color code of a. MRE11 binds to the coiled-coil of RAD50, adjacent to the RAD50 ABC domains. The RAD50 antiparallel coiled-coil domains protrude from the ABC domains. The apex of the coiled-coil domains contains the zinc hook, implicated in metal-mediated intramolecular or intermolecular joining of Mre11 complexes via the RAD50 coiled-coils. The precise location of NBS1 (modeled in this illustration) has not been revealed. c Hypothetical model for the interaction of the Mre11 complex with an ATM dimer and DNA. In principle, a single MRE112/RAD502/NBS12 complex could link the sister chromatids to both DNA ends utilizing combined RAD50, MRE11 and NBS1 DNA-binding sites (Chen et al. 2001). Alternatively, the Mre11 complex binds a single end, and tethering of ends proceeds by intermolecular joining of hook structures (de Jager et al. (2001); see text for details and Fig. 2 for the alternative model) Mre11 and Rad50 are well conserved in evolution (Aravind et al. 1999; Blinov et al. 1989; Gorbalenya and Koonin 1990) and form an ATP-stimulated 3′-5′ exonuclease that can act on most types of DNA ends and hairpins (see below). The remarkable evolutionary conservation suggests that the Mre11/Rad50 complex is an ancient DSB detection/repair factor that probably co-evolved with recombination processes. The third subunit of the eukaryotic Mre11 complex, NBS1 (or its functional homolog Xrs2 in Saccharomyces cerevisiae) is much less sequence conserved and only found in eukaryotes (for recent reviews see, e.g., D’Amours and Jackson 2002; Tauchi et al. 2002a,b). NBS1/Xrs2 probably joined MRE11/RAD50 in eukaryotes with the evolution of cell cycle checkpoints and adoption of new roles of the Mre11 complex in monitoring genome integrity.  Besides these three core subunits, the Mre11 complex has multiple functional and physical interactions with additional repair and checkpoint factors that link the core Mre11 complex activities to a variety of cellular processes (Table 2). Table 2 Functional or physical interactions of the Mre11 complex with other DNA repair/checkpoint proteins. (ATM, Ataxia telangiectasia mutated; BLM, Bloom syndrome; WRN, Werner Syndrome) Partner Via Remarks References Eukaryotes  ATM Multiple sites within MRE11 complex ATM phosphorylates NBS1, Mre11 complex stimulates ATM, colocalization Andegeko et al. (2001), Lee and Paull (2004), Lim et al. (2000), Uziel et al. (2003), Zhao et al. (2000)  WRN NBS1 Colocalization and association Cheng et al. (2004)  BLM BASC BLM required for relocalization of Mre11 complex, colocalization Franchitto and Pichierri (2002), Wang et al. (2000)  FANCD2 NBS1 Required for focus formation Nakanishi et al. (2002), Pichierri et al. (2002)  E2F1 NBS1 Colocalization Maser et al. (2001)  MCD1 NBS1(?) Colocalization Goldberg et al. (2003), Stewart et al. (2003), Xu and Stern (2003)  BRCA1 BASC Colocalization Wang et al. (2000)  SMC1 NBS1(?) NBS1 is required for ATM dependent phosphorylation of SMC1 Kim et al. (2002), Yazdi et al. (2002)  H2AX-γ NBS1 Direct binding Kobayashi et al. (2002) Archaea  NurA Operon in archaea 5′–3′ Exonuclease Constantinesco et al. (2002)  MlaA/HerA Operon in archaea Secondary structure binding ATPase/helicase Constantinesco et al. (2004), Manzan et al. (2004) Bacteriophage T4  UsvY  \\t Rad52 functional homolog Bleuit et al. (2001), Kreuzer (2000) From structure to structural functions It is clear from the nature of DSBs that all DSB repair pathways require a structural and an enzymatic component. The structural activity captures and aligns the broken ends with each other or with the homologous stretch on the sister chromatid, while the enzymatic activity processes the ends to enable subsequent repair steps. The Mre11 complex has the remarkable feature of combining these requirements in a single multisubunit complex. Electron microscopy and scanning force microscopy studies of the Mre11/Rad50 complex revealed a striking architecture that immediately suggested mechanisms for structural functions of the complex (Anderson et al. 2001; Chen et al. 2001; Connelly et al. 1998; de Jager et al. 2001; Hopfner et al. 2001). The Mre11 complex appears as a bipolar structure with a global head and two long tails (Fig. 1b). The head consists of two RAD50 ABC ATPase domains, along with an MRE11 dimer, and harbors the ATP-stimulated nuclease activity. Although not directly imaged yet, mutagenesis and biophysical data suggest that NBS1/Xrs2 binds to the MRE112:RAD502 head, possibly forming a hexameric or even higher order RAD502/MRE112/NBS12 complex in eukaryotes (Lee et al. 2003).  The most striking feature of the Mre11 complex are the up to 60 nm long antiparallel coiled-coil tails that protrudes from the RAD50 ABC ATPase domains (Anderson et al. 2001; Chen et al. 2001; Connelly et al. 1998; de Jager et al. 2001; Hopfner et al. 2001; van Noort et al. 2003). At the apex of these coiled-coil domains are intriguing hook structures, which can form interlocked hook/zinc/hook bridges and join two RAD50 coiled-coils (Fig. 1b,c) (de Jager et al. 2001; Hopfner et al. 2002). Although the precise nature of the supermolecular architecture of these joined coiled-coils remains to be established, a possible model could be the formation of large proteinaceous loops in homologous recombination. These loops, formed by a single MRE112/RAD502 heterotetramer, are closed at one end by the global head, and at the other end by the interlocked hooks (Fig. 1c). This structure is of remarkable similarity to the ring model proposed for cohesin, which contains the RAD50-related structural maintenance of chromosome (SMC) ATPases (Haering et al. 2002). It will be quite interesting to test whether DNA is trapped within coiled-coil structures of RAD50, analogously to the model proposed for the related cohesin complex (Gruber et al. 2003). A different model, consistent with experimental data, proposes intermolecular joining of two MRE112/RAD502 complexes. In this model, two head domains (each formed by an MRE112/RAD502 complex) bind to opposing DNA ends or sister chromatids and are linked to each other by zinc hook-mediated coiled-coil links (de Jager et al. 2001; Hopfner et al. 2002). Both models provide a compelling mechanism for an architectural function of the Mre11 complex in recombinational sister chromatid interaction, as well as for direct joining of two DNA ends in NHEJ.  The structure and function of NBS1/Xrs2 is the least understood among the Mre11 complex proteins. NBS1 modulates the ATP-stimulated nuclease activity of human MRE11/RAD50 and is required for the hairpin-processing activity of the complex (Lee et al. 2003; Paull and Gellert 1999). Recently, it was found that Xrs2 binds to DNA secondary structures and helps to target MRE11/RAD50 to DSBs (Trujillo et al. 2003). These data suggest that NBS1/Xrs2 could be directly involved in DNA damage detection functions of the Mre11 complex. In addition, NBS1/Xrs2 probably mediates protein–protein interactions of the Mre11 complex with other repair/signaling factors at the DSB (Table 2). NBS1 has an N-terminal forkhead-associated (FHA) domain followed by a breast cancer-associated C-terminus (BRCT) domain (Fig. 1a). Both domains typically bind to phosphopeptides (Durocher et al. 1999; Manke et al. 2003; Yu et al. 2003). Since phosphorylation is a key modification in the DNA damage checkpoint response, NBS1 could be involved in targeting the Mre11 complex to damaged sites, in attracting other repair factors to damaged sites or in interacting with damage-induced chromatin markers at DSBs (Bradbury and Jackson 2003; Kobayashi et al. 2002).  Since all three eukaryotic components (MRE11, RAD50, NBS1) are implicated in direct DNA binding, the close interaction of MRE11, NBS1 and the RAD50 ABC domain suggest that all three components probably form a single DNA recognition surface that detects and processes breaks/secondary structures in both repair and checkpoint activation. A model for the eukaryotic complex consistent with current data is shown in Fig. 1b.  Biochemistry Biochemically, the Mre11 complex is an ATP-stimulated nuclease that acts endonucleolytically and exonucleolytically on ssDNA and hairpins, and exonucleolytically on various types of dsDNA ends (Arthur et al. 2004; Connelly et al. 1997; Hopfner et al. 2000a,b; Ohta et al. 1998; Paull and Gellert 1998; Trujillo et al. 1998). With the exception of gene products 46/47, the bacteriophage T4 homologs of MRE11/RAD50, all MRE11/RAD50 proteins assayed degrade dsDNA in the 3′→5′ direction (Connelly et al. 1999; Hopfner et al. 2001; Paull and Gellert 1998; Trujillo et al. 1998). This proofreading-like nuclease activity argues that the Mre11 complex is not directly involved in the formation of 3′-tails in recombination but is likely involved in the processing of DNA secondary structures and misfolded DNA ends (Connelly et al. 2003; Farah et al. 2002; Lobachev et al. 2002; Paull and Gellert 1999; Trujillo and Sung 2001). Recent findings suggest that the Escherichia coli homologs of the Mre11 complex, SbcCD, can directly remove covalently attached proteins from DNA ends (Connelly et al. 2003), an activity that could explain the function of the Mre11 complex in the removal of covalently attached Spo11 from meiotic DSBs (Borde et al. 2004; Keeney 2001; Moreau et al. 1999). The processing of protein-bound DNA ends requires ATP and proceeds by introducing a DSB close to the end. It appears that the Mre11 complex can process virtually all types of DNA ends and hairpins in an ATP-stimulated manner, resulting in a clean 3′-OH for subsequent DNA synthesis steps in HR.  The role of RAD50 ATP hydrolysis in the 3′–5′ nuclease activity is not fully established. Recent results suggest that ATP controls mainly the binding of DNA to the Mre11 complex (Connelly et al. 2003; de Jager et al. 2002; Paull and Gellert 1999; Trujillo and Sung 2001). Such an activity has interesting similarities to the role of ATP in cohesin, in which ATP hydrolysis was found to be important for association of DNA with the SMC protein rings (Arumugam et al. 2003; Weitzer et al. 2003). In addition to DNA end recognition, ATP-induced structural switches in the RAD50 ABC domains might partially melt DNA, to facilitate DNA end processing by MRE11 (Paull and Gellert 1999). Since the Mre11 complex presumably has two MRE11 subunits, ATP-stimulated DNA melting by RAD50 could expose both strands for simultaneous processing by the MRE11 dimer (Paull and Gellert 2000). Such an activity could account for the observed generation of a DSB close to the end in the case of protein-bound DNA ends.  Checkpoint activation and ATM stimulation The probably least understood function of the Mre11 complex, until recently, is its role in DNA damage checkpoint activation. The DSB-mediated checkpoint response is triggered mainly by the ATM kinase. ATM activation initiates a signaling cascade that leads to cell cycle arrest, DNA repair and/or apoptosis (see, e.g., Shiloh 2003 and references therein). ATM phosphorylates a variety of targets in response to DSBs. ATM substrates include proteins involved in G1, intra-S-phase and G2/M checkpoints like p53 (Banin et al. 1998; Canman et al. 1998), MDM2 (Maya et al. 2001), CHK2 (Chehab et al. 2000; Matsuoka et al. 2000), NBS1 (Lim et al. 2000; Zhao et al. 2000), BRCA1 (Xu et al. 2002), FANCD2 (Taniguchi et al. 2002), SMC1 (Kim et al. 2002; Yazdi et al. 2002), hRAD17 (Bao et al. 2001), 53BP1 (Rappold et al. 2001) and MDC1 (Stewart et al. 2003).  ATM autoactivation and the phosphorylation of downstream substrates is dependent on the Mre11 complex (Carson et al. 2003; Horejsi et al. 2004; Lee and Paull 2004; Uziel et al. 2003). NBS1 itself is a substrate of ATM. NBS1 phosphorylation at Ser343 is required for ATM dependent phosphorylation of substrates like CHK2 (Buscemi et al. 2001; Lee and Paull 2004), histone H2AX (Lee and Paull 2004), SMC1 (Horejsi et al. 2004; Yazdi et al. 2002), and FANCD2 (Nakanishi et al. 2002). In contrast, phosphorylation of p53 requires the presence of the Mre11 complex but not NBS1 phosphophorylation (Lee and Paull 2004).  The linkage of the Mre11 complex to the ATM-dependent DNA damage response was suggested early on, based on the similarity of AT, ATLD and NBS, and a variety of genetic and cellular studies [see, e.g., Petrini (2000) and references therein]. These studies revealed that the eukaryotic Mre11 complex participates in the activation of ATM, possibly in conjunction with other checkpoint factors such as the mediator of damage checkpoint 1 (MDC1) (Carson et al. 2003; D’Amours and Jackson 2001; Gatei et al. 2000; Goldberg et al. 2003; Lou et al. 2003; Nakada et al. 2003; Stewart et al. 1999; Stewart et al. 2003; Usui et al. 2001).  ATM is a large 360 kDa kinase that resides in the nucleus as an inactive dimer or higher order multimer (Bakkenist and Kastan 2003). Upon introduction of DSBs and possibly other changes in the chromatin structure, ATM autophosphorylates itself at Ser1981. This phosphorylation leads to ATM dimer dissociation into monomers (Bakkenist and Kastan 2003). The mechanistic basis for the activation of ATM by the Mre11 complex has begun to be revealed. Exciting recent results suggest that the Mre11 complex physically interacts with ATM, resulting in direct stimulation of ATM kinase activity (Lee and Paull 2004). Although mechanistic details need to be revealed, the Mre11 complex-dependent stimulation of ATM seems to have several subcomponents. These include multiple interaction sites, substrate recruitment and allosteric stimulation of the kinase activity (Lee and Paull 2004). NBS1 could sequester specific substrates in a process that is dependent on phosphorylation of Ser343 by ATM. For instance, NBS1 could attract several substrates by phosphopeptide interactions, e.g., via the CHK2 FHA domain and thus lead to activation of a specific subset of targets, such as CHK2, SMC1 and others. Alternatively, the Ser343 phosphorylation on NBS1 might induce a structural switch that could alter the binding surface properties of the MRE11/ATM complex. A hypothetical model for the interaction of ATM with the Mre11 complex is shown in Fig. 1c.  In future studies, it will be important specifically to address the biochemical and structural activities of the Mre11 complex in ATM activation. For instance, the surface cluster of the yeast Rad50S mutations is located in the vicinity of the putative DNA-binding region of RAD50. The mechanistic role of these mutations has not been revealed, but their specific location on the three-dimensional structure suggests that they are involved in macromolecular interaction. One effect of the Rad50S mutations could be the modulation of the DNA recognition properties of the RAD50/MRE11 DNA processing machinery, but it will be interesting to see whether they have any effect in direct activation of ATM (or the yeast homologue Tel1). However, the RAD50S mutation in mouse apparently does not have an effect on cell cycle regulation, so it is unlikely to influence ATM activation (Bender et al. 2002). In addition, the role of ATP hydrolysis by RAD50 in the activation of ATM needs to be addressed. Since ATP binding to RAD50 modulates its interaction with DNA, the ATP-driven conformational changes in RAD50 are probably an important stereochemical event in the damage recognition functions of the Mre11 complex (Alani et al. 1990; Connelly et al. 2003; de Jager et al. 2002; Hopfner et al. 2000a,b). In addition, ATP-driven structural changes in the RAD50 ATPase domain could directly participate in ATM activation, such as to facilitate structural changes in ATM leading to autophosphorylation.  Likewise, it will be interesting to see whether DNA degradation by MRE11 contributes to the activation of ATM, since the nuclease activity of the Mre11 complex was found to be involved in the DNA damage checkpoint (D’Amours and Jackson 2001). In contrast, in vitro stimulation of the ATM kinase by the Mre11 complex does not depend on DNA (Lee and Paull 2004). Therefore DSBs might not be a direct co-factor for the activation of ATM in the DNA damage checkpoint response. Possibly, the Mre11 complex has a twofold role in this aspect. One function is probably the direct activation of ATM. The other function, which has to be further addressed, could be the sequestering of ATM to sites of DNA damage.  Novel nuclease/helicase alliances One of the remaining questions regarding the Mre11 complex activity is the role of the nuclease activity of the Mre11 complex in mitotic recombination. Early results suggested only a minor role of nuclease activity in mitotic homologous recombination (Bressan et al. 1999; Moreau et al. 1999). However, recent data indicate that some nuclease-deficient Mre11 mutations can interfere with recombination in mitotic cells (Lewis et al. 2004).  Considering both the directionality and magnitude of the Mre11 complex nuclease, it is unlikely that the Mre11 complex has a direct role in generating the long, invading 3′-tails in homologous recombination. This suggests functional and/or physical contributions from other nucleases. In support of this idea, recent findings indicate that the Mre11 complex functionally or structurally interacts with nucleases that could potentially be involved in 5′–3′ processing in recombination or replication fork restart. In archaea, Mre11 and Rad50 are polycistronically co-expressed with two proteins, a 5′–3′ exonuclease (termed NurA), and an ATPase (termed MlaA or HerA) that binds to secondary structures and has DNA unwinding activity (Constantinesco et al. 2002, 2004; Manzan et al. 2004). Combined, the nuclease and DNA unwinding activities could resect DNA ends or process stalled replication forks to prime strand invasion and D-loop formation for subsequent recombinational repair or replication fork restart.  Likewise, in eukaryotes the Mre11 complex functionally and physically interacts with RecQ helicases such as the Bloom Syndrome and Werner Syndrome Proteins (BLM, WRN). Werner syndrome (WS) and NBS share several phenotypic features, including genomic instability, cancer predisposition, sensitivity to genotoxic agents, telomere abnormalities and cell cycle defects [see, e.g., Pichierri and Franchitto (2004) and references therein]. The protein mutated in WS, the RecQ helicase family protein WRN, binds to NBS1 (Cheng et al. 2004), while the Mre11 complex and BLM colocalize in a large repair complex (Wang et al. 2000). WRN contains a 5′–3′/3′–5′ nuclease domain and a 3′–5′ helicase domain and has been suggested to be involved in the repair of stalled replication forks (Constantinou et al. 2000; Rodriguez-Lopez et al. 2002). Association of WRN with the Mre11 complex stimulates WRN helicase activity, suggesting that both proteins could form a functional unit at stalled replication forks (Cheng et al. 2004).  Although WRN and NurA/MlaA have no detectable sequence homology, both factors evidently share a combination of nuclease, helicase and secondary structure binding activity. In the case of WRN, these activities reside on a single polypeptide chain while in the case of NurA/MlaA, they reside on two polypeptide chains. Although it remains to be shown whether NurA and MlaA directly cooperate with Mre11/Rad50, it is interesting to speculate that these activities specifically link the Mre11 complex to rescue processes arising at stalled replication forks.  Mre11 complex and replication A key role of the Mre11 complex in the repair of DSBs directly at replication forks is suggested by several lines of evidence. Experiments with cell-free Xenopus egg extracts showed that in the absence of the Mre11 complex, DNA replication resulted in many DSBs (Costanzo et al. 2001). In E. coli, SbcCD is required to process hairpins that arise during lagging strand synthesis and at inverted repeats (Connelly and Leach 1996). Likewise, propagation of inverted repeats in eukaryotes requires the Mre11 complex (Lobachev et al. 2002). Gene products 46/47, the bacteriophage T4 homologs of Mre11/Rad50, are required for recombination-induced replication (Bleuit et al. 2001; George et al. 2001; Kreuzer 2000). In eukaryotes, NBS1 binds to E2F transcription factors near replication origins (Maser et al. 2001) and the Mre11 complex co-localizes with replication forks and is loaded onto nascent sister chromatids in S-phase (Mirzoeva and Petrini 2003).  Figure 2 unifies our current understanding of functional and structural activities of the Mre11 complex into a replication fork-associated model. Replication over DNA damage or nicks could specifically lead to a detached nascent sister chromatid. The Mre11 complex could be loaded onto chromosomes before replication or onto nascent sister chromatids during replication. The precise association of the Mre11 complex with DNA needs to be established, but a possible model is the entrapment of a sister chromatid into the coiled-coil ring. Such a loose but stable association could allow translocation by diffusion along DNA. In the case of a break, the Mre11 complex could specifically locate to the DNA end, joining the end to the sister chromatid. Conformational changes or simply liberation of previously bound Mre11 complex from broken sister chromatids might lead to activation of ATM. Phosphorylation of NBS1 by ATM will promote to an active S-phase checkpoint complex that will phosphorylate specific downstream targets. If replication resumes, a DSB will be created. A second role of the Mre11 complex could be cohesion of both DNA ends with each other, similar to a role in NHEJ. This might be done by simultaneous binding of both ends to a single head (Fig. 1c; Chen et al. 2001), or by binding of two heads to each end (de Jager et al. 2001), followed by intramolecular joining of coiled-coil ends by the hook structures (Fig. 2). The precise mechanism has to be revealed in future studies but recent results suggest that such a clustering of the Mre11 complex and ATM at DNA ends occurs in vitro (Costanzo et al. 2004). Fig. 2 Hypothetical model for replication-linked architectural and S-phase checkpoint functions of the Mre11 complex. Replication over nicked DNA (or resected hairpins) by the replisome (yellow) will lead to a detached nascent chromatid. In this model the replisome is drawn stationary while the DNA moves (arrow). Mre11 complexes (orange heads, green dimerized hooks, grey coiled-coils), loaded onto nascent chromatids in the wake of replication forks, could specifically capture the free DNA end and degrade any secondary structures. In addition DNA-bound or liberated Mre11 complex might activate ATM (light blue). ATM will phosphorylate NBS1 and, after assembly of foci, the ATM/Mre11 complexes could phosphorylate S-phase checkpoint targets. Repair proceeds by recruitment of additional factors such as WRN helicase/nuclease or other nucleases. Resection of ends will lead to strand exchange and D-loop formation, followed by repair synthesis Conclusions and outlook The Mre11 complex has at least three critical functions. First, the complex is a primary DNA damage recognition factor at DSBs, single DNA ends arising from stalled replication forks and otherwise misfolded or obstructed DNA. Using ATP, the Mre11 complex most likely binds to these ends/structures and induces a signal that directly assists the activation of the ATM kinase. Second, the ATP-stimulated nuclease activity is probably involved to process these misfolded/blocked DNA ends or to degrade secondary structures/hairpins that arise during replication. This function might be important to create a “common intermediate” that funnels heterogeneity arising at a break in a processed, clean DNA end. The newly formed end could be further processed by specialized nucleases in homologous recombination and replication fork rescue, such as WRN, BLM or the prokaryotic NurA. Finally, the perhaps most important function of the Mre11 complex is to link DNA ends with each other or to sister chromatids, possibly in the wake of replication forks. Such a linker function could also be a preeminent step prior to NHEJ, meiotic recombination or formation of T-loops and D-loops at telomeres.  Although we begin to understand the central role of the Mre11 complex in genome stability, the molecular mechanism still needs to be clarified. In particular, the ATP-driven mechanism of association with DNA substrates and the process and timing of loading onto and unloading from sister chromatids is still fairly unresolved. One of the key questions, however, will be the molecular events leading to ATM kinase activation by the Mre11 complex. Exciting steps have been taken into this direction and we look forward to the combined efforts of cell biology, genetics, biochemistry and structural biology in unravelling the complex activities of the Mre11 complex in genome maintenance over the next few years.'"
      ]
     },
     "execution_count": 256,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "del_or_ins = data_all[(data_all['del_or_ins_var']==1)]\n",
    "del_or_ins.Text[del_or_ins.index[2]]\n",
    "deletions.Text[deletions.index[3]]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 257,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Class</th>\n",
       "      <th>Gene</th>\n",
       "      <th>ID</th>\n",
       "      <th>Text</th>\n",
       "      <th>Variation</th>\n",
       "      <th>gene_fusion_var</th>\n",
       "      <th>Deletion_var</th>\n",
       "      <th>del_or_ins_var</th>\n",
       "      <th>Fusion_var</th>\n",
       "      <th>Amplification_var</th>\n",
       "      <th>Truncation_var</th>\n",
       "      <th>exon_var</th>\n",
       "      <th>frameshift_var</th>\n",
       "      <th>dup_var</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>1.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>7</td>\n",
       "      <td>CBL is a negative regulator of activated recep...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>1.0</td>\n",
       "      <td>DICER1</td>\n",
       "      <td>43</td>\n",
       "      <td>Mesenchymal cell populations contribute to mic...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>1.0</td>\n",
       "      <td>SHQ1</td>\n",
       "      <td>70</td>\n",
       "      <td>Assembly of H/ACA RNPs in yeast is aided by at...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>1.0</td>\n",
       "      <td>RAD50</td>\n",
       "      <td>74</td>\n",
       "      <td>Mre11, Rad50, and Nbs1 function in a protein c...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101</th>\n",
       "      <td>1.0</td>\n",
       "      <td>TGFBR2</td>\n",
       "      <td>101</td>\n",
       "      <td>Melanoma patients treated with oncogenic BRAF ...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>107</th>\n",
       "      <td>1.0</td>\n",
       "      <td>MSH6</td>\n",
       "      <td>107</td>\n",
       "      <td>A hypermutated subtype of advanced prostate ca...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>115</th>\n",
       "      <td>1.0</td>\n",
       "      <td>KMT2D</td>\n",
       "      <td>115</td>\n",
       "      <td>Kabuki syndrome (KS) is a multiple congenital ...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>119</th>\n",
       "      <td>1.0</td>\n",
       "      <td>PBRM1</td>\n",
       "      <td>119</td>\n",
       "      <td>Screening for tumor suppressor genes in breast...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>137</th>\n",
       "      <td>4.0</td>\n",
       "      <td>LATS2</td>\n",
       "      <td>137</td>\n",
       "      <td>Malignant mesothelioma (MM) is an aggressive n...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>286</th>\n",
       "      <td>1.0</td>\n",
       "      <td>ARID2</td>\n",
       "      <td>286</td>\n",
       "      <td>In eukaryotic cells, DNA is packaged into chro...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>291</th>\n",
       "      <td>1.0</td>\n",
       "      <td>ERRFI1</td>\n",
       "      <td>291</td>\n",
       "      <td>The growing number of recently identified nega...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>343</th>\n",
       "      <td>1.0</td>\n",
       "      <td>CDH1</td>\n",
       "      <td>343</td>\n",
       "      <td>A 41-year-old man with no familial history of ...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>359</th>\n",
       "      <td>1.0</td>\n",
       "      <td>EP300</td>\n",
       "      <td>359</td>\n",
       "      <td>The EP300 protein is a histone acetyltransfera...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>533</th>\n",
       "      <td>1.0</td>\n",
       "      <td>TP53BP1</td>\n",
       "      <td>533</td>\n",
       "      <td>p53 binding protein 1 (53BP1) was identified i...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>539</th>\n",
       "      <td>4.0</td>\n",
       "      <td>SMAD2</td>\n",
       "      <td>539</td>\n",
       "      <td>Transforming growth factor- (TGF-) plays a com...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>559</th>\n",
       "      <td>1.0</td>\n",
       "      <td>SMAD3</td>\n",
       "      <td>559</td>\n",
       "      <td>Members of the transforming growth factor-beta...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>597</th>\n",
       "      <td>4.0</td>\n",
       "      <td>SMAD4</td>\n",
       "      <td>597</td>\n",
       "      <td>TGFβ signaling Smads (Smad2, 3, and 4) were su...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>629</th>\n",
       "      <td>1.0</td>\n",
       "      <td>FBXW7</td>\n",
       "      <td>629</td>\n",
       "      <td>Tumor suppressors with widespread impact on ca...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>635</th>\n",
       "      <td>1.0</td>\n",
       "      <td>CDKN1A</td>\n",
       "      <td>635</td>\n",
       "      <td>The cell cycle inhibitor p21CDKN1A induces cel...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>638</th>\n",
       "      <td>4.0</td>\n",
       "      <td>CDKN1B</td>\n",
       "      <td>638</td>\n",
       "      <td>Introduction A series of cyclin and cyclin-dep...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>664</th>\n",
       "      <td>4.0</td>\n",
       "      <td>CDKN2A</td>\n",
       "      <td>664</td>\n",
       "      <td>Determining how somatic copy-number alteration...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>695</th>\n",
       "      <td>4.0</td>\n",
       "      <td>CDKN2B</td>\n",
       "      <td>695</td>\n",
       "      <td>Introduction  Since the first proposal of a po...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>701</th>\n",
       "      <td>1.0</td>\n",
       "      <td>CDKN2C</td>\n",
       "      <td>701</td>\n",
       "      <td>Deletions of chromosome 1 have been described ...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>702</th>\n",
       "      <td>1.0</td>\n",
       "      <td>ASXL1</td>\n",
       "      <td>702</td>\n",
       "      <td>ASXL1 is mutated/deleted with high frequencies...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>827</th>\n",
       "      <td>1.0</td>\n",
       "      <td>BRIP1</td>\n",
       "      <td>827</td>\n",
       "      <td>BRIP1 is a DNA helicase that directly interact...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>832</th>\n",
       "      <td>1.0</td>\n",
       "      <td>ERCC4</td>\n",
       "      <td>832</td>\n",
       "      <td>Xeroderma pigmentosum (XP) complementation gro...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>955</th>\n",
       "      <td>1.0</td>\n",
       "      <td>RBM10</td>\n",
       "      <td>955</td>\n",
       "      <td>RBM10 is an RNA binding protein and alternativ...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1009</th>\n",
       "      <td>4.0</td>\n",
       "      <td>IKZF1</td>\n",
       "      <td>1009</td>\n",
       "      <td>Interrogate the impact of IKZF1 deletion on th...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1036</th>\n",
       "      <td>1.0</td>\n",
       "      <td>TSC2</td>\n",
       "      <td>1036</td>\n",
       "      <td>Abstract Large TSC gene rearrangements are not...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1073</th>\n",
       "      <td>1.0</td>\n",
       "      <td>FOXA1</td>\n",
       "      <td>1073</td>\n",
       "      <td>The forkhead box (Fox) superfamily of transcri...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2368</th>\n",
       "      <td>4.0</td>\n",
       "      <td>PTPN11</td>\n",
       "      <td>2368</td>\n",
       "      <td>PTPN11, which encodes tyrosine phosphatase Shp...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2408</th>\n",
       "      <td>4.0</td>\n",
       "      <td>NF1</td>\n",
       "      <td>2408</td>\n",
       "      <td>Germline heterozygous alterations of the tumor...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2416</th>\n",
       "      <td>4.0</td>\n",
       "      <td>PTPRD</td>\n",
       "      <td>2416</td>\n",
       "      <td>Tyrosine phosphorylation plays a critical role...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2439</th>\n",
       "      <td>1.0</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>2439</td>\n",
       "      <td>BRCA1 and BRCA2 mutations are estimated to be ...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2840</th>\n",
       "      <td>4.0</td>\n",
       "      <td>BRCA2</td>\n",
       "      <td>2840</td>\n",
       "      <td>BRCA1 and BRCA2 mutations are estimated to be ...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2950</th>\n",
       "      <td>1.0</td>\n",
       "      <td>BCL10</td>\n",
       "      <td>2950</td>\n",
       "      <td>Mucosa-associated lymphoid tissue (MALT) lymph...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2958</th>\n",
       "      <td>1.0</td>\n",
       "      <td>MGA</td>\n",
       "      <td>2958</td>\n",
       "      <td>The basic‐helix‐loop‐helix‐leucine zipper (bHL...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3064</th>\n",
       "      <td>1.0</td>\n",
       "      <td>MED12</td>\n",
       "      <td>3064</td>\n",
       "      <td>Inhibitors of the ALK and EGF receptor tyrosin...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3106</th>\n",
       "      <td>1.0</td>\n",
       "      <td>DNMT3A</td>\n",
       "      <td>3106</td>\n",
       "      <td>The genetic alterations responsible for an adv...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3110</th>\n",
       "      <td>1.0</td>\n",
       "      <td>RAD21</td>\n",
       "      <td>3110</td>\n",
       "      <td>ASXL1 is frequently mutated in a spectrum of m...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3161</th>\n",
       "      <td>1.0</td>\n",
       "      <td>NPM1</td>\n",
       "      <td>3161</td>\n",
       "      <td>Nucleophosmin (NPM1) gene has been heavily imp...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3201</th>\n",
       "      <td>1.0</td>\n",
       "      <td>RASA1</td>\n",
       "      <td>3201</td>\n",
       "      <td>Phospholipase C gamma 1 (PLC gamma 1) and p21r...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3203</th>\n",
       "      <td>1.0</td>\n",
       "      <td>BCL2L11</td>\n",
       "      <td>3203</td>\n",
       "      <td>Tyrosine kinase inhibitors (TKIs) elicit high...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3213</th>\n",
       "      <td>1.0</td>\n",
       "      <td>RB1</td>\n",
       "      <td>3213</td>\n",
       "      <td>Quantitative multiplex PCR and genomic real-ti...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3239</th>\n",
       "      <td>6.0</td>\n",
       "      <td>DUSP4</td>\n",
       "      <td>3239</td>\n",
       "      <td>Mitogen-activated protein kinase phosphatase-2...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3253</th>\n",
       "      <td>1.0</td>\n",
       "      <td>CASP8</td>\n",
       "      <td>3253</td>\n",
       "      <td>Evidence exists that alterations of the genes ...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4143</th>\n",
       "      <td>NaN</td>\n",
       "      <td>PMS2</td>\n",
       "      <td>822</td>\n",
       "      <td>Lynch syndrome (LS; also known as hereditary n...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4505</th>\n",
       "      <td>NaN</td>\n",
       "      <td>SOX9</td>\n",
       "      <td>1184</td>\n",
       "      <td>In humans, mutations in SOX9 result in a skele...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4594</th>\n",
       "      <td>NaN</td>\n",
       "      <td>LATS1</td>\n",
       "      <td>1273</td>\n",
       "      <td>The lats gene has been identified as a tumour ...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5573</th>\n",
       "      <td>NaN</td>\n",
       "      <td>TSC1</td>\n",
       "      <td>2252</td>\n",
       "      <td>Tuberous sclerosis complex (TSC) is an autosom...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5611</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NOTCH1</td>\n",
       "      <td>2290</td>\n",
       "      <td>Analysis of the chronic lymphocytic leukemia (...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5953</th>\n",
       "      <td>NaN</td>\n",
       "      <td>SMARCA4</td>\n",
       "      <td>2632</td>\n",
       "      <td>Eukaryotic organisms package DNA into chromati...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6634</th>\n",
       "      <td>NaN</td>\n",
       "      <td>TP53</td>\n",
       "      <td>3313</td>\n",
       "      <td>Mutations disabling the TP53 tumour suppressor...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6837</th>\n",
       "      <td>NaN</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>3516</td>\n",
       "      <td>Mamo et al. (2011) identified ARID1A as a cand...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7153</th>\n",
       "      <td>NaN</td>\n",
       "      <td>ASXL2</td>\n",
       "      <td>3832</td>\n",
       "      <td>Acute myeloid leukemia (AML) with t(8;21) (q22...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7834</th>\n",
       "      <td>NaN</td>\n",
       "      <td>SOX17</td>\n",
       "      <td>4513</td>\n",
       "      <td>SUMMARY Fetal stem cells differ phenotypicall...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8217</th>\n",
       "      <td>NaN</td>\n",
       "      <td>INPP4B</td>\n",
       "      <td>4896</td>\n",
       "      <td>Inositol polyphosphate 4-phosphatase B (INPP4B...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8255</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NF2</td>\n",
       "      <td>4934</td>\n",
       "      <td>The FERM domain protein Merlin, encoded by the...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8637</th>\n",
       "      <td>NaN</td>\n",
       "      <td>GATA3</td>\n",
       "      <td>5316</td>\n",
       "      <td>Terminal deletions of chromosome 10p result in...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8662</th>\n",
       "      <td>NaN</td>\n",
       "      <td>HNF1A</td>\n",
       "      <td>5341</td>\n",
       "      <td>Mutations in the hepatocyte nuclear factor (HN...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>88 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Class     Gene    ID                                               Text  \\\n",
       "7       1.0      CBL     7  CBL is a negative regulator of activated recep...   \n",
       "43      1.0   DICER1    43  Mesenchymal cell populations contribute to mic...   \n",
       "70      1.0     SHQ1    70  Assembly of H/ACA RNPs in yeast is aided by at...   \n",
       "74      1.0    RAD50    74  Mre11, Rad50, and Nbs1 function in a protein c...   \n",
       "101     1.0   TGFBR2   101  Melanoma patients treated with oncogenic BRAF ...   \n",
       "107     1.0     MSH6   107  A hypermutated subtype of advanced prostate ca...   \n",
       "115     1.0    KMT2D   115  Kabuki syndrome (KS) is a multiple congenital ...   \n",
       "119     1.0    PBRM1   119  Screening for tumor suppressor genes in breast...   \n",
       "137     4.0    LATS2   137  Malignant mesothelioma (MM) is an aggressive n...   \n",
       "286     1.0    ARID2   286  In eukaryotic cells, DNA is packaged into chro...   \n",
       "291     1.0   ERRFI1   291  The growing number of recently identified nega...   \n",
       "343     1.0     CDH1   343  A 41-year-old man with no familial history of ...   \n",
       "359     1.0    EP300   359  The EP300 protein is a histone acetyltransfera...   \n",
       "533     1.0  TP53BP1   533  p53 binding protein 1 (53BP1) was identified i...   \n",
       "539     4.0    SMAD2   539  Transforming growth factor- (TGF-) plays a com...   \n",
       "559     1.0    SMAD3   559  Members of the transforming growth factor-beta...   \n",
       "597     4.0    SMAD4   597  TGFβ signaling Smads (Smad2, 3, and 4) were su...   \n",
       "629     1.0    FBXW7   629  Tumor suppressors with widespread impact on ca...   \n",
       "635     1.0   CDKN1A   635  The cell cycle inhibitor p21CDKN1A induces cel...   \n",
       "638     4.0   CDKN1B   638  Introduction A series of cyclin and cyclin-dep...   \n",
       "664     4.0   CDKN2A   664  Determining how somatic copy-number alteration...   \n",
       "695     4.0   CDKN2B   695  Introduction  Since the first proposal of a po...   \n",
       "701     1.0   CDKN2C   701  Deletions of chromosome 1 have been described ...   \n",
       "702     1.0    ASXL1   702  ASXL1 is mutated/deleted with high frequencies...   \n",
       "827     1.0    BRIP1   827  BRIP1 is a DNA helicase that directly interact...   \n",
       "832     1.0    ERCC4   832  Xeroderma pigmentosum (XP) complementation gro...   \n",
       "955     1.0    RBM10   955  RBM10 is an RNA binding protein and alternativ...   \n",
       "1009    4.0    IKZF1  1009  Interrogate the impact of IKZF1 deletion on th...   \n",
       "1036    1.0     TSC2  1036  Abstract Large TSC gene rearrangements are not...   \n",
       "1073    1.0    FOXA1  1073  The forkhead box (Fox) superfamily of transcri...   \n",
       "...     ...      ...   ...                                                ...   \n",
       "2368    4.0   PTPN11  2368  PTPN11, which encodes tyrosine phosphatase Shp...   \n",
       "2408    4.0      NF1  2408  Germline heterozygous alterations of the tumor...   \n",
       "2416    4.0    PTPRD  2416  Tyrosine phosphorylation plays a critical role...   \n",
       "2439    1.0    BRCA1  2439  BRCA1 and BRCA2 mutations are estimated to be ...   \n",
       "2840    4.0    BRCA2  2840  BRCA1 and BRCA2 mutations are estimated to be ...   \n",
       "2950    1.0    BCL10  2950  Mucosa-associated lymphoid tissue (MALT) lymph...   \n",
       "2958    1.0      MGA  2958  The basic‐helix‐loop‐helix‐leucine zipper (bHL...   \n",
       "3064    1.0    MED12  3064  Inhibitors of the ALK and EGF receptor tyrosin...   \n",
       "3106    1.0   DNMT3A  3106  The genetic alterations responsible for an adv...   \n",
       "3110    1.0    RAD21  3110  ASXL1 is frequently mutated in a spectrum of m...   \n",
       "3161    1.0     NPM1  3161  Nucleophosmin (NPM1) gene has been heavily imp...   \n",
       "3201    1.0    RASA1  3201  Phospholipase C gamma 1 (PLC gamma 1) and p21r...   \n",
       "3203    1.0  BCL2L11  3203   Tyrosine kinase inhibitors (TKIs) elicit high...   \n",
       "3213    1.0      RB1  3213  Quantitative multiplex PCR and genomic real-ti...   \n",
       "3239    6.0    DUSP4  3239  Mitogen-activated protein kinase phosphatase-2...   \n",
       "3253    1.0    CASP8  3253  Evidence exists that alterations of the genes ...   \n",
       "4143    NaN     PMS2   822  Lynch syndrome (LS; also known as hereditary n...   \n",
       "4505    NaN     SOX9  1184  In humans, mutations in SOX9 result in a skele...   \n",
       "4594    NaN    LATS1  1273  The lats gene has been identified as a tumour ...   \n",
       "5573    NaN     TSC1  2252  Tuberous sclerosis complex (TSC) is an autosom...   \n",
       "5611    NaN   NOTCH1  2290  Analysis of the chronic lymphocytic leukemia (...   \n",
       "5953    NaN  SMARCA4  2632  Eukaryotic organisms package DNA into chromati...   \n",
       "6634    NaN     TP53  3313  Mutations disabling the TP53 tumour suppressor...   \n",
       "6837    NaN   ARID1A  3516  Mamo et al. (2011) identified ARID1A as a cand...   \n",
       "7153    NaN    ASXL2  3832  Acute myeloid leukemia (AML) with t(8;21) (q22...   \n",
       "7834    NaN    SOX17  4513   SUMMARY Fetal stem cells differ phenotypicall...   \n",
       "8217    NaN   INPP4B  4896  Inositol polyphosphate 4-phosphatase B (INPP4B...   \n",
       "8255    NaN      NF2  4934  The FERM domain protein Merlin, encoded by the...   \n",
       "8637    NaN    GATA3  5316  Terminal deletions of chromosome 10p result in...   \n",
       "8662    NaN    HNF1A  5341  Mutations in the hepatocyte nuclear factor (HN...   \n",
       "\n",
       "     Variation  gene_fusion_var  Deletion_var  del_or_ins_var  Fusion_var  \\\n",
       "7     Deletion                0             1               0           0   \n",
       "43    Deletion                0             1               0           0   \n",
       "70    Deletion                0             1               0           0   \n",
       "74    Deletion                0             1               0           0   \n",
       "101   Deletion                0             1               0           0   \n",
       "107   Deletion                0             1               0           0   \n",
       "115   Deletion                0             1               0           0   \n",
       "119   Deletion                0             1               0           0   \n",
       "137   Deletion                0             1               0           0   \n",
       "286   Deletion                0             1               0           0   \n",
       "291   Deletion                0             1               0           0   \n",
       "343   Deletion                0             1               0           0   \n",
       "359   Deletion                0             1               0           0   \n",
       "533   Deletion                0             1               0           0   \n",
       "539   Deletion                0             1               0           0   \n",
       "559   Deletion                0             1               0           0   \n",
       "597   Deletion                0             1               0           0   \n",
       "629   Deletion                0             1               0           0   \n",
       "635   Deletion                0             1               0           0   \n",
       "638   Deletion                0             1               0           0   \n",
       "664   Deletion                0             1               0           0   \n",
       "695   Deletion                0             1               0           0   \n",
       "701   Deletion                0             1               0           0   \n",
       "702   Deletion                0             1               0           0   \n",
       "827   Deletion                0             1               0           0   \n",
       "832   Deletion                0             1               0           0   \n",
       "955   Deletion                0             1               0           0   \n",
       "1009  Deletion                0             1               0           0   \n",
       "1036  Deletion                0             1               0           0   \n",
       "1073  Deletion                0             1               0           0   \n",
       "...        ...              ...           ...             ...         ...   \n",
       "2368  Deletion                0             1               0           0   \n",
       "2408  Deletion                0             1               0           0   \n",
       "2416  Deletion                0             1               0           0   \n",
       "2439  Deletion                0             1               0           0   \n",
       "2840  Deletion                0             1               0           0   \n",
       "2950  Deletion                0             1               0           0   \n",
       "2958  Deletion                0             1               0           0   \n",
       "3064  Deletion                0             1               0           0   \n",
       "3106  Deletion                0             1               0           0   \n",
       "3110  Deletion                0             1               0           0   \n",
       "3161  Deletion                0             1               0           0   \n",
       "3201  Deletion                0             1               0           0   \n",
       "3203  Deletion                0             1               0           0   \n",
       "3213  Deletion                0             1               0           0   \n",
       "3239  Deletion                0             1               0           0   \n",
       "3253  Deletion                0             1               0           0   \n",
       "4143  Deletion                0             1               0           0   \n",
       "4505  Deletion                0             1               0           0   \n",
       "4594  Deletion                0             1               0           0   \n",
       "5573  Deletion                0             1               0           0   \n",
       "5611  Deletion                0             1               0           0   \n",
       "5953  Deletion                0             1               0           0   \n",
       "6634  Deletion                0             1               0           0   \n",
       "6837  Deletion                0             1               0           0   \n",
       "7153  Deletion                0             1               0           0   \n",
       "7834  Deletion                0             1               0           0   \n",
       "8217  Deletion                0             1               0           0   \n",
       "8255  Deletion                0             1               0           0   \n",
       "8637  Deletion                0             1               0           0   \n",
       "8662  Deletion                0             1               0           0   \n",
       "\n",
       "      Amplification_var  Truncation_var  exon_var  frameshift_var  dup_var  \n",
       "7                     0               0         0               0        0  \n",
       "43                    0               0         0               0        0  \n",
       "70                    0               0         0               0        0  \n",
       "74                    0               0         0               0        0  \n",
       "101                   0               0         0               0        0  \n",
       "107                   0               0         0               0        0  \n",
       "115                   0               0         0               0        0  \n",
       "119                   0               0         0               0        0  \n",
       "137                   0               0         0               0        0  \n",
       "286                   0               0         0               0        0  \n",
       "291                   0               0         0               0        0  \n",
       "343                   0               0         0               0        0  \n",
       "359                   0               0         0               0        0  \n",
       "533                   0               0         0               0        0  \n",
       "539                   0               0         0               0        0  \n",
       "559                   0               0         0               0        0  \n",
       "597                   0               0         0               0        0  \n",
       "629                   0               0         0               0        0  \n",
       "635                   0               0         0               0        0  \n",
       "638                   0               0         0               0        0  \n",
       "664                   0               0         0               0        0  \n",
       "695                   0               0         0               0        0  \n",
       "701                   0               0         0               0        0  \n",
       "702                   0               0         0               0        0  \n",
       "827                   0               0         0               0        0  \n",
       "832                   0               0         0               0        0  \n",
       "955                   0               0         0               0        0  \n",
       "1009                  0               0         0               0        0  \n",
       "1036                  0               0         0               0        0  \n",
       "1073                  0               0         0               0        0  \n",
       "...                 ...             ...       ...             ...      ...  \n",
       "2368                  0               0         0               0        0  \n",
       "2408                  0               0         0               0        0  \n",
       "2416                  0               0         0               0        0  \n",
       "2439                  0               0         0               0        0  \n",
       "2840                  0               0         0               0        0  \n",
       "2950                  0               0         0               0        0  \n",
       "2958                  0               0         0               0        0  \n",
       "3064                  0               0         0               0        0  \n",
       "3106                  0               0         0               0        0  \n",
       "3110                  0               0         0               0        0  \n",
       "3161                  0               0         0               0        0  \n",
       "3201                  0               0         0               0        0  \n",
       "3203                  0               0         0               0        0  \n",
       "3213                  0               0         0               0        0  \n",
       "3239                  0               0         0               0        0  \n",
       "3253                  0               0         0               0        0  \n",
       "4143                  0               0         0               0        0  \n",
       "4505                  0               0         0               0        0  \n",
       "4594                  0               0         0               0        0  \n",
       "5573                  0               0         0               0        0  \n",
       "5611                  0               0         0               0        0  \n",
       "5953                  0               0         0               0        0  \n",
       "6634                  0               0         0               0        0  \n",
       "6837                  0               0         0               0        0  \n",
       "7153                  0               0         0               0        0  \n",
       "7834                  0               0         0               0        0  \n",
       "8217                  0               0         0               0        0  \n",
       "8255                  0               0         0               0        0  \n",
       "8637                  0               0         0               0        0  \n",
       "8662                  0               0         0               0        0  \n",
       "\n",
       "[88 rows x 14 columns]"
      ]
     },
     "execution_count": 257,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "deletions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "164     Oncogenic mutations in the epidermal growth fa...\n",
       "268     Oncogenic mutations in the epidermal growth fa...\n",
       "279     For 40% of pediatric T-ALL cases, underlying o...\n",
       "280     For 40% of pediatric T-ALL cases, underlying o...\n",
       "283     For 40% of pediatric T-ALL cases, underlying o...\n",
       "289     NUT midline carcinoma (NMC) is a genetically d...\n",
       "300      Abstract The discovery of recurrent gene fusi...\n",
       "301     Recurrent gene fusions involving oncogenic ETS...\n",
       "302     Although common in hematologic and mesenchymal...\n",
       "319     Echinoderm microtubule associated protein like...\n",
       "321     Oncogenic gene fusions involving the 3’ region...\n",
       "322     Chromosomal rearrangements of the gene encodin...\n",
       "326     We report the results of whole genome and tran...\n",
       "327     Oncogenic gene fusions involving the 3’ region...\n",
       "329     Chromosomal rearrangements of the gene encodin...\n",
       "331     Echinoderm microtubule associated protein like...\n",
       "332     Echinoderm microtubule associated protein like...\n",
       "333     Inflammatory myofibroblastic tumor (IMT) is a ...\n",
       "335     Kinases activated by gene fusions represent an...\n",
       "336     Oncogenic gene fusions involving the 3’ region...\n",
       "338     All cancers harbor molecular alterations in th...\n",
       "340     The transmembrane proto-oncogene receptor tyro...\n",
       "356     We report on the fusion of the monocytic leuke...\n",
       "358     The p300 protein shows a striking homology wit...\n",
       "784     Purpose: To identify druggable oncogenic fusio...\n",
       "837     The BCR-ABL oncogene is generated by the Phila...\n",
       "864     TMPRSS2-ERG gene fusions occur in about 50% of...\n",
       "865     Sarcomas constitute some of the most aggressiv...\n",
       "866     Thet(16;21)(pll;q22)translocationIsa recurrent...\n",
       "876     Gene rearrangement in the form of an intrageni...\n",
       "                              ...                        \n",
       "3219     Abstract Glioblastoma multiforme (GBM) is the...\n",
       "3220     Abstract Glioblastoma multiforme (GBM) is the...\n",
       "3223    In this study, we utilized retroviral transfer...\n",
       "3224    We identified novel gene fusions in patients w...\n",
       "3225    A human oncogene formed by the fusion of trunc...\n",
       "3226     Abstract Glioblastoma multiforme (GBM) is the...\n",
       "3228    Oncogenic rearrangements of the NTRK1 gene (al...\n",
       "3229    Spitzoid neoplasms are a group of melanocytic ...\n",
       "3230    We identified novel gene fusions in patients w...\n",
       "3234    The congenital fibrosarcoma t(12;15)(p13;q25) ...\n",
       "3235    Pediatric high-grade glioma (HGG) is a devasta...\n",
       "3237    Pediatric high-grade glioma (HGG) is a devasta...\n",
       "3250    The identification of specific chromosome abno...\n",
       "3257    Rearrangements of the proto-oncogene RET are n...\n",
       "3284    A considerable proportion of LADCs, the most c...\n",
       "3311    The AML1/EVI-1 chimeric gene is generated by t...\n",
       "3312    Balanced chromosomal translocations are freque...\n",
       "3315    The RUNX gene family includes three evolutiona...\n",
       "3703    Ets-related gene (ERG), which encodes a member...\n",
       "4296    Conventional epithelioid hemangioendotheliomas...\n",
       "4356    A fusion of the PDGFRA and FIP1L1 genes can be...\n",
       "5375    Inflammatory myofibroblastic tumor (IMT) is a ...\n",
       "5462    Echinoderm microtubule associated protein like...\n",
       "6027    Introduction  Chromosomal translocations invol...\n",
       "6183    Chromosomal translocations involving the plate...\n",
       "6453     Abstract Through a prospective clinical seque...\n",
       "6517    The ETS related gene, ERG, is one of 20 or mor...\n",
       "8307     Purpose: Oncogenic fusions consisting of fibr...\n",
       "8587    Chronic myeloproliferative disorders with rear...\n",
       "8772    High-grade serous ovarian cancer (HGSC) is the...\n",
       "Name: Text, Length: 160, dtype: object"
      ]
     },
     "execution_count": 96,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_all.Text[data_all['gene_fusion_var']==1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "NLTK_nosub = [sent_tokenize(no_sub_yes.Text[i]) for i in no_sub_yes.index]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "collapsed": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "no_sub_yes,no_sub_no=custom_find(no_sub)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Class</th>\n",
       "      <th>Gene</th>\n",
       "      <th>ID</th>\n",
       "      <th>Text</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Fusion_var</th>\n",
       "      <th>gene_fusion_var</th>\n",
       "      <th>Deletion_var</th>\n",
       "      <th>del_or_ins_var</th>\n",
       "      <th>Amplification_var</th>\n",
       "      <th>Truncation_var</th>\n",
       "      <th>exon_var</th>\n",
       "      <th>frameshift_var</th>\n",
       "      <th>dup_var</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1.0</td>\n",
       "      <td>FAM58A</td>\n",
       "      <td>0</td>\n",
       "      <td>Cyclin-dependent kinases (CDKs) regulate a var...</td>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>1</td>\n",
       "      <td>Abstract Background  Non-small cell lung canc...</td>\n",
       "      <td>W802*</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>2</td>\n",
       "      <td>Abstract Background  Non-small cell lung canc...</td>\n",
       "      <td>Q249E</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>3</td>\n",
       "      <td>Recent evidence has demonstrated that acquired...</td>\n",
       "      <td>N454D</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>4</td>\n",
       "      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n",
       "      <td>L399V</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>4.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>5</td>\n",
       "      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n",
       "      <td>V391I</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>5.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>6</td>\n",
       "      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n",
       "      <td>V430M</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>1.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>7</td>\n",
       "      <td>CBL is a negative regulator of activated recep...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>4.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>8</td>\n",
       "      <td>Abstract Juvenile myelomonocytic leukemia (JM...</td>\n",
       "      <td>Y371H</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>4.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>9</td>\n",
       "      <td>Abstract Juvenile myelomonocytic leukemia (JM...</td>\n",
       "      <td>C384R</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>4.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>10</td>\n",
       "      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n",
       "      <td>P395A</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>4.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>11</td>\n",
       "      <td>Noonan syndrome is an autosomal dominant conge...</td>\n",
       "      <td>K382E</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>4.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>12</td>\n",
       "      <td>Noonan syndrome is an autosomal dominant conge...</td>\n",
       "      <td>R420Q</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>4.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>13</td>\n",
       "      <td>Noonan syndrome is an autosomal dominant conge...</td>\n",
       "      <td>C381A</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>5.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>14</td>\n",
       "      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n",
       "      <td>P428L</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>4.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>15</td>\n",
       "      <td>Noonan syndrome is an autosomal dominant conge...</td>\n",
       "      <td>D390Y</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>1.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>16</td>\n",
       "      <td>To determine if residual cylindrical refractiv...</td>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>4.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>17</td>\n",
       "      <td>Acquired uniparental disomy (aUPD) is a common...</td>\n",
       "      <td>Q367P</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>5.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>18</td>\n",
       "      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n",
       "      <td>M374V</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>4.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>19</td>\n",
       "      <td>Acquired uniparental disomy (aUPD) is a common...</td>\n",
       "      <td>Y371S</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>6.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>20</td>\n",
       "      <td>Abstract Background  Non-small cell lung canc...</td>\n",
       "      <td>H94Y</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>4.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>21</td>\n",
       "      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n",
       "      <td>C396R</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>4.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>22</td>\n",
       "      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n",
       "      <td>G375P</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>4.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>23</td>\n",
       "      <td>Recent evidence has demonstrated that acquired...</td>\n",
       "      <td>S376F</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>4.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>24</td>\n",
       "      <td>Recent evidence has demonstrated that acquired...</td>\n",
       "      <td>P417A</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>4.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>25</td>\n",
       "      <td>Recent evidence has demonstrated that acquired...</td>\n",
       "      <td>H398Y</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>4.0</td>\n",
       "      <td>SHOC2</td>\n",
       "      <td>26</td>\n",
       "      <td>Abstract N-myristoylation is a common form of...</td>\n",
       "      <td>S2G</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>4.0</td>\n",
       "      <td>TERT</td>\n",
       "      <td>27</td>\n",
       "      <td>Heterozygous mutations in the telomerase compo...</td>\n",
       "      <td>Y846C</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>7.0</td>\n",
       "      <td>TERT</td>\n",
       "      <td>28</td>\n",
       "      <td>Sequencing studies have identified many recurr...</td>\n",
       "      <td>C228T</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>4.0</td>\n",
       "      <td>TERT</td>\n",
       "      <td>29</td>\n",
       "      <td>Heterozygous mutations in the telomerase compo...</td>\n",
       "      <td>H412Y</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8959</th>\n",
       "      <td>NaN</td>\n",
       "      <td>VSX1</td>\n",
       "      <td>5638</td>\n",
       "      <td>Using a DNA microarray approach to screen for ...</td>\n",
       "      <td>R166W</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8960</th>\n",
       "      <td>NaN</td>\n",
       "      <td>MTM1</td>\n",
       "      <td>5639</td>\n",
       "      <td>The tumor suppressor protein p53 is inactivate...</td>\n",
       "      <td>E157K</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8961</th>\n",
       "      <td>NaN</td>\n",
       "      <td>D2HGDH</td>\n",
       "      <td>5640</td>\n",
       "      <td>Mutational analysis of oncogenes is critical f...</td>\n",
       "      <td>V444A</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8962</th>\n",
       "      <td>NaN</td>\n",
       "      <td>DMD</td>\n",
       "      <td>5641</td>\n",
       "      <td>The serine/threonine protein kinase encoded by...</td>\n",
       "      <td>Y231N</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8963</th>\n",
       "      <td>NaN</td>\n",
       "      <td>ANKH</td>\n",
       "      <td>5642</td>\n",
       "      <td>The common participation of oncogenic KRAS pro...</td>\n",
       "      <td>G389R</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8964</th>\n",
       "      <td>NaN</td>\n",
       "      <td>DCX</td>\n",
       "      <td>5643</td>\n",
       "      <td>EZH2 (enhancer of zeste homolog 2) is a critic...</td>\n",
       "      <td>R59L</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8965</th>\n",
       "      <td>NaN</td>\n",
       "      <td>ADSL</td>\n",
       "      <td>5644</td>\n",
       "      <td>Mutations in the metabolic enzymes isocitrate ...</td>\n",
       "      <td>R190Q</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8966</th>\n",
       "      <td>NaN</td>\n",
       "      <td>HSD17B3</td>\n",
       "      <td>5645</td>\n",
       "      <td>Pancreatic carcinomas with acinar differentiat...</td>\n",
       "      <td>A56T</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8967</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CD96</td>\n",
       "      <td>5646</td>\n",
       "      <td>β-Catenin-mediated signaling can be constituti...</td>\n",
       "      <td>T280M</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8968</th>\n",
       "      <td>NaN</td>\n",
       "      <td>HPS3</td>\n",
       "      <td>5647</td>\n",
       "      <td>Summary The genetic abnormalities underlying h...</td>\n",
       "      <td>R397W</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8969</th>\n",
       "      <td>NaN</td>\n",
       "      <td>FKTN</td>\n",
       "      <td>5648</td>\n",
       "      <td>Lung cancer is the leading cause of cancer-rel...</td>\n",
       "      <td>R179T</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8970</th>\n",
       "      <td>NaN</td>\n",
       "      <td>DARS2</td>\n",
       "      <td>5649</td>\n",
       "      <td>Summary During the past decade, the treatment ...</td>\n",
       "      <td>L613F</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8971</th>\n",
       "      <td>NaN</td>\n",
       "      <td>TP53</td>\n",
       "      <td>5650</td>\n",
       "      <td>The transcription factor and tumor suppressor ...</td>\n",
       "      <td>G245C</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8972</th>\n",
       "      <td>NaN</td>\n",
       "      <td>ALG12</td>\n",
       "      <td>5651</td>\n",
       "      <td>Protein tyrosine phosphatase receptor type D (...</td>\n",
       "      <td>T67M</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8973</th>\n",
       "      <td>NaN</td>\n",
       "      <td>ACOX1</td>\n",
       "      <td>5652</td>\n",
       "      <td>Sensitizing activating mutations in the tyrosi...</td>\n",
       "      <td>Q309R</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8974</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CLDN19</td>\n",
       "      <td>5653</td>\n",
       "      <td>Structural rearrangements of chromosome 10 are...</td>\n",
       "      <td>Q57E</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8975</th>\n",
       "      <td>NaN</td>\n",
       "      <td>MLH3</td>\n",
       "      <td>5654</td>\n",
       "      <td>Introduction  Production of fertile gametes is...</td>\n",
       "      <td>N499S</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8976</th>\n",
       "      <td>NaN</td>\n",
       "      <td>GJB1</td>\n",
       "      <td>5655</td>\n",
       "      <td>Checkpoint kinase 2 (CHEK2, Chk2) emerges as a...</td>\n",
       "      <td>F235C</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8977</th>\n",
       "      <td>NaN</td>\n",
       "      <td>LRP5</td>\n",
       "      <td>5656</td>\n",
       "      <td>Introduction The telomere sequences at the chr...</td>\n",
       "      <td>G171V</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8978</th>\n",
       "      <td>NaN</td>\n",
       "      <td>TGFBI</td>\n",
       "      <td>5657</td>\n",
       "      <td>There are an estimated 1 million cases of brea...</td>\n",
       "      <td>R124S</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8979</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CYP2C9</td>\n",
       "      <td>5658</td>\n",
       "      <td>As of the occurring responders expression Tabl...</td>\n",
       "      <td>I359L</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8980</th>\n",
       "      <td>NaN</td>\n",
       "      <td>HSD17B3</td>\n",
       "      <td>5659</td>\n",
       "      <td>Background &amp; Aims: Inherited deleterious muta...</td>\n",
       "      <td>M235V</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8981</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CAV3</td>\n",
       "      <td>5660</td>\n",
       "      <td>Glioblastoma multiforme (GBM) is a lethal brai...</td>\n",
       "      <td>A46V</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8982</th>\n",
       "      <td>NaN</td>\n",
       "      <td>ABHD5</td>\n",
       "      <td>5661</td>\n",
       "      <td>Diffuse large B cell lymphoma (DLBCL) is a com...</td>\n",
       "      <td>E260K</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8983</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NR3C2</td>\n",
       "      <td>5662</td>\n",
       "      <td>Figure largeDownload these has S818L clones ch...</td>\n",
       "      <td>S818L</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8984</th>\n",
       "      <td>NaN</td>\n",
       "      <td>SLC46A1</td>\n",
       "      <td>5663</td>\n",
       "      <td>The realization in the late 1970s that RAS har...</td>\n",
       "      <td>R113S</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8985</th>\n",
       "      <td>NaN</td>\n",
       "      <td>FOXC1</td>\n",
       "      <td>5664</td>\n",
       "      <td>Hemizygous deletions are common molecular abno...</td>\n",
       "      <td>L130F</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8986</th>\n",
       "      <td>NaN</td>\n",
       "      <td>GSS</td>\n",
       "      <td>5665</td>\n",
       "      <td>All most R267W of has with to SMARTpool invest...</td>\n",
       "      <td>R267W</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8987</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CTSK</td>\n",
       "      <td>5666</td>\n",
       "      <td>Abstract Blood samples from 125 unrelated fami...</td>\n",
       "      <td>G79E</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8988</th>\n",
       "      <td>NaN</td>\n",
       "      <td>DFNB59</td>\n",
       "      <td>5667</td>\n",
       "      <td>Loss of DNA mismatch repair (MMR) in humans, m...</td>\n",
       "      <td>T54I</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8989 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Class     Gene    ID                                               Text  \\\n",
       "0       1.0   FAM58A     0  Cyclin-dependent kinases (CDKs) regulate a var...   \n",
       "1       2.0      CBL     1   Abstract Background  Non-small cell lung canc...   \n",
       "2       2.0      CBL     2   Abstract Background  Non-small cell lung canc...   \n",
       "3       3.0      CBL     3  Recent evidence has demonstrated that acquired...   \n",
       "4       4.0      CBL     4  Oncogenic mutations in the monomeric Casitas B...   \n",
       "5       4.0      CBL     5  Oncogenic mutations in the monomeric Casitas B...   \n",
       "6       5.0      CBL     6  Oncogenic mutations in the monomeric Casitas B...   \n",
       "7       1.0      CBL     7  CBL is a negative regulator of activated recep...   \n",
       "8       4.0      CBL     8   Abstract Juvenile myelomonocytic leukemia (JM...   \n",
       "9       4.0      CBL     9   Abstract Juvenile myelomonocytic leukemia (JM...   \n",
       "10      4.0      CBL    10  Oncogenic mutations in the monomeric Casitas B...   \n",
       "11      4.0      CBL    11  Noonan syndrome is an autosomal dominant conge...   \n",
       "12      4.0      CBL    12  Noonan syndrome is an autosomal dominant conge...   \n",
       "13      4.0      CBL    13  Noonan syndrome is an autosomal dominant conge...   \n",
       "14      5.0      CBL    14  Oncogenic mutations in the monomeric Casitas B...   \n",
       "15      4.0      CBL    15  Noonan syndrome is an autosomal dominant conge...   \n",
       "16      1.0      CBL    16  To determine if residual cylindrical refractiv...   \n",
       "17      4.0      CBL    17  Acquired uniparental disomy (aUPD) is a common...   \n",
       "18      5.0      CBL    18  Oncogenic mutations in the monomeric Casitas B...   \n",
       "19      4.0      CBL    19  Acquired uniparental disomy (aUPD) is a common...   \n",
       "20      6.0      CBL    20   Abstract Background  Non-small cell lung canc...   \n",
       "21      4.0      CBL    21  Oncogenic mutations in the monomeric Casitas B...   \n",
       "22      4.0      CBL    22  Oncogenic mutations in the monomeric Casitas B...   \n",
       "23      4.0      CBL    23  Recent evidence has demonstrated that acquired...   \n",
       "24      4.0      CBL    24  Recent evidence has demonstrated that acquired...   \n",
       "25      4.0      CBL    25  Recent evidence has demonstrated that acquired...   \n",
       "26      4.0    SHOC2    26   Abstract N-myristoylation is a common form of...   \n",
       "27      4.0     TERT    27  Heterozygous mutations in the telomerase compo...   \n",
       "28      7.0     TERT    28  Sequencing studies have identified many recurr...   \n",
       "29      4.0     TERT    29  Heterozygous mutations in the telomerase compo...   \n",
       "...     ...      ...   ...                                                ...   \n",
       "8959    NaN     VSX1  5638  Using a DNA microarray approach to screen for ...   \n",
       "8960    NaN     MTM1  5639  The tumor suppressor protein p53 is inactivate...   \n",
       "8961    NaN   D2HGDH  5640  Mutational analysis of oncogenes is critical f...   \n",
       "8962    NaN      DMD  5641  The serine/threonine protein kinase encoded by...   \n",
       "8963    NaN     ANKH  5642  The common participation of oncogenic KRAS pro...   \n",
       "8964    NaN      DCX  5643  EZH2 (enhancer of zeste homolog 2) is a critic...   \n",
       "8965    NaN     ADSL  5644  Mutations in the metabolic enzymes isocitrate ...   \n",
       "8966    NaN  HSD17B3  5645  Pancreatic carcinomas with acinar differentiat...   \n",
       "8967    NaN     CD96  5646  β-Catenin-mediated signaling can be constituti...   \n",
       "8968    NaN     HPS3  5647  Summary The genetic abnormalities underlying h...   \n",
       "8969    NaN     FKTN  5648  Lung cancer is the leading cause of cancer-rel...   \n",
       "8970    NaN    DARS2  5649  Summary During the past decade, the treatment ...   \n",
       "8971    NaN     TP53  5650  The transcription factor and tumor suppressor ...   \n",
       "8972    NaN    ALG12  5651  Protein tyrosine phosphatase receptor type D (...   \n",
       "8973    NaN    ACOX1  5652  Sensitizing activating mutations in the tyrosi...   \n",
       "8974    NaN   CLDN19  5653  Structural rearrangements of chromosome 10 are...   \n",
       "8975    NaN     MLH3  5654  Introduction  Production of fertile gametes is...   \n",
       "8976    NaN     GJB1  5655  Checkpoint kinase 2 (CHEK2, Chk2) emerges as a...   \n",
       "8977    NaN     LRP5  5656  Introduction The telomere sequences at the chr...   \n",
       "8978    NaN    TGFBI  5657  There are an estimated 1 million cases of brea...   \n",
       "8979    NaN   CYP2C9  5658  As of the occurring responders expression Tabl...   \n",
       "8980    NaN  HSD17B3  5659   Background & Aims: Inherited deleterious muta...   \n",
       "8981    NaN     CAV3  5660  Glioblastoma multiforme (GBM) is a lethal brai...   \n",
       "8982    NaN    ABHD5  5661  Diffuse large B cell lymphoma (DLBCL) is a com...   \n",
       "8983    NaN    NR3C2  5662  Figure largeDownload these has S818L clones ch...   \n",
       "8984    NaN  SLC46A1  5663  The realization in the late 1970s that RAS har...   \n",
       "8985    NaN    FOXC1  5664  Hemizygous deletions are common molecular abno...   \n",
       "8986    NaN      GSS  5665  All most R267W of has with to SMARTpool invest...   \n",
       "8987    NaN     CTSK  5666  Abstract Blood samples from 125 unrelated fami...   \n",
       "8988    NaN   DFNB59  5667  Loss of DNA mismatch repair (MMR) in humans, m...   \n",
       "\n",
       "                 Variation  Fusion_var  gene_fusion_var  Deletion_var  \\\n",
       "0     Truncating Mutations           0              0.0           0.0   \n",
       "1                    W802*           0              0.0           0.0   \n",
       "2                    Q249E           0              0.0           0.0   \n",
       "3                    N454D           0              0.0           0.0   \n",
       "4                    L399V           0              0.0           0.0   \n",
       "5                    V391I           0              0.0           0.0   \n",
       "6                    V430M           0              0.0           0.0   \n",
       "7                 Deletion           0              0.0           1.0   \n",
       "8                    Y371H           0              0.0           0.0   \n",
       "9                    C384R           0              0.0           0.0   \n",
       "10                   P395A           0              0.0           0.0   \n",
       "11                   K382E           0              0.0           0.0   \n",
       "12                   R420Q           0              0.0           0.0   \n",
       "13                   C381A           0              0.0           0.0   \n",
       "14                   P428L           0              0.0           0.0   \n",
       "15                   D390Y           0              0.0           0.0   \n",
       "16    Truncating Mutations           0              0.0           0.0   \n",
       "17                   Q367P           0              0.0           0.0   \n",
       "18                   M374V           0              0.0           0.0   \n",
       "19                   Y371S           0              0.0           0.0   \n",
       "20                    H94Y           0              0.0           0.0   \n",
       "21                   C396R           0              0.0           0.0   \n",
       "22                   G375P           0              0.0           0.0   \n",
       "23                   S376F           0              0.0           0.0   \n",
       "24                   P417A           0              0.0           0.0   \n",
       "25                   H398Y           0              0.0           0.0   \n",
       "26                     S2G           0              0.0           0.0   \n",
       "27                   Y846C           0              0.0           0.0   \n",
       "28                   C228T           0              0.0           0.0   \n",
       "29                   H412Y           0              0.0           0.0   \n",
       "...                    ...         ...              ...           ...   \n",
       "8959                 R166W           0              0.0           0.0   \n",
       "8960                 E157K           0              0.0           0.0   \n",
       "8961                 V444A           0              0.0           0.0   \n",
       "8962                 Y231N           0              0.0           0.0   \n",
       "8963                 G389R           0              0.0           0.0   \n",
       "8964                  R59L           0              0.0           0.0   \n",
       "8965                 R190Q           0              0.0           0.0   \n",
       "8966                  A56T           0              0.0           0.0   \n",
       "8967                 T280M           0              0.0           0.0   \n",
       "8968                 R397W           0              0.0           0.0   \n",
       "8969                 R179T           0              0.0           0.0   \n",
       "8970                 L613F           0              0.0           0.0   \n",
       "8971                 G245C           0              0.0           0.0   \n",
       "8972                  T67M           0              0.0           0.0   \n",
       "8973                 Q309R           0              0.0           0.0   \n",
       "8974                  Q57E           0              0.0           0.0   \n",
       "8975                 N499S           0              0.0           0.0   \n",
       "8976                 F235C           0              0.0           0.0   \n",
       "8977                 G171V           0              0.0           0.0   \n",
       "8978                 R124S           0              0.0           0.0   \n",
       "8979                 I359L           0              0.0           0.0   \n",
       "8980                 M235V           0              0.0           0.0   \n",
       "8981                  A46V           0              0.0           0.0   \n",
       "8982                 E260K           0              0.0           0.0   \n",
       "8983                 S818L           0              0.0           0.0   \n",
       "8984                 R113S           0              0.0           0.0   \n",
       "8985                 L130F           0              0.0           0.0   \n",
       "8986                 R267W           0              0.0           0.0   \n",
       "8987                  G79E           0              0.0           0.0   \n",
       "8988                  T54I           0              0.0           0.0   \n",
       "\n",
       "      del_or_ins_var  Amplification_var  Truncation_var  exon_var  \\\n",
       "0                0.0                  0               1         0   \n",
       "1                0.0                  0               0         0   \n",
       "2                0.0                  0               0         0   \n",
       "3                0.0                  0               0         0   \n",
       "4                0.0                  0               0         0   \n",
       "5                0.0                  0               0         0   \n",
       "6                0.0                  0               0         0   \n",
       "7                NaN                  0               0         0   \n",
       "8                0.0                  0               0         0   \n",
       "9                0.0                  0               0         0   \n",
       "10               0.0                  0               0         0   \n",
       "11               0.0                  0               0         0   \n",
       "12               0.0                  0               0         0   \n",
       "13               0.0                  0               0         0   \n",
       "14               0.0                  0               0         0   \n",
       "15               0.0                  0               0         0   \n",
       "16               0.0                  0               1         0   \n",
       "17               0.0                  0               0         0   \n",
       "18               0.0                  0               0         0   \n",
       "19               0.0                  0               0         0   \n",
       "20               0.0                  0               0         0   \n",
       "21               0.0                  0               0         0   \n",
       "22               0.0                  0               0         0   \n",
       "23               0.0                  0               0         0   \n",
       "24               0.0                  0               0         0   \n",
       "25               0.0                  0               0         0   \n",
       "26               0.0                  0               0         0   \n",
       "27               0.0                  0               0         0   \n",
       "28               0.0                  0               0         0   \n",
       "29               0.0                  0               0         0   \n",
       "...              ...                ...             ...       ...   \n",
       "8959             0.0                  0               0         0   \n",
       "8960             0.0                  0               0         0   \n",
       "8961             0.0                  0               0         0   \n",
       "8962             0.0                  0               0         0   \n",
       "8963             0.0                  0               0         0   \n",
       "8964             0.0                  0               0         0   \n",
       "8965             0.0                  0               0         0   \n",
       "8966             0.0                  0               0         0   \n",
       "8967             0.0                  0               0         0   \n",
       "8968             0.0                  0               0         0   \n",
       "8969             0.0                  0               0         0   \n",
       "8970             0.0                  0               0         0   \n",
       "8971             0.0                  0               0         0   \n",
       "8972             0.0                  0               0         0   \n",
       "8973             0.0                  0               0         0   \n",
       "8974             0.0                  0               0         0   \n",
       "8975             0.0                  0               0         0   \n",
       "8976             0.0                  0               0         0   \n",
       "8977             0.0                  0               0         0   \n",
       "8978             0.0                  0               0         0   \n",
       "8979             0.0                  0               0         0   \n",
       "8980             0.0                  0               0         0   \n",
       "8981             0.0                  0               0         0   \n",
       "8982             0.0                  0               0         0   \n",
       "8983             0.0                  0               0         0   \n",
       "8984             0.0                  0               0         0   \n",
       "8985             0.0                  0               0         0   \n",
       "8986             0.0                  0               0         0   \n",
       "8987             0.0                  0               0         0   \n",
       "8988             0.0                  0               0         0   \n",
       "\n",
       "      frameshift_var  dup_var  \n",
       "0                  0        0  \n",
       "1                  0        0  \n",
       "2                  0        0  \n",
       "3                  0        0  \n",
       "4                  0        0  \n",
       "5                  0        0  \n",
       "6                  0        0  \n",
       "7                  0        0  \n",
       "8                  0        0  \n",
       "9                  0        0  \n",
       "10                 0        0  \n",
       "11                 0        0  \n",
       "12                 0        0  \n",
       "13                 0        0  \n",
       "14                 0        0  \n",
       "15                 0        0  \n",
       "16                 0        0  \n",
       "17                 0        0  \n",
       "18                 0        0  \n",
       "19                 0        0  \n",
       "20                 0        0  \n",
       "21                 0        0  \n",
       "22                 0        0  \n",
       "23                 0        0  \n",
       "24                 0        0  \n",
       "25                 0        0  \n",
       "26                 0        0  \n",
       "27                 0        0  \n",
       "28                 0        0  \n",
       "29                 0        0  \n",
       "...              ...      ...  \n",
       "8959               0        0  \n",
       "8960               0        0  \n",
       "8961               0        0  \n",
       "8962               0        0  \n",
       "8963               0        0  \n",
       "8964               0        0  \n",
       "8965               0        0  \n",
       "8966               0        0  \n",
       "8967               0        0  \n",
       "8968               0        0  \n",
       "8969               0        0  \n",
       "8970               0        0  \n",
       "8971               0        0  \n",
       "8972               0        0  \n",
       "8973               0        0  \n",
       "8974               0        0  \n",
       "8975               0        0  \n",
       "8976               0        0  \n",
       "8977               0        0  \n",
       "8978               0        0  \n",
       "8979               0        0  \n",
       "8980               0        0  \n",
       "8981               0        0  \n",
       "8982               0        0  \n",
       "8983               0        0  \n",
       "8984               0        0  \n",
       "8985               0        0  \n",
       "8986               0        0  \n",
       "8987               0        0  \n",
       "8988               0        0  \n",
       "\n",
       "[8989 rows x 14 columns]"
      ]
     },
     "execution_count": 92,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_all"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "NLTK_sub_again_notext = [sent_tokenize(sub_again_yes.Text[i]) for i in sub_again_yes.index]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "nosub_sentences = get_sentences_nosub(no_sub_yes, NLTK_nosub, window_left = 0, window_right = 0) # Retrieves sentences where keyword is included\n",
    "nosub_sentences = [sorted(set(sentences), key = sentences.index) for sentences in nosub_sentences] # only use unique \n",
    "nosub_background = [sorted(set(sentences), key = sentences.index) for sentences in nosub_background] # only use unique \n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "sub_again_sentences,sub_again_background = get_sentences_sub_again_notext(sub_again_yes, NLTK_sub_again_notext, window_left = 0, window_right = 0) # Retrieves sentences where keyword is included\n",
    "sub_again_sentences = [sorted(set(sentences), key = sentences.index) for sentences in sub_again_sentences] # only use unique \n",
    "sub_again_background = [sorted(set(sentences), key = sentences.index) for sentences in sub_again_background] # only use unique \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "NLTK_sub_noText = [sent_tokenize(sub_noText.Text[i]) for i in sub_noText.index]\n",
    "sub_noText_sentences,sub_noText_background = get_sentences_sub_noText(sub_noText, NLTK_sub_noText, window_left = 0, window_right = 0) # Retrieves sentences where subsitution mutation is included\n",
    "sub_noText_sentences = [sorted(set(sentences), key = sentences.index) for sentences in sub_noText_sentences] # only use unique sentences\n",
    "sub_noText_background = [sorted(set(sentences), key = sentences.index) for sentences in sub_noText_background] # only use unique \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "NLTK_sub = [sent_tokenize(sub_in_text.Text[i]) for i in sub_in_text.index] # takes a long time to run tokenizer => use pickle to save\n",
    "sub_sentences,sub_background = get_sentences_sub(sub_in_text, NLTK_sub, window_left = 0, window_right = 0) \n",
    "# Retrieves sentences where subsitution mutation is included.\n",
    "# window_left and window_right specify which sentences to keep at the left side or right side of the sub sentences.\n",
    "# IMPORTANT: I used also placeholderMutation to replace the original sub mutations here\n",
    "sub_sentences = [sorted(set(sentences), key = sentences.index) for sentences in sub_sentences]\n",
    "sub_background = [sorted(set(sentences), key = sentences.index) for sentences in sub_background]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "sub_sentences_string = sentences_to_string(sub_sentences)\n",
    "sub_noText_string = sentences_to_string(sub_noText_sentences)\n",
    "nosub_sentences_string=sentences_to_string(nosub_sentences)\n",
    "sub_again_string=sentences_to_string(sub_again_sentences)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Creating the background text (Text is now the frontline text)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "data_all.loc[sub_in_text.index,\"Background\"]=sentences_to_string(sub_background)\n",
    "data_all.loc[sub_noText.index,\"Background\"]=sentences_to_string(sub_noText_background)\n",
    "data_all.loc[no_sub_yes.index,\"Background\"]=sentences_to_string(nosub_background)\n",
    "data_all.loc[sub_again_yes.index,\"Background\"]=sentences_to_string(sub_again_background)\n",
    "data_all.loc[sub_again_no.index,\"Background\"]=\"nbckgr\"\n",
    "data_all.loc[no_sub_no.index,\"Background\"]=\"nbckgr\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\Franck\\Documents\\Continuum\\Anaconda3\\lib\\site-packages\\ipykernel_launcher.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \"\"\"Entry point for launching an IPython kernel.\n",
      "C:\\Users\\Franck\\Documents\\Continuum\\Anaconda3\\lib\\site-packages\\ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n"
     ]
    }
   ],
   "source": [
    "data_all.Background[data_all.Background==\"\"]=\"nbckgr\"\n",
    "data_all.Background[data_all.Background.isnull()]=\"nbckgr\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\Franck\\Documents\\Continuum\\Anaconda3\\lib\\site-packages\\ipykernel_launcher.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \"\"\"Entry point for launching an IPython kernel.\n",
      "C:\\Users\\Franck\\Documents\\Continuum\\Anaconda3\\lib\\site-packages\\ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "C:\\Users\\Franck\\Documents\\Continuum\\Anaconda3\\lib\\site-packages\\ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "C:\\Users\\Franck\\Documents\\Continuum\\Anaconda3\\lib\\site-packages\\ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "C:\\Users\\Franck\\Documents\\Continuum\\Anaconda3\\lib\\site-packages\\ipykernel_launcher.py:5: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \"\"\"\n",
      "C:\\Users\\Franck\\Documents\\Continuum\\Anaconda3\\lib\\site-packages\\ipykernel_launcher.py:6: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "C:\\Users\\Franck\\Documents\\Continuum\\Anaconda3\\lib\\site-packages\\ipykernel_launcher.py:7: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  import sys\n",
      "C:\\Users\\Franck\\Documents\\Continuum\\Anaconda3\\lib\\site-packages\\ipykernel_launcher.py:8: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n"
     ]
    }
   ],
   "source": [
    "data_all.Text[sub_in_text.index] = sub_sentences_string\n",
    "data_all.Text[sub_noText.index] = sub_noText_string\n",
    "data_all.Text[no_sub_yes.index] = nosub_sentences_string\n",
    "data_all.Text[sub_again_yes.index] = sub_again_string\n",
    "data_all.Text[sub_again_no.index]=\"null\"\n",
    "data_all.Text[no_sub_no.index]=\"null\"\n",
    "data_all.Text[data_all.Text==\"\"]=\"null\"\n",
    "data_all.Text[data_all.Text.isnull()]=\"null\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#TODO : #Sentence Tokenizer for non-subs that are not in text, AND\n",
    "#subs that are not in text"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Not used\n",
    "################ The dummy variables for Gene and Text ##################\n",
    "## TODO: also use dummy for Text? There are 135 shared Genes and 142 shared Text between train and Leaks!  len(set(train.Text) & set(Leaks.Text))\n",
    "#data_all_dummy = data_all_backup[['Gene', 'Text']] # drop those columns we don't need as dummy.\n",
    "#X_dummy = pd.get_dummies(data_all_dummy) # converts categorical variables into dummy variable. From len set => 269 genes + 2090 texts\n",
    "#X_dummy_train = X_dummy[:train.shape[0]]\n",
    "#X_dummy_test = X_dummy[train.shape[0]:]\n",
    "#dummy_names = X_dummy.columns.values #### To remember names if you want to check again what Gene or Text used\n",
    "#X_dummy = X_dummy.values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "###### Use the variation types \n",
    "#variation_types = data_all.drop(['ID', 'Gene', 'Class', 'Text', 'Variation'], axis =1)\n",
    "#X_variation_train = variation_types[:train.shape[0]]\n",
    "#X_variation_test = variation_types[train.shape[0]:]\n",
    "#variation_names = variation_types.columns.values "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Cleaning "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "stop = set(stopwords.words('english')) + \n",
    "exclude = set('!\"#$%&\\'()*+:;<=>?@[\\\\]^_`{|}~') \n",
    "lemma = WordNetLemmatizer()\n",
    "def clean(doc,lemmatiz=False):\n",
    "    stop_free = \" \".join([i for i in doc.lower().split() if i not in stop])\n",
    "    punc_free_0 = [re.sub(\",|\\.|/\",\" \",ch) for ch in stop_free]\n",
    "    if lemmatiz:\n",
    "        punc_free_lem=\"\".join(ch for ch in punc_free_0 if ch not in exclude)\n",
    "        normalized = \" \".join(lemma.lemmatize(word) for word in punc_free_lem.split())\n",
    "        return normalized\n",
    "    else:\n",
    "        punc_free = \"\".join(ch for ch in punc_free_0 if ch not in exclude)\n",
    "        return punc_free"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#No lemmatization for the moment, be careful not to lemmatize then w2vec\n",
    "data_all.Text = [clean(doc) for doc in data_all.Text]  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "train = data_all.iloc[:len(train)]\n",
    "test = data_all.iloc[len(train):]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "data_all=pd.concat((train, test), axis=0, ignore_index=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Some more feature engineering"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 463,
   "metadata": {
    "collapsed": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# Feature based on length of the text\n",
    "data_all[\"Text_words\"] = data_all[\"Text\"].map(lambda x: len(str(x).split(\" \")))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 195,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['null',\n",
       " 'null',\n",
       " 'null',\n",
       " 'The case of a 40-year-old woman with severe edema of the face and neck after the injection of a local dental anesthetic is presented. The reaction is attributed to the presence of sodium metabisulfite, and antioxidant, in the local anesthetic. Both the anesthetic and the sodium metabisulfite gave a delayed positive patch-test response.',\n",
       " 'null',\n",
       " 'null',\n",
       " 'null']"
      ]
     },
     "execution_count": 195,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list(data_all.Text[data_all.Text_words<100])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 465,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Class</th>\n",
       "      <th>Gene</th>\n",
       "      <th>ID</th>\n",
       "      <th>Text</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Substitutions_var</th>\n",
       "      <th>Stop_codon_var</th>\n",
       "      <th>gene_fusion_var</th>\n",
       "      <th>Deletion_var</th>\n",
       "      <th>del_or_ins_var</th>\n",
       "      <th>Fusion_var</th>\n",
       "      <th>Amplification_var</th>\n",
       "      <th>Truncation_var</th>\n",
       "      <th>exon_var</th>\n",
       "      <th>frameshift_var</th>\n",
       "      <th>dup_var</th>\n",
       "      <th>Text_words</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1.0</td>\n",
       "      <td>FAM58A</td>\n",
       "      <td>0</td>\n",
       "      <td>Cyclin-dependent kinases (CDKs) regulate a var...</td>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>1</td>\n",
       "      <td>Abstract Background  Non-small cell lung canc...</td>\n",
       "      <td>W802*</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5783</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>2</td>\n",
       "      <td>Abstract Background  Non-small cell lung canc...</td>\n",
       "      <td>Q249E</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5783</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>3</td>\n",
       "      <td>Recent evidence has demonstrated that acquired...</td>\n",
       "      <td>N454D</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5625</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>4</td>\n",
       "      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n",
       "      <td>L399V</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6248</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>4.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>5</td>\n",
       "      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n",
       "      <td>V391I</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6248</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>5.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>6</td>\n",
       "      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n",
       "      <td>V430M</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6248</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>1.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>7</td>\n",
       "      <td>CBL is a negative regulator of activated recep...</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14709</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>4.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>8</td>\n",
       "      <td>Abstract Juvenile myelomonocytic leukemia (JM...</td>\n",
       "      <td>Y371H</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>12009</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>4.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>9</td>\n",
       "      <td>Abstract Juvenile myelomonocytic leukemia (JM...</td>\n",
       "      <td>C384R</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5761</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>4.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>10</td>\n",
       "      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n",
       "      <td>P395A</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6248</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>4.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>11</td>\n",
       "      <td>Noonan syndrome is an autosomal dominant conge...</td>\n",
       "      <td>K382E</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>12251</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>4.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>12</td>\n",
       "      <td>Noonan syndrome is an autosomal dominant conge...</td>\n",
       "      <td>R420Q</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>25745</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>4.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>13</td>\n",
       "      <td>Noonan syndrome is an autosomal dominant conge...</td>\n",
       "      <td>C381A</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>12251</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>5.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>14</td>\n",
       "      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n",
       "      <td>P428L</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6248</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>4.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>15</td>\n",
       "      <td>Noonan syndrome is an autosomal dominant conge...</td>\n",
       "      <td>D390Y</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6003</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>1.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>16</td>\n",
       "      <td>To determine if residual cylindrical refractiv...</td>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>8206</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>4.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>17</td>\n",
       "      <td>Acquired uniparental disomy (aUPD) is a common...</td>\n",
       "      <td>Q367P</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3013</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>5.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>18</td>\n",
       "      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n",
       "      <td>M374V</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6248</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>4.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>19</td>\n",
       "      <td>Acquired uniparental disomy (aUPD) is a common...</td>\n",
       "      <td>Y371S</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3013</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>6.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>20</td>\n",
       "      <td>Abstract Background  Non-small cell lung canc...</td>\n",
       "      <td>H94Y</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>12031</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>4.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>21</td>\n",
       "      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n",
       "      <td>C396R</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6248</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>4.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>22</td>\n",
       "      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n",
       "      <td>G375P</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6248</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>4.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>23</td>\n",
       "      <td>Recent evidence has demonstrated that acquired...</td>\n",
       "      <td>S376F</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5625</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>4.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>24</td>\n",
       "      <td>Recent evidence has demonstrated that acquired...</td>\n",
       "      <td>P417A</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5625</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>4.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>25</td>\n",
       "      <td>Recent evidence has demonstrated that acquired...</td>\n",
       "      <td>H398Y</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5625</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>4.0</td>\n",
       "      <td>SHOC2</td>\n",
       "      <td>26</td>\n",
       "      <td>Abstract N-myristoylation is a common form of...</td>\n",
       "      <td>S2G</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6693</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>4.0</td>\n",
       "      <td>TERT</td>\n",
       "      <td>27</td>\n",
       "      <td>Heterozygous mutations in the telomerase compo...</td>\n",
       "      <td>Y846C</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1422</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>7.0</td>\n",
       "      <td>TERT</td>\n",
       "      <td>28</td>\n",
       "      <td>Sequencing studies have identified many recurr...</td>\n",
       "      <td>C228T</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>13862</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>4.0</td>\n",
       "      <td>TERT</td>\n",
       "      <td>29</td>\n",
       "      <td>Heterozygous mutations in the telomerase compo...</td>\n",
       "      <td>H412Y</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1422</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3291</th>\n",
       "      <td>7.0</td>\n",
       "      <td>RET</td>\n",
       "      <td>3291</td>\n",
       "      <td>We investigatedthe transformingactivityofthe r...</td>\n",
       "      <td>C620R</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1913</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3292</th>\n",
       "      <td>7.0</td>\n",
       "      <td>RET</td>\n",
       "      <td>3292</td>\n",
       "      <td>We investigatedthe transformingactivityofthe r...</td>\n",
       "      <td>C634Y</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1913</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3293</th>\n",
       "      <td>1.0</td>\n",
       "      <td>RET</td>\n",
       "      <td>3293</td>\n",
       "      <td>RET is a transmembrane tyrosine kinase partici...</td>\n",
       "      <td>V804G</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3889</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3294</th>\n",
       "      <td>7.0</td>\n",
       "      <td>RET</td>\n",
       "      <td>3294</td>\n",
       "      <td>Introduction  Inherited germ line activating m...</td>\n",
       "      <td>R886W</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5333</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3295</th>\n",
       "      <td>4.0</td>\n",
       "      <td>RET</td>\n",
       "      <td>3295</td>\n",
       "      <td>Many missense mutations in the RET proto-oncog...</td>\n",
       "      <td>F893L</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4733</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3296</th>\n",
       "      <td>7.0</td>\n",
       "      <td>RET</td>\n",
       "      <td>3296</td>\n",
       "      <td>The RET proto-oncogene encodes a receptor tyro...</td>\n",
       "      <td>Y791F</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>18412</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3297</th>\n",
       "      <td>4.0</td>\n",
       "      <td>RUNX1</td>\n",
       "      <td>3297</td>\n",
       "      <td>THE AML1 GENE IS KNOWN as the most frequent ta...</td>\n",
       "      <td>R177*</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>15154</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3298</th>\n",
       "      <td>4.0</td>\n",
       "      <td>RUNX1</td>\n",
       "      <td>3298</td>\n",
       "      <td>Introduction  Myelodysplastic syndromes (MDS) ...</td>\n",
       "      <td>Y113*</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6118</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3299</th>\n",
       "      <td>4.0</td>\n",
       "      <td>RUNX1</td>\n",
       "      <td>3299</td>\n",
       "      <td>The BCR-ABL fusion protein generated by t(9;22...</td>\n",
       "      <td>R139G</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6392</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3300</th>\n",
       "      <td>4.0</td>\n",
       "      <td>RUNX1</td>\n",
       "      <td>3300</td>\n",
       "      <td>The most frequent mutations associated with le...</td>\n",
       "      <td>K83N</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>9846</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3301</th>\n",
       "      <td>4.0</td>\n",
       "      <td>RUNX1</td>\n",
       "      <td>3301</td>\n",
       "      <td>The most frequent mutations associated with le...</td>\n",
       "      <td>R177Q</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>11051</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3302</th>\n",
       "      <td>4.0</td>\n",
       "      <td>RUNX1</td>\n",
       "      <td>3302</td>\n",
       "      <td>Familial platelet disorder with predisposition...</td>\n",
       "      <td>R166Q</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5896</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3303</th>\n",
       "      <td>4.0</td>\n",
       "      <td>RUNX1</td>\n",
       "      <td>3303</td>\n",
       "      <td>Introduction  Myelodysplastic syndromes (MDS) ...</td>\n",
       "      <td>P173S</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6118</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3304</th>\n",
       "      <td>4.0</td>\n",
       "      <td>RUNX1</td>\n",
       "      <td>3304</td>\n",
       "      <td>Familial platelet disorder with predisposition...</td>\n",
       "      <td>R201Q</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5896</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3305</th>\n",
       "      <td>4.0</td>\n",
       "      <td>RUNX1</td>\n",
       "      <td>3305</td>\n",
       "      <td>Introduction  Myelodysplastic syndromes (MDS) ...</td>\n",
       "      <td>S70fsX93</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>6118</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3306</th>\n",
       "      <td>1.0</td>\n",
       "      <td>RUNX1</td>\n",
       "      <td>3306</td>\n",
       "      <td>Here we report two new RUNX1 mutations in one ...</td>\n",
       "      <td>W279*</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1777</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3307</th>\n",
       "      <td>1.0</td>\n",
       "      <td>RUNX1</td>\n",
       "      <td>3307</td>\n",
       "      <td>The RUNX genes have come to prominence recentl...</td>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14955</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3308</th>\n",
       "      <td>7.0</td>\n",
       "      <td>RUNX1</td>\n",
       "      <td>3308</td>\n",
       "      <td>Familial platelet disorder with propensity to...</td>\n",
       "      <td>R174*</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6067</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3309</th>\n",
       "      <td>1.0</td>\n",
       "      <td>RUNX1</td>\n",
       "      <td>3309</td>\n",
       "      <td>The BCR-ABL fusion protein generated by t(9;22...</td>\n",
       "      <td>D171G</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6392</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3310</th>\n",
       "      <td>7.0</td>\n",
       "      <td>RUNX1</td>\n",
       "      <td>3310</td>\n",
       "      <td>RUNX proteins belong to a family of metazoan t...</td>\n",
       "      <td>Amplification</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>9132</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3311</th>\n",
       "      <td>4.0</td>\n",
       "      <td>RUNX1</td>\n",
       "      <td>3311</td>\n",
       "      <td>The AML1/EVI-1 chimeric gene is generated by t...</td>\n",
       "      <td>RUNX1-EVI1 Fusion</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7946</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3312</th>\n",
       "      <td>4.0</td>\n",
       "      <td>RUNX1</td>\n",
       "      <td>3312</td>\n",
       "      <td>Balanced chromosomal translocations are freque...</td>\n",
       "      <td>TEL-RUNX1 Fusion</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4988</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3313</th>\n",
       "      <td>4.0</td>\n",
       "      <td>RUNX1</td>\n",
       "      <td>3313</td>\n",
       "      <td>The BCR-ABL fusion protein generated by t(9;22...</td>\n",
       "      <td>H78Q</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6392</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3314</th>\n",
       "      <td>6.0</td>\n",
       "      <td>RUNX1</td>\n",
       "      <td>3314</td>\n",
       "      <td>Introduction  Myelodysplastic syndromes (MDS) ...</td>\n",
       "      <td>G42R</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14249</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3315</th>\n",
       "      <td>4.0</td>\n",
       "      <td>RUNX1</td>\n",
       "      <td>3315</td>\n",
       "      <td>The RUNX gene family includes three evolutiona...</td>\n",
       "      <td>RUNX1-RUNX1T1 Fusion</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>11857</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3316</th>\n",
       "      <td>4.0</td>\n",
       "      <td>RUNX1</td>\n",
       "      <td>3316</td>\n",
       "      <td>Introduction  Myelodysplastic syndromes (MDS) ...</td>\n",
       "      <td>D171N</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>11112</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3317</th>\n",
       "      <td>1.0</td>\n",
       "      <td>RUNX1</td>\n",
       "      <td>3317</td>\n",
       "      <td>Introduction  Myelodysplastic syndromes (MDS) ...</td>\n",
       "      <td>A122*</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6118</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3318</th>\n",
       "      <td>1.0</td>\n",
       "      <td>RUNX1</td>\n",
       "      <td>3318</td>\n",
       "      <td>The Runt-related transcription factor 1 gene (...</td>\n",
       "      <td>Fusions</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5119</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3319</th>\n",
       "      <td>4.0</td>\n",
       "      <td>RUNX1</td>\n",
       "      <td>3319</td>\n",
       "      <td>The RUNX1/AML1 gene is the most frequent targe...</td>\n",
       "      <td>R80C</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4913</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3320</th>\n",
       "      <td>4.0</td>\n",
       "      <td>RUNX1</td>\n",
       "      <td>3320</td>\n",
       "      <td>The most frequent mutations associated with le...</td>\n",
       "      <td>K83E</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>9846</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3321 rows × 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Class    Gene    ID                                               Text  \\\n",
       "0       1.0  FAM58A     0  Cyclin-dependent kinases (CDKs) regulate a var...   \n",
       "1       2.0     CBL     1   Abstract Background  Non-small cell lung canc...   \n",
       "2       2.0     CBL     2   Abstract Background  Non-small cell lung canc...   \n",
       "3       3.0     CBL     3  Recent evidence has demonstrated that acquired...   \n",
       "4       4.0     CBL     4  Oncogenic mutations in the monomeric Casitas B...   \n",
       "5       4.0     CBL     5  Oncogenic mutations in the monomeric Casitas B...   \n",
       "6       5.0     CBL     6  Oncogenic mutations in the monomeric Casitas B...   \n",
       "7       1.0     CBL     7  CBL is a negative regulator of activated recep...   \n",
       "8       4.0     CBL     8   Abstract Juvenile myelomonocytic leukemia (JM...   \n",
       "9       4.0     CBL     9   Abstract Juvenile myelomonocytic leukemia (JM...   \n",
       "10      4.0     CBL    10  Oncogenic mutations in the monomeric Casitas B...   \n",
       "11      4.0     CBL    11  Noonan syndrome is an autosomal dominant conge...   \n",
       "12      4.0     CBL    12  Noonan syndrome is an autosomal dominant conge...   \n",
       "13      4.0     CBL    13  Noonan syndrome is an autosomal dominant conge...   \n",
       "14      5.0     CBL    14  Oncogenic mutations in the monomeric Casitas B...   \n",
       "15      4.0     CBL    15  Noonan syndrome is an autosomal dominant conge...   \n",
       "16      1.0     CBL    16  To determine if residual cylindrical refractiv...   \n",
       "17      4.0     CBL    17  Acquired uniparental disomy (aUPD) is a common...   \n",
       "18      5.0     CBL    18  Oncogenic mutations in the monomeric Casitas B...   \n",
       "19      4.0     CBL    19  Acquired uniparental disomy (aUPD) is a common...   \n",
       "20      6.0     CBL    20   Abstract Background  Non-small cell lung canc...   \n",
       "21      4.0     CBL    21  Oncogenic mutations in the monomeric Casitas B...   \n",
       "22      4.0     CBL    22  Oncogenic mutations in the monomeric Casitas B...   \n",
       "23      4.0     CBL    23  Recent evidence has demonstrated that acquired...   \n",
       "24      4.0     CBL    24  Recent evidence has demonstrated that acquired...   \n",
       "25      4.0     CBL    25  Recent evidence has demonstrated that acquired...   \n",
       "26      4.0   SHOC2    26   Abstract N-myristoylation is a common form of...   \n",
       "27      4.0    TERT    27  Heterozygous mutations in the telomerase compo...   \n",
       "28      7.0    TERT    28  Sequencing studies have identified many recurr...   \n",
       "29      4.0    TERT    29  Heterozygous mutations in the telomerase compo...   \n",
       "...     ...     ...   ...                                                ...   \n",
       "3291    7.0     RET  3291  We investigatedthe transformingactivityofthe r...   \n",
       "3292    7.0     RET  3292  We investigatedthe transformingactivityofthe r...   \n",
       "3293    1.0     RET  3293  RET is a transmembrane tyrosine kinase partici...   \n",
       "3294    7.0     RET  3294  Introduction  Inherited germ line activating m...   \n",
       "3295    4.0     RET  3295  Many missense mutations in the RET proto-oncog...   \n",
       "3296    7.0     RET  3296  The RET proto-oncogene encodes a receptor tyro...   \n",
       "3297    4.0   RUNX1  3297  THE AML1 GENE IS KNOWN as the most frequent ta...   \n",
       "3298    4.0   RUNX1  3298  Introduction  Myelodysplastic syndromes (MDS) ...   \n",
       "3299    4.0   RUNX1  3299  The BCR-ABL fusion protein generated by t(9;22...   \n",
       "3300    4.0   RUNX1  3300  The most frequent mutations associated with le...   \n",
       "3301    4.0   RUNX1  3301  The most frequent mutations associated with le...   \n",
       "3302    4.0   RUNX1  3302  Familial platelet disorder with predisposition...   \n",
       "3303    4.0   RUNX1  3303  Introduction  Myelodysplastic syndromes (MDS) ...   \n",
       "3304    4.0   RUNX1  3304  Familial platelet disorder with predisposition...   \n",
       "3305    4.0   RUNX1  3305  Introduction  Myelodysplastic syndromes (MDS) ...   \n",
       "3306    1.0   RUNX1  3306  Here we report two new RUNX1 mutations in one ...   \n",
       "3307    1.0   RUNX1  3307  The RUNX genes have come to prominence recentl...   \n",
       "3308    7.0   RUNX1  3308   Familial platelet disorder with propensity to...   \n",
       "3309    1.0   RUNX1  3309  The BCR-ABL fusion protein generated by t(9;22...   \n",
       "3310    7.0   RUNX1  3310  RUNX proteins belong to a family of metazoan t...   \n",
       "3311    4.0   RUNX1  3311  The AML1/EVI-1 chimeric gene is generated by t...   \n",
       "3312    4.0   RUNX1  3312  Balanced chromosomal translocations are freque...   \n",
       "3313    4.0   RUNX1  3313  The BCR-ABL fusion protein generated by t(9;22...   \n",
       "3314    6.0   RUNX1  3314  Introduction  Myelodysplastic syndromes (MDS) ...   \n",
       "3315    4.0   RUNX1  3315  The RUNX gene family includes three evolutiona...   \n",
       "3316    4.0   RUNX1  3316  Introduction  Myelodysplastic syndromes (MDS) ...   \n",
       "3317    1.0   RUNX1  3317  Introduction  Myelodysplastic syndromes (MDS) ...   \n",
       "3318    1.0   RUNX1  3318  The Runt-related transcription factor 1 gene (...   \n",
       "3319    4.0   RUNX1  3319  The RUNX1/AML1 gene is the most frequent targe...   \n",
       "3320    4.0   RUNX1  3320  The most frequent mutations associated with le...   \n",
       "\n",
       "                 Variation  Substitutions_var  Stop_codon_var  \\\n",
       "0     Truncating Mutations                  0               0   \n",
       "1                    W802*                  1               1   \n",
       "2                    Q249E                  1               0   \n",
       "3                    N454D                  1               0   \n",
       "4                    L399V                  1               0   \n",
       "5                    V391I                  1               0   \n",
       "6                    V430M                  1               0   \n",
       "7                 Deletion                  0               0   \n",
       "8                    Y371H                  1               0   \n",
       "9                    C384R                  1               0   \n",
       "10                   P395A                  1               0   \n",
       "11                   K382E                  1               0   \n",
       "12                   R420Q                  1               0   \n",
       "13                   C381A                  1               0   \n",
       "14                   P428L                  1               0   \n",
       "15                   D390Y                  1               0   \n",
       "16    Truncating Mutations                  0               0   \n",
       "17                   Q367P                  1               0   \n",
       "18                   M374V                  1               0   \n",
       "19                   Y371S                  1               0   \n",
       "20                    H94Y                  1               0   \n",
       "21                   C396R                  1               0   \n",
       "22                   G375P                  1               0   \n",
       "23                   S376F                  1               0   \n",
       "24                   P417A                  1               0   \n",
       "25                   H398Y                  1               0   \n",
       "26                     S2G                  1               0   \n",
       "27                   Y846C                  1               0   \n",
       "28                   C228T                  1               0   \n",
       "29                   H412Y                  1               0   \n",
       "...                    ...                ...             ...   \n",
       "3291                 C620R                  1               0   \n",
       "3292                 C634Y                  1               0   \n",
       "3293                 V804G                  1               0   \n",
       "3294                 R886W                  1               0   \n",
       "3295                 F893L                  1               0   \n",
       "3296                 Y791F                  1               0   \n",
       "3297                 R177*                  1               1   \n",
       "3298                 Y113*                  1               1   \n",
       "3299                 R139G                  1               0   \n",
       "3300                  K83N                  1               0   \n",
       "3301                 R177Q                  1               0   \n",
       "3302                 R166Q                  1               0   \n",
       "3303                 P173S                  1               0   \n",
       "3304                 R201Q                  1               0   \n",
       "3305              S70fsX93                  0               0   \n",
       "3306                 W279*                  1               1   \n",
       "3307  Truncating Mutations                  0               0   \n",
       "3308                 R174*                  1               1   \n",
       "3309                 D171G                  1               0   \n",
       "3310         Amplification                  0               0   \n",
       "3311     RUNX1-EVI1 Fusion                  0               0   \n",
       "3312      TEL-RUNX1 Fusion                  0               0   \n",
       "3313                  H78Q                  1               0   \n",
       "3314                  G42R                  1               0   \n",
       "3315  RUNX1-RUNX1T1 Fusion                  0               0   \n",
       "3316                 D171N                  1               0   \n",
       "3317                 A122*                  1               1   \n",
       "3318               Fusions                  0               0   \n",
       "3319                  R80C                  1               0   \n",
       "3320                  K83E                  1               0   \n",
       "\n",
       "      gene_fusion_var  Deletion_var  del_or_ins_var  Fusion_var  \\\n",
       "0                   0             0               0           0   \n",
       "1                   0             0               0           0   \n",
       "2                   0             0               0           0   \n",
       "3                   0             0               0           0   \n",
       "4                   0             0               0           0   \n",
       "5                   0             0               0           0   \n",
       "6                   0             0               0           0   \n",
       "7                   0             1               0           0   \n",
       "8                   0             0               0           0   \n",
       "9                   0             0               0           0   \n",
       "10                  0             0               0           0   \n",
       "11                  0             0               0           0   \n",
       "12                  0             0               0           0   \n",
       "13                  0             0               0           0   \n",
       "14                  0             0               0           0   \n",
       "15                  0             0               0           0   \n",
       "16                  0             0               0           0   \n",
       "17                  0             0               0           0   \n",
       "18                  0             0               0           0   \n",
       "19                  0             0               0           0   \n",
       "20                  0             0               0           0   \n",
       "21                  0             0               0           0   \n",
       "22                  0             0               0           0   \n",
       "23                  0             0               0           0   \n",
       "24                  0             0               0           0   \n",
       "25                  0             0               0           0   \n",
       "26                  0             0               0           0   \n",
       "27                  0             0               0           0   \n",
       "28                  0             0               0           0   \n",
       "29                  0             0               0           0   \n",
       "...               ...           ...             ...         ...   \n",
       "3291                0             0               0           0   \n",
       "3292                0             0               0           0   \n",
       "3293                0             0               0           0   \n",
       "3294                0             0               0           0   \n",
       "3295                0             0               0           0   \n",
       "3296                0             0               0           0   \n",
       "3297                0             0               0           0   \n",
       "3298                0             0               0           0   \n",
       "3299                0             0               0           0   \n",
       "3300                0             0               0           0   \n",
       "3301                0             0               0           0   \n",
       "3302                0             0               0           0   \n",
       "3303                0             0               0           0   \n",
       "3304                0             0               0           0   \n",
       "3305                0             0               0           0   \n",
       "3306                0             0               0           0   \n",
       "3307                0             0               0           0   \n",
       "3308                0             0               0           0   \n",
       "3309                0             0               0           0   \n",
       "3310                0             0               0           0   \n",
       "3311                1             0               0           0   \n",
       "3312                1             0               0           0   \n",
       "3313                0             0               0           0   \n",
       "3314                0             0               0           0   \n",
       "3315                1             0               0           0   \n",
       "3316                0             0               0           0   \n",
       "3317                0             0               0           0   \n",
       "3318                0             0               0           1   \n",
       "3319                0             0               0           0   \n",
       "3320                0             0               0           0   \n",
       "\n",
       "      Amplification_var  Truncation_var  exon_var  frameshift_var  dup_var  \\\n",
       "0                     0               1         0               0        0   \n",
       "1                     0               0         0               0        0   \n",
       "2                     0               0         0               0        0   \n",
       "3                     0               0         0               0        0   \n",
       "4                     0               0         0               0        0   \n",
       "5                     0               0         0               0        0   \n",
       "6                     0               0         0               0        0   \n",
       "7                     0               0         0               0        0   \n",
       "8                     0               0         0               0        0   \n",
       "9                     0               0         0               0        0   \n",
       "10                    0               0         0               0        0   \n",
       "11                    0               0         0               0        0   \n",
       "12                    0               0         0               0        0   \n",
       "13                    0               0         0               0        0   \n",
       "14                    0               0         0               0        0   \n",
       "15                    0               0         0               0        0   \n",
       "16                    0               1         0               0        0   \n",
       "17                    0               0         0               0        0   \n",
       "18                    0               0         0               0        0   \n",
       "19                    0               0         0               0        0   \n",
       "20                    0               0         0               0        0   \n",
       "21                    0               0         0               0        0   \n",
       "22                    0               0         0               0        0   \n",
       "23                    0               0         0               0        0   \n",
       "24                    0               0         0               0        0   \n",
       "25                    0               0         0               0        0   \n",
       "26                    0               0         0               0        0   \n",
       "27                    0               0         0               0        0   \n",
       "28                    0               0         0               0        0   \n",
       "29                    0               0         0               0        0   \n",
       "...                 ...             ...       ...             ...      ...   \n",
       "3291                  0               0         0               0        0   \n",
       "3292                  0               0         0               0        0   \n",
       "3293                  0               0         0               0        0   \n",
       "3294                  0               0         0               0        0   \n",
       "3295                  0               0         0               0        0   \n",
       "3296                  0               0         0               0        0   \n",
       "3297                  0               0         0               0        0   \n",
       "3298                  0               0         0               0        0   \n",
       "3299                  0               0         0               0        0   \n",
       "3300                  0               0         0               0        0   \n",
       "3301                  0               0         0               0        0   \n",
       "3302                  0               0         0               0        0   \n",
       "3303                  0               0         0               0        0   \n",
       "3304                  0               0         0               0        0   \n",
       "3305                  0               0         0               1        0   \n",
       "3306                  0               0         0               0        0   \n",
       "3307                  0               1         0               0        0   \n",
       "3308                  0               0         0               0        0   \n",
       "3309                  0               0         0               0        0   \n",
       "3310                  1               0         0               0        0   \n",
       "3311                  0               0         0               0        0   \n",
       "3312                  0               0         0               0        0   \n",
       "3313                  0               0         0               0        0   \n",
       "3314                  0               0         0               0        0   \n",
       "3315                  0               0         0               0        0   \n",
       "3316                  0               0         0               0        0   \n",
       "3317                  0               0         0               0        0   \n",
       "3318                  0               0         0               0        0   \n",
       "3319                  0               0         0               0        0   \n",
       "3320                  0               0         0               0        0   \n",
       "\n",
       "      Text_words  \n",
       "0           6105  \n",
       "1           5783  \n",
       "2           5783  \n",
       "3           5625  \n",
       "4           6248  \n",
       "5           6248  \n",
       "6           6248  \n",
       "7          14709  \n",
       "8          12009  \n",
       "9           5761  \n",
       "10          6248  \n",
       "11         12251  \n",
       "12         25745  \n",
       "13         12251  \n",
       "14          6248  \n",
       "15          6003  \n",
       "16          8206  \n",
       "17          3013  \n",
       "18          6248  \n",
       "19          3013  \n",
       "20         12031  \n",
       "21          6248  \n",
       "22          6248  \n",
       "23          5625  \n",
       "24          5625  \n",
       "25          5625  \n",
       "26          6693  \n",
       "27          1422  \n",
       "28         13862  \n",
       "29          1422  \n",
       "...          ...  \n",
       "3291        1913  \n",
       "3292        1913  \n",
       "3293        3889  \n",
       "3294        5333  \n",
       "3295        4733  \n",
       "3296       18412  \n",
       "3297       15154  \n",
       "3298        6118  \n",
       "3299        6392  \n",
       "3300        9846  \n",
       "3301       11051  \n",
       "3302        5896  \n",
       "3303        6118  \n",
       "3304        5896  \n",
       "3305        6118  \n",
       "3306        1777  \n",
       "3307       14955  \n",
       "3308        6067  \n",
       "3309        6392  \n",
       "3310        9132  \n",
       "3311        7946  \n",
       "3312        4988  \n",
       "3313        6392  \n",
       "3314       14249  \n",
       "3315       11857  \n",
       "3316       11112  \n",
       "3317        6118  \n",
       "3318        5119  \n",
       "3319        4913  \n",
       "3320        9846  \n",
       "\n",
       "[3321 rows x 17 columns]"
      ]
     },
     "execution_count": 465,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train_new = data_all.iloc[:len(train)]\n",
    "train_new"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "train_2=pd.get_dummies(train_1,columns=[\"Gene\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 145,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "test_2=pd.get_dummies(test_1,columns=[\"Gene\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 146,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#add and remove gene dummies on test set to match train"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 155,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "train_2.to_csv(\"checkpoints_databases/w_working_train.csv\",index=False,encoding=\"utf8\")\n",
    "test_3.to_csv(\"checkpoints_databases/w_working_test.csv\",index=False,encoding=\"utf8\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 488,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.921350579565\n"
     ]
    }
   ],
   "source": [
    "Gene_INPUT_DIM=25\n",
    "svd = TruncatedSVD(n_components=125, n_iter=Gene_INPUT_DIM, random_state=12)\n",
    "\n",
    "\n",
    "\n",
    "one_hot_gene = pd.get_dummies(train['Gene'])\n",
    "truncated_one_hot_gene = svd.fit_transform(one_hot_gene.values)\n",
    "print(svd.explained_variance_ratio_.sum())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 474,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>...</th>\n",
       "      <th>15</th>\n",
       "      <th>16</th>\n",
       "      <th>17</th>\n",
       "      <th>18</th>\n",
       "      <th>19</th>\n",
       "      <th>20</th>\n",
       "      <th>21</th>\n",
       "      <th>22</th>\n",
       "      <th>23</th>\n",
       "      <th>24</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1.931671e-24</td>\n",
       "      <td>-1.600352e-19</td>\n",
       "      <td>1.825182e-20</td>\n",
       "      <td>-7.212122e-22</td>\n",
       "      <td>1.511184e-21</td>\n",
       "      <td>3.909289e-24</td>\n",
       "      <td>-2.613487e-25</td>\n",
       "      <td>-1.948228e-28</td>\n",
       "      <td>4.273498e-28</td>\n",
       "      <td>-1.097003e-29</td>\n",
       "      <td>...</td>\n",
       "      <td>2.388333e-26</td>\n",
       "      <td>1.908868e-26</td>\n",
       "      <td>-1.092443e-25</td>\n",
       "      <td>-1.228134e-25</td>\n",
       "      <td>1.805306e-25</td>\n",
       "      <td>2.782531e-26</td>\n",
       "      <td>8.946231e-24</td>\n",
       "      <td>-2.819339e-23</td>\n",
       "      <td>-1.824502e-23</td>\n",
       "      <td>-1.427539e-23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>6.598803e-21</td>\n",
       "      <td>-9.421152e-16</td>\n",
       "      <td>1.260674e-16</td>\n",
       "      <td>2.947655e-16</td>\n",
       "      <td>-1.344321e-15</td>\n",
       "      <td>-8.430348e-16</td>\n",
       "      <td>-4.814295e-16</td>\n",
       "      <td>1.744864e-13</td>\n",
       "      <td>2.077607e-13</td>\n",
       "      <td>-1.292588e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.976270e-08</td>\n",
       "      <td>6.557776e-09</td>\n",
       "      <td>-7.656762e-08</td>\n",
       "      <td>-6.647731e-07</td>\n",
       "      <td>3.739187e-07</td>\n",
       "      <td>4.182728e-07</td>\n",
       "      <td>1.039008e-05</td>\n",
       "      <td>-3.644007e-07</td>\n",
       "      <td>-7.565462e-06</td>\n",
       "      <td>-9.850639e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>6.509848e-21</td>\n",
       "      <td>-9.544203e-16</td>\n",
       "      <td>-1.625777e-16</td>\n",
       "      <td>1.878908e-16</td>\n",
       "      <td>-1.394798e-15</td>\n",
       "      <td>-8.424694e-16</td>\n",
       "      <td>-4.808551e-16</td>\n",
       "      <td>1.744864e-13</td>\n",
       "      <td>2.077607e-13</td>\n",
       "      <td>-1.292588e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.976270e-08</td>\n",
       "      <td>6.557776e-09</td>\n",
       "      <td>-7.656762e-08</td>\n",
       "      <td>-6.647731e-07</td>\n",
       "      <td>3.739187e-07</td>\n",
       "      <td>4.182728e-07</td>\n",
       "      <td>1.039008e-05</td>\n",
       "      <td>-3.644007e-07</td>\n",
       "      <td>-7.565462e-06</td>\n",
       "      <td>-9.850639e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>6.534243e-21</td>\n",
       "      <td>-9.626240e-16</td>\n",
       "      <td>-3.870358e-16</td>\n",
       "      <td>6.629769e-15</td>\n",
       "      <td>1.754405e-15</td>\n",
       "      <td>-8.755560e-16</td>\n",
       "      <td>-4.850829e-16</td>\n",
       "      <td>1.744864e-13</td>\n",
       "      <td>2.077607e-13</td>\n",
       "      <td>-1.292588e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.976270e-08</td>\n",
       "      <td>6.557776e-09</td>\n",
       "      <td>-7.656762e-08</td>\n",
       "      <td>-6.647731e-07</td>\n",
       "      <td>3.739187e-07</td>\n",
       "      <td>4.182728e-07</td>\n",
       "      <td>1.039008e-05</td>\n",
       "      <td>-3.644007e-07</td>\n",
       "      <td>-7.565462e-06</td>\n",
       "      <td>-9.850639e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>6.510358e-21</td>\n",
       "      <td>-9.545605e-16</td>\n",
       "      <td>-1.666234e-16</td>\n",
       "      <td>3.084059e-16</td>\n",
       "      <td>-1.334077e-15</td>\n",
       "      <td>-8.455937e-16</td>\n",
       "      <td>-4.685449e-16</td>\n",
       "      <td>1.744864e-13</td>\n",
       "      <td>2.077607e-13</td>\n",
       "      <td>-1.292588e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.976270e-08</td>\n",
       "      <td>6.557776e-09</td>\n",
       "      <td>-7.656762e-08</td>\n",
       "      <td>-6.647731e-07</td>\n",
       "      <td>3.739187e-07</td>\n",
       "      <td>4.182728e-07</td>\n",
       "      <td>1.039008e-05</td>\n",
       "      <td>-3.644007e-07</td>\n",
       "      <td>-7.565462e-06</td>\n",
       "      <td>-9.850639e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>6.510372e-21</td>\n",
       "      <td>-9.545717e-16</td>\n",
       "      <td>-1.668053e-16</td>\n",
       "      <td>3.113815e-16</td>\n",
       "      <td>-1.335866e-15</td>\n",
       "      <td>2.642914e-16</td>\n",
       "      <td>-5.166400e-16</td>\n",
       "      <td>1.744857e-13</td>\n",
       "      <td>2.077607e-13</td>\n",
       "      <td>-1.292588e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.976270e-08</td>\n",
       "      <td>6.557776e-09</td>\n",
       "      <td>-7.656762e-08</td>\n",
       "      <td>-6.647731e-07</td>\n",
       "      <td>3.739187e-07</td>\n",
       "      <td>4.182728e-07</td>\n",
       "      <td>1.039008e-05</td>\n",
       "      <td>-3.644007e-07</td>\n",
       "      <td>-7.565462e-06</td>\n",
       "      <td>-9.850639e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>6.510358e-21</td>\n",
       "      <td>-9.545604e-16</td>\n",
       "      <td>-1.666299e-16</td>\n",
       "      <td>3.085683e-16</td>\n",
       "      <td>-1.333880e-15</td>\n",
       "      <td>-8.722547e-16</td>\n",
       "      <td>-2.650174e-16</td>\n",
       "      <td>1.744836e-13</td>\n",
       "      <td>2.077649e-13</td>\n",
       "      <td>-1.292588e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.976270e-08</td>\n",
       "      <td>6.557776e-09</td>\n",
       "      <td>-7.656762e-08</td>\n",
       "      <td>-6.647731e-07</td>\n",
       "      <td>3.739187e-07</td>\n",
       "      <td>4.182728e-07</td>\n",
       "      <td>1.039008e-05</td>\n",
       "      <td>-3.644007e-07</td>\n",
       "      <td>-7.565462e-06</td>\n",
       "      <td>-9.850639e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>6.510357e-21</td>\n",
       "      <td>-9.545605e-16</td>\n",
       "      <td>-1.666224e-16</td>\n",
       "      <td>3.083764e-16</td>\n",
       "      <td>-1.334101e-15</td>\n",
       "      <td>-8.428125e-16</td>\n",
       "      <td>-4.812189e-16</td>\n",
       "      <td>1.762190e-13</td>\n",
       "      <td>2.077205e-13</td>\n",
       "      <td>-1.292604e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.976270e-08</td>\n",
       "      <td>6.557776e-09</td>\n",
       "      <td>-7.656762e-08</td>\n",
       "      <td>-6.647731e-07</td>\n",
       "      <td>3.739187e-07</td>\n",
       "      <td>4.182728e-07</td>\n",
       "      <td>1.039008e-05</td>\n",
       "      <td>-3.644007e-07</td>\n",
       "      <td>-7.565462e-06</td>\n",
       "      <td>-9.850639e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>6.510357e-21</td>\n",
       "      <td>-9.545605e-16</td>\n",
       "      <td>-1.666224e-16</td>\n",
       "      <td>3.083766e-16</td>\n",
       "      <td>-1.334101e-15</td>\n",
       "      <td>-8.429561e-16</td>\n",
       "      <td>-4.812583e-16</td>\n",
       "      <td>1.738943e-13</td>\n",
       "      <td>2.078936e-13</td>\n",
       "      <td>-1.292626e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.976270e-08</td>\n",
       "      <td>6.557776e-09</td>\n",
       "      <td>-7.656762e-08</td>\n",
       "      <td>-6.647731e-07</td>\n",
       "      <td>3.739187e-07</td>\n",
       "      <td>4.182728e-07</td>\n",
       "      <td>1.039008e-05</td>\n",
       "      <td>-3.644007e-07</td>\n",
       "      <td>-7.565462e-06</td>\n",
       "      <td>-9.850639e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>6.510357e-21</td>\n",
       "      <td>-9.545605e-16</td>\n",
       "      <td>-1.666224e-16</td>\n",
       "      <td>3.083765e-16</td>\n",
       "      <td>-1.334101e-15</td>\n",
       "      <td>-8.428841e-16</td>\n",
       "      <td>-4.813957e-16</td>\n",
       "      <td>1.745164e-13</td>\n",
       "      <td>2.077656e-13</td>\n",
       "      <td>-1.292648e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.976270e-08</td>\n",
       "      <td>6.557776e-09</td>\n",
       "      <td>-7.656762e-08</td>\n",
       "      <td>-6.647731e-07</td>\n",
       "      <td>3.739187e-07</td>\n",
       "      <td>4.182728e-07</td>\n",
       "      <td>1.039008e-05</td>\n",
       "      <td>-3.644007e-07</td>\n",
       "      <td>-7.565462e-06</td>\n",
       "      <td>-9.850639e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>6.510357e-21</td>\n",
       "      <td>-9.545605e-16</td>\n",
       "      <td>-1.666224e-16</td>\n",
       "      <td>3.083765e-16</td>\n",
       "      <td>-1.334101e-15</td>\n",
       "      <td>-8.428847e-16</td>\n",
       "      <td>-4.814162e-16</td>\n",
       "      <td>1.744752e-13</td>\n",
       "      <td>2.077611e-13</td>\n",
       "      <td>-1.292563e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.976270e-08</td>\n",
       "      <td>6.557776e-09</td>\n",
       "      <td>-7.656762e-08</td>\n",
       "      <td>-6.647731e-07</td>\n",
       "      <td>3.739187e-07</td>\n",
       "      <td>4.182728e-07</td>\n",
       "      <td>1.039008e-05</td>\n",
       "      <td>-3.644007e-07</td>\n",
       "      <td>-7.565462e-06</td>\n",
       "      <td>-9.850639e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>6.510357e-21</td>\n",
       "      <td>-9.545605e-16</td>\n",
       "      <td>-1.666224e-16</td>\n",
       "      <td>3.083764e-16</td>\n",
       "      <td>-1.334101e-15</td>\n",
       "      <td>-8.428841e-16</td>\n",
       "      <td>-4.814136e-16</td>\n",
       "      <td>1.744854e-13</td>\n",
       "      <td>2.077612e-13</td>\n",
       "      <td>-1.292583e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.976270e-08</td>\n",
       "      <td>6.557776e-09</td>\n",
       "      <td>-7.656762e-08</td>\n",
       "      <td>-6.647731e-07</td>\n",
       "      <td>3.739187e-07</td>\n",
       "      <td>4.182728e-07</td>\n",
       "      <td>1.039008e-05</td>\n",
       "      <td>-3.644007e-07</td>\n",
       "      <td>-7.565462e-06</td>\n",
       "      <td>-9.850639e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>6.510357e-21</td>\n",
       "      <td>-9.545605e-16</td>\n",
       "      <td>-1.666224e-16</td>\n",
       "      <td>3.083764e-16</td>\n",
       "      <td>-1.334101e-15</td>\n",
       "      <td>-8.428840e-16</td>\n",
       "      <td>-4.814136e-16</td>\n",
       "      <td>1.744857e-13</td>\n",
       "      <td>2.077601e-13</td>\n",
       "      <td>-1.292589e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.976270e-08</td>\n",
       "      <td>6.557777e-09</td>\n",
       "      <td>-7.656762e-08</td>\n",
       "      <td>-6.647731e-07</td>\n",
       "      <td>3.739187e-07</td>\n",
       "      <td>4.182728e-07</td>\n",
       "      <td>1.039008e-05</td>\n",
       "      <td>-3.644007e-07</td>\n",
       "      <td>-7.565462e-06</td>\n",
       "      <td>-9.850639e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>6.510357e-21</td>\n",
       "      <td>-9.545605e-16</td>\n",
       "      <td>-1.666224e-16</td>\n",
       "      <td>3.083764e-16</td>\n",
       "      <td>-1.334101e-15</td>\n",
       "      <td>-8.428840e-16</td>\n",
       "      <td>-4.814131e-16</td>\n",
       "      <td>1.744864e-13</td>\n",
       "      <td>2.077608e-13</td>\n",
       "      <td>-1.292587e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.976270e-08</td>\n",
       "      <td>6.557777e-09</td>\n",
       "      <td>-7.656762e-08</td>\n",
       "      <td>-6.647731e-07</td>\n",
       "      <td>3.739187e-07</td>\n",
       "      <td>4.182728e-07</td>\n",
       "      <td>1.039008e-05</td>\n",
       "      <td>-3.644007e-07</td>\n",
       "      <td>-7.565462e-06</td>\n",
       "      <td>-9.850639e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>6.510357e-21</td>\n",
       "      <td>-9.545605e-16</td>\n",
       "      <td>-1.666224e-16</td>\n",
       "      <td>3.083764e-16</td>\n",
       "      <td>-1.334101e-15</td>\n",
       "      <td>-8.428840e-16</td>\n",
       "      <td>-4.814135e-16</td>\n",
       "      <td>1.744865e-13</td>\n",
       "      <td>2.077603e-13</td>\n",
       "      <td>-1.292587e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.976271e-08</td>\n",
       "      <td>6.557778e-09</td>\n",
       "      <td>-7.656762e-08</td>\n",
       "      <td>-6.647731e-07</td>\n",
       "      <td>3.739187e-07</td>\n",
       "      <td>4.182728e-07</td>\n",
       "      <td>1.039008e-05</td>\n",
       "      <td>-3.644007e-07</td>\n",
       "      <td>-7.565462e-06</td>\n",
       "      <td>-9.850639e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>6.510357e-21</td>\n",
       "      <td>-9.545605e-16</td>\n",
       "      <td>-1.666224e-16</td>\n",
       "      <td>3.083764e-16</td>\n",
       "      <td>-1.334101e-15</td>\n",
       "      <td>-8.428840e-16</td>\n",
       "      <td>-4.814132e-16</td>\n",
       "      <td>1.744863e-13</td>\n",
       "      <td>2.077607e-13</td>\n",
       "      <td>-1.292588e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.976270e-08</td>\n",
       "      <td>6.557777e-09</td>\n",
       "      <td>-7.656763e-08</td>\n",
       "      <td>-6.647731e-07</td>\n",
       "      <td>3.739187e-07</td>\n",
       "      <td>4.182728e-07</td>\n",
       "      <td>1.039008e-05</td>\n",
       "      <td>-3.644007e-07</td>\n",
       "      <td>-7.565462e-06</td>\n",
       "      <td>-9.850639e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>6.510357e-21</td>\n",
       "      <td>-9.545605e-16</td>\n",
       "      <td>-1.666224e-16</td>\n",
       "      <td>3.083764e-16</td>\n",
       "      <td>-1.334101e-15</td>\n",
       "      <td>-8.428840e-16</td>\n",
       "      <td>-4.814132e-16</td>\n",
       "      <td>1.744863e-13</td>\n",
       "      <td>2.077607e-13</td>\n",
       "      <td>-1.292588e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.976270e-08</td>\n",
       "      <td>6.557775e-09</td>\n",
       "      <td>-7.656763e-08</td>\n",
       "      <td>-6.647731e-07</td>\n",
       "      <td>3.739187e-07</td>\n",
       "      <td>4.182728e-07</td>\n",
       "      <td>1.039008e-05</td>\n",
       "      <td>-3.644007e-07</td>\n",
       "      <td>-7.565462e-06</td>\n",
       "      <td>-9.850639e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>6.510357e-21</td>\n",
       "      <td>-9.545605e-16</td>\n",
       "      <td>-1.666224e-16</td>\n",
       "      <td>3.083764e-16</td>\n",
       "      <td>-1.334101e-15</td>\n",
       "      <td>-8.428840e-16</td>\n",
       "      <td>-4.814131e-16</td>\n",
       "      <td>1.744862e-13</td>\n",
       "      <td>2.077608e-13</td>\n",
       "      <td>-1.292588e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.976270e-08</td>\n",
       "      <td>6.557773e-09</td>\n",
       "      <td>-7.656762e-08</td>\n",
       "      <td>-6.647731e-07</td>\n",
       "      <td>3.739187e-07</td>\n",
       "      <td>4.182728e-07</td>\n",
       "      <td>1.039008e-05</td>\n",
       "      <td>-3.644007e-07</td>\n",
       "      <td>-7.565462e-06</td>\n",
       "      <td>-9.850639e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>6.510357e-21</td>\n",
       "      <td>-9.545605e-16</td>\n",
       "      <td>-1.666224e-16</td>\n",
       "      <td>3.083764e-16</td>\n",
       "      <td>-1.334101e-15</td>\n",
       "      <td>-8.428840e-16</td>\n",
       "      <td>-4.814131e-16</td>\n",
       "      <td>1.744861e-13</td>\n",
       "      <td>2.077609e-13</td>\n",
       "      <td>-1.292589e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.976270e-08</td>\n",
       "      <td>6.557770e-09</td>\n",
       "      <td>-7.656763e-08</td>\n",
       "      <td>-6.647731e-07</td>\n",
       "      <td>3.739187e-07</td>\n",
       "      <td>4.182728e-07</td>\n",
       "      <td>1.039008e-05</td>\n",
       "      <td>-3.644007e-07</td>\n",
       "      <td>-7.565462e-06</td>\n",
       "      <td>-9.850639e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>6.510357e-21</td>\n",
       "      <td>-9.545605e-16</td>\n",
       "      <td>-1.666224e-16</td>\n",
       "      <td>3.083764e-16</td>\n",
       "      <td>-1.334101e-15</td>\n",
       "      <td>-8.428840e-16</td>\n",
       "      <td>-4.814132e-16</td>\n",
       "      <td>1.744864e-13</td>\n",
       "      <td>2.077607e-13</td>\n",
       "      <td>-1.292588e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.976270e-08</td>\n",
       "      <td>6.557777e-09</td>\n",
       "      <td>-7.656762e-08</td>\n",
       "      <td>-6.647731e-07</td>\n",
       "      <td>3.739187e-07</td>\n",
       "      <td>4.182728e-07</td>\n",
       "      <td>1.039008e-05</td>\n",
       "      <td>-3.644007e-07</td>\n",
       "      <td>-7.565462e-06</td>\n",
       "      <td>-9.850639e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>6.510357e-21</td>\n",
       "      <td>-9.545605e-16</td>\n",
       "      <td>-1.666224e-16</td>\n",
       "      <td>3.083764e-16</td>\n",
       "      <td>-1.334101e-15</td>\n",
       "      <td>-8.428840e-16</td>\n",
       "      <td>-4.814132e-16</td>\n",
       "      <td>1.744864e-13</td>\n",
       "      <td>2.077607e-13</td>\n",
       "      <td>-1.292588e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.976270e-08</td>\n",
       "      <td>6.557776e-09</td>\n",
       "      <td>-7.656762e-08</td>\n",
       "      <td>-6.647731e-07</td>\n",
       "      <td>3.739187e-07</td>\n",
       "      <td>4.182728e-07</td>\n",
       "      <td>1.039008e-05</td>\n",
       "      <td>-3.644007e-07</td>\n",
       "      <td>-7.565462e-06</td>\n",
       "      <td>-9.850639e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>6.510357e-21</td>\n",
       "      <td>-9.545605e-16</td>\n",
       "      <td>-1.666224e-16</td>\n",
       "      <td>3.083764e-16</td>\n",
       "      <td>-1.334101e-15</td>\n",
       "      <td>-8.428840e-16</td>\n",
       "      <td>-4.814132e-16</td>\n",
       "      <td>1.744864e-13</td>\n",
       "      <td>2.077607e-13</td>\n",
       "      <td>-1.292588e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.976270e-08</td>\n",
       "      <td>6.557776e-09</td>\n",
       "      <td>-7.656762e-08</td>\n",
       "      <td>-6.647731e-07</td>\n",
       "      <td>3.739187e-07</td>\n",
       "      <td>4.182728e-07</td>\n",
       "      <td>1.039008e-05</td>\n",
       "      <td>-3.644007e-07</td>\n",
       "      <td>-7.565462e-06</td>\n",
       "      <td>-9.850639e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>6.510357e-21</td>\n",
       "      <td>-9.545605e-16</td>\n",
       "      <td>-1.666224e-16</td>\n",
       "      <td>3.083764e-16</td>\n",
       "      <td>-1.334101e-15</td>\n",
       "      <td>-8.428840e-16</td>\n",
       "      <td>-4.814132e-16</td>\n",
       "      <td>1.744864e-13</td>\n",
       "      <td>2.077607e-13</td>\n",
       "      <td>-1.292588e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.976270e-08</td>\n",
       "      <td>6.557776e-09</td>\n",
       "      <td>-7.656762e-08</td>\n",
       "      <td>-6.647731e-07</td>\n",
       "      <td>3.739187e-07</td>\n",
       "      <td>4.182728e-07</td>\n",
       "      <td>1.039008e-05</td>\n",
       "      <td>-3.644007e-07</td>\n",
       "      <td>-7.565462e-06</td>\n",
       "      <td>-9.850639e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>6.510357e-21</td>\n",
       "      <td>-9.545605e-16</td>\n",
       "      <td>-1.666224e-16</td>\n",
       "      <td>3.083764e-16</td>\n",
       "      <td>-1.334101e-15</td>\n",
       "      <td>-8.428840e-16</td>\n",
       "      <td>-4.814132e-16</td>\n",
       "      <td>1.744864e-13</td>\n",
       "      <td>2.077607e-13</td>\n",
       "      <td>-1.292588e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.976270e-08</td>\n",
       "      <td>6.557776e-09</td>\n",
       "      <td>-7.656762e-08</td>\n",
       "      <td>-6.647731e-07</td>\n",
       "      <td>3.739187e-07</td>\n",
       "      <td>4.182728e-07</td>\n",
       "      <td>1.039008e-05</td>\n",
       "      <td>-3.644007e-07</td>\n",
       "      <td>-7.565462e-06</td>\n",
       "      <td>-9.850639e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>6.510357e-21</td>\n",
       "      <td>-9.545605e-16</td>\n",
       "      <td>-1.666224e-16</td>\n",
       "      <td>3.083764e-16</td>\n",
       "      <td>-1.334101e-15</td>\n",
       "      <td>-8.428840e-16</td>\n",
       "      <td>-4.814132e-16</td>\n",
       "      <td>1.744864e-13</td>\n",
       "      <td>2.077607e-13</td>\n",
       "      <td>-1.292588e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.976270e-08</td>\n",
       "      <td>6.557776e-09</td>\n",
       "      <td>-7.656762e-08</td>\n",
       "      <td>-6.647731e-07</td>\n",
       "      <td>3.739187e-07</td>\n",
       "      <td>4.182728e-07</td>\n",
       "      <td>1.039008e-05</td>\n",
       "      <td>-3.644007e-07</td>\n",
       "      <td>-7.565462e-06</td>\n",
       "      <td>-9.850639e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>6.510357e-21</td>\n",
       "      <td>-9.545605e-16</td>\n",
       "      <td>-1.666224e-16</td>\n",
       "      <td>3.083764e-16</td>\n",
       "      <td>-1.334101e-15</td>\n",
       "      <td>-8.428840e-16</td>\n",
       "      <td>-4.814132e-16</td>\n",
       "      <td>1.744864e-13</td>\n",
       "      <td>2.077607e-13</td>\n",
       "      <td>-1.292588e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.976270e-08</td>\n",
       "      <td>6.557776e-09</td>\n",
       "      <td>-7.656762e-08</td>\n",
       "      <td>-6.647731e-07</td>\n",
       "      <td>3.739187e-07</td>\n",
       "      <td>4.182728e-07</td>\n",
       "      <td>1.039008e-05</td>\n",
       "      <td>-3.644007e-07</td>\n",
       "      <td>-7.565462e-06</td>\n",
       "      <td>-9.850639e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>4.179436e-36</td>\n",
       "      <td>-6.097422e-31</td>\n",
       "      <td>-6.004571e-32</td>\n",
       "      <td>2.365377e-31</td>\n",
       "      <td>-6.170225e-31</td>\n",
       "      <td>-1.303360e-30</td>\n",
       "      <td>1.318258e-29</td>\n",
       "      <td>5.632740e-27</td>\n",
       "      <td>2.140855e-26</td>\n",
       "      <td>-3.252649e-25</td>\n",
       "      <td>...</td>\n",
       "      <td>-5.441986e-22</td>\n",
       "      <td>-1.647734e-21</td>\n",
       "      <td>1.927727e-21</td>\n",
       "      <td>1.246537e-20</td>\n",
       "      <td>-9.073663e-22</td>\n",
       "      <td>-1.621543e-21</td>\n",
       "      <td>-3.123233e-19</td>\n",
       "      <td>1.220446e-19</td>\n",
       "      <td>1.482846e-18</td>\n",
       "      <td>1.691810e-18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>5.365775e-33</td>\n",
       "      <td>-7.997832e-28</td>\n",
       "      <td>-3.113990e-28</td>\n",
       "      <td>1.130510e-27</td>\n",
       "      <td>-1.888098e-27</td>\n",
       "      <td>-2.182882e-27</td>\n",
       "      <td>2.253275e-27</td>\n",
       "      <td>-4.541894e-24</td>\n",
       "      <td>3.941416e-25</td>\n",
       "      <td>-8.699311e-24</td>\n",
       "      <td>...</td>\n",
       "      <td>2.086698e-19</td>\n",
       "      <td>-3.203639e-19</td>\n",
       "      <td>4.882944e-19</td>\n",
       "      <td>5.813657e-18</td>\n",
       "      <td>1.174506e-18</td>\n",
       "      <td>-4.018295e-18</td>\n",
       "      <td>-5.797398e-17</td>\n",
       "      <td>-1.058983e-16</td>\n",
       "      <td>3.558215e-17</td>\n",
       "      <td>4.617533e-16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>5.380511e-33</td>\n",
       "      <td>-8.018302e-28</td>\n",
       "      <td>-3.098445e-28</td>\n",
       "      <td>1.132430e-27</td>\n",
       "      <td>-1.890351e-27</td>\n",
       "      <td>-2.204075e-27</td>\n",
       "      <td>2.395725e-27</td>\n",
       "      <td>-4.519462e-24</td>\n",
       "      <td>5.923673e-25</td>\n",
       "      <td>-1.019871e-23</td>\n",
       "      <td>...</td>\n",
       "      <td>2.015424e-19</td>\n",
       "      <td>-3.382423e-19</td>\n",
       "      <td>5.060429e-19</td>\n",
       "      <td>5.966223e-18</td>\n",
       "      <td>1.156722e-18</td>\n",
       "      <td>-4.185278e-18</td>\n",
       "      <td>-6.111000e-17</td>\n",
       "      <td>-1.085493e-16</td>\n",
       "      <td>5.246566e-17</td>\n",
       "      <td>4.842430e-16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>5.309321e-33</td>\n",
       "      <td>-7.914965e-28</td>\n",
       "      <td>-3.095246e-28</td>\n",
       "      <td>1.129011e-27</td>\n",
       "      <td>-1.883365e-27</td>\n",
       "      <td>-2.218351e-27</td>\n",
       "      <td>2.476181e-27</td>\n",
       "      <td>-4.574451e-24</td>\n",
       "      <td>7.231830e-25</td>\n",
       "      <td>-1.088885e-23</td>\n",
       "      <td>...</td>\n",
       "      <td>1.955753e-19</td>\n",
       "      <td>-3.490741e-19</td>\n",
       "      <td>5.102199e-19</td>\n",
       "      <td>6.040520e-18</td>\n",
       "      <td>1.199614e-18</td>\n",
       "      <td>-4.428682e-18</td>\n",
       "      <td>-6.211727e-17</td>\n",
       "      <td>-1.130633e-16</td>\n",
       "      <td>5.951330e-17</td>\n",
       "      <td>4.997088e-16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3291</th>\n",
       "      <td>2.736553e-21</td>\n",
       "      <td>-5.314687e-22</td>\n",
       "      <td>-1.351511e-21</td>\n",
       "      <td>1.183130e-17</td>\n",
       "      <td>1.901717e-18</td>\n",
       "      <td>7.440883e-21</td>\n",
       "      <td>6.833630e-21</td>\n",
       "      <td>1.030664e-16</td>\n",
       "      <td>-4.953025e-18</td>\n",
       "      <td>5.255226e-17</td>\n",
       "      <td>...</td>\n",
       "      <td>2.289379e-13</td>\n",
       "      <td>5.132134e-14</td>\n",
       "      <td>1.120862e-12</td>\n",
       "      <td>2.178092e-11</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.346882e-11</td>\n",
       "      <td>4.072145e-11</td>\n",
       "      <td>-3.527472e-12</td>\n",
       "      <td>1.526174e-10</td>\n",
       "      <td>-2.107420e-10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3292</th>\n",
       "      <td>2.736553e-21</td>\n",
       "      <td>-5.314687e-22</td>\n",
       "      <td>-1.351511e-21</td>\n",
       "      <td>1.183130e-17</td>\n",
       "      <td>1.901717e-18</td>\n",
       "      <td>7.440883e-21</td>\n",
       "      <td>6.833630e-21</td>\n",
       "      <td>1.030664e-16</td>\n",
       "      <td>-4.953025e-18</td>\n",
       "      <td>5.255226e-17</td>\n",
       "      <td>...</td>\n",
       "      <td>2.289379e-13</td>\n",
       "      <td>5.132134e-14</td>\n",
       "      <td>1.120862e-12</td>\n",
       "      <td>2.178092e-11</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.346882e-11</td>\n",
       "      <td>4.072145e-11</td>\n",
       "      <td>-3.527472e-12</td>\n",
       "      <td>1.526174e-10</td>\n",
       "      <td>-2.107420e-10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3293</th>\n",
       "      <td>2.736553e-21</td>\n",
       "      <td>-5.314687e-22</td>\n",
       "      <td>-1.351511e-21</td>\n",
       "      <td>1.183130e-17</td>\n",
       "      <td>1.901717e-18</td>\n",
       "      <td>7.440883e-21</td>\n",
       "      <td>6.833630e-21</td>\n",
       "      <td>1.030664e-16</td>\n",
       "      <td>-4.953025e-18</td>\n",
       "      <td>5.255226e-17</td>\n",
       "      <td>...</td>\n",
       "      <td>2.289379e-13</td>\n",
       "      <td>5.132134e-14</td>\n",
       "      <td>1.120862e-12</td>\n",
       "      <td>2.178092e-11</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.346882e-11</td>\n",
       "      <td>4.072145e-11</td>\n",
       "      <td>-3.527472e-12</td>\n",
       "      <td>1.526174e-10</td>\n",
       "      <td>-2.107420e-10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3294</th>\n",
       "      <td>2.736553e-21</td>\n",
       "      <td>-5.314687e-22</td>\n",
       "      <td>-1.351511e-21</td>\n",
       "      <td>1.183130e-17</td>\n",
       "      <td>1.901717e-18</td>\n",
       "      <td>7.440883e-21</td>\n",
       "      <td>6.833630e-21</td>\n",
       "      <td>1.030664e-16</td>\n",
       "      <td>-4.953025e-18</td>\n",
       "      <td>5.255226e-17</td>\n",
       "      <td>...</td>\n",
       "      <td>2.289379e-13</td>\n",
       "      <td>5.132134e-14</td>\n",
       "      <td>1.120862e-12</td>\n",
       "      <td>2.178092e-11</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.346882e-11</td>\n",
       "      <td>4.072145e-11</td>\n",
       "      <td>-3.527472e-12</td>\n",
       "      <td>1.526174e-10</td>\n",
       "      <td>-2.107420e-10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3295</th>\n",
       "      <td>2.736553e-21</td>\n",
       "      <td>-5.314687e-22</td>\n",
       "      <td>-1.351511e-21</td>\n",
       "      <td>1.183130e-17</td>\n",
       "      <td>1.901717e-18</td>\n",
       "      <td>7.440883e-21</td>\n",
       "      <td>6.833630e-21</td>\n",
       "      <td>1.030664e-16</td>\n",
       "      <td>-4.953025e-18</td>\n",
       "      <td>5.255226e-17</td>\n",
       "      <td>...</td>\n",
       "      <td>2.289379e-13</td>\n",
       "      <td>5.132134e-14</td>\n",
       "      <td>1.120862e-12</td>\n",
       "      <td>2.178092e-11</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.346882e-11</td>\n",
       "      <td>4.072145e-11</td>\n",
       "      <td>-3.527472e-12</td>\n",
       "      <td>1.526174e-10</td>\n",
       "      <td>-2.107420e-10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3296</th>\n",
       "      <td>2.736553e-21</td>\n",
       "      <td>-5.314687e-22</td>\n",
       "      <td>-1.351511e-21</td>\n",
       "      <td>1.183130e-17</td>\n",
       "      <td>1.901717e-18</td>\n",
       "      <td>7.440883e-21</td>\n",
       "      <td>6.833630e-21</td>\n",
       "      <td>1.030664e-16</td>\n",
       "      <td>-4.953025e-18</td>\n",
       "      <td>5.255226e-17</td>\n",
       "      <td>...</td>\n",
       "      <td>2.289379e-13</td>\n",
       "      <td>5.132134e-14</td>\n",
       "      <td>1.120862e-12</td>\n",
       "      <td>2.178092e-11</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.346882e-11</td>\n",
       "      <td>4.072145e-11</td>\n",
       "      <td>-3.527472e-12</td>\n",
       "      <td>1.526174e-10</td>\n",
       "      <td>-2.107420e-10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3297</th>\n",
       "      <td>1.094315e-21</td>\n",
       "      <td>-1.665559e-16</td>\n",
       "      <td>-1.156251e-16</td>\n",
       "      <td>1.633978e-16</td>\n",
       "      <td>-5.329973e-16</td>\n",
       "      <td>-4.530368e-16</td>\n",
       "      <td>4.606909e-16</td>\n",
       "      <td>-3.547177e-13</td>\n",
       "      <td>-3.975436e-13</td>\n",
       "      <td>2.473201e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>2.704580e-08</td>\n",
       "      <td>-1.476486e-08</td>\n",
       "      <td>1.292417e-07</td>\n",
       "      <td>1.116902e-06</td>\n",
       "      <td>-6.315828e-07</td>\n",
       "      <td>-6.652809e-07</td>\n",
       "      <td>-1.385228e-05</td>\n",
       "      <td>6.167965e-07</td>\n",
       "      <td>9.738967e-06</td>\n",
       "      <td>1.746492e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3298</th>\n",
       "      <td>1.094315e-21</td>\n",
       "      <td>-1.665559e-16</td>\n",
       "      <td>-1.156251e-16</td>\n",
       "      <td>1.633978e-16</td>\n",
       "      <td>-5.329973e-16</td>\n",
       "      <td>-4.530368e-16</td>\n",
       "      <td>4.606909e-16</td>\n",
       "      <td>-3.547177e-13</td>\n",
       "      <td>-3.975436e-13</td>\n",
       "      <td>2.473201e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>2.704580e-08</td>\n",
       "      <td>-1.476486e-08</td>\n",
       "      <td>1.292417e-07</td>\n",
       "      <td>1.116902e-06</td>\n",
       "      <td>-6.315828e-07</td>\n",
       "      <td>-6.652809e-07</td>\n",
       "      <td>-1.385228e-05</td>\n",
       "      <td>6.167965e-07</td>\n",
       "      <td>9.738967e-06</td>\n",
       "      <td>1.746492e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3299</th>\n",
       "      <td>1.094315e-21</td>\n",
       "      <td>-1.665559e-16</td>\n",
       "      <td>-1.156251e-16</td>\n",
       "      <td>1.633978e-16</td>\n",
       "      <td>-5.329973e-16</td>\n",
       "      <td>-4.530368e-16</td>\n",
       "      <td>4.606909e-16</td>\n",
       "      <td>-3.547177e-13</td>\n",
       "      <td>-3.975436e-13</td>\n",
       "      <td>2.473201e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>2.704580e-08</td>\n",
       "      <td>-1.476486e-08</td>\n",
       "      <td>1.292417e-07</td>\n",
       "      <td>1.116902e-06</td>\n",
       "      <td>-6.315828e-07</td>\n",
       "      <td>-6.652809e-07</td>\n",
       "      <td>-1.385228e-05</td>\n",
       "      <td>6.167965e-07</td>\n",
       "      <td>9.738967e-06</td>\n",
       "      <td>1.746492e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3300</th>\n",
       "      <td>1.094315e-21</td>\n",
       "      <td>-1.665559e-16</td>\n",
       "      <td>-1.156251e-16</td>\n",
       "      <td>1.633978e-16</td>\n",
       "      <td>-5.329973e-16</td>\n",
       "      <td>-4.530368e-16</td>\n",
       "      <td>4.606909e-16</td>\n",
       "      <td>-3.547177e-13</td>\n",
       "      <td>-3.975436e-13</td>\n",
       "      <td>2.473201e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>2.704580e-08</td>\n",
       "      <td>-1.476486e-08</td>\n",
       "      <td>1.292417e-07</td>\n",
       "      <td>1.116902e-06</td>\n",
       "      <td>-6.315828e-07</td>\n",
       "      <td>-6.652809e-07</td>\n",
       "      <td>-1.385228e-05</td>\n",
       "      <td>6.167965e-07</td>\n",
       "      <td>9.738967e-06</td>\n",
       "      <td>1.746492e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3301</th>\n",
       "      <td>1.094315e-21</td>\n",
       "      <td>-1.665559e-16</td>\n",
       "      <td>-1.156251e-16</td>\n",
       "      <td>1.633978e-16</td>\n",
       "      <td>-5.329973e-16</td>\n",
       "      <td>-4.530368e-16</td>\n",
       "      <td>4.606909e-16</td>\n",
       "      <td>-3.547177e-13</td>\n",
       "      <td>-3.975436e-13</td>\n",
       "      <td>2.473201e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>2.704580e-08</td>\n",
       "      <td>-1.476486e-08</td>\n",
       "      <td>1.292417e-07</td>\n",
       "      <td>1.116902e-06</td>\n",
       "      <td>-6.315828e-07</td>\n",
       "      <td>-6.652809e-07</td>\n",
       "      <td>-1.385228e-05</td>\n",
       "      <td>6.167965e-07</td>\n",
       "      <td>9.738967e-06</td>\n",
       "      <td>1.746492e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3302</th>\n",
       "      <td>1.094315e-21</td>\n",
       "      <td>-1.665559e-16</td>\n",
       "      <td>-1.156251e-16</td>\n",
       "      <td>1.633978e-16</td>\n",
       "      <td>-5.329973e-16</td>\n",
       "      <td>-4.530368e-16</td>\n",
       "      <td>4.606909e-16</td>\n",
       "      <td>-3.547177e-13</td>\n",
       "      <td>-3.975436e-13</td>\n",
       "      <td>2.473201e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>2.704580e-08</td>\n",
       "      <td>-1.476486e-08</td>\n",
       "      <td>1.292417e-07</td>\n",
       "      <td>1.116902e-06</td>\n",
       "      <td>-6.315828e-07</td>\n",
       "      <td>-6.652809e-07</td>\n",
       "      <td>-1.385228e-05</td>\n",
       "      <td>6.167965e-07</td>\n",
       "      <td>9.738967e-06</td>\n",
       "      <td>1.746492e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3303</th>\n",
       "      <td>1.094315e-21</td>\n",
       "      <td>-1.665559e-16</td>\n",
       "      <td>-1.156251e-16</td>\n",
       "      <td>1.633978e-16</td>\n",
       "      <td>-5.329973e-16</td>\n",
       "      <td>-4.530368e-16</td>\n",
       "      <td>4.606909e-16</td>\n",
       "      <td>-3.547177e-13</td>\n",
       "      <td>-3.975436e-13</td>\n",
       "      <td>2.473201e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>2.704580e-08</td>\n",
       "      <td>-1.476486e-08</td>\n",
       "      <td>1.292417e-07</td>\n",
       "      <td>1.116902e-06</td>\n",
       "      <td>-6.315828e-07</td>\n",
       "      <td>-6.652809e-07</td>\n",
       "      <td>-1.385228e-05</td>\n",
       "      <td>6.167965e-07</td>\n",
       "      <td>9.738967e-06</td>\n",
       "      <td>1.746492e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3304</th>\n",
       "      <td>1.094315e-21</td>\n",
       "      <td>-1.665559e-16</td>\n",
       "      <td>-1.156251e-16</td>\n",
       "      <td>1.633978e-16</td>\n",
       "      <td>-5.329973e-16</td>\n",
       "      <td>-4.530368e-16</td>\n",
       "      <td>4.606909e-16</td>\n",
       "      <td>-3.547177e-13</td>\n",
       "      <td>-3.975436e-13</td>\n",
       "      <td>2.473201e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>2.704580e-08</td>\n",
       "      <td>-1.476486e-08</td>\n",
       "      <td>1.292417e-07</td>\n",
       "      <td>1.116902e-06</td>\n",
       "      <td>-6.315828e-07</td>\n",
       "      <td>-6.652809e-07</td>\n",
       "      <td>-1.385228e-05</td>\n",
       "      <td>6.167965e-07</td>\n",
       "      <td>9.738967e-06</td>\n",
       "      <td>1.746492e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3305</th>\n",
       "      <td>1.094315e-21</td>\n",
       "      <td>-1.665559e-16</td>\n",
       "      <td>-1.156251e-16</td>\n",
       "      <td>1.633978e-16</td>\n",
       "      <td>-5.329973e-16</td>\n",
       "      <td>-4.530368e-16</td>\n",
       "      <td>4.606909e-16</td>\n",
       "      <td>-3.547177e-13</td>\n",
       "      <td>-3.975436e-13</td>\n",
       "      <td>2.473201e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>2.704580e-08</td>\n",
       "      <td>-1.476486e-08</td>\n",
       "      <td>1.292417e-07</td>\n",
       "      <td>1.116902e-06</td>\n",
       "      <td>-6.315828e-07</td>\n",
       "      <td>-6.652809e-07</td>\n",
       "      <td>-1.385228e-05</td>\n",
       "      <td>6.167965e-07</td>\n",
       "      <td>9.738967e-06</td>\n",
       "      <td>1.746492e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3306</th>\n",
       "      <td>1.094315e-21</td>\n",
       "      <td>-1.665559e-16</td>\n",
       "      <td>-1.156251e-16</td>\n",
       "      <td>1.633978e-16</td>\n",
       "      <td>-5.329973e-16</td>\n",
       "      <td>-4.530368e-16</td>\n",
       "      <td>4.606909e-16</td>\n",
       "      <td>-3.547177e-13</td>\n",
       "      <td>-3.975436e-13</td>\n",
       "      <td>2.473201e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>2.704580e-08</td>\n",
       "      <td>-1.476486e-08</td>\n",
       "      <td>1.292417e-07</td>\n",
       "      <td>1.116902e-06</td>\n",
       "      <td>-6.315828e-07</td>\n",
       "      <td>-6.652809e-07</td>\n",
       "      <td>-1.385228e-05</td>\n",
       "      <td>6.167965e-07</td>\n",
       "      <td>9.738967e-06</td>\n",
       "      <td>1.746492e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3307</th>\n",
       "      <td>1.094315e-21</td>\n",
       "      <td>-1.665559e-16</td>\n",
       "      <td>-1.156251e-16</td>\n",
       "      <td>1.633978e-16</td>\n",
       "      <td>-5.329973e-16</td>\n",
       "      <td>-4.530368e-16</td>\n",
       "      <td>4.606909e-16</td>\n",
       "      <td>-3.547177e-13</td>\n",
       "      <td>-3.975436e-13</td>\n",
       "      <td>2.473201e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>2.704580e-08</td>\n",
       "      <td>-1.476486e-08</td>\n",
       "      <td>1.292417e-07</td>\n",
       "      <td>1.116902e-06</td>\n",
       "      <td>-6.315828e-07</td>\n",
       "      <td>-6.652809e-07</td>\n",
       "      <td>-1.385228e-05</td>\n",
       "      <td>6.167965e-07</td>\n",
       "      <td>9.738967e-06</td>\n",
       "      <td>1.746492e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3308</th>\n",
       "      <td>1.094315e-21</td>\n",
       "      <td>-1.665559e-16</td>\n",
       "      <td>-1.156251e-16</td>\n",
       "      <td>1.633978e-16</td>\n",
       "      <td>-5.329973e-16</td>\n",
       "      <td>-4.530368e-16</td>\n",
       "      <td>4.606909e-16</td>\n",
       "      <td>-3.547177e-13</td>\n",
       "      <td>-3.975436e-13</td>\n",
       "      <td>2.473201e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>2.704580e-08</td>\n",
       "      <td>-1.476486e-08</td>\n",
       "      <td>1.292417e-07</td>\n",
       "      <td>1.116902e-06</td>\n",
       "      <td>-6.315828e-07</td>\n",
       "      <td>-6.652809e-07</td>\n",
       "      <td>-1.385228e-05</td>\n",
       "      <td>6.167965e-07</td>\n",
       "      <td>9.738967e-06</td>\n",
       "      <td>1.746492e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3309</th>\n",
       "      <td>1.094315e-21</td>\n",
       "      <td>-1.665559e-16</td>\n",
       "      <td>-1.156251e-16</td>\n",
       "      <td>1.633978e-16</td>\n",
       "      <td>-5.329973e-16</td>\n",
       "      <td>-4.530368e-16</td>\n",
       "      <td>4.606909e-16</td>\n",
       "      <td>-3.547177e-13</td>\n",
       "      <td>-3.975436e-13</td>\n",
       "      <td>2.473201e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>2.704580e-08</td>\n",
       "      <td>-1.476486e-08</td>\n",
       "      <td>1.292417e-07</td>\n",
       "      <td>1.116902e-06</td>\n",
       "      <td>-6.315828e-07</td>\n",
       "      <td>-6.652809e-07</td>\n",
       "      <td>-1.385228e-05</td>\n",
       "      <td>6.167965e-07</td>\n",
       "      <td>9.738967e-06</td>\n",
       "      <td>1.746492e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3310</th>\n",
       "      <td>1.094315e-21</td>\n",
       "      <td>-1.665559e-16</td>\n",
       "      <td>-1.156251e-16</td>\n",
       "      <td>1.633978e-16</td>\n",
       "      <td>-5.329973e-16</td>\n",
       "      <td>-4.530368e-16</td>\n",
       "      <td>4.606909e-16</td>\n",
       "      <td>-3.547177e-13</td>\n",
       "      <td>-3.975436e-13</td>\n",
       "      <td>2.473201e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>2.704580e-08</td>\n",
       "      <td>-1.476486e-08</td>\n",
       "      <td>1.292417e-07</td>\n",
       "      <td>1.116902e-06</td>\n",
       "      <td>-6.315828e-07</td>\n",
       "      <td>-6.652809e-07</td>\n",
       "      <td>-1.385228e-05</td>\n",
       "      <td>6.167965e-07</td>\n",
       "      <td>9.738967e-06</td>\n",
       "      <td>1.746492e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3311</th>\n",
       "      <td>1.094315e-21</td>\n",
       "      <td>-1.665559e-16</td>\n",
       "      <td>-1.156251e-16</td>\n",
       "      <td>1.633978e-16</td>\n",
       "      <td>-5.329973e-16</td>\n",
       "      <td>-4.530368e-16</td>\n",
       "      <td>4.606909e-16</td>\n",
       "      <td>-3.547177e-13</td>\n",
       "      <td>-3.975436e-13</td>\n",
       "      <td>2.473201e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>2.704580e-08</td>\n",
       "      <td>-1.476486e-08</td>\n",
       "      <td>1.292417e-07</td>\n",
       "      <td>1.116902e-06</td>\n",
       "      <td>-6.315828e-07</td>\n",
       "      <td>-6.652809e-07</td>\n",
       "      <td>-1.385228e-05</td>\n",
       "      <td>6.167965e-07</td>\n",
       "      <td>9.738967e-06</td>\n",
       "      <td>1.746492e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3312</th>\n",
       "      <td>1.094315e-21</td>\n",
       "      <td>-1.665559e-16</td>\n",
       "      <td>-1.156251e-16</td>\n",
       "      <td>1.633978e-16</td>\n",
       "      <td>-5.329973e-16</td>\n",
       "      <td>-4.530368e-16</td>\n",
       "      <td>4.606909e-16</td>\n",
       "      <td>-3.547177e-13</td>\n",
       "      <td>-3.975436e-13</td>\n",
       "      <td>2.473201e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>2.704580e-08</td>\n",
       "      <td>-1.476486e-08</td>\n",
       "      <td>1.292417e-07</td>\n",
       "      <td>1.116902e-06</td>\n",
       "      <td>-6.315828e-07</td>\n",
       "      <td>-6.652809e-07</td>\n",
       "      <td>-1.385228e-05</td>\n",
       "      <td>6.167965e-07</td>\n",
       "      <td>9.738967e-06</td>\n",
       "      <td>1.746492e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3313</th>\n",
       "      <td>1.094315e-21</td>\n",
       "      <td>-1.665559e-16</td>\n",
       "      <td>-1.156251e-16</td>\n",
       "      <td>1.633978e-16</td>\n",
       "      <td>-5.329973e-16</td>\n",
       "      <td>-4.530368e-16</td>\n",
       "      <td>4.606909e-16</td>\n",
       "      <td>-3.547177e-13</td>\n",
       "      <td>-3.975436e-13</td>\n",
       "      <td>2.473201e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>2.704580e-08</td>\n",
       "      <td>-1.476486e-08</td>\n",
       "      <td>1.292417e-07</td>\n",
       "      <td>1.116902e-06</td>\n",
       "      <td>-6.315828e-07</td>\n",
       "      <td>-6.652809e-07</td>\n",
       "      <td>-1.385228e-05</td>\n",
       "      <td>6.167965e-07</td>\n",
       "      <td>9.738967e-06</td>\n",
       "      <td>1.746492e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3314</th>\n",
       "      <td>1.094315e-21</td>\n",
       "      <td>-1.665559e-16</td>\n",
       "      <td>-1.156251e-16</td>\n",
       "      <td>1.633978e-16</td>\n",
       "      <td>-5.329973e-16</td>\n",
       "      <td>-4.530368e-16</td>\n",
       "      <td>4.606909e-16</td>\n",
       "      <td>-3.547177e-13</td>\n",
       "      <td>-3.975436e-13</td>\n",
       "      <td>2.473201e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>2.704580e-08</td>\n",
       "      <td>-1.476486e-08</td>\n",
       "      <td>1.292417e-07</td>\n",
       "      <td>1.116902e-06</td>\n",
       "      <td>-6.315828e-07</td>\n",
       "      <td>-6.652809e-07</td>\n",
       "      <td>-1.385228e-05</td>\n",
       "      <td>6.167965e-07</td>\n",
       "      <td>9.738967e-06</td>\n",
       "      <td>1.746492e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3315</th>\n",
       "      <td>1.094315e-21</td>\n",
       "      <td>-1.665559e-16</td>\n",
       "      <td>-1.156251e-16</td>\n",
       "      <td>1.633978e-16</td>\n",
       "      <td>-5.329973e-16</td>\n",
       "      <td>-4.530368e-16</td>\n",
       "      <td>4.606909e-16</td>\n",
       "      <td>-3.547177e-13</td>\n",
       "      <td>-3.975436e-13</td>\n",
       "      <td>2.473201e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>2.704580e-08</td>\n",
       "      <td>-1.476486e-08</td>\n",
       "      <td>1.292417e-07</td>\n",
       "      <td>1.116902e-06</td>\n",
       "      <td>-6.315828e-07</td>\n",
       "      <td>-6.652809e-07</td>\n",
       "      <td>-1.385228e-05</td>\n",
       "      <td>6.167965e-07</td>\n",
       "      <td>9.738967e-06</td>\n",
       "      <td>1.746492e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3316</th>\n",
       "      <td>1.094315e-21</td>\n",
       "      <td>-1.665559e-16</td>\n",
       "      <td>-1.156251e-16</td>\n",
       "      <td>1.633978e-16</td>\n",
       "      <td>-5.329973e-16</td>\n",
       "      <td>-4.530368e-16</td>\n",
       "      <td>4.606909e-16</td>\n",
       "      <td>-3.547177e-13</td>\n",
       "      <td>-3.975436e-13</td>\n",
       "      <td>2.473201e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>2.704580e-08</td>\n",
       "      <td>-1.476486e-08</td>\n",
       "      <td>1.292417e-07</td>\n",
       "      <td>1.116902e-06</td>\n",
       "      <td>-6.315828e-07</td>\n",
       "      <td>-6.652809e-07</td>\n",
       "      <td>-1.385228e-05</td>\n",
       "      <td>6.167965e-07</td>\n",
       "      <td>9.738967e-06</td>\n",
       "      <td>1.746492e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3317</th>\n",
       "      <td>1.094315e-21</td>\n",
       "      <td>-1.665559e-16</td>\n",
       "      <td>-1.156251e-16</td>\n",
       "      <td>1.633978e-16</td>\n",
       "      <td>-5.329973e-16</td>\n",
       "      <td>-4.530368e-16</td>\n",
       "      <td>4.606909e-16</td>\n",
       "      <td>-3.547177e-13</td>\n",
       "      <td>-3.975436e-13</td>\n",
       "      <td>2.473201e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>2.704580e-08</td>\n",
       "      <td>-1.476486e-08</td>\n",
       "      <td>1.292417e-07</td>\n",
       "      <td>1.116902e-06</td>\n",
       "      <td>-6.315828e-07</td>\n",
       "      <td>-6.652809e-07</td>\n",
       "      <td>-1.385228e-05</td>\n",
       "      <td>6.167965e-07</td>\n",
       "      <td>9.738967e-06</td>\n",
       "      <td>1.746492e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3318</th>\n",
       "      <td>1.094315e-21</td>\n",
       "      <td>-1.665559e-16</td>\n",
       "      <td>-1.156251e-16</td>\n",
       "      <td>1.633978e-16</td>\n",
       "      <td>-5.329973e-16</td>\n",
       "      <td>-4.530368e-16</td>\n",
       "      <td>4.606909e-16</td>\n",
       "      <td>-3.547177e-13</td>\n",
       "      <td>-3.975436e-13</td>\n",
       "      <td>2.473201e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>2.704580e-08</td>\n",
       "      <td>-1.476486e-08</td>\n",
       "      <td>1.292417e-07</td>\n",
       "      <td>1.116902e-06</td>\n",
       "      <td>-6.315828e-07</td>\n",
       "      <td>-6.652809e-07</td>\n",
       "      <td>-1.385228e-05</td>\n",
       "      <td>6.167965e-07</td>\n",
       "      <td>9.738967e-06</td>\n",
       "      <td>1.746492e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3319</th>\n",
       "      <td>1.094315e-21</td>\n",
       "      <td>-1.665559e-16</td>\n",
       "      <td>-1.156251e-16</td>\n",
       "      <td>1.633978e-16</td>\n",
       "      <td>-5.329973e-16</td>\n",
       "      <td>-4.530368e-16</td>\n",
       "      <td>4.606909e-16</td>\n",
       "      <td>-3.547177e-13</td>\n",
       "      <td>-3.975436e-13</td>\n",
       "      <td>2.473201e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>2.704580e-08</td>\n",
       "      <td>-1.476486e-08</td>\n",
       "      <td>1.292417e-07</td>\n",
       "      <td>1.116902e-06</td>\n",
       "      <td>-6.315828e-07</td>\n",
       "      <td>-6.652809e-07</td>\n",
       "      <td>-1.385228e-05</td>\n",
       "      <td>6.167965e-07</td>\n",
       "      <td>9.738967e-06</td>\n",
       "      <td>1.746492e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3320</th>\n",
       "      <td>1.094315e-21</td>\n",
       "      <td>-1.665559e-16</td>\n",
       "      <td>-1.156251e-16</td>\n",
       "      <td>1.633978e-16</td>\n",
       "      <td>-5.329973e-16</td>\n",
       "      <td>-4.530368e-16</td>\n",
       "      <td>4.606909e-16</td>\n",
       "      <td>-3.547177e-13</td>\n",
       "      <td>-3.975436e-13</td>\n",
       "      <td>2.473201e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>2.704580e-08</td>\n",
       "      <td>-1.476486e-08</td>\n",
       "      <td>1.292417e-07</td>\n",
       "      <td>1.116902e-06</td>\n",
       "      <td>-6.315828e-07</td>\n",
       "      <td>-6.652809e-07</td>\n",
       "      <td>-1.385228e-05</td>\n",
       "      <td>6.167965e-07</td>\n",
       "      <td>9.738967e-06</td>\n",
       "      <td>1.746492e-04</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3321 rows × 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                0             1             2             3             4   \\\n",
       "0     1.931671e-24 -1.600352e-19  1.825182e-20 -7.212122e-22  1.511184e-21   \n",
       "1     6.598803e-21 -9.421152e-16  1.260674e-16  2.947655e-16 -1.344321e-15   \n",
       "2     6.509848e-21 -9.544203e-16 -1.625777e-16  1.878908e-16 -1.394798e-15   \n",
       "3     6.534243e-21 -9.626240e-16 -3.870358e-16  6.629769e-15  1.754405e-15   \n",
       "4     6.510358e-21 -9.545605e-16 -1.666234e-16  3.084059e-16 -1.334077e-15   \n",
       "5     6.510372e-21 -9.545717e-16 -1.668053e-16  3.113815e-16 -1.335866e-15   \n",
       "6     6.510358e-21 -9.545604e-16 -1.666299e-16  3.085683e-16 -1.333880e-15   \n",
       "7     6.510357e-21 -9.545605e-16 -1.666224e-16  3.083764e-16 -1.334101e-15   \n",
       "8     6.510357e-21 -9.545605e-16 -1.666224e-16  3.083766e-16 -1.334101e-15   \n",
       "9     6.510357e-21 -9.545605e-16 -1.666224e-16  3.083765e-16 -1.334101e-15   \n",
       "10    6.510357e-21 -9.545605e-16 -1.666224e-16  3.083765e-16 -1.334101e-15   \n",
       "11    6.510357e-21 -9.545605e-16 -1.666224e-16  3.083764e-16 -1.334101e-15   \n",
       "12    6.510357e-21 -9.545605e-16 -1.666224e-16  3.083764e-16 -1.334101e-15   \n",
       "13    6.510357e-21 -9.545605e-16 -1.666224e-16  3.083764e-16 -1.334101e-15   \n",
       "14    6.510357e-21 -9.545605e-16 -1.666224e-16  3.083764e-16 -1.334101e-15   \n",
       "15    6.510357e-21 -9.545605e-16 -1.666224e-16  3.083764e-16 -1.334101e-15   \n",
       "16    6.510357e-21 -9.545605e-16 -1.666224e-16  3.083764e-16 -1.334101e-15   \n",
       "17    6.510357e-21 -9.545605e-16 -1.666224e-16  3.083764e-16 -1.334101e-15   \n",
       "18    6.510357e-21 -9.545605e-16 -1.666224e-16  3.083764e-16 -1.334101e-15   \n",
       "19    6.510357e-21 -9.545605e-16 -1.666224e-16  3.083764e-16 -1.334101e-15   \n",
       "20    6.510357e-21 -9.545605e-16 -1.666224e-16  3.083764e-16 -1.334101e-15   \n",
       "21    6.510357e-21 -9.545605e-16 -1.666224e-16  3.083764e-16 -1.334101e-15   \n",
       "22    6.510357e-21 -9.545605e-16 -1.666224e-16  3.083764e-16 -1.334101e-15   \n",
       "23    6.510357e-21 -9.545605e-16 -1.666224e-16  3.083764e-16 -1.334101e-15   \n",
       "24    6.510357e-21 -9.545605e-16 -1.666224e-16  3.083764e-16 -1.334101e-15   \n",
       "25    6.510357e-21 -9.545605e-16 -1.666224e-16  3.083764e-16 -1.334101e-15   \n",
       "26    4.179436e-36 -6.097422e-31 -6.004571e-32  2.365377e-31 -6.170225e-31   \n",
       "27    5.365775e-33 -7.997832e-28 -3.113990e-28  1.130510e-27 -1.888098e-27   \n",
       "28    5.380511e-33 -8.018302e-28 -3.098445e-28  1.132430e-27 -1.890351e-27   \n",
       "29    5.309321e-33 -7.914965e-28 -3.095246e-28  1.129011e-27 -1.883365e-27   \n",
       "...            ...           ...           ...           ...           ...   \n",
       "3291  2.736553e-21 -5.314687e-22 -1.351511e-21  1.183130e-17  1.901717e-18   \n",
       "3292  2.736553e-21 -5.314687e-22 -1.351511e-21  1.183130e-17  1.901717e-18   \n",
       "3293  2.736553e-21 -5.314687e-22 -1.351511e-21  1.183130e-17  1.901717e-18   \n",
       "3294  2.736553e-21 -5.314687e-22 -1.351511e-21  1.183130e-17  1.901717e-18   \n",
       "3295  2.736553e-21 -5.314687e-22 -1.351511e-21  1.183130e-17  1.901717e-18   \n",
       "3296  2.736553e-21 -5.314687e-22 -1.351511e-21  1.183130e-17  1.901717e-18   \n",
       "3297  1.094315e-21 -1.665559e-16 -1.156251e-16  1.633978e-16 -5.329973e-16   \n",
       "3298  1.094315e-21 -1.665559e-16 -1.156251e-16  1.633978e-16 -5.329973e-16   \n",
       "3299  1.094315e-21 -1.665559e-16 -1.156251e-16  1.633978e-16 -5.329973e-16   \n",
       "3300  1.094315e-21 -1.665559e-16 -1.156251e-16  1.633978e-16 -5.329973e-16   \n",
       "3301  1.094315e-21 -1.665559e-16 -1.156251e-16  1.633978e-16 -5.329973e-16   \n",
       "3302  1.094315e-21 -1.665559e-16 -1.156251e-16  1.633978e-16 -5.329973e-16   \n",
       "3303  1.094315e-21 -1.665559e-16 -1.156251e-16  1.633978e-16 -5.329973e-16   \n",
       "3304  1.094315e-21 -1.665559e-16 -1.156251e-16  1.633978e-16 -5.329973e-16   \n",
       "3305  1.094315e-21 -1.665559e-16 -1.156251e-16  1.633978e-16 -5.329973e-16   \n",
       "3306  1.094315e-21 -1.665559e-16 -1.156251e-16  1.633978e-16 -5.329973e-16   \n",
       "3307  1.094315e-21 -1.665559e-16 -1.156251e-16  1.633978e-16 -5.329973e-16   \n",
       "3308  1.094315e-21 -1.665559e-16 -1.156251e-16  1.633978e-16 -5.329973e-16   \n",
       "3309  1.094315e-21 -1.665559e-16 -1.156251e-16  1.633978e-16 -5.329973e-16   \n",
       "3310  1.094315e-21 -1.665559e-16 -1.156251e-16  1.633978e-16 -5.329973e-16   \n",
       "3311  1.094315e-21 -1.665559e-16 -1.156251e-16  1.633978e-16 -5.329973e-16   \n",
       "3312  1.094315e-21 -1.665559e-16 -1.156251e-16  1.633978e-16 -5.329973e-16   \n",
       "3313  1.094315e-21 -1.665559e-16 -1.156251e-16  1.633978e-16 -5.329973e-16   \n",
       "3314  1.094315e-21 -1.665559e-16 -1.156251e-16  1.633978e-16 -5.329973e-16   \n",
       "3315  1.094315e-21 -1.665559e-16 -1.156251e-16  1.633978e-16 -5.329973e-16   \n",
       "3316  1.094315e-21 -1.665559e-16 -1.156251e-16  1.633978e-16 -5.329973e-16   \n",
       "3317  1.094315e-21 -1.665559e-16 -1.156251e-16  1.633978e-16 -5.329973e-16   \n",
       "3318  1.094315e-21 -1.665559e-16 -1.156251e-16  1.633978e-16 -5.329973e-16   \n",
       "3319  1.094315e-21 -1.665559e-16 -1.156251e-16  1.633978e-16 -5.329973e-16   \n",
       "3320  1.094315e-21 -1.665559e-16 -1.156251e-16  1.633978e-16 -5.329973e-16   \n",
       "\n",
       "                5             6             7             8             9   \\\n",
       "0     3.909289e-24 -2.613487e-25 -1.948228e-28  4.273498e-28 -1.097003e-29   \n",
       "1    -8.430348e-16 -4.814295e-16  1.744864e-13  2.077607e-13 -1.292588e-11   \n",
       "2    -8.424694e-16 -4.808551e-16  1.744864e-13  2.077607e-13 -1.292588e-11   \n",
       "3    -8.755560e-16 -4.850829e-16  1.744864e-13  2.077607e-13 -1.292588e-11   \n",
       "4    -8.455937e-16 -4.685449e-16  1.744864e-13  2.077607e-13 -1.292588e-11   \n",
       "5     2.642914e-16 -5.166400e-16  1.744857e-13  2.077607e-13 -1.292588e-11   \n",
       "6    -8.722547e-16 -2.650174e-16  1.744836e-13  2.077649e-13 -1.292588e-11   \n",
       "7    -8.428125e-16 -4.812189e-16  1.762190e-13  2.077205e-13 -1.292604e-11   \n",
       "8    -8.429561e-16 -4.812583e-16  1.738943e-13  2.078936e-13 -1.292626e-11   \n",
       "9    -8.428841e-16 -4.813957e-16  1.745164e-13  2.077656e-13 -1.292648e-11   \n",
       "10   -8.428847e-16 -4.814162e-16  1.744752e-13  2.077611e-13 -1.292563e-11   \n",
       "11   -8.428841e-16 -4.814136e-16  1.744854e-13  2.077612e-13 -1.292583e-11   \n",
       "12   -8.428840e-16 -4.814136e-16  1.744857e-13  2.077601e-13 -1.292589e-11   \n",
       "13   -8.428840e-16 -4.814131e-16  1.744864e-13  2.077608e-13 -1.292587e-11   \n",
       "14   -8.428840e-16 -4.814135e-16  1.744865e-13  2.077603e-13 -1.292587e-11   \n",
       "15   -8.428840e-16 -4.814132e-16  1.744863e-13  2.077607e-13 -1.292588e-11   \n",
       "16   -8.428840e-16 -4.814132e-16  1.744863e-13  2.077607e-13 -1.292588e-11   \n",
       "17   -8.428840e-16 -4.814131e-16  1.744862e-13  2.077608e-13 -1.292588e-11   \n",
       "18   -8.428840e-16 -4.814131e-16  1.744861e-13  2.077609e-13 -1.292589e-11   \n",
       "19   -8.428840e-16 -4.814132e-16  1.744864e-13  2.077607e-13 -1.292588e-11   \n",
       "20   -8.428840e-16 -4.814132e-16  1.744864e-13  2.077607e-13 -1.292588e-11   \n",
       "21   -8.428840e-16 -4.814132e-16  1.744864e-13  2.077607e-13 -1.292588e-11   \n",
       "22   -8.428840e-16 -4.814132e-16  1.744864e-13  2.077607e-13 -1.292588e-11   \n",
       "23   -8.428840e-16 -4.814132e-16  1.744864e-13  2.077607e-13 -1.292588e-11   \n",
       "24   -8.428840e-16 -4.814132e-16  1.744864e-13  2.077607e-13 -1.292588e-11   \n",
       "25   -8.428840e-16 -4.814132e-16  1.744864e-13  2.077607e-13 -1.292588e-11   \n",
       "26   -1.303360e-30  1.318258e-29  5.632740e-27  2.140855e-26 -3.252649e-25   \n",
       "27   -2.182882e-27  2.253275e-27 -4.541894e-24  3.941416e-25 -8.699311e-24   \n",
       "28   -2.204075e-27  2.395725e-27 -4.519462e-24  5.923673e-25 -1.019871e-23   \n",
       "29   -2.218351e-27  2.476181e-27 -4.574451e-24  7.231830e-25 -1.088885e-23   \n",
       "...            ...           ...           ...           ...           ...   \n",
       "3291  7.440883e-21  6.833630e-21  1.030664e-16 -4.953025e-18  5.255226e-17   \n",
       "3292  7.440883e-21  6.833630e-21  1.030664e-16 -4.953025e-18  5.255226e-17   \n",
       "3293  7.440883e-21  6.833630e-21  1.030664e-16 -4.953025e-18  5.255226e-17   \n",
       "3294  7.440883e-21  6.833630e-21  1.030664e-16 -4.953025e-18  5.255226e-17   \n",
       "3295  7.440883e-21  6.833630e-21  1.030664e-16 -4.953025e-18  5.255226e-17   \n",
       "3296  7.440883e-21  6.833630e-21  1.030664e-16 -4.953025e-18  5.255226e-17   \n",
       "3297 -4.530368e-16  4.606909e-16 -3.547177e-13 -3.975436e-13  2.473201e-11   \n",
       "3298 -4.530368e-16  4.606909e-16 -3.547177e-13 -3.975436e-13  2.473201e-11   \n",
       "3299 -4.530368e-16  4.606909e-16 -3.547177e-13 -3.975436e-13  2.473201e-11   \n",
       "3300 -4.530368e-16  4.606909e-16 -3.547177e-13 -3.975436e-13  2.473201e-11   \n",
       "3301 -4.530368e-16  4.606909e-16 -3.547177e-13 -3.975436e-13  2.473201e-11   \n",
       "3302 -4.530368e-16  4.606909e-16 -3.547177e-13 -3.975436e-13  2.473201e-11   \n",
       "3303 -4.530368e-16  4.606909e-16 -3.547177e-13 -3.975436e-13  2.473201e-11   \n",
       "3304 -4.530368e-16  4.606909e-16 -3.547177e-13 -3.975436e-13  2.473201e-11   \n",
       "3305 -4.530368e-16  4.606909e-16 -3.547177e-13 -3.975436e-13  2.473201e-11   \n",
       "3306 -4.530368e-16  4.606909e-16 -3.547177e-13 -3.975436e-13  2.473201e-11   \n",
       "3307 -4.530368e-16  4.606909e-16 -3.547177e-13 -3.975436e-13  2.473201e-11   \n",
       "3308 -4.530368e-16  4.606909e-16 -3.547177e-13 -3.975436e-13  2.473201e-11   \n",
       "3309 -4.530368e-16  4.606909e-16 -3.547177e-13 -3.975436e-13  2.473201e-11   \n",
       "3310 -4.530368e-16  4.606909e-16 -3.547177e-13 -3.975436e-13  2.473201e-11   \n",
       "3311 -4.530368e-16  4.606909e-16 -3.547177e-13 -3.975436e-13  2.473201e-11   \n",
       "3312 -4.530368e-16  4.606909e-16 -3.547177e-13 -3.975436e-13  2.473201e-11   \n",
       "3313 -4.530368e-16  4.606909e-16 -3.547177e-13 -3.975436e-13  2.473201e-11   \n",
       "3314 -4.530368e-16  4.606909e-16 -3.547177e-13 -3.975436e-13  2.473201e-11   \n",
       "3315 -4.530368e-16  4.606909e-16 -3.547177e-13 -3.975436e-13  2.473201e-11   \n",
       "3316 -4.530368e-16  4.606909e-16 -3.547177e-13 -3.975436e-13  2.473201e-11   \n",
       "3317 -4.530368e-16  4.606909e-16 -3.547177e-13 -3.975436e-13  2.473201e-11   \n",
       "3318 -4.530368e-16  4.606909e-16 -3.547177e-13 -3.975436e-13  2.473201e-11   \n",
       "3319 -4.530368e-16  4.606909e-16 -3.547177e-13 -3.975436e-13  2.473201e-11   \n",
       "3320 -4.530368e-16  4.606909e-16 -3.547177e-13 -3.975436e-13  2.473201e-11   \n",
       "\n",
       "          ...                 15            16            17            18  \\\n",
       "0         ...       2.388333e-26  1.908868e-26 -1.092443e-25 -1.228134e-25   \n",
       "1         ...      -1.976270e-08  6.557776e-09 -7.656762e-08 -6.647731e-07   \n",
       "2         ...      -1.976270e-08  6.557776e-09 -7.656762e-08 -6.647731e-07   \n",
       "3         ...      -1.976270e-08  6.557776e-09 -7.656762e-08 -6.647731e-07   \n",
       "4         ...      -1.976270e-08  6.557776e-09 -7.656762e-08 -6.647731e-07   \n",
       "5         ...      -1.976270e-08  6.557776e-09 -7.656762e-08 -6.647731e-07   \n",
       "6         ...      -1.976270e-08  6.557776e-09 -7.656762e-08 -6.647731e-07   \n",
       "7         ...      -1.976270e-08  6.557776e-09 -7.656762e-08 -6.647731e-07   \n",
       "8         ...      -1.976270e-08  6.557776e-09 -7.656762e-08 -6.647731e-07   \n",
       "9         ...      -1.976270e-08  6.557776e-09 -7.656762e-08 -6.647731e-07   \n",
       "10        ...      -1.976270e-08  6.557776e-09 -7.656762e-08 -6.647731e-07   \n",
       "11        ...      -1.976270e-08  6.557776e-09 -7.656762e-08 -6.647731e-07   \n",
       "12        ...      -1.976270e-08  6.557777e-09 -7.656762e-08 -6.647731e-07   \n",
       "13        ...      -1.976270e-08  6.557777e-09 -7.656762e-08 -6.647731e-07   \n",
       "14        ...      -1.976271e-08  6.557778e-09 -7.656762e-08 -6.647731e-07   \n",
       "15        ...      -1.976270e-08  6.557777e-09 -7.656763e-08 -6.647731e-07   \n",
       "16        ...      -1.976270e-08  6.557775e-09 -7.656763e-08 -6.647731e-07   \n",
       "17        ...      -1.976270e-08  6.557773e-09 -7.656762e-08 -6.647731e-07   \n",
       "18        ...      -1.976270e-08  6.557770e-09 -7.656763e-08 -6.647731e-07   \n",
       "19        ...      -1.976270e-08  6.557777e-09 -7.656762e-08 -6.647731e-07   \n",
       "20        ...      -1.976270e-08  6.557776e-09 -7.656762e-08 -6.647731e-07   \n",
       "21        ...      -1.976270e-08  6.557776e-09 -7.656762e-08 -6.647731e-07   \n",
       "22        ...      -1.976270e-08  6.557776e-09 -7.656762e-08 -6.647731e-07   \n",
       "23        ...      -1.976270e-08  6.557776e-09 -7.656762e-08 -6.647731e-07   \n",
       "24        ...      -1.976270e-08  6.557776e-09 -7.656762e-08 -6.647731e-07   \n",
       "25        ...      -1.976270e-08  6.557776e-09 -7.656762e-08 -6.647731e-07   \n",
       "26        ...      -5.441986e-22 -1.647734e-21  1.927727e-21  1.246537e-20   \n",
       "27        ...       2.086698e-19 -3.203639e-19  4.882944e-19  5.813657e-18   \n",
       "28        ...       2.015424e-19 -3.382423e-19  5.060429e-19  5.966223e-18   \n",
       "29        ...       1.955753e-19 -3.490741e-19  5.102199e-19  6.040520e-18   \n",
       "...       ...                ...           ...           ...           ...   \n",
       "3291      ...       2.289379e-13  5.132134e-14  1.120862e-12  2.178092e-11   \n",
       "3292      ...       2.289379e-13  5.132134e-14  1.120862e-12  2.178092e-11   \n",
       "3293      ...       2.289379e-13  5.132134e-14  1.120862e-12  2.178092e-11   \n",
       "3294      ...       2.289379e-13  5.132134e-14  1.120862e-12  2.178092e-11   \n",
       "3295      ...       2.289379e-13  5.132134e-14  1.120862e-12  2.178092e-11   \n",
       "3296      ...       2.289379e-13  5.132134e-14  1.120862e-12  2.178092e-11   \n",
       "3297      ...       2.704580e-08 -1.476486e-08  1.292417e-07  1.116902e-06   \n",
       "3298      ...       2.704580e-08 -1.476486e-08  1.292417e-07  1.116902e-06   \n",
       "3299      ...       2.704580e-08 -1.476486e-08  1.292417e-07  1.116902e-06   \n",
       "3300      ...       2.704580e-08 -1.476486e-08  1.292417e-07  1.116902e-06   \n",
       "3301      ...       2.704580e-08 -1.476486e-08  1.292417e-07  1.116902e-06   \n",
       "3302      ...       2.704580e-08 -1.476486e-08  1.292417e-07  1.116902e-06   \n",
       "3303      ...       2.704580e-08 -1.476486e-08  1.292417e-07  1.116902e-06   \n",
       "3304      ...       2.704580e-08 -1.476486e-08  1.292417e-07  1.116902e-06   \n",
       "3305      ...       2.704580e-08 -1.476486e-08  1.292417e-07  1.116902e-06   \n",
       "3306      ...       2.704580e-08 -1.476486e-08  1.292417e-07  1.116902e-06   \n",
       "3307      ...       2.704580e-08 -1.476486e-08  1.292417e-07  1.116902e-06   \n",
       "3308      ...       2.704580e-08 -1.476486e-08  1.292417e-07  1.116902e-06   \n",
       "3309      ...       2.704580e-08 -1.476486e-08  1.292417e-07  1.116902e-06   \n",
       "3310      ...       2.704580e-08 -1.476486e-08  1.292417e-07  1.116902e-06   \n",
       "3311      ...       2.704580e-08 -1.476486e-08  1.292417e-07  1.116902e-06   \n",
       "3312      ...       2.704580e-08 -1.476486e-08  1.292417e-07  1.116902e-06   \n",
       "3313      ...       2.704580e-08 -1.476486e-08  1.292417e-07  1.116902e-06   \n",
       "3314      ...       2.704580e-08 -1.476486e-08  1.292417e-07  1.116902e-06   \n",
       "3315      ...       2.704580e-08 -1.476486e-08  1.292417e-07  1.116902e-06   \n",
       "3316      ...       2.704580e-08 -1.476486e-08  1.292417e-07  1.116902e-06   \n",
       "3317      ...       2.704580e-08 -1.476486e-08  1.292417e-07  1.116902e-06   \n",
       "3318      ...       2.704580e-08 -1.476486e-08  1.292417e-07  1.116902e-06   \n",
       "3319      ...       2.704580e-08 -1.476486e-08  1.292417e-07  1.116902e-06   \n",
       "3320      ...       2.704580e-08 -1.476486e-08  1.292417e-07  1.116902e-06   \n",
       "\n",
       "                19            20            21            22            23  \\\n",
       "0     1.805306e-25  2.782531e-26  8.946231e-24 -2.819339e-23 -1.824502e-23   \n",
       "1     3.739187e-07  4.182728e-07  1.039008e-05 -3.644007e-07 -7.565462e-06   \n",
       "2     3.739187e-07  4.182728e-07  1.039008e-05 -3.644007e-07 -7.565462e-06   \n",
       "3     3.739187e-07  4.182728e-07  1.039008e-05 -3.644007e-07 -7.565462e-06   \n",
       "4     3.739187e-07  4.182728e-07  1.039008e-05 -3.644007e-07 -7.565462e-06   \n",
       "5     3.739187e-07  4.182728e-07  1.039008e-05 -3.644007e-07 -7.565462e-06   \n",
       "6     3.739187e-07  4.182728e-07  1.039008e-05 -3.644007e-07 -7.565462e-06   \n",
       "7     3.739187e-07  4.182728e-07  1.039008e-05 -3.644007e-07 -7.565462e-06   \n",
       "8     3.739187e-07  4.182728e-07  1.039008e-05 -3.644007e-07 -7.565462e-06   \n",
       "9     3.739187e-07  4.182728e-07  1.039008e-05 -3.644007e-07 -7.565462e-06   \n",
       "10    3.739187e-07  4.182728e-07  1.039008e-05 -3.644007e-07 -7.565462e-06   \n",
       "11    3.739187e-07  4.182728e-07  1.039008e-05 -3.644007e-07 -7.565462e-06   \n",
       "12    3.739187e-07  4.182728e-07  1.039008e-05 -3.644007e-07 -7.565462e-06   \n",
       "13    3.739187e-07  4.182728e-07  1.039008e-05 -3.644007e-07 -7.565462e-06   \n",
       "14    3.739187e-07  4.182728e-07  1.039008e-05 -3.644007e-07 -7.565462e-06   \n",
       "15    3.739187e-07  4.182728e-07  1.039008e-05 -3.644007e-07 -7.565462e-06   \n",
       "16    3.739187e-07  4.182728e-07  1.039008e-05 -3.644007e-07 -7.565462e-06   \n",
       "17    3.739187e-07  4.182728e-07  1.039008e-05 -3.644007e-07 -7.565462e-06   \n",
       "18    3.739187e-07  4.182728e-07  1.039008e-05 -3.644007e-07 -7.565462e-06   \n",
       "19    3.739187e-07  4.182728e-07  1.039008e-05 -3.644007e-07 -7.565462e-06   \n",
       "20    3.739187e-07  4.182728e-07  1.039008e-05 -3.644007e-07 -7.565462e-06   \n",
       "21    3.739187e-07  4.182728e-07  1.039008e-05 -3.644007e-07 -7.565462e-06   \n",
       "22    3.739187e-07  4.182728e-07  1.039008e-05 -3.644007e-07 -7.565462e-06   \n",
       "23    3.739187e-07  4.182728e-07  1.039008e-05 -3.644007e-07 -7.565462e-06   \n",
       "24    3.739187e-07  4.182728e-07  1.039008e-05 -3.644007e-07 -7.565462e-06   \n",
       "25    3.739187e-07  4.182728e-07  1.039008e-05 -3.644007e-07 -7.565462e-06   \n",
       "26   -9.073663e-22 -1.621543e-21 -3.123233e-19  1.220446e-19  1.482846e-18   \n",
       "27    1.174506e-18 -4.018295e-18 -5.797398e-17 -1.058983e-16  3.558215e-17   \n",
       "28    1.156722e-18 -4.185278e-18 -6.111000e-17 -1.085493e-16  5.246566e-17   \n",
       "29    1.199614e-18 -4.428682e-18 -6.211727e-17 -1.130633e-16  5.951330e-17   \n",
       "...            ...           ...           ...           ...           ...   \n",
       "3291  1.000000e+00  1.346882e-11  4.072145e-11 -3.527472e-12  1.526174e-10   \n",
       "3292  1.000000e+00  1.346882e-11  4.072145e-11 -3.527472e-12  1.526174e-10   \n",
       "3293  1.000000e+00  1.346882e-11  4.072145e-11 -3.527472e-12  1.526174e-10   \n",
       "3294  1.000000e+00  1.346882e-11  4.072145e-11 -3.527472e-12  1.526174e-10   \n",
       "3295  1.000000e+00  1.346882e-11  4.072145e-11 -3.527472e-12  1.526174e-10   \n",
       "3296  1.000000e+00  1.346882e-11  4.072145e-11 -3.527472e-12  1.526174e-10   \n",
       "3297 -6.315828e-07 -6.652809e-07 -1.385228e-05  6.167965e-07  9.738967e-06   \n",
       "3298 -6.315828e-07 -6.652809e-07 -1.385228e-05  6.167965e-07  9.738967e-06   \n",
       "3299 -6.315828e-07 -6.652809e-07 -1.385228e-05  6.167965e-07  9.738967e-06   \n",
       "3300 -6.315828e-07 -6.652809e-07 -1.385228e-05  6.167965e-07  9.738967e-06   \n",
       "3301 -6.315828e-07 -6.652809e-07 -1.385228e-05  6.167965e-07  9.738967e-06   \n",
       "3302 -6.315828e-07 -6.652809e-07 -1.385228e-05  6.167965e-07  9.738967e-06   \n",
       "3303 -6.315828e-07 -6.652809e-07 -1.385228e-05  6.167965e-07  9.738967e-06   \n",
       "3304 -6.315828e-07 -6.652809e-07 -1.385228e-05  6.167965e-07  9.738967e-06   \n",
       "3305 -6.315828e-07 -6.652809e-07 -1.385228e-05  6.167965e-07  9.738967e-06   \n",
       "3306 -6.315828e-07 -6.652809e-07 -1.385228e-05  6.167965e-07  9.738967e-06   \n",
       "3307 -6.315828e-07 -6.652809e-07 -1.385228e-05  6.167965e-07  9.738967e-06   \n",
       "3308 -6.315828e-07 -6.652809e-07 -1.385228e-05  6.167965e-07  9.738967e-06   \n",
       "3309 -6.315828e-07 -6.652809e-07 -1.385228e-05  6.167965e-07  9.738967e-06   \n",
       "3310 -6.315828e-07 -6.652809e-07 -1.385228e-05  6.167965e-07  9.738967e-06   \n",
       "3311 -6.315828e-07 -6.652809e-07 -1.385228e-05  6.167965e-07  9.738967e-06   \n",
       "3312 -6.315828e-07 -6.652809e-07 -1.385228e-05  6.167965e-07  9.738967e-06   \n",
       "3313 -6.315828e-07 -6.652809e-07 -1.385228e-05  6.167965e-07  9.738967e-06   \n",
       "3314 -6.315828e-07 -6.652809e-07 -1.385228e-05  6.167965e-07  9.738967e-06   \n",
       "3315 -6.315828e-07 -6.652809e-07 -1.385228e-05  6.167965e-07  9.738967e-06   \n",
       "3316 -6.315828e-07 -6.652809e-07 -1.385228e-05  6.167965e-07  9.738967e-06   \n",
       "3317 -6.315828e-07 -6.652809e-07 -1.385228e-05  6.167965e-07  9.738967e-06   \n",
       "3318 -6.315828e-07 -6.652809e-07 -1.385228e-05  6.167965e-07  9.738967e-06   \n",
       "3319 -6.315828e-07 -6.652809e-07 -1.385228e-05  6.167965e-07  9.738967e-06   \n",
       "3320 -6.315828e-07 -6.652809e-07 -1.385228e-05  6.167965e-07  9.738967e-06   \n",
       "\n",
       "                24  \n",
       "0    -1.427539e-23  \n",
       "1    -9.850639e-05  \n",
       "2    -9.850639e-05  \n",
       "3    -9.850639e-05  \n",
       "4    -9.850639e-05  \n",
       "5    -9.850639e-05  \n",
       "6    -9.850639e-05  \n",
       "7    -9.850639e-05  \n",
       "8    -9.850639e-05  \n",
       "9    -9.850639e-05  \n",
       "10   -9.850639e-05  \n",
       "11   -9.850639e-05  \n",
       "12   -9.850639e-05  \n",
       "13   -9.850639e-05  \n",
       "14   -9.850639e-05  \n",
       "15   -9.850639e-05  \n",
       "16   -9.850639e-05  \n",
       "17   -9.850639e-05  \n",
       "18   -9.850639e-05  \n",
       "19   -9.850639e-05  \n",
       "20   -9.850639e-05  \n",
       "21   -9.850639e-05  \n",
       "22   -9.850639e-05  \n",
       "23   -9.850639e-05  \n",
       "24   -9.850639e-05  \n",
       "25   -9.850639e-05  \n",
       "26    1.691810e-18  \n",
       "27    4.617533e-16  \n",
       "28    4.842430e-16  \n",
       "29    4.997088e-16  \n",
       "...            ...  \n",
       "3291 -2.107420e-10  \n",
       "3292 -2.107420e-10  \n",
       "3293 -2.107420e-10  \n",
       "3294 -2.107420e-10  \n",
       "3295 -2.107420e-10  \n",
       "3296 -2.107420e-10  \n",
       "3297  1.746492e-04  \n",
       "3298  1.746492e-04  \n",
       "3299  1.746492e-04  \n",
       "3300  1.746492e-04  \n",
       "3301  1.746492e-04  \n",
       "3302  1.746492e-04  \n",
       "3303  1.746492e-04  \n",
       "3304  1.746492e-04  \n",
       "3305  1.746492e-04  \n",
       "3306  1.746492e-04  \n",
       "3307  1.746492e-04  \n",
       "3308  1.746492e-04  \n",
       "3309  1.746492e-04  \n",
       "3310  1.746492e-04  \n",
       "3311  1.746492e-04  \n",
       "3312  1.746492e-04  \n",
       "3313  1.746492e-04  \n",
       "3314  1.746492e-04  \n",
       "3315  1.746492e-04  \n",
       "3316  1.746492e-04  \n",
       "3317  1.746492e-04  \n",
       "3318  1.746492e-04  \n",
       "3319  1.746492e-04  \n",
       "3320  1.746492e-04  \n",
       "\n",
       "[3321 rows x 25 columns]"
      ]
     },
     "execution_count": 474,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.DataFrame(truncated_one_hot_gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
