{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import string\n",
    "import os\n",
    "from collections import Counter\n",
    "\n",
    "import nltk\n",
    "from nltk.tokenize import sent_tokenize, word_tokenize\n",
    "from nltk.corpus import stopwords\n",
    "from nltk.stem import WordNetLemmatizer\n",
    "from sklearn.decomposition import TruncatedSVD\n",
    "\n",
    "import re"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Creating base"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "train = pd.read_csv('..//bases/training_variants')\n",
    "test = pd.read_csv('..//bases/test_variants')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "train_texts = pd.read_csv('..//bases/training_text', sep=\"\\|\\|\", engine='python', header=None, skiprows=1, names=[\"ID\",\"Text\"], encoding = \"utf-8\")\n",
    "test_texts = pd.read_csv('..//bases/test_text', sep=\"\\|\\|\", engine='python', header=None, skiprows=1, names=[\"ID\",\"Text\"], encoding = \"utf-8\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_test=pd.read_csv('..//bases/new_test_variants.csv')\n",
    "new_test_texts = pd.read_csv('..//bases/new_test_text.csv', sep=\"\\|\\|\", engine='python', header=None, skiprows=1, names=[\"ID\",\"Text\"], encoding = \"utf-8\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "train = pd.merge(train, train_texts, how='left', on='ID')\n",
    "test = pd.merge(test, test_texts, how='left', on='ID')\n",
    "new_test_final=pd.merge(new_test,new_test_texts,how=\"left\",on=\"ID\")\n",
    "leaks=pd.read_csv('..//bases/s1_add_train.csv')\n",
    "leaks_1=pd.DataFrame([leaks[\"ID\"],leaks.drop(\"ID\",axis=1).idxmax(axis=1).map(lambda x: x.lstrip('class'))])\n",
    "leaks_2=leaks_1.T\n",
    "leaks_2.columns=[\"ID\",\"Class\"]\n",
    "leaks_3=pd.merge(leaks_2,test[test.ID.isin(leaks_2.ID)])\n",
    "leaks_final=pd.merge(leaks_3,test_texts[test_texts.ID.isin(leaks_3.ID)])\n",
    "train_final=pd.concat([train,leaks_final]) #adding first stage\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "del train,test,leaks,leaks_1,leaks_2,leaks_3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Transform Gene Letter to their abbreviation in order to find them in the text\n",
    "One_to_Three_AA = {'C': 'Cys', 'D': 'Asp', 'S': 'Ser', 'Q': 'Gln', 'K': 'Lys',\n",
    "         'I': 'Ile', 'P': 'Pro', 'T': 'Thr', 'F': 'Phe', 'N': 'Asn', \n",
    "         'G': 'Gly', 'H': 'His', 'L': 'Leu', 'R': 'Arg', 'W': 'Trp', \n",
    "         'A': 'Ala', 'V': 'Val', 'E': 'Glu', 'Y': 'Tyr', 'M': 'Met'}\n",
    "pattern = re.compile('|'.join(One_to_Three_AA.keys()))\n",
    "##### Get variation types by using regex\n",
    "def variation_regex(data, pattern): # if you want to not ignore cases, add extra argument to function\n",
    "    Boolean = [not bool(re.search(pattern, i, re.IGNORECASE)) for i in data.Variation]\n",
    "    data_no_regex = data[Boolean]  # 182 Fusions => 495 over \n",
    "    not_Boolean = [not i for i in Boolean]  \n",
    "    data_regex = data[not_Boolean]\n",
    "    \n",
    "    return (data_regex, data_no_regex)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "#### process the train and test set together\n",
    "data_all = pd.concat((train_final, new_test_final), axis=0, ignore_index=True)\n",
    "data_all_backup = data_all[:] ##### We keep backup because we want dummy variables of Gene & Text \n",
    "# TODO maybe also use Variation function of Gene from a database, and other suggestions. Also can use Count_sub as feature\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(4675, 5)"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_all.shape"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Functions for pre-processing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "def find_sub(data):\n",
    "\n",
    "    ##### The normal case is around 2080 out of the 2644\n",
    "    \n",
    "    \n",
    "    Boolean = [data.Variation[i] in data.Text[i] or #normal case\n",
    "               data.Variation[i][:-1] in data.Text[i] or #case 1.\n",
    "               pattern.sub(lambda x: One_to_Three_AA[x.group()], data.Variation[i][:-1]) # case2\n",
    "               in data.Text[i]  for i in data.index] ## because new indexing we use \n",
    "    \n",
    "    #TODO could also match insensitive as a next step for more info.\n",
    "    #Shorter Boolean below = the normal version\n",
    "    \n",
    "    #Boolean = [trainSub.Variation[i] in trainSub.Text[i] #normal case\n",
    "    #           for i in trainSub.ID] ## because new indexing we use ID\n",
    "    #           \n",
    "            \n",
    "    sub_in_text = data[Boolean]\n",
    "    not_Boolean = [not i for i in Boolean]  \n",
    "\n",
    "    sub_not_in_text = data[not_Boolean]\n",
    "#    sub_in_text['Count'] = [sub_in_text.Text[i].count(sub_in_text.Variation[i][:-1])\n",
    "#                    +sub_in_text.Text[i].count(pattern.sub(lambda x: One_to_Three_AA[x.group()], sub_in_text.Variation[i][:-1]))\n",
    "#                    for i in sub_in_text.index]\n",
    "    \n",
    "    return sub_in_text, sub_not_in_text\n",
    "##### For subs that are not find in text: use regex to account for a different number\n",
    "##### TODO: things you can further try - with AA name replacement, searching for the number only etc.\n",
    "def find_sub_noText(data):\n",
    "    Booleans = []\n",
    "    for i in data.index:\n",
    "        split_variation = re.split('(\\d+)', data.Variation[i]) # split based on a number\n",
    "        first_Amino = re.escape(split_variation[0]) #re.escpae uses variable as regex\n",
    "        last_Amino = re.escape(split_variation[-1])\n",
    "        #first_number = re.escape(split_variation[1][0])\n",
    "        #new_regex = r\"[^a-zA-Z0-9]\" + first_Amino + first_number\n",
    "        new_regex  = first_Amino + r\"\\d+\" + last_Amino\n",
    "        Boolean = bool(re.search(new_regex, data.Text[i]))\n",
    "        Booleans.append(Boolean)\n",
    "    \n",
    "    sub_number_in_text = data[Booleans]\n",
    "    not_Boolean = [not i for i in Booleans]  \n",
    "\n",
    "    sub_again_no_text = data[not_Boolean]\n",
    "    return sub_again_no_text, sub_number_in_text\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "#### Converts list of sentences into one string of sentences for each document => to use for tfidf etc.\n",
    "def sentences_to_string(sentences_list):\n",
    "    sentence_strings = []\n",
    "    for sentences in sentences_list:\n",
    "        sentence_string =  ' '.join(str(sentence) for sentence in sentences)\n",
    "        sentence_strings.append(sentence_string)\n",
    "    \n",
    "    return sentence_strings ### This doesn't take such a long time to run\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Subtitutions (subs) processing of data set "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "######### First find those that have the format of being a substitution in data\n",
    "data_all['Substitutions_var'] = data_all.Variation.apply(lambda x: bool(re.search('^[A-Z]\\\\d+[A-Z*]$', x))*1) #multiplying by 1 converts True to 1, False to 0 => Maybe modify this later?\n",
    "data_all['Stop_codon_var'] = data_all.Variation.apply(lambda x: bool(re.search('[*]', x))*1) #multiplying by 1 converts True to 1, False to 0\n",
    "data_sub = data_all[data_all['Substitutions_var']==1] ### Now we know the index of where a substitution occurs - the data_sub"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "sub_in_text, sub_not_in_text = find_sub(data_sub)\n",
    "sub_in_text_backup = sub_in_text[:] ## index gets changed by text_processing if we don't make a copy\n",
    "##### INVESTIGATION: Why do some subs don't appear in Text?: Try to automize this and find out\n",
    "### Substitutions can appear as SAME_PREFIX - Other number - SAME_SUFFIX\n",
    "\n",
    "sub_again_no_Text, sub_noText = find_sub_noText(sub_not_in_text) # 108 such cases out of 411 = nice improvement\n",
    "sub_noText_backup = sub_noText[:]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Working on variations who have the 2 letters right but not same numbers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "#nltk.download(\"popular\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [],
   "source": [
    "##############################################################################################################################\n",
    "############## Non-subs preprocessing of data set #######################\n",
    "\n",
    "#def find_mutation_type(row, pattern):  ##### TODO: make clearer by using a function instead of lambda\n",
    "#    return bool(re.search('^fusion', row, re.IGNORECASE)) *1. Also for subs\n",
    "\n",
    "####### Fusions : 'Fusions' ############\n",
    "data_all['Fusion_var'] = data_all.Variation.apply(lambda x: bool(re.search('^fusion', x, re.IGNORECASE))*1) #multiplying by 1 converts True to 1, False to 0\n",
    "new_fusion, new_data_all = variation_regex(data_all, '^fusion') \n",
    "\n",
    "###### Fusions: 'Gene-Gene fusion' ########\n",
    "data_all['gene_fusion_var'] = new_data_all.Variation.apply(lambda x: bool(re.search('fusion', x, re.IGNORECASE))*1) \n",
    "_ , new_data_all = variation_regex(new_data_all, 'fusion') \n",
    "###### Notice that NaN introduced for places where splicing occured => replace after NaN with 0's when complete\n",
    "\n",
    "####### Deletions: 'Deletions' ############\n",
    "data_all['Deletion_var'] = new_data_all.Variation.apply(lambda x: bool(re.search('^del', x, re.IGNORECASE))*1) \n",
    "new_del, new_data_all = variation_regex(new_data_all, '^del') \n",
    "\n",
    "####### Deletions & Insertions wheteher together or seperately (doesn't make a big difference IMO)\n",
    "data_all['del_or_ins_var'] = new_data_all.Variation.apply(lambda x: bool(re.search('del|ins', x, re.IGNORECASE))*1) \n",
    "# we could also later divide it into del, ins if we want to\n",
    "\n",
    "###### Amplifications #########\n",
    "data_all['Amplification_var'] = data_all.Variation.apply(lambda x: bool(re.search('ampl', x, re.IGNORECASE))*1) \n",
    "\n",
    "###### Truncations ########### Don't forget there are 'Truncating mutations' = 95 and '_trunc' = 4\n",
    "data_all['Truncation_var'] = data_all.Variation.apply(lambda x: bool(re.search('trunc', x, re.IGNORECASE))*1) \n",
    "\n",
    "####### Exons #########\n",
    "data_all['exon_var'] = data_all.Variation.apply(lambda x: bool(re.search('exon', x, re.IGNORECASE))*1) \n",
    "\n",
    "####### Frameshift mutations ########\n",
    "data_all['frameshift_var'] = data_all.Variation.apply(lambda x: bool(re.search('fs', x, re.IGNORECASE))*1) \n",
    "\n",
    "####### Duplications ##############\n",
    "data_all['dup_var'] = data_all.Variation.apply(lambda x: bool(re.search('dup', x, re.IGNORECASE))*1) \n",
    "\n",
    "data_all.fillna(0, inplace = True)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [],
   "source": [
    "#TODO : #Sentence Tokenizer for non-subs that are not in text, AND\n",
    "#subs that are not in text"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Not used\n",
    "################ The dummy variables for Gene and Text ##################\n",
    "## TODO: also use dummy for Text? There are 135 shared Genes and 142 shared Text between train and Leaks!  len(set(train.Text) & set(Leaks.Text))\n",
    "#data_all_dummy = data_all_backup[['Gene', 'Text']] # drop those columns we don't need as dummy.\n",
    "#X_dummy = pd.get_dummies(data_all_dummy) # converts categorical variables into dummy variable. From len set => 269 genes + 2090 texts\n",
    "#X_dummy_train = X_dummy[:train.shape[0]]\n",
    "#X_dummy_test = X_dummy[train.shape[0]:]\n",
    "#dummy_names = X_dummy.columns.values #### To remember names if you want to check again what Gene or Text used\n",
    "#X_dummy = X_dummy.values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "###### Use the variation types \n",
    "#variation_types = data_all.drop(['ID', 'Gene', 'Class', 'Text', 'Variation'], axis =1)\n",
    "#X_variation_train = variation_types[:train.shape[0]]\n",
    "#X_variation_test = variation_types[train.shape[0]:]\n",
    "#variation_names = variation_types.columns.values "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Cleaning "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "stop = set(stopwords.words('english'))\n",
    "exclude = set('!\"#$%&\\'()*+:;<=>?@[\\\\]^_`{|}~0123456789') \n",
    "lemma = WordNetLemmatizer()\n",
    "def clean(doc,lemmatiz=False):\n",
    "    stop_free = \" \".join([i for i in doc.lower().split() if i not in stop])\n",
    "    punc_free_0 = [re.sub(\",|\\.|/\",\" \",ch) for ch in stop_free]\n",
    "    if lemmatiz:\n",
    "        punc_free_lem=\"\".join(ch for ch in punc_free_0 if ch not in exclude)\n",
    "        normalized = \" \".join(lemma.lemmatize(word) for word in punc_free_lem.split())\n",
    "        return normalized\n",
    "    else:\n",
    "        punc_free = \"\".join(ch for ch in punc_free_0 if ch not in exclude)\n",
    "        return punc_free"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "#No lemmatization for the moment, be careful not to lemmatize then w2vec\n",
    "data_all.Text = [clean(doc) for doc in data_all.Text]  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "abstract\n",
      "background\n",
      "non-small\n",
      "cell\n",
      "lung\n",
      "cancer\n",
      "(nsclc)\n",
      "is\n",
      "a\n",
      "heterogeneous\n",
      "group\n",
      "of\n",
      "disorders\n",
      "with\n",
      "a\n",
      "number\n",
      "of\n",
      "genetic\n",
      "and\n",
      "proteomic\n",
      "alterations.\n",
      "c-cbl\n",
      "is\n",
      "an\n",
      "e3\n",
      "ubiquitin\n",
      "ligase\n",
      "and\n",
      "adaptor\n",
      "molecule\n",
      "important\n",
      "in\n",
      "normal\n",
      "homeostasis\n",
      "and\n",
      "cancer.\n",
      "we\n",
      "determined\n",
      "the\n",
      "genetic\n",
      "variations\n",
      "of\n",
      "c-cbl,\n",
      "relationship\n",
      "to\n",
      "receptor\n",
      "tyrosine\n",
      "kinases\n",
      "(egfr\n",
      "and\n",
      "met),\n",
      "and\n",
      "functionality\n",
      "in\n",
      "nsclc.\n",
      "methods\n",
      "and\n",
      "findings\n",
      "using\n",
      "archival\n",
      "formalin-fixed\n",
      "paraffin\n",
      "embedded\n",
      "(ffpe)\n",
      "extracted\n",
      "genomic\n",
      "dna,\n",
      "we\n",
      "show\n",
      "that\n",
      "c-cbl\n",
      "mutations\n",
      "occur\n",
      "in\n",
      "somatic\n",
      "fashion\n",
      "for\n",
      "lung\n",
      "cancers.\n",
      "c-cbl\n",
      "mutations\n",
      "were\n",
      "not\n",
      "mutually\n",
      "exclusive\n",
      "of\n",
      "met\n",
      "or\n",
      "egfr\n",
      "mutations;\n",
      "however\n",
      "they\n",
      "were\n",
      "independent\n",
      "of\n",
      "p53\n",
      "and\n",
      "kras\n",
      "mutations.\n",
      "in\n",
      "normal/tumor\n",
      "pairwise\n",
      "analysis,\n",
      "there\n",
      "was\n",
      "significant\n",
      "loss\n",
      "of\n",
      "heterozygosity\n",
      "(loh)\n",
      "for\n",
      "the\n",
      "c-cbl\n",
      "locus\n",
      "(22%,\n",
      "n\n",
      "=\n",
      "8/37)\n",
      "and\n",
      "none\n",
      "of\n",
      "these\n",
      "samples\n",
      "revealed\n",
      "any\n",
      "mutation\n",
      "in\n",
      "the\n",
      "remaining\n",
      "copy\n",
      "of\n",
      "c-cbl.\n",
      "the\n",
      "c-cbl\n",
      "loh\n",
      "also\n",
      "positively\n",
      "correlated\n",
      "with\n",
      "egfr\n",
      "and\n",
      "met\n",
      "mutations\n",
      "observed\n",
      "in\n",
      "the\n",
      "same\n",
      "samples.\n",
      "using\n",
      "select\n",
      "c-cbl\n",
      "somatic\n",
      "mutations\n",
      "such\n",
      "as\n",
      "s80n/h94y,\n",
      "q249e\n",
      "and\n",
      "w802*\n",
      "(obtained\n",
      "from\n",
      "caucasian,\n",
      "taiwanese\n",
      "and\n",
      "african-american\n",
      "samples,\n",
      "respectively)\n",
      "transfected\n",
      "in\n",
      "nsclc\n",
      "cell\n",
      "lines,\n",
      "there\n",
      "was\n",
      "increased\n",
      "cell\n",
      "viability\n",
      "and\n",
      "cell\n",
      "motility.\n",
      "conclusions\n",
      "taking\n",
      "the\n",
      "overall\n",
      "mutation\n",
      "rate\n",
      "of\n",
      "c-cbl\n",
      "to\n",
      "be\n",
      "a\n",
      "combination\n",
      "as\n",
      "somatic\n",
      "missense\n",
      "mutation\n",
      "and\n",
      "loh,\n",
      "it\n",
      "is\n",
      "clear\n",
      "that\n",
      "c-cbl\n",
      "is\n",
      "highly\n",
      "mutated\n",
      "in\n",
      "lung\n",
      "cancers\n",
      "and\n",
      "may\n",
      "play\n",
      "an\n",
      "essential\n",
      "role\n",
      "in\n",
      "lung\n",
      "tumorigenesis\n",
      "and\n",
      "metastasis.\n",
      "go\n",
      "to:\n",
      "introduction\n",
      "in\n",
      "the\n",
      "us\n",
      "alone,\n",
      "each\n",
      "year\n",
      "approximately\n",
      "219,400\n",
      "people\n",
      "are\n",
      "diagnosed\n",
      "with\n",
      "lung\n",
      "cancers,\n",
      "out\n",
      "of\n",
      "which\n",
      "more\n",
      "than\n",
      "145,000\n",
      "of\n",
      "them\n",
      "succumb\n",
      "to\n",
      "the\n",
      "disease\n",
      "[1].\n",
      "this\n",
      "number\n",
      "is\n",
      "roughly\n",
      "equivalent\n",
      "to\n",
      "the\n",
      "combined\n",
      "mortality\n",
      "rates\n",
      "of\n",
      "cancers\n",
      "of\n",
      "the\n",
      "breast,\n",
      "prostate,\n",
      "colon,\n",
      "liver,\n",
      "kidney\n",
      "and\n",
      "melanoma\n",
      "[1].\n",
      "in\n",
      "addition\n",
      "the\n",
      "prognosis\n",
      "is\n",
      "usually\n",
      "poor\n",
      "and\n",
      "the\n",
      "five-year\n",
      "survival\n",
      "rate\n",
      "is\n",
      "less\n",
      "than\n",
      "15%.\n",
      "there\n",
      "are\n",
      "also\n",
      "significant\n",
      "ethnic\n",
      "differences\n",
      "for\n",
      "lung\n",
      "cancer,\n",
      "and\n",
      "the\n",
      "outcome\n",
      "is\n",
      "worse\n",
      "for\n",
      "blacks\n",
      "compared\n",
      "to\n",
      "whites.\n",
      "gender\n",
      "differences\n",
      "are\n",
      "also\n",
      "striking\n",
      "with\n",
      "women\n",
      "having\n",
      "significantly\n",
      "better\n",
      "prognosis\n",
      "as\n",
      "compared\n",
      "to\n",
      "men.\n",
      "there\n",
      "are\n",
      "a\n",
      "number\n",
      "of\n",
      "genetic\n",
      "alterations\n",
      "that\n",
      "can\n",
      "occur\n",
      "in\n",
      "lung\n",
      "cancer.\n",
      "as\n",
      "an\n",
      "example,\n",
      "in\n",
      "nsclc,\n",
      "mutations\n",
      "in\n",
      "kras,\n",
      "p53,\n",
      "egfr\n",
      "and\n",
      "met\n",
      "have\n",
      "been\n",
      "identified.\n",
      "many\n",
      "of\n",
      "these\n",
      "pathways,\n",
      "especially\n",
      "receptor\n",
      "tyrosine\n",
      "kinases\n",
      "(rtks)\n",
      "are\n",
      "controlled\n",
      "by\n",
      "c-cbl.\n",
      "cbl\n",
      "(casitas\n",
      "b-lineage\n",
      "lymphoma)\n",
      "is\n",
      "a\n",
      "mammalian\n",
      "gene\n",
      "located\n",
      "on\n",
      "human\n",
      "chromosome\n",
      "11q23.3\n",
      "[2]\n",
      "and\n",
      "is\n",
      "involved\n",
      "in\n",
      "cell\n",
      "signaling\n",
      "and\n",
      "protein\n",
      "ubiquitination\n",
      "[3].\n",
      "cbl\n",
      "proteins\n",
      "belong\n",
      "to\n",
      "the\n",
      "ring\n",
      "finger\n",
      "class\n",
      "of\n",
      "ubiquitin\n",
      "ligases\n",
      "(e3)\n",
      "and\n",
      "there\n",
      "are\n",
      "three\n",
      "homologues\n",
      "c-cbl,\n",
      "cbl-b,\n",
      "cbl-3\n",
      "[4].\n",
      "the\n",
      "c-cbl\n",
      "and\n",
      "cbl-b\n",
      "genes\n",
      "are\n",
      "ubiquitously\n",
      "expressed\n",
      "with\n",
      "the\n",
      "highest\n",
      "levels\n",
      "in\n",
      "hematopoietic\n",
      "tissues\n",
      "[5].\n",
      "c-cbl\n",
      "consists\n",
      "of\n",
      "four\n",
      "regions\n",
      "encoding\n",
      "for\n",
      "functionally\n",
      "distinct\n",
      "protein\n",
      "domains:\n",
      "the\n",
      "n-terminal\n",
      "tyrosine\n",
      "kinase\n",
      "binding\n",
      "(tkb)\n",
      "domain,\n",
      "the\n",
      "linker\n",
      "region,\n",
      "the\n",
      "catalytic\n",
      "ring\n",
      "finger\n",
      "domain,\n",
      "the\n",
      "proline-rich\n",
      "region\n",
      "and\n",
      "the\n",
      "c-terminal\n",
      "ubiquitin-associated\n",
      "(uba)\n",
      "domain\n",
      "that\n",
      "also\n",
      "overlaps\n",
      "with\n",
      "a\n",
      "leucine-zipper\n",
      "(lz)\n",
      "domain\n",
      "[3].\n",
      "both\n",
      "tkb\n",
      "and\n",
      "ring\n",
      "finger\n",
      "domains\n",
      "are\n",
      "essential\n",
      "for\n",
      "ligand-induced\n",
      "ubiquitination\n",
      "of\n",
      "rtks\n",
      "[6],\n",
      "[7],\n",
      "[8],\n",
      "[9].\n",
      "the\n",
      "ring\n",
      "finger\n",
      "domain\n",
      "is\n",
      "required\n",
      "for\n",
      "the\n",
      "recruitment\n",
      "of\n",
      "e2\n",
      "ubiquitin-conjugating\n",
      "enzymes.\n",
      "the\n",
      "tkb\n",
      "domain\n",
      "includes\n",
      "four-helix\n",
      "bundle\n",
      "(4h),\n",
      "a\n",
      "calcium-biding\n",
      "ef\n",
      "hand,\n",
      "and\n",
      "a\n",
      "modified\n",
      "sh2\n",
      "domain,\n",
      "which\n",
      "binds\n",
      "to\n",
      "phosphotyrosine\n",
      "residues\n",
      "[3],\n",
      "[10],\n",
      "[11],\n",
      "[12].\n",
      "in\n",
      "addition,\n",
      "the\n",
      "proline-rich\n",
      "region\n",
      "of\n",
      "c-cbl\n",
      "can\n",
      "associate\n",
      "with\n",
      "the\n",
      "sh3\n",
      "domain\n",
      "of\n",
      "grb2,\n",
      "which\n",
      "can\n",
      "indirectly\n",
      "recruit\n",
      "c-cbl\n",
      "to\n",
      "rtks\n",
      "via\n",
      "the\n",
      "grb2\n",
      "adaptor\n",
      "protein\n",
      "[7],\n",
      "[13],\n",
      "[14].\n",
      "c-cbl\n",
      "also\n",
      "binds\n",
      "to\n",
      "egfr\n",
      "and\n",
      "acts\n",
      "as\n",
      "the\n",
      "e3\n",
      "that\n",
      "targets\n",
      "egfr\n",
      "for\n",
      "ubiquitination\n",
      "and\n",
      "degradation.\n",
      "furthermore,\n",
      "cbl\n",
      "desensitizes\n",
      "egf\n",
      "signaling\n",
      "and\n",
      "opposes\n",
      "cellular\n",
      "proliferation\n",
      "induced\n",
      "by\n",
      "egf\n",
      "[15].\n",
      "egf\n",
      "activation\n",
      "also\n",
      "appears\n",
      "to\n",
      "activate\n",
      "the\n",
      "tyrosine\n",
      "kinase\n",
      "src,\n",
      "which\n",
      "phosphorylates\n",
      "c-cbl\n",
      "and\n",
      "in\n",
      "turn\n",
      "activates\n",
      "the\n",
      "ubiquitination\n",
      "and\n",
      "degradation\n",
      "of\n",
      "egfr\n",
      "[16],\n",
      "[17],\n",
      "[18].\n",
      "a\n",
      "recent\n",
      "study\n",
      "shows\n",
      "that\n",
      "defective\n",
      "endocytosis\n",
      "of\n",
      "egfr\n",
      "is\n",
      "characterized\n",
      "by\n",
      "a\n",
      "deletion\n",
      "mutant\n",
      "and\n",
      "the\n",
      "point\n",
      "mutation\n",
      "l858r,\n",
      "whereby\n",
      "its\n",
      "association\n",
      "with\n",
      "c-cbl\n",
      "and\n",
      "subsequent\n",
      "ubiquitination\n",
      "are\n",
      "impaired\n",
      "[19].\n",
      "recently,\n",
      "the\n",
      "first\n",
      "human\n",
      "c-cbl\n",
      "mutations\n",
      "were\n",
      "reported\n",
      "in\n",
      "acute\n",
      "myeloid\n",
      "leukemia\n",
      "(aml)\n",
      "patients\n",
      "[20].\n",
      "the\n",
      "mutation\n",
      "r420q\n",
      "inhibits\n",
      "fms-like\n",
      "tyrosine\n",
      "kinase\n",
      "3\n",
      "(flt3)\n",
      "internalization\n",
      "and\n",
      "ubiquitination\n",
      "[20].\n",
      "not\n",
      "only\n",
      "can\n",
      "e3\n",
      "activity\n",
      "be\n",
      "important\n",
      "in\n",
      "oncogenesis,\n",
      "c-cbl\n",
      "has\n",
      "a\n",
      "dual\n",
      "but\n",
      "separate\n",
      "function\n",
      "as\n",
      "a\n",
      "signal\n",
      "transduction\n",
      "molecule.\n",
      "we\n",
      "have\n",
      "previously\n",
      "shown\n",
      "that\n",
      "c-cbl\n",
      "is\n",
      "important\n",
      "in\n",
      "binding\n",
      "crkl\n",
      "and\n",
      "bcr/abl\n",
      "in\n",
      "hematopoietic\n",
      "cells.\n",
      "also,\n",
      "it\n",
      "can\n",
      "bind\n",
      "and\n",
      "modulate\n",
      "functions\n",
      "of\n",
      "cytoskeleton\n",
      "by\n",
      "binding\n",
      "to\n",
      "proteins\n",
      "like\n",
      "talin\n",
      "and\n",
      "paxillin.\n",
      "the\n",
      "tkb\n",
      "domain\n",
      "is\n",
      "important\n",
      "in\n",
      "binding\n",
      "to\n",
      "a\n",
      "number\n",
      "of\n",
      "molecules,\n",
      "and\n",
      "they\n",
      "then\n",
      "function\n",
      "in\n",
      "signal\n",
      "transduction.\n",
      "given\n",
      "the\n",
      "critical\n",
      "role\n",
      "of\n",
      "cbl\n",
      "in\n",
      "normal\n",
      "homeostasis\n",
      "and\n",
      "cancer,\n",
      "we\n",
      "hypothesized\n",
      "that\n",
      "it\n",
      "might\n",
      "be\n",
      "mutated\n",
      "in\n",
      "lung\n",
      "cancers.\n",
      "in\n",
      "this\n",
      "study,\n",
      "we\n",
      "report\n",
      "novel\n",
      "c-cbl\n",
      "somatic\n",
      "mutations\n",
      "s80n/h94y,\n",
      "q249e\n",
      "and\n",
      "w802*\n",
      "in\n",
      "caucasian,\n",
      "taiwanese\n",
      "and\n",
      "african-american\n",
      "lung\n",
      "cancer\n",
      "patients,\n",
      "respectively.\n",
      "expressing\n",
      "these\n",
      "mutations\n",
      "in\n",
      "nsclc\n",
      "cell\n",
      "lines\n",
      "lead\n",
      "to\n",
      "increased\n",
      "proliferation\n",
      "and\n",
      "cell\n",
      "motility.\n",
      "we\n",
      "show\n",
      "that\n",
      "c-cbl\n",
      "mutations\n",
      "occur\n",
      "with\n",
      "or\n",
      "without\n",
      "met\n",
      "or\n",
      "egfr\n",
      "mutations\n",
      "but\n",
      "are\n",
      "mutually\n",
      "exclusive\n",
      "of\n",
      "a\n",
      "loh\n",
      "at\n",
      "the\n",
      "c-cbl\n",
      "locus.\n",
      "additionally,\n",
      "c-cbl\n",
      "loh\n",
      "is\n",
      "associated\n",
      "with\n",
      "either\n",
      "met\n",
      "or\n",
      "egfr\n",
      "mutations.\n",
      "we\n",
      "thus\n",
      "hypothesize\n",
      "that\n",
      "c-cbl\n",
      "mutations\n",
      "might\n",
      "contribute\n",
      "to\n",
      "the\n",
      "oncogenic\n",
      "potential\n",
      "of\n",
      "met\n",
      "and\n",
      "egfr\n",
      "in\n",
      "lung\n",
      "cancer.\n",
      "go\n",
      "to:\n",
      "methods\n",
      "ethics\n",
      "statement\n",
      "written\n",
      "consent\n",
      "on\n",
      "all\n",
      "research\n",
      "on\n",
      "human\n",
      "subjects\n",
      "has\n",
      "been\n",
      "obtained\n",
      "from\n",
      "the\n",
      "institutional\n",
      "review\n",
      "board,\n",
      "university\n",
      "of\n",
      "chicago\n",
      "and\n",
      "covers\n",
      "all\n",
      "research\n",
      "performed\n",
      "in\n",
      "the\n",
      "laboratory.\n",
      "the\n",
      "following\n",
      "is\n",
      "their\n",
      "contact\n",
      "information:\n",
      "institutional\n",
      "review\n",
      "board,\n",
      "the\n",
      "university\n",
      "of\n",
      "chicago,\n",
      "mcgiffert\n",
      "hall,\n",
      "5751\n",
      "s.\n",
      "woodlawn\n",
      "ave.,\n",
      "2nd\n",
      "floor,\n",
      "chicago,\n",
      "il\n",
      "60637.\n",
      "written\n",
      "informed\n",
      "consents\n",
      "were\n",
      "received\n",
      "from\n",
      "all\n",
      "patients\n",
      "whose\n",
      "tissue\n",
      "samples\n",
      "were\n",
      "used\n",
      "for\n",
      "this\n",
      "study.\n",
      "tissue\n",
      "samples\n",
      "lung\n",
      "cancer\n",
      "tissue\n",
      "and\n",
      "paired\n",
      "adjacent\n",
      "normal\n",
      "lung\n",
      "tissues\n",
      "were\n",
      "obtained\n",
      "from\n",
      "50\n",
      "caucasian,\n",
      "29\n",
      "african-americans\n",
      "and\n",
      "40\n",
      "taiwanese\n",
      "nsclc\n",
      "patients\n",
      "who\n",
      "were\n",
      "recruited\n",
      "at\n",
      "the\n",
      "university\n",
      "of\n",
      "chicago\n",
      "hospital\n",
      "(chicago,\n",
      "usa)\n",
      "(caucasian\n",
      "and\n",
      "african-american\n",
      "patients)\n",
      "and\n",
      "taipei\n",
      "veterans\n",
      "general\n",
      "hospital\n",
      "of\n",
      "taiwan\n",
      "(taiwanese\n",
      "patients)\n",
      "after\n",
      "obtaining\n",
      "appropriate\n",
      "institutional\n",
      "review\n",
      "board\n",
      "permission\n",
      "and\n",
      "informed\n",
      "consent\n",
      "from\n",
      "the\n",
      "patients.\n",
      "out\n",
      "of\n",
      "119\n",
      "samples,\n",
      "77\n",
      "were\n",
      "men,\n",
      "38\n",
      "were\n",
      "women\n",
      "and\n",
      "4\n",
      "were\n",
      "unknown\n",
      "with\n",
      "age\n",
      "at\n",
      "diagnosis\n",
      "ranging\n",
      "from\n",
      "47\n",
      "to\n",
      "90\n",
      "years.\n",
      "in\n",
      "terms\n",
      "of\n",
      "tumor\n",
      "types,\n",
      "53\n",
      "were\n",
      "adenocarcinoma,\n",
      "32\n",
      "were\n",
      "squamous\n",
      "cell\n",
      "carcinoma\n",
      "and\n",
      "34\n",
      "were\n",
      "large\n",
      "cell\n",
      "carcinoma.\n",
      "49\n",
      "were\n",
      "stage\n",
      "i,\n",
      "14\n",
      "were\n",
      "stage\n",
      "ii,\n",
      "34\n",
      "were\n",
      "stage\n",
      "iii,\n",
      "and\n",
      "13\n",
      "were\n",
      "stage\n",
      "iv\n",
      "(table\n",
      "s1).\n",
      "cell\n",
      "culture\n",
      "human\n",
      "non-small\n",
      "cell\n",
      "lung\n",
      "carcinoma\n",
      "cells\n",
      "a549\n",
      "and\n",
      "h358\n",
      "were\n",
      "maintained\n",
      "in\n",
      "dmem\n",
      "and\n",
      "rpmi-1640,\n",
      "respectively.\n",
      "human\n",
      "embryonic\n",
      "kidney\n",
      "293t\n",
      "cells\n",
      "were\n",
      "cultured\n",
      "in\n",
      "dmem.\n",
      "media\n",
      "were\n",
      "supplemented\n",
      "with\n",
      "10%\n",
      "fetal\n",
      "bovine\n",
      "serum,\n",
      "100\n",
      "units/ml\n",
      "of\n",
      "penicillin,\n",
      "and\n",
      "100\n",
      "µg/ml\n",
      "of\n",
      "streptomycin\n",
      "(invitrogen,\n",
      "carlsbad,\n",
      "ca).\n",
      "cells\n",
      "were\n",
      "cultured\n",
      "at\n",
      "37°c\n",
      "in\n",
      "a\n",
      "humidified\n",
      "incubator\n",
      "containing\n",
      "5%\n",
      "co2.\n",
      "c-cbl\n",
      "gene\n",
      "mutational\n",
      "analysis\n",
      "exons\n",
      "2\n",
      "to\n",
      "16\n",
      "of\n",
      "c-cbl\n",
      "gene\n",
      "were\n",
      "individually\n",
      "amplified\n",
      "by\n",
      "polymerase\n",
      "chain\n",
      "reaction\n",
      "(pcr).\n",
      "primers\n",
      "are\n",
      "listed\n",
      "in\n",
      "table\n",
      "s2.\n",
      "pcr\n",
      "conditions\n",
      "were\n",
      "1\n",
      "cycle\n",
      "of\n",
      "95°c\n",
      "for\n",
      "5\n",
      "minutes;\n",
      "35\n",
      "cycles\n",
      "of\n",
      "94°c\n",
      "for\n",
      "30\n",
      "seconds,\n",
      "58°c\n",
      "for\n",
      "30\n",
      "seconds\n",
      "and\n",
      "72°c\n",
      "for\n",
      "2\n",
      "minutes;\n",
      "and\n",
      "one\n",
      "cycle\n",
      "of\n",
      "72°c\n",
      "for\n",
      "10\n",
      "minutes.\n",
      "pcr\n",
      "products\n",
      "were\n",
      "treated\n",
      "with\n",
      "exosap-it\n",
      "(usb\n",
      "corporation,\n",
      "cleveland,\n",
      "oh)\n",
      "and\n",
      "sequenced\n",
      "by\n",
      "big-dye\n",
      "terminator\n",
      "chemistry\n",
      "(applied\n",
      "biosystems,\n",
      "foster\n",
      "city,\n",
      "ca).\n",
      "sequencing\n",
      "was\n",
      "performed\n",
      "on\n",
      "the\n",
      "forward\n",
      "coding\n",
      "strand\n",
      "with\n",
      "confirmation\n",
      "of\n",
      "c-cbl\n",
      "alterations\n",
      "performed\n",
      "by\n",
      "sequencing\n",
      "the\n",
      "reverse\n",
      "strand\n",
      "as\n",
      "well.\n",
      "chromatograms\n",
      "were\n",
      "analyzed\n",
      "for\n",
      "mutations\n",
      "using\n",
      "mutation\n",
      "surveyor\n",
      "v2.61\n",
      "(softgenetics,\n",
      "state\n",
      "college,\n",
      "pa).\n",
      "plasmid\n",
      "constructs\n",
      "and\n",
      "site-directed\n",
      "mutagenesis\n",
      "the\n",
      "wild-type\n",
      "c-cbl\n",
      "cdna\n",
      "insert\n",
      "was\n",
      "subcloned\n",
      "into\n",
      "the\n",
      "paltermax\n",
      "expression\n",
      "vector\n",
      "using\n",
      "xhoi\n",
      "and\n",
      "sali\n",
      "restriction\n",
      "enzyme\n",
      "sites\n",
      "(promega,\n",
      "madison,\n",
      "wi).\n",
      "using\n",
      "this\n",
      "parental\n",
      "plasmid\n",
      "paltermax-c-cbl,\n",
      "the\n",
      "tkb\n",
      "domain\n",
      "double\n",
      "mutation\n",
      "(s80n/h94y),\n",
      "the\n",
      "point\n",
      "mutation\n",
      "(q249e),\n",
      "and\n",
      "the\n",
      "c-terminal\n",
      "point\n",
      "mutation\n",
      "w802*\n",
      "of\n",
      "c-cbl\n",
      "were\n",
      "created\n",
      "using\n",
      "the\n",
      "following\n",
      "primers:\n",
      "5′-gctggcgctaaagaataacccaccttatatcttagac-3′\n",
      "and\n",
      "5′-ctaccagatacctaccagtatctccgtactatcttgtc-3′\n",
      "for\n",
      "the\n",
      "double\n",
      "mutation\n",
      "s80n/h94y;\n",
      "5′-ctttacccgactctttgagccctggtcctctttgc-3′\n",
      "for\n",
      "q249e,\n",
      "and\n",
      "5′-cagctcctcctttggctgattgtctctggatggtgatc-3′\n",
      "for\n",
      "w802*\n",
      "along\n",
      "with\n",
      "their\n",
      "complementary\n",
      "primers\n",
      "using\n",
      "the\n",
      "quickchange\n",
      "site-directed\n",
      "mutagenesis\n",
      "xl\n",
      "kit\n",
      "(stratagene,\n",
      "la\n",
      "jolla,\n",
      "ca)\n",
      "according\n",
      "to\n",
      "the\n",
      "manufacturer's\n",
      "instructions.\n",
      "the\n",
      "constructs\n",
      "were\n",
      "confirmed\n",
      "for\n",
      "the\n",
      "point\n",
      "mutations\n",
      "by\n",
      "standard\n",
      "dna\n",
      "sequencing\n",
      "of\n",
      "both\n",
      "strands.\n",
      "loss\n",
      "of\n",
      "heterozygosity\n",
      "(loh)\n",
      "analysis\n",
      "five\n",
      "microsatellites\n",
      "on\n",
      "chromosome\n",
      "11\n",
      "(3\n",
      "on\n",
      "11q\n",
      "at\n",
      "or\n",
      "within\n",
      "200\n",
      "kb\n",
      "up\n",
      "or\n",
      "downstream\n",
      "of\n",
      "the\n",
      "c-cbl\n",
      "gene\n",
      "and\n",
      "2\n",
      "control\n",
      "markers\n",
      "on\n",
      "11p)\n",
      "were\n",
      "selected\n",
      "for\n",
      "analysis\n",
      "(table\n",
      "s3).\n",
      "established\n",
      "microsatellite\n",
      "markers\n",
      "and\n",
      "respective\n",
      "primer\n",
      "sequences\n",
      "were\n",
      "selected\n",
      "from\n",
      "the\n",
      "geneloc\n",
      "database\n",
      "(http://genecards.weizmann.ac.il/geneloc/index.shtml,\n",
      "weizmann\n",
      "institute\n",
      "of\n",
      "science,\n",
      "rehovot,\n",
      "israel).\n",
      "primers\n",
      "were\n",
      "custom\n",
      "designed\n",
      "and\n",
      "each\n",
      "forward\n",
      "primer\n",
      "was\n",
      "fluorescently\n",
      "labeled\n",
      "at\n",
      "the\n",
      "5′\n",
      "end\n",
      "with\n",
      "fam,\n",
      "pet,\n",
      "ned,\n",
      "or\n",
      "vic\n",
      "(applied\n",
      "biosystems).\n",
      "primer\n",
      "annealing\n",
      "temperatures\n",
      "and\n",
      "duplex\n",
      "scores\n",
      "were\n",
      "evaluated\n",
      "with\n",
      "nist\n",
      "primer\n",
      "tools\n",
      "(http://yellow.nist.gov:8444/dnaanalysis/primertoolspage.do;\n",
      "national\n",
      "institute\n",
      "of\n",
      "standards\n",
      "and\n",
      "technology,\n",
      "gaithersburg,\n",
      "md).\n",
      "primers\n",
      "were\n",
      "verified\n",
      "by\n",
      "performing\n",
      "pcr\n",
      "with\n",
      "control\n",
      "dna\n",
      "(isolated\n",
      "from\n",
      "tk6\n",
      "cells)\n",
      "and\n",
      "resolving\n",
      "the\n",
      "products\n",
      "on\n",
      "agarose\n",
      "gels.\n",
      "bands\n",
      "were\n",
      "visualized\n",
      "with\n",
      "an\n",
      "uv\n",
      "transilluminator.\n",
      "genomic\n",
      "dna\n",
      "was\n",
      "extracted\n",
      "from\n",
      "tumor\n",
      "samples\n",
      "and\n",
      "paired\n",
      "normal\n",
      "lung\n",
      "tissue.\n",
      "primers\n",
      "were\n",
      "grouped\n",
      "into\n",
      "multiplex\n",
      "combinations\n",
      "shown\n",
      "in\n",
      "table\n",
      "s4.\n",
      "marker\n",
      "d11s929\n",
      "served\n",
      "as\n",
      "an\n",
      "internal\n",
      "control\n",
      "to\n",
      "check\n",
      "for\n",
      "consistency\n",
      "in\n",
      "pcrs\n",
      "and\n",
      "of\n",
      "peaks\n",
      "from\n",
      "capillary\n",
      "electrophoresis.\n",
      "multiplex\n",
      "pcrs\n",
      "were\n",
      "carried\n",
      "out\n",
      "in\n",
      "a\n",
      "volume\n",
      "of\n",
      "10\n",
      "µl\n",
      "that\n",
      "contained\n",
      "1\n",
      "µl\n",
      "genomic\n",
      "dna\n",
      "(20–50\n",
      "ng),\n",
      "0.5\n",
      "µm\n",
      "of\n",
      "each\n",
      "primer\n",
      "(1.0\n",
      "µm\n",
      "total\n",
      "for\n",
      "each\n",
      "primer\n",
      "pair),\n",
      "400\n",
      "µm\n",
      "dntps,\n",
      "1x\n",
      "pcr\n",
      "buffer\n",
      "containing\n",
      "mgcl2,\n",
      "and\n",
      "0.2\n",
      "u\n",
      "taq\n",
      "dna\n",
      "polymerase.\n",
      "pcr\n",
      "was\n",
      "performed\n",
      "on\n",
      "the\n",
      "abi\n",
      "geneamp\n",
      "9700\n",
      "pcr\n",
      "system\n",
      "under\n",
      "the\n",
      "following\n",
      "conditions:\n",
      "5\n",
      "min\n",
      "at\n",
      "94°c;\n",
      "30\n",
      "cycles\n",
      "of\n",
      "30\n",
      "sec\n",
      "at\n",
      "94°c,\n",
      "1\n",
      "min\n",
      "at\n",
      "60°c,\n",
      "1\n",
      "min\n",
      "at\n",
      "72°c;\n",
      "and\n",
      "5\n",
      "min\n",
      "at\n",
      "72°c.\n",
      "the\n",
      "pcr\n",
      "products\n",
      "were\n",
      "separated\n",
      "by\n",
      "capillary\n",
      "electrophoresis\n",
      "on\n",
      "an\n",
      "abi\n",
      "3130xl\n",
      "dna\n",
      "analyzer.\n",
      "chromatograms\n",
      "were\n",
      "analyzed\n",
      "with\n",
      "peak\n",
      "scanner\n",
      "1.0\n",
      "and\n",
      "genemapper\n",
      "3.7\n",
      "software\n",
      "(applied\n",
      "biosystems)\n",
      "for\n",
      "allelic\n",
      "alterations.\n",
      "the\n",
      "area\n",
      "of\n",
      "the\n",
      "peaks\n",
      "produced\n",
      "by\n",
      "the\n",
      "dna\n",
      "pcr\n",
      "products\n",
      "was\n",
      "quantified\n",
      "for\n",
      "each\n",
      "allele.\n",
      "the\n",
      "ratio\n",
      "of\n",
      "the\n",
      "allelic\n",
      "areas\n",
      "was\n",
      "calculated\n",
      "for\n",
      "each\n",
      "tumor\n",
      "and\n",
      "paired\n",
      "normal\n",
      "dna\n",
      "sample.\n",
      "when\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "the\n",
      "qloh\n",
      "(allelic\n",
      "ratio\n",
      "for\n",
      "the\n",
      "tumor\n",
      "peaks\n",
      "divided\n",
      "by\n",
      "the\n",
      "allelic\n",
      "ratio\n",
      "of\n",
      "paired\n",
      "normal\n",
      "sample)\n",
      "was\n",
      "≤0.5\n",
      "or\n",
      "≥2.0\n",
      "for\n",
      "c-cbl\n",
      "and\n",
      "at\n",
      "least\n",
      "one\n",
      "other\n",
      "11q\n",
      "marker\n",
      "in\n",
      "at\n",
      "least\n",
      "two\n",
      "separate\n",
      "experiments,\n",
      "the\n",
      "sample\n",
      "was\n",
      "considered\n",
      "as\n",
      "having\n",
      "an\n",
      "allelic\n",
      "imbalance\n",
      "and\n",
      "interpreted\n",
      "as\n",
      "loh.\n",
      "samples\n",
      "were\n",
      "evaluated\n",
      "in\n",
      "at\n",
      "least\n",
      "two\n",
      "separate\n",
      "experiments\n",
      "and\n",
      "samples\n",
      "showing\n",
      "prospective\n",
      "loh\n",
      "at\n",
      "c-cbl\n",
      "repeated\n",
      "a\n",
      "third\n",
      "time\n",
      "which\n",
      "included\n",
      "a\n",
      "new\n",
      "control\n",
      "marker\n",
      "at\n",
      "the\n",
      "bax\n",
      "locus\n",
      "(data\n",
      "not\n",
      "shown)\n",
      "on\n",
      "chromosome\n",
      "19\n",
      "to\n",
      "verify\n",
      "integrity\n",
      "of\n",
      "sample\n",
      "dna.\n",
      "transfection\n",
      "of\n",
      "c-cbl\n",
      "constructs\n",
      "the\n",
      "a549\n",
      "cell\n",
      "line\n",
      "was\n",
      "transfected\n",
      "using\n",
      "the\n",
      "fugene\n",
      "hd\n",
      "(roche,\n",
      "nutley,\n",
      "nj)\n",
      "reagent\n",
      "according\n",
      "to\n",
      "the\n",
      "manufacturer's\n",
      "instructions.\n",
      "eight\n",
      "µg\n",
      "of\n",
      "plasmid\n",
      "dna,\n",
      "containing\n",
      "either\n",
      "no\n",
      "insert\n",
      "(empty\n",
      "vector),\n",
      "wild-type\n",
      "c-cbl,\n",
      "s80n/h94y\n",
      "c-cbl,\n",
      "q249e\n",
      "c-cbl\n",
      "or\n",
      "w802*\n",
      "cbl\n",
      "was\n",
      "used\n",
      "for\n",
      "transfection\n",
      "in\n",
      "a\n",
      "6-well\n",
      "culture\n",
      "plate.\n",
      "cells\n",
      "were\n",
      "harvested\n",
      "48\n",
      "h\n",
      "after\n",
      "transfection\n",
      "and\n",
      "analyzed\n",
      "for\n",
      "expression.\n",
      "c-cbl\n",
      "knockdown\n",
      "c-cbl\n",
      "knockdown\n",
      "was\n",
      "performed\n",
      "using\n",
      "lentiviral\n",
      "transduction\n",
      "using\n",
      "mission\n",
      "lentiviral\n",
      "transduction\n",
      "particles\n",
      "(sigma-aldrich,\n",
      "st.\n",
      "louis,\n",
      "mo)\n",
      "as\n",
      "per\n",
      "manufacturer's\n",
      "instructions.\n",
      "briefly,\n",
      "1×105\n",
      "h358\n",
      "cells/well\n",
      "were\n",
      "seeded\n",
      "in\n",
      "6-well\n",
      "plates\n",
      "and\n",
      "infected\n",
      "the\n",
      "following\n",
      "day\n",
      "with\n",
      "c-cbl\n",
      "lentiviral\n",
      "shrna\n",
      "constructs.\n",
      "to\n",
      "generate\n",
      "stable\n",
      "c-cbl\n",
      "knockdown\n",
      "cell\n",
      "lines,\n",
      "cells\n",
      "were\n",
      "selected\n",
      "for\n",
      "2\n",
      "days\n",
      "with\n",
      "1\n",
      "µg/ml\n",
      "puromycin.\n",
      "c-cbl\n",
      "levels\n",
      "were\n",
      "determined\n",
      "using\n",
      "whole\n",
      "cell\n",
      "lysates\n",
      "by\n",
      "immunoblotting\n",
      "with\n",
      "anti-cbl\n",
      "antibody\n",
      "(santa\n",
      "cruz\n",
      "biotechnologies,\n",
      "santa\n",
      "cruz,\n",
      "ca).\n",
      "cell\n",
      "viability\n",
      "assay\n",
      "cells\n",
      "were\n",
      "transfected\n",
      "as\n",
      "described\n",
      "above\n",
      "in\n",
      "the\n",
      "transfection\n",
      "assay.\n",
      "forty-eight\n",
      "hours\n",
      "after\n",
      "transfection,\n",
      "viability\n",
      "of\n",
      "cells\n",
      "was\n",
      "assessed\n",
      "using\n",
      "trypan\n",
      "blue\n",
      "exclusion.\n",
      "wound\n",
      "healing\n",
      "assay\n",
      "a549\n",
      "cells\n",
      "were\n",
      "seeded\n",
      "in\n",
      "6-well\n",
      "plates\n",
      "and\n",
      "cultured\n",
      "for\n",
      "48\n",
      "h\n",
      "until\n",
      "100%\n",
      "confluent.\n",
      "the\n",
      "medium\n",
      "was\n",
      "then\n",
      "changed\n",
      "and\n",
      "the\n",
      "cells\n",
      "were\n",
      "transfected\n",
      "as\n",
      "described\n",
      "in\n",
      "the\n",
      "transfection\n",
      "assay.\n",
      "twelve\n",
      "hours\n",
      "after\n",
      "transfection,\n",
      "a\n",
      "straight\n",
      "scratch\n",
      "was\n",
      "made\n",
      "across\n",
      "the\n",
      "cell\n",
      "layer\n",
      "using\n",
      "a\n",
      "1\n",
      "ml\n",
      "pipette\n",
      "tip.\n",
      "the\n",
      "cells\n",
      "were\n",
      "then\n",
      "gently\n",
      "washed\n",
      "with\n",
      "1×\n",
      "pbs\n",
      "to\n",
      "remove\n",
      "cellular\n",
      "debris\n",
      "and\n",
      "the\n",
      "media\n",
      "was\n",
      "replaced.\n",
      "photographs\n",
      "were\n",
      "taken\n",
      "of\n",
      "the\n",
      "wound\n",
      "region\n",
      "every\n",
      "12\n",
      "h\n",
      "until\n",
      "48\n",
      "h.\n",
      "western\n",
      "blot\n",
      "analysis\n",
      "forty\n",
      "eight\n",
      "hours\n",
      "after\n",
      "transfection,\n",
      "cells\n",
      "were\n",
      "collected\n",
      "and\n",
      "washed\n",
      "twice\n",
      "in\n",
      "1x\n",
      "pbs,\n",
      "then\n",
      "lysed\n",
      "in\n",
      "ice-cold\n",
      "lysis\n",
      "buffer\n",
      "(0.5m\n",
      "tris-hcl\n",
      "with\n",
      "ph\n",
      "7.4,\n",
      "1.5\n",
      "m\n",
      "nacl,\n",
      "2.5%\n",
      "deoxycholic\n",
      "acid,\n",
      "10\n",
      "mm\n",
      "edta,\n",
      "10%\n",
      "np-40,\n",
      "0.5\n",
      "mm\n",
      "dtt,\n",
      "1\n",
      "mm\n",
      "phenylmethylsulfonyl\n",
      "fluoride,\n",
      "5\n",
      "µg/ml\n",
      "leupeptin,\n",
      "and\n",
      "10\n",
      "µg/ml\n",
      "aprotinin)\n",
      "for\n",
      "5\n",
      "minutes.\n",
      "the\n",
      "lysate\n",
      "was\n",
      "centrifuged\n",
      "at\n",
      "13,000\n",
      "rpm\n",
      "for\n",
      "20\n",
      "minutes\n",
      "at\n",
      "4°c,\n",
      "and\n",
      "protein\n",
      "content\n",
      "of\n",
      "the\n",
      "supernatant\n",
      "was\n",
      "measured.\n",
      "total\n",
      "cell\n",
      "lysates\n",
      "(50\n",
      "µg/well)\n",
      "were\n",
      "separated\n",
      "by\n",
      "sds-page\n",
      "electrophoresis\n",
      "and\n",
      "the\n",
      "gels\n",
      "transferred\n",
      "onto\n",
      "nitrocellulose\n",
      "membranes\n",
      "(whatman,\n",
      "piscataway,\n",
      "nj).\n",
      "membranes\n",
      "were\n",
      "blocked\n",
      "with\n",
      "5%\n",
      "non-fat\n",
      "dry\n",
      "milk\n",
      "in\n",
      "phosphate-buffered\n",
      "saline\n",
      "containing\n",
      "tween-20\n",
      "(pbst)\n",
      "(1x\n",
      "pbs,\n",
      "0.1%\n",
      "tween-20)\n",
      "for\n",
      "1\n",
      "h\n",
      "at\n",
      "room\n",
      "temperature\n",
      "and\n",
      "incubated\n",
      "with\n",
      "the\n",
      "appropriate\n",
      "primary\n",
      "antibody\n",
      "at\n",
      "4°c\n",
      "overnight.\n",
      "membranes\n",
      "then\n",
      "were\n",
      "washed\n",
      "three\n",
      "times\n",
      "with\n",
      "pbst\n",
      "and\n",
      "probed\n",
      "with\n",
      "appropriate\n",
      "horseradish\n",
      "peroxidase\n",
      "(hrp)-conjugated\n",
      "secondary\n",
      "antibody\n",
      "for\n",
      "1\n",
      "h\n",
      "at\n",
      "room\n",
      "temperature.\n",
      "the\n",
      "membranes\n",
      "were\n",
      "again\n",
      "washed\n",
      "three\n",
      "times\n",
      "in\n",
      "pbst\n",
      "and\n",
      "bands\n",
      "were\n",
      "visualized\n",
      "using\n",
      "western\n",
      "blot\n",
      "chemiluminescence\n",
      "reagent\n",
      "(biorad,\n",
      "valencia,\n",
      "ca)\n",
      "on\n",
      "a\n",
      "chemidoc\n",
      "gel\n",
      "documentation\n",
      "system\n",
      "(biorad,\n",
      "valencia,\n",
      "ca).\n",
      "antibodies\n",
      "were\n",
      "obtained\n",
      "from\n",
      "santa\n",
      "cruz\n",
      "biotechnologies\n",
      "and\n",
      "used\n",
      "at\n",
      "the\n",
      "following\n",
      "dilutions\n",
      "(c-cbl,\n",
      "1∶5000;\n",
      "c-met,\n",
      "1∶5000;\n",
      "egfr,\n",
      "1∶5000;\n",
      "ubiquitin,\n",
      "1∶1000;\n",
      "ha,\n",
      "1∶5000\n",
      "and\n",
      "β-actin,\n",
      "1∶10,000).\n",
      "flow\n",
      "cytometry\n",
      "cell\n",
      "cycle\n",
      "analysis\n",
      "was\n",
      "carried\n",
      "out\n",
      "by\n",
      "flow\n",
      "cytometry.\n",
      "approximately\n",
      "2×106\n",
      "cells\n",
      "were\n",
      "grown\n",
      "in\n",
      "media\n",
      "containing\n",
      "10%\n",
      "fbs.\n",
      "cells\n",
      "were\n",
      "harvested\n",
      "by\n",
      "trypsin/edta\n",
      "treatment,\n",
      "washed\n",
      "with\n",
      "1x\n",
      "pbs\n",
      "three\n",
      "times\n",
      "and\n",
      "fixed\n",
      "with\n",
      "ice-cold\n",
      "70%\n",
      "ethanol\n",
      "for\n",
      "2\n",
      "h.\n",
      "cells\n",
      "were\n",
      "washed\n",
      "again\n",
      "with\n",
      "cold\n",
      "pbs\n",
      "and\n",
      "stained\n",
      "with\n",
      "a\n",
      "solution\n",
      "containing\n",
      "25\n",
      "µg/ml\n",
      "propidium\n",
      "iodide,\n",
      "200\n",
      "µg/ml\n",
      "rnase\n",
      "a,\n",
      "and\n",
      "0.1%\n",
      "triton\n",
      "x-100\n",
      "for\n",
      "30\n",
      "minutes\n",
      "in\n",
      "the\n",
      "dark.\n",
      "cell\n",
      "cycle\n",
      "analysis\n",
      "was\n",
      "performed\n",
      "using\n",
      "a\n",
      "guava\n",
      "pca-96\n",
      "flow\n",
      "cytometer\n",
      "(guava\n",
      "technologies,\n",
      "millipore,\n",
      "billerica,\n",
      "ma).\n",
      "ubiquitin\n",
      "ligase\n",
      "activity\n",
      "293t\n",
      "cells\n",
      "were\n",
      "maintained\n",
      "in\n",
      "culture\n",
      "in\n",
      "dmem\n",
      "supplemented\n",
      "with\n",
      "10%\n",
      "fbs\n",
      "and\n",
      "1%\n",
      "penicillin\n",
      "(100\n",
      "units/ml)\n",
      "and\n",
      "streptomycin\n",
      "(100\n",
      "µg/ml)\n",
      "were\n",
      "transfected\n",
      "with\n",
      "0.2\n",
      "µg\n",
      "egfr-pcdna3\n",
      "and\n",
      "2\n",
      "µg\n",
      "ha-tagged\n",
      "c-cbl\n",
      "constructs\n",
      "as\n",
      "indicated\n",
      "using\n",
      "calcium\n",
      "phosphate\n",
      "according\n",
      "to\n",
      "manufacturer's\n",
      "protocol\n",
      "(profection,\n",
      "promega,\n",
      "madison,\n",
      "wi).\n",
      "twenty-four\n",
      "hours\n",
      "post-transfection,\n",
      "cells\n",
      "were\n",
      "starved\n",
      "overnight\n",
      "in\n",
      "dmem\n",
      "supplemented\n",
      "with\n",
      "0.5%\n",
      "fbs,\n",
      "and\n",
      "then\n",
      "treated\n",
      "with\n",
      "or\n",
      "without\n",
      "egf\n",
      "(100\n",
      "ng/ml)\n",
      "for\n",
      "15\n",
      "min.\n",
      "the\n",
      "cells\n",
      "were\n",
      "collected\n",
      "and\n",
      "washed\n",
      "two\n",
      "times\n",
      "in\n",
      "ice-cold\n",
      "pbs\n",
      "containing\n",
      "0.2\n",
      "mm\n",
      "sodium\n",
      "orthovanadate\n",
      "then\n",
      "lysed\n",
      "in\n",
      "ice-cold\n",
      "lysis\n",
      "buffer\n",
      "(10\n",
      "mm\n",
      "tris\n",
      "hcl,\n",
      "ph\n",
      "7.5,\n",
      "150\n",
      "mm\n",
      "nacl,\n",
      "5\n",
      "mm\n",
      "edta,\n",
      "1%\n",
      "triton\n",
      "x100,\n",
      "10%\n",
      "glycerol,\n",
      "2\n",
      "mm\n",
      "sodium\n",
      "orthovanadate\n",
      "and\n",
      "protease\n",
      "inhibitors).\n",
      "lysates\n",
      "were\n",
      "cleared\n",
      "of\n",
      "debris\n",
      "by\n",
      "centrifugation\n",
      "at\n",
      "16,000\n",
      "g\n",
      "for\n",
      "10\n",
      "min\n",
      "at\n",
      "4°c.\n",
      "egfr\n",
      "immunoprecipitations\n",
      "were\n",
      "performed\n",
      "on\n",
      "200\n",
      "µg\n",
      "of\n",
      "cleared\n",
      "lysate\n",
      "using\n",
      "250\n",
      "ng\n",
      "of\n",
      "rabbit-anti-egfr\n",
      "and\n",
      "protein\n",
      "a/g\n",
      "plus\n",
      "sepharose\n",
      "overnight\n",
      "at\n",
      "4°c.\n",
      "precipitations\n",
      "were\n",
      "washed\n",
      "5\n",
      "times\n",
      "in\n",
      "lysis\n",
      "buffer\n",
      "before\n",
      "boiling\n",
      "in\n",
      "laemmli\n",
      "buffer.\n",
      "elutions\n",
      "were\n",
      "immunoblotted\n",
      "with\n",
      "anti-ubiquitin\n",
      "and\n",
      "egfr.\n",
      "twenty\n",
      "micrograms\n",
      "of\n",
      "cleared\n",
      "lysate\n",
      "were\n",
      "immunoblotted\n",
      "for\n",
      "each\n",
      "of\n",
      "the\n",
      "c-cbl\n",
      "constructs\n",
      "using\n",
      "anti-ha.\n",
      "statistical\n",
      "analysis\n",
      "mutation\n",
      "rates\n",
      "between\n",
      "different\n",
      "groups\n",
      "were\n",
      "compared\n",
      "using\n",
      "fisher's\n",
      "exact\n",
      "test.\n",
      "for\n",
      "continuous\n",
      "variables,\n",
      "group\n",
      "comparisons\n",
      "were\n",
      "performed\n",
      "using\n",
      "analysis\n",
      "of\n",
      "variance\n",
      "(anova)\n",
      "followed\n",
      "by\n",
      "sidak's\n",
      "adjustment\n",
      "for\n",
      "multiple\n",
      "comparisons.\n",
      "experiments\n",
      "involving\n",
      "repeated\n",
      "measurements\n",
      "over\n",
      "time\n",
      "were\n",
      "analyzed\n",
      "using\n",
      "repeated\n",
      "measures\n",
      "anova\n",
      "with\n",
      "the\n",
      "greenhouse-geisser\n",
      "adjustment\n",
      "to\n",
      "the\n",
      "degrees\n",
      "of\n",
      "freedom.\n",
      "analyses\n",
      "were\n",
      "conducted\n",
      "using\n",
      "stata\n",
      "(v10.1)\n",
      "software\n",
      "(stata\n",
      "corporation,\n",
      "college\n",
      "station,\n",
      "tx).\n",
      "go\n",
      "to:\n",
      "results\n",
      "c-cbl\n",
      "gene\n",
      "mutations\n",
      "in\n",
      "lung\n",
      "cancer\n",
      "to\n",
      "investigate\n",
      "the\n",
      "role\n",
      "of\n",
      "c-cbl\n",
      "in\n",
      "lung\n",
      "cancer,\n",
      "we\n",
      "analyzed\n",
      "its\n",
      "genomic\n",
      "dna\n",
      "in\n",
      "tumor\n",
      "and\n",
      "paired\n",
      "normal\n",
      "samples\n",
      "drawn\n",
      "from\n",
      "multiple\n",
      "ethnicities.\n",
      "the\n",
      "lung\n",
      "tumor\n",
      "samples\n",
      "represented\n",
      "caucasians\n",
      "(n\n",
      "=\n",
      "50),\n",
      "african-americans\n",
      "(n\n",
      "=\n",
      "29),\n",
      "and\n",
      "taiwanese\n",
      "(n\n",
      "=\n",
      "40)\n",
      "lung\n",
      "cancer\n",
      "patients.\n",
      "we\n",
      "designed\n",
      "12\n",
      "pairs\n",
      "of\n",
      "primers\n",
      "to\n",
      "sequence\n",
      "the\n",
      "coding\n",
      "region\n",
      "of\n",
      "c-cbl\n",
      "gene\n",
      "that\n",
      "spans\n",
      "exons\n",
      "2\n",
      "to\n",
      "16\n",
      "(table\n",
      "s2).\n",
      "we\n",
      "identified\n",
      "8\n",
      "unique\n",
      "somatic\n",
      "mutations\n",
      "in\n",
      "c-cbl\n",
      "exons\n",
      "among\n",
      "8\n",
      "different\n",
      "patients.\n",
      "a\n",
      "variation\n",
      "l620f,\n",
      "a\n",
      "known\n",
      "snp\n",
      "(rs2227988)\n",
      "in\n",
      "exon\n",
      "11\n",
      "was\n",
      "also\n",
      "detected.\n",
      "importantly,\n",
      "the\n",
      "eight\n",
      "novel\n",
      "non-synonymous\n",
      "mutations\n",
      "were\n",
      "confirmed\n",
      "by\n",
      "sequencing\n",
      "both\n",
      "strands\n",
      "of\n",
      "c-cbl\n",
      "genomic\n",
      "dna\n",
      "obtained\n",
      "from\n",
      "lung\n",
      "tumor\n",
      "samples\n",
      "(table\n",
      "1).\n",
      "moreover,\n",
      "none\n",
      "of\n",
      "the\n",
      "8\n",
      "mutations\n",
      "were\n",
      "detected\n",
      "in\n",
      "the\n",
      "corresponding\n",
      "normal\n",
      "tissue,\n",
      "indicating\n",
      "that\n",
      "these\n",
      "were\n",
      "somatic\n",
      "mutations.\n",
      "four\n",
      "synonymous\n",
      "single\n",
      "nucleotide\n",
      "variations\n",
      "(snvs)\n",
      "were\n",
      "also\n",
      "identified\n",
      "but\n",
      "were\n",
      "not\n",
      "used\n",
      "further\n",
      "in\n",
      "this\n",
      "study.\n",
      "table\n",
      "1\n",
      "table\n",
      "1\n",
      "c-cbl\n",
      "mutation\n",
      "analysis\n",
      "in\n",
      "119\n",
      "lung\n",
      "cancer\n",
      "patient\n",
      "tumor\n",
      "tissues.\n",
      "three\n",
      "of\n",
      "the\n",
      "8\n",
      "novel\n",
      "non-synonymous\n",
      "mutations\n",
      "were\n",
      "located\n",
      "in\n",
      "the\n",
      "tkb\n",
      "(tyrosine\n",
      "kinase\n",
      "binding)\n",
      "domain\n",
      "(s80n,\n",
      "h94y,\n",
      "and\n",
      "q249e),\n",
      "one\n",
      "in\n",
      "the\n",
      "ring\n",
      "finger\n",
      "domain\n",
      "(v391i),\n",
      "one\n",
      "in\n",
      "the\n",
      "proline-rich\n",
      "region\n",
      "(72515_72517\n",
      "del\n",
      "atg),\n",
      "and\n",
      "three\n",
      "in\n",
      "the\n",
      "c-terminal\n",
      "region\n",
      "(w802*,\n",
      "r830k,\n",
      "and\n",
      "a848t)\n",
      "of\n",
      "the\n",
      "c-cbl\n",
      "protein\n",
      "(figure\n",
      "1a\n",
      "and\n",
      "figure\n",
      "s1).\n",
      "in\n",
      "figure\n",
      "1b,\n",
      "we\n",
      "show\n",
      "model\n",
      "chromatograms\n",
      "of\n",
      "representative\n",
      "samples.\n",
      "figure\n",
      "1\n",
      "figure\n",
      "1\n",
      "c-cbl\n",
      "mutations\n",
      "and\n",
      "loh\n",
      "in\n",
      "non-small\n",
      "cell\n",
      "lung\n",
      "cancer.\n",
      "11q\n",
      "loh\n",
      "of\n",
      "c-cbl\n",
      "gene\n",
      "paired\n",
      "lung\n",
      "tumor\n",
      "and\n",
      "normal\n",
      "lung\n",
      "tissue\n",
      "samples\n",
      "from\n",
      "taiwanese\n",
      "patients\n",
      "(n\n",
      "=\n",
      "37)\n",
      "were\n",
      "investigated\n",
      "for\n",
      "loh.\n",
      "eight\n",
      "(21.6%)\n",
      "showed\n",
      "loh\n",
      "at\n",
      "the\n",
      "c-cbl\n",
      "locus\n",
      "on\n",
      "chromosome\n",
      "11\n",
      "while\n",
      "29\n",
      "samples\n",
      "(78.4%)\n",
      "revealed\n",
      "normal\n",
      "allelic\n",
      "contribution\n",
      "at\n",
      "the\n",
      "microsatellite\n",
      "markers\n",
      "(figures\n",
      "1c\n",
      "and\n",
      "d).\n",
      "c-cbl\n",
      "mutations\n",
      "in\n",
      "different\n",
      "ethnic\n",
      "groups\n",
      "the\n",
      "c-cbl\n",
      "double\n",
      "mutant\n",
      "s80n/h94y\n",
      "was\n",
      "found\n",
      "in\n",
      "the\n",
      "same\n",
      "patient\n",
      "and\n",
      "the\n",
      "overall\n",
      "mutation\n",
      "rate\n",
      "for\n",
      "c-cbl\n",
      "in\n",
      "lung\n",
      "tumors\n",
      "was\n",
      "6.7%\n",
      "(8/119).\n",
      "the\n",
      "frequency\n",
      "of\n",
      "c-cbl\n",
      "mutation\n",
      "was\n",
      "highest\n",
      "in\n",
      "large\n",
      "cell\n",
      "carcinoma\n",
      "(14.7%;\n",
      "5\n",
      "of\n",
      "34\n",
      "patients)\n",
      "followed\n",
      "by\n",
      "squamous\n",
      "carcinoma\n",
      "(6.3%;\n",
      "2\n",
      "of\n",
      "32\n",
      "patients)\n",
      "and\n",
      "the\n",
      "least\n",
      "was\n",
      "observed\n",
      "in\n",
      "adenocarcinoma\n",
      "(ad)\n",
      "(1.8%;\n",
      "1\n",
      "of\n",
      "53\n",
      "patients),\n",
      "although\n",
      "these\n",
      "rates\n",
      "were\n",
      "not\n",
      "statistically\n",
      "significant\n",
      "(p\n",
      "=\n",
      "0.292).\n",
      "mutation\n",
      "rates\n",
      "were\n",
      "6.0%\n",
      "among\n",
      "caucasians\n",
      "(0\n",
      "of\n",
      "20\n",
      "in\n",
      "ad;\n",
      "0\n",
      "of\n",
      "10\n",
      "in\n",
      "sq;\n",
      "and\n",
      "3\n",
      "of\n",
      "20\n",
      "in\n",
      "lc),\n",
      "13.8%\n",
      "in\n",
      "african-americans\n",
      "(1\n",
      "of\n",
      "10\n",
      "in\n",
      "ad;\n",
      "1\n",
      "of\n",
      "10\n",
      "in\n",
      "sq;\n",
      "and\n",
      "2\n",
      "of\n",
      "9\n",
      "in\n",
      "lc),\n",
      "and\n",
      "2.5%\n",
      "(0\n",
      "of\n",
      "23\n",
      "in\n",
      "ad;\n",
      "1\n",
      "of\n",
      "12\n",
      "in\n",
      "sq;\n",
      "and\n",
      "0\n",
      "of\n",
      "5\n",
      "in\n",
      "lc)\n",
      "in\n",
      "the\n",
      "taiwanese\n",
      "population.\n",
      "additionally\n",
      "two\n",
      "taiwanese\n",
      "patients\n",
      "with\n",
      "lung\n",
      "cancer\n",
      "(one\n",
      "squamous\n",
      "and\n",
      "one\n",
      "adenocarcinoma)\n",
      "had\n",
      "the\n",
      "known\n",
      "snp\n",
      "l620f.\n",
      "ethnic\n",
      "differences\n",
      "were\n",
      "not\n",
      "statistically\n",
      "significant,\n",
      "however\n",
      "the\n",
      "power\n",
      "to\n",
      "detect\n",
      "differences\n",
      "was\n",
      "low.\n",
      "mutations\n",
      "in\n",
      "met\n",
      "and\n",
      "egfr\n",
      "can\n",
      "be\n",
      "co-associated\n",
      "with\n",
      "c-cbl\n",
      "alterations\n",
      "since\n",
      "east\n",
      "asians\n",
      "with\n",
      "lung\n",
      "cancer\n",
      "have\n",
      "a\n",
      "higher\n",
      "frequency\n",
      "of\n",
      "egfr\n",
      "and\n",
      "met\n",
      "mutations\n",
      "in\n",
      "lung\n",
      "tumors\n",
      "[21],\n",
      "[22],\n",
      "we\n",
      "also\n",
      "determined\n",
      "mutations\n",
      "in\n",
      "egfr\n",
      "and\n",
      "met\n",
      "in\n",
      "the\n",
      "same\n",
      "taiwanese\n",
      "cohort\n",
      "samples\n",
      "and\n",
      "compared\n",
      "the\n",
      "results\n",
      "with\n",
      "the\n",
      "observed\n",
      "c-cbl\n",
      "alterations\n",
      "(loh\n",
      "and/or\n",
      "mutations).\n",
      "in\n",
      "the\n",
      "37\n",
      "samples\n",
      "tested,\n",
      "we\n",
      "did\n",
      "not\n",
      "find\n",
      "any\n",
      "overlap\n",
      "between\n",
      "c-cbl\n",
      "mutations\n",
      "and\n",
      "c-cbl\n",
      "loh\n",
      "(figure\n",
      "2).\n",
      "of\n",
      "the\n",
      "three\n",
      "c-cbl\n",
      "mutants\n",
      "(including\n",
      "the\n",
      "known\n",
      "l620f\n",
      "snp,\n",
      "rs2227988),\n",
      "one\n",
      "of\n",
      "the\n",
      "samples\n",
      "had\n",
      "a\n",
      "met\n",
      "mutation\n",
      "(n375s)\n",
      "and\n",
      "the\n",
      "other\n",
      "had\n",
      "an\n",
      "egfr\n",
      "mutation\n",
      "(l858r).\n",
      "among\n",
      "the\n",
      "8\n",
      "samples\n",
      "that\n",
      "had\n",
      "a\n",
      "loh\n",
      "at\n",
      "the\n",
      "c-cbl\n",
      "locus\n",
      "5\n",
      "had\n",
      "an\n",
      "additional\n",
      "mutation\n",
      "in\n",
      "met\n",
      "(n375s)\n",
      "and\n",
      "2\n",
      "had\n",
      "an\n",
      "egfr\n",
      "exon\n",
      "19\n",
      "deletion.\n",
      "twenty-six\n",
      "samples\n",
      "had\n",
      "neither\n",
      "c-cbl\n",
      "mutation\n",
      "nor\n",
      "c-cbl\n",
      "loh\n",
      "(3\n",
      "patients\n",
      "had\n",
      "a\n",
      "c-cbl\n",
      "mutation\n",
      "but\n",
      "no\n",
      "c-cbl\n",
      "loh).\n",
      "among\n",
      "these\n",
      "26\n",
      "samples\n",
      "9\n",
      "had\n",
      "a\n",
      "met\n",
      "mutation\n",
      "(8\n",
      "n375s,\n",
      "1\n",
      "l211w),\n",
      "13\n",
      "had\n",
      "an\n",
      "egfr\n",
      "mutation\n",
      "(7\n",
      "exon\n",
      "9\n",
      "deletion,\n",
      "6\n",
      "l858r)\n",
      "and\n",
      "4\n",
      "had\n",
      "no\n",
      "other\n",
      "met\n",
      "or\n",
      "egfr\n",
      "mutation.\n",
      "thus\n",
      "the\n",
      "rate\n",
      "of\n",
      "met\n",
      "or\n",
      "egfr\n",
      "mutations\n",
      "among\n",
      "patients\n",
      "with\n",
      "loh\n",
      "at\n",
      "the\n",
      "c-cbl\n",
      "locus\n",
      "(7\n",
      "of\n",
      "8)\n",
      "was\n",
      "similar\n",
      "to\n",
      "that\n",
      "seen\n",
      "in\n",
      "patients\n",
      "without\n",
      "c-cbl\n",
      "mutation\n",
      "or\n",
      "loh\n",
      "(22\n",
      "of\n",
      "26\n",
      "patients)\n",
      "(p\n",
      "=\n",
      "0.99).\n",
      "these\n",
      "4\n",
      "patients\n",
      "with\n",
      "no\n",
      "identifiable\n",
      "mutation\n",
      "in\n",
      "c-cbl,\n",
      "met\n",
      "or\n",
      "egfr\n",
      "represented\n",
      "10.8%\n",
      "of\n",
      "the\n",
      "37\n",
      "patients\n",
      "analyzed\n",
      "in\n",
      "the\n",
      "taiwanese\n",
      "patient\n",
      "cohort.\n",
      "conversely,\n",
      "89.2%\n",
      "taiwanese\n",
      "lung\n",
      "cancer\n",
      "patients\n",
      "have\n",
      "an\n",
      "identifiable\n",
      "mutation\n",
      "in\n",
      "either\n",
      "c-cbl,\n",
      "met\n",
      "or\n",
      "egfr\n",
      "or\n",
      "a\n",
      "combination\n",
      "of\n",
      "the\n",
      "three\n",
      "genes\n",
      "(figure\n",
      "2).\n",
      "additionally,\n",
      "we\n",
      "determined\n",
      "p53\n",
      "and\n",
      "kras\n",
      "mutations\n",
      "in\n",
      "these\n",
      "taiwanese\n",
      "cohorts.\n",
      "two\n",
      "p53\n",
      "and\n",
      "1\n",
      "kras\n",
      "mutation\n",
      "were\n",
      "detected.\n",
      "the\n",
      "single\n",
      "kras\n",
      "mutation\n",
      "overlapped\n",
      "with\n",
      "one\n",
      "p53\n",
      "mutation.\n",
      "this\n",
      "patient\n",
      "also\n",
      "had\n",
      "the\n",
      "egfr\n",
      "exon\n",
      "19\n",
      "deletion\n",
      "but\n",
      "had\n",
      "no\n",
      "c-cbl\n",
      "mutation.\n",
      "the\n",
      "other\n",
      "p53\n",
      "mutation\n",
      "sample\n",
      "had\n",
      "a\n",
      "c-cbl\n",
      "loh\n",
      "with\n",
      "concurrent\n",
      "met\n",
      "n375s\n",
      "mutation.\n",
      "thus,\n",
      "in\n",
      "the\n",
      "taiwanese\n",
      "samples\n",
      "analyzed,\n",
      "p53/kras\n",
      "mutations\n",
      "and\n",
      "c-cbl\n",
      "mutations\n",
      "were\n",
      "mutually\n",
      "exclusive\n",
      "(data\n",
      "not\n",
      "shown).\n",
      "figure\n",
      "2\n",
      "figure\n",
      "2\n",
      "c-cbl\n",
      "mutations\n",
      "and\n",
      "relationship\n",
      "to\n",
      "met\n",
      "and\n",
      "egfr\n",
      "mutations\n",
      "in\n",
      "lung\n",
      "cancer.\n",
      "cellular\n",
      "functions\n",
      "of\n",
      "c-cbl\n",
      "alterations\n",
      "in\n",
      "the\n",
      "context\n",
      "of\n",
      "lung\n",
      "tumorigenesis\n",
      "a.\n",
      "e3\n",
      "activity\n",
      "is\n",
      "intact\n",
      "in\n",
      "the\n",
      "mutant\n",
      "c-cbl\n",
      "proteins\n",
      "to\n",
      "investigate\n",
      "whether\n",
      "the\n",
      "different\n",
      "c-cbl\n",
      "mutations\n",
      "affect\n",
      "the\n",
      "e3\n",
      "activity,\n",
      "egfr\n",
      "was\n",
      "chosen\n",
      "as\n",
      "a\n",
      "model\n",
      "substrate\n",
      "for\n",
      "c-cbl\n",
      "e3\n",
      "function.\n",
      "all\n",
      "of\n",
      "the\n",
      "c-cbl\n",
      "mutants\n",
      "tested\n",
      "enhanced\n",
      "ubiquitination\n",
      "of\n",
      "the\n",
      "activated\n",
      "egfr\n",
      "similar\n",
      "to\n",
      "the\n",
      "wild-type\n",
      "c-cbl\n",
      "protein.\n",
      "this\n",
      "result\n",
      "demonstrates\n",
      "that\n",
      "the\n",
      "catalytic\n",
      "activity\n",
      "of\n",
      "the\n",
      "c-cbl\n",
      "mutants\n",
      "is\n",
      "not\n",
      "impaired\n",
      "when\n",
      "egfr\n",
      "was\n",
      "the\n",
      "substrate.\n",
      "(figure\n",
      "3a).\n",
      "figure\n",
      "3\n",
      "figure\n",
      "3\n",
      "ubiquitination,\n",
      "viability,\n",
      "expression\n",
      "and\n",
      "cell\n",
      "cycle\n",
      "analysis\n",
      "of\n",
      "various\n",
      "c-cbl\n",
      "mutants.\n",
      "b.\n",
      "effect\n",
      "on\n",
      "lung\n",
      "cancer\n",
      "cell\n",
      "viability\n",
      "the\n",
      "effect\n",
      "of\n",
      "a\n",
      "representative\n",
      "c-cbl\n",
      "mutant\n",
      "from\n",
      "each\n",
      "of\n",
      "the\n",
      "three\n",
      "ethnic\n",
      "backgrounds\n",
      "on\n",
      "lung\n",
      "cancer\n",
      "cell\n",
      "viability\n",
      "in\n",
      "cell\n",
      "lines\n",
      "was\n",
      "determined.\n",
      "s80n/h94y\n",
      "double\n",
      "mutation,\n",
      "q249e,\n",
      "and\n",
      "w802*\n",
      "were\n",
      "identified\n",
      "in\n",
      "lung\n",
      "tumor\n",
      "samples\n",
      "obtained\n",
      "from\n",
      "a\n",
      "caucasian,\n",
      "a\n",
      "taiwanese\n",
      "and\n",
      "an\n",
      "african-american,\n",
      "respectively.\n",
      "as\n",
      "described\n",
      "in\n",
      "methods,\n",
      "the\n",
      "c-cbl\n",
      "wild-type\n",
      "(wt)\n",
      "and\n",
      "the\n",
      "above\n",
      "three\n",
      "mutants\n",
      "were\n",
      "expressed\n",
      "after\n",
      "cloning\n",
      "them\n",
      "into\n",
      "paltermax\n",
      "vector\n",
      "in\n",
      "a549\n",
      "cells.\n",
      "these\n",
      "cells\n",
      "express\n",
      "relatively\n",
      "low\n",
      "basal\n",
      "levels\n",
      "of\n",
      "endogenous\n",
      "c-cbl\n",
      "(data\n",
      "not\n",
      "shown).\n",
      "transfection\n",
      "efficiency\n",
      "was\n",
      "comparable\n",
      "between\n",
      "different\n",
      "groups\n",
      "and\n",
      "the\n",
      "number\n",
      "of\n",
      "cells\n",
      "transfected\n",
      "with\n",
      "c-cbl\n",
      "wild-type\n",
      "construct\n",
      "was\n",
      "about\n",
      "70%\n",
      "compared\n",
      "to\n",
      "control\n",
      "cells\n",
      "that\n",
      "were\n",
      "transfected\n",
      "with\n",
      "the\n",
      "empty\n",
      "vector.\n",
      "cells\n",
      "transfected\n",
      "with\n",
      "s80n/h94y,\n",
      "q249e\n",
      "and\n",
      "w802*\n",
      "c-cbl\n",
      "mutant\n",
      "constructs\n",
      "resulted\n",
      "in\n",
      "increased\n",
      "number\n",
      "of\n",
      "viable\n",
      "cells\n",
      "that\n",
      "was\n",
      "132.3%,\n",
      "120.8%\n",
      "and\n",
      "147.9%\n",
      "higher\n",
      "respectively,\n",
      "relative\n",
      "to\n",
      "the\n",
      "empty\n",
      "vector\n",
      "control\n",
      "transfected\n",
      "cells\n",
      "and\n",
      "significantly\n",
      "different\n",
      "from\n",
      "the\n",
      "wild-type\n",
      "construct\n",
      "(p\n",
      "=\n",
      "0.022,\n",
      "p\n",
      "=\n",
      "0.049,\n",
      "and\n",
      "p\n",
      "=\n",
      "0.008,\n",
      "respectively)\n",
      "(figure\n",
      "3b).\n",
      "relative\n",
      "levels\n",
      "of\n",
      "c-cbl\n",
      "protein\n",
      "in\n",
      "whole\n",
      "cell\n",
      "lysates\n",
      "prepared\n",
      "from\n",
      "samples\n",
      "obtained\n",
      "from\n",
      "a\n",
      "parallel\n",
      "experiment\n",
      "was\n",
      "determined.\n",
      "the\n",
      "c-cbl\n",
      "protein\n",
      "levels\n",
      "in\n",
      "samples\n",
      "representing\n",
      "untransfected\n",
      "and\n",
      "empty\n",
      "vector\n",
      "transfected\n",
      "cells\n",
      "were\n",
      "comparable\n",
      "and\n",
      "those\n",
      "representing\n",
      "the\n",
      "c-cbl\n",
      "wt\n",
      "and\n",
      "the\n",
      "three\n",
      "c-cbl\n",
      "mutants\n",
      "were\n",
      "comparable\n",
      "(figure\n",
      "3c).\n",
      "c.\n",
      "effect\n",
      "on\n",
      "cell\n",
      "cycle\n",
      "to\n",
      "investigate\n",
      "if\n",
      "the\n",
      "increases\n",
      "in\n",
      "cell\n",
      "viability\n",
      "in\n",
      "different\n",
      "c-cbl\n",
      "mutants\n",
      "are\n",
      "due\n",
      "to\n",
      "increased\n",
      "cellular\n",
      "proliferation,\n",
      "a\n",
      "cell\n",
      "cycle\n",
      "analysis\n",
      "was\n",
      "performed.\n",
      "a549\n",
      "cells\n",
      "were\n",
      "transfected\n",
      "with\n",
      "the\n",
      "c-cbl\n",
      "wt\n",
      "or\n",
      "the\n",
      "three\n",
      "different\n",
      "mutants:\n",
      "s80n/h94y,\n",
      "q249e\n",
      "and\n",
      "w802*.\n",
      "the\n",
      "empty\n",
      "vector\n",
      "transfectant\n",
      "was\n",
      "used\n",
      "as\n",
      "a\n",
      "control.\n",
      "forty-eight\n",
      "hours\n",
      "after\n",
      "transfection\n",
      "cell\n",
      "cycle\n",
      "analysis\n",
      "was\n",
      "performed\n",
      "as\n",
      "described\n",
      "in\n",
      "materials\n",
      "and\n",
      "methods.\n",
      "there\n",
      "was\n",
      "no\n",
      "significant\n",
      "change\n",
      "in\n",
      "the\n",
      "subg1,\n",
      "g1\n",
      "or\n",
      "the\n",
      "s\n",
      "phase\n",
      "of\n",
      "the\n",
      "cell\n",
      "cycle\n",
      "among\n",
      "the\n",
      "different\n",
      "mutants\n",
      "compared\n",
      "to\n",
      "the\n",
      "wt\n",
      "construct\n",
      "(p\n",
      "=\n",
      "0.64,\n",
      "p\n",
      "=\n",
      "0.40,\n",
      "and\n",
      "p\n",
      "=\n",
      "0.28,\n",
      "respectively).\n",
      "the\n",
      "g2/m\n",
      "phase\n",
      "of\n",
      "the\n",
      "cell\n",
      "cycle\n",
      "showed\n",
      "an\n",
      "increase\n",
      "in\n",
      "cell\n",
      "numbers\n",
      "for\n",
      "the\n",
      "three\n",
      "mutants,\n",
      "s80n/h94y,\n",
      "q249e\n",
      "and\n",
      "w802*,\n",
      "when\n",
      "compared\n",
      "to\n",
      "the\n",
      "wt\n",
      "but\n",
      "again\n",
      "the\n",
      "difference\n",
      "was\n",
      "not\n",
      "statistically\n",
      "significant\n",
      "(p\n",
      "=\n",
      "0.25)\n",
      "(figure\n",
      "3d).\n",
      "d.\n",
      "effect\n",
      "on\n",
      "cell\n",
      "motility\n",
      "to\n",
      "investigate\n",
      "the\n",
      "effect\n",
      "of\n",
      "the\n",
      "expression\n",
      "of\n",
      "the\n",
      "above\n",
      "three\n",
      "c-cbl\n",
      "mutants\n",
      "on\n",
      "cell\n",
      "migration,\n",
      "we\n",
      "carried\n",
      "out\n",
      "wound\n",
      "healing\n",
      "assay\n",
      "as\n",
      "described\n",
      "in\n",
      "materials\n",
      "and\n",
      "methods.\n",
      "the\n",
      "closing\n",
      "of\n",
      "the\n",
      "scratch\n",
      "or\n",
      "the\n",
      "wound\n",
      "was\n",
      "monitored\n",
      "at\n",
      "0,\n",
      "12,\n",
      "24,\n",
      "36,\n",
      "and\n",
      "48\n",
      "h.\n",
      "(figure\n",
      "4a).\n",
      "in\n",
      "all\n",
      "the\n",
      "samples,\n",
      "that\n",
      "represented\n",
      "cells\n",
      "transfected\n",
      "with\n",
      "mutants,\n",
      "the\n",
      "wound\n",
      "gap\n",
      "was\n",
      "much\n",
      "smaller\n",
      "than\n",
      "that\n",
      "seen\n",
      "in\n",
      "the\n",
      "sample\n",
      "that\n",
      "represented\n",
      "cells\n",
      "transfected\n",
      "with\n",
      "c-cbl\n",
      "wt\n",
      "(p<0.001).\n",
      "we\n",
      "also\n",
      "determined\n",
      "the\n",
      "rate\n",
      "of\n",
      "wound\n",
      "closure\n",
      "for\n",
      "all\n",
      "the\n",
      "five\n",
      "groups.\n",
      "at\n",
      "48\n",
      "h,\n",
      "wild-type\n",
      "c-cbl\n",
      "transfectants\n",
      "showed\n",
      "61.1%\n",
      "open\n",
      "wound\n",
      "while\n",
      "the\n",
      "s80n/h94y,\n",
      "q249e\n",
      "and\n",
      "w802*\n",
      "mutants\n",
      "showed\n",
      "18.7%,\n",
      "23.9%\n",
      "and\n",
      "34.3%\n",
      "open\n",
      "wound\n",
      "respectively\n",
      "(p<0.001)\n",
      "(figure\n",
      "4b).\n",
      "figure\n",
      "4\n",
      "figure\n",
      "4\n",
      "c-cbl\n",
      "mutations\n",
      "affect\n",
      "wound\n",
      "healing\n",
      "in\n",
      "a549\n",
      "cells.\n",
      "e.\n",
      "c-cbl\n",
      "knockdown\n",
      "increases\n",
      "cell\n",
      "viability\n",
      "it\n",
      "is\n",
      "hypothesized\n",
      "that\n",
      "the\n",
      "loh\n",
      "seen\n",
      "in\n",
      "our\n",
      "samples\n",
      "could\n",
      "lead\n",
      "to\n",
      "decreased\n",
      "expression\n",
      "of\n",
      "c-cbl.\n",
      "thus\n",
      "we\n",
      "tested\n",
      "the\n",
      "effect\n",
      "of\n",
      "c-cbl\n",
      "knockdown\n",
      "in\n",
      "lung\n",
      "cancer\n",
      "cells.\n",
      "compared\n",
      "to\n",
      "a549,\n",
      "h358\n",
      "lung\n",
      "cancer\n",
      "cells\n",
      "express\n",
      "relatively\n",
      "high\n",
      "levels\n",
      "of\n",
      "endogenous\n",
      "c-cbl\n",
      "(data\n",
      "not\n",
      "shown).\n",
      "c-cbl\n",
      "expression\n",
      "was\n",
      "knocked\n",
      "down\n",
      "using\n",
      "lentiviral\n",
      "construct\n",
      "that\n",
      "expressed\n",
      "c-cbl\n",
      "specific\n",
      "shrna\n",
      "and\n",
      "compared\n",
      "the\n",
      "results\n",
      "with\n",
      "those\n",
      "that\n",
      "were\n",
      "transduced\n",
      "with\n",
      "scrambled\n",
      "shrna\n",
      "and\n",
      "the\n",
      "results\n",
      "are\n",
      "shown\n",
      "in\n",
      "figure\n",
      "5.\n",
      "we\n",
      "identified\n",
      "several\n",
      "clones\n",
      "that\n",
      "revealed\n",
      "varying\n",
      "degrees\n",
      "of\n",
      "c-cbl\n",
      "knockdown\n",
      "showing\n",
      "different\n",
      "sets\n",
      "of\n",
      "c-cbl\n",
      "lentiviral\n",
      "shrna\n",
      "knockdown\n",
      "efficiency\n",
      "(figure\n",
      "5a).\n",
      "of\n",
      "all\n",
      "the\n",
      "clones\n",
      "tested,\n",
      "clone\n",
      "27\n",
      "was\n",
      "chosen\n",
      "for\n",
      "further\n",
      "experiments.\n",
      "equal\n",
      "amount\n",
      "of\n",
      "cells\n",
      "were\n",
      "seeded\n",
      "in\n",
      "a\n",
      "6-well\n",
      "plate\n",
      "and\n",
      "the\n",
      "cell\n",
      "proliferation\n",
      "was\n",
      "measured\n",
      "at\n",
      "various\n",
      "times\n",
      "and\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "the\n",
      "results\n",
      "are\n",
      "depicted\n",
      "in\n",
      "figure\n",
      "5b.\n",
      "as\n",
      "expected,\n",
      "number\n",
      "of\n",
      "cells\n",
      "increased\n",
      "in\n",
      "a\n",
      "time\n",
      "dependent\n",
      "fashion\n",
      "from\n",
      "100\n",
      "to\n",
      "190%\n",
      "relative\n",
      "to\n",
      "scrambled\n",
      "shrna\n",
      "as\n",
      "control\n",
      "in\n",
      "a\n",
      "span\n",
      "of\n",
      "48\n",
      "h\n",
      "(p\n",
      "=\n",
      "0.0002)\n",
      "(figure\n",
      "5b).\n",
      "the\n",
      "cell\n",
      "cycle\n",
      "phases\n",
      "in\n",
      "h358\n",
      "cells\n",
      "that\n",
      "were\n",
      "knocked\n",
      "down\n",
      "with\n",
      "c-cbl\n",
      "shrna\n",
      "were\n",
      "looked\n",
      "at\n",
      "and\n",
      "compared\n",
      "with\n",
      "the\n",
      "scrambled\n",
      "shrna.\n",
      "there\n",
      "were\n",
      "no\n",
      "discernable\n",
      "differences\n",
      "between\n",
      "these\n",
      "two\n",
      "constructs\n",
      "in\n",
      "the\n",
      "different\n",
      "phases\n",
      "of\n",
      "the\n",
      "cell\n",
      "cycle\n",
      "(data\n",
      "not\n",
      "shown).\n",
      "figure\n",
      "5\n",
      "figure\n",
      "5\n",
      "knockdown\n",
      "of\n",
      "c-cbl\n",
      "using\n",
      "an\n",
      "shrna\n",
      "increases\n",
      "cell\n",
      "proliferation.\n",
      "go\n",
      "to:\n",
      "discussion\n",
      "our\n",
      "results\n",
      "demonstrate\n",
      "that\n",
      "c-cbl\n",
      "is\n",
      "somatically\n",
      "mutated\n",
      "(or\n",
      "has\n",
      "loh)\n",
      "in\n",
      "lung\n",
      "cancers,\n",
      "and\n",
      "can\n",
      "significantly\n",
      "contribute\n",
      "to\n",
      "enhanced\n",
      "cell\n",
      "viability\n",
      "and\n",
      "motility.\n",
      "there\n",
      "was\n",
      "also\n",
      "a\n",
      "high\n",
      "prevalence\n",
      "of\n",
      "loh\n",
      "with\n",
      "respect\n",
      "to\n",
      "c-cbl\n",
      "in\n",
      "lung\n",
      "tumors\n",
      "that\n",
      "harbored\n",
      "met\n",
      "or\n",
      "egfr\n",
      "mutation.\n",
      "in\n",
      "the\n",
      "present\n",
      "study,\n",
      "we\n",
      "have\n",
      "demonstrated\n",
      "the\n",
      "occurrence\n",
      "of\n",
      "c-cbl\n",
      "mutations\n",
      "in\n",
      "lung\n",
      "cancer\n",
      "patients,\n",
      "especially\n",
      "with\n",
      "different\n",
      "ancestral\n",
      "variations.\n",
      "mutations\n",
      "in\n",
      "c-cbl\n",
      "have\n",
      "been\n",
      "recently\n",
      "reported\n",
      "in\n",
      "juvenile\n",
      "myelomonocytic\n",
      "leukemia\n",
      "and\n",
      "myeloid\n",
      "malignancies.\n",
      "in\n",
      "the\n",
      "aml\n",
      "study,\n",
      "the\n",
      "mutation\n",
      "r420q\n",
      "located\n",
      "in\n",
      "the\n",
      "junction\n",
      "of\n",
      "the\n",
      "ring\n",
      "finger\n",
      "and\n",
      "the\n",
      "linker\n",
      "region\n",
      "inhibited\n",
      "fms-like\n",
      "tyrosine\n",
      "kinase\n",
      "3\n",
      "(flt3)\n",
      "internalization\n",
      "and\n",
      "ubiquitination\n",
      "[20],\n",
      "thus\n",
      "contributing\n",
      "to\n",
      "the\n",
      "gain-in-function\n",
      "for\n",
      "the\n",
      "rtk.\n",
      "in\n",
      "addition,\n",
      "mutations\n",
      "such\n",
      "as\n",
      "h398y,\n",
      "c384r,\n",
      "and\n",
      "l380p\n",
      "mapped\n",
      "to\n",
      "the\n",
      "ring\n",
      "finger\n",
      "domain\n",
      "and\n",
      "the\n",
      "linker\n",
      "region\n",
      "of\n",
      "c-cbl\n",
      "that\n",
      "is\n",
      "required\n",
      "for\n",
      "its\n",
      "e3\n",
      "activity\n",
      "[23],\n",
      "[24],\n",
      "[25],\n",
      "[26],\n",
      "[27].\n",
      "additionally,\n",
      "homozygous\n",
      "mutations\n",
      "in\n",
      "the\n",
      "ring\n",
      "finger\n",
      "domain\n",
      "of\n",
      "the\n",
      "c-cbl\n",
      "gene\n",
      "were\n",
      "described\n",
      "as\n",
      "a\n",
      "result\n",
      "of\n",
      "acquired\n",
      "uniparental\n",
      "disomy\n",
      "(upd)\n",
      "[26].\n",
      "it\n",
      "is\n",
      "important\n",
      "to\n",
      "note\n",
      "that\n",
      "our\n",
      "results\n",
      "indicate\n",
      "loh\n",
      "at\n",
      "11q23\n",
      "locus\n",
      "and\n",
      "these\n",
      "are\n",
      "mutually\n",
      "exclusive\n",
      "from\n",
      "missense\n",
      "mutations\n",
      "of\n",
      "c-cbl.\n",
      "the\n",
      "somatic\n",
      "mutations\n",
      "were\n",
      "all\n",
      "heterozygous.\n",
      "the\n",
      "mutations\n",
      "in\n",
      "aml\n",
      "led\n",
      "to\n",
      "abrogation\n",
      "of\n",
      "the\n",
      "e3\n",
      "activity\n",
      "leading\n",
      "to\n",
      "prolonged\n",
      "rtk\n",
      "activation.\n",
      "in\n",
      "addition\n",
      "mutants\n",
      "located\n",
      "on\n",
      "the\n",
      "linker\n",
      "region\n",
      "surrounding\n",
      "the\n",
      "ring\n",
      "finger\n",
      "domain\n",
      "exhibited\n",
      "enhanced\n",
      "akt\n",
      "signaling\n",
      "in\n",
      "response\n",
      "to\n",
      "cytokine\n",
      "stimulation\n",
      "[26].\n",
      "in\n",
      "addition,\n",
      "it\n",
      "was\n",
      "shown\n",
      "in\n",
      "nh3t3\n",
      "cells,\n",
      "that\n",
      "neither\n",
      "mutations\n",
      "in\n",
      "the\n",
      "ring\n",
      "finger\n",
      "nor\n",
      "the\n",
      "linker\n",
      "region\n",
      "causes\n",
      "transformation,\n",
      "however\n",
      "while\n",
      "certain\n",
      "mutations\n",
      "perturbs\n",
      "the\n",
      "ubiquitination,\n",
      "others\n",
      "affect\n",
      "receptor\n",
      "recycling\n",
      "and\n",
      "prolong\n",
      "kinase\n",
      "activity\n",
      "[28].\n",
      "we\n",
      "report\n",
      "here\n",
      "c-cbl\n",
      "mutations\n",
      "that\n",
      "mapped\n",
      "not\n",
      "only\n",
      "to\n",
      "the\n",
      "ring\n",
      "finger\n",
      "domain,\n",
      "but\n",
      "also\n",
      "to\n",
      "the\n",
      "tkb\n",
      "domain,\n",
      "proline-rich\n",
      "domain\n",
      "and\n",
      "the\n",
      "c-terminal\n",
      "region,\n",
      "but\n",
      "none\n",
      "mapped\n",
      "to\n",
      "the\n",
      "linker\n",
      "region\n",
      "as\n",
      "reported\n",
      "in\n",
      "the\n",
      "aml\n",
      "studies\n",
      "described\n",
      "above\n",
      "[23],\n",
      "[24],\n",
      "[25],\n",
      "[26],\n",
      "[29].\n",
      "in\n",
      "addition,\n",
      "8\n",
      "mutants\n",
      "that\n",
      "we\n",
      "detected\n",
      "were\n",
      "found\n",
      "in\n",
      "different\n",
      "ethnic\n",
      "backgrounds.\n",
      "for\n",
      "example,\n",
      "s80n/h94y,\n",
      "q249e,\n",
      "w802*\n",
      "were\n",
      "detected\n",
      "in\n",
      "caucasians,\n",
      "taiwanese\n",
      "and\n",
      "african-americans,\n",
      "respectively.\n",
      "the\n",
      "results\n",
      "point\n",
      "out\n",
      "not\n",
      "only\n",
      "the\n",
      "difference\n",
      "between\n",
      "lung\n",
      "cancer\n",
      "and\n",
      "other\n",
      "cancers,\n",
      "but\n",
      "also\n",
      "genetic\n",
      "polymorphism\n",
      "among\n",
      "different\n",
      "races\n",
      "in\n",
      "the\n",
      "same\n",
      "cancer.\n",
      "interestingly,\n",
      "there\n",
      "is\n",
      "a\n",
      "large\n",
      "disparity\n",
      "between\n",
      "african-american\n",
      "and\n",
      "other\n",
      "ethnic\n",
      "populations\n",
      "with\n",
      "lung\n",
      "cancer\n",
      "[30].\n",
      "we\n",
      "have\n",
      "previously\n",
      "shown\n",
      "that\n",
      "there\n",
      "was\n",
      "a\n",
      "low\n",
      "frequency\n",
      "of\n",
      "egfr\n",
      "and\n",
      "met\n",
      "mutation\n",
      "in\n",
      "african-americans\n",
      "as\n",
      "compared\n",
      "with\n",
      "taiwanese\n",
      "and\n",
      "caucasians\n",
      "[31].\n",
      "in\n",
      "this\n",
      "study,\n",
      "the\n",
      "number\n",
      "of\n",
      "african-american\n",
      "samples\n",
      "analyzed\n",
      "was\n",
      "relatively\n",
      "fewer\n",
      "and\n",
      "we\n",
      "found\n",
      "3\n",
      "mutations\n",
      "that\n",
      "are\n",
      "unique\n",
      "to\n",
      "this\n",
      "ethnicity.\n",
      "it\n",
      "would\n",
      "now\n",
      "behoove\n",
      "us\n",
      "to\n",
      "further\n",
      "study\n",
      "the\n",
      "genetic\n",
      "alterations\n",
      "that\n",
      "can\n",
      "occur\n",
      "and\n",
      "determine\n",
      "the\n",
      "targeted\n",
      "therapeutics\n",
      "for\n",
      "african-americans.\n",
      "our\n",
      "results\n",
      "provide\n",
      "evidence\n",
      "of\n",
      "the\n",
      "importance\n",
      "of\n",
      "c-cbl\n",
      "in\n",
      "tumorigenesis\n",
      "and\n",
      "potential\n",
      "signaling.\n",
      "our\n",
      "prediction,\n",
      "based\n",
      "on\n",
      "the\n",
      "aml\n",
      "data,\n",
      "would\n",
      "be\n",
      "that\n",
      "the\n",
      "v391i\n",
      "ring\n",
      "finger\n",
      "domain\n",
      "mutation\n",
      "would\n",
      "affect\n",
      "the\n",
      "e3\n",
      "activity.\n",
      "also,\n",
      "it\n",
      "will\n",
      "be\n",
      "important\n",
      "to\n",
      "determine\n",
      "the\n",
      "binding\n",
      "partners\n",
      "for\n",
      "c-cbl\n",
      "in\n",
      "the\n",
      "tkb\n",
      "domain\n",
      "and\n",
      "proline\n",
      "rich\n",
      "domain\n",
      "mutations.\n",
      "it\n",
      "has\n",
      "previously\n",
      "been\n",
      "shown\n",
      "that\n",
      "the\n",
      "tkb\n",
      "domain\n",
      "can\n",
      "bind\n",
      "to\n",
      "growth\n",
      "factor\n",
      "receptors\n",
      "and\n",
      "it\n",
      "will\n",
      "be\n",
      "important\n",
      "to\n",
      "determine\n",
      "the\n",
      "cross-binding\n",
      "of\n",
      "these\n",
      "mutants\n",
      "to\n",
      "met\n",
      "and\n",
      "or\n",
      "egfr.\n",
      "it\n",
      "would\n",
      "also\n",
      "be\n",
      "important\n",
      "in\n",
      "the\n",
      "future\n",
      "to\n",
      "look\n",
      "at\n",
      "fluorescence\n",
      "in\n",
      "situ\n",
      "hybridization/copy\n",
      "number\n",
      "changes\n",
      "in\n",
      "c-cbl\n",
      "in\n",
      "lung\n",
      "cancer.\n",
      "c-cbl\n",
      "plays\n",
      "an\n",
      "important\n",
      "role\n",
      "in\n",
      "down\n",
      "regulating\n",
      "rtk-mediated\n",
      "signaling\n",
      "through\n",
      "k63\n",
      "poly-ubiquitination\n",
      "and\n",
      "subsequent\n",
      "downregulation\n",
      "of\n",
      "rtks\n",
      "followed\n",
      "by\n",
      "lysosomal\n",
      "degredation\n",
      "[3].\n",
      "mono-ubiquitination\n",
      "or\n",
      "ubiquitinated\n",
      "with\n",
      "k63-linked\n",
      "chains\n",
      "of\n",
      "substrates\n",
      "by\n",
      "c-cbl\n",
      "may\n",
      "lead\n",
      "to\n",
      "enhancement\n",
      "of\n",
      "biological\n",
      "and\n",
      "biochemical\n",
      "functions\n",
      "(reviewed\n",
      "in\n",
      "hermann\n",
      "et\n",
      "al,\n",
      "2007\n",
      "[32]).\n",
      "the\n",
      "mutations\n",
      "that\n",
      "we\n",
      "analyzed\n",
      "in\n",
      "our\n",
      "studies\n",
      "all\n",
      "point\n",
      "out\n",
      "to\n",
      "the\n",
      "fact\n",
      "that\n",
      "the\n",
      "e3\n",
      "activity\n",
      "of\n",
      "c-cbl\n",
      "on\n",
      "egfr\n",
      "is\n",
      "intact;\n",
      "the\n",
      "egfr\n",
      "levels\n",
      "in\n",
      "the\n",
      "various\n",
      "mutants\n",
      "remain\n",
      "same\n",
      "(figure\n",
      "s2).\n",
      "multiple\n",
      "kinases,\n",
      "both\n",
      "rtks\n",
      "and\n",
      "non-rtks\n",
      "could\n",
      "be\n",
      "acted\n",
      "upon\n",
      "by\n",
      "c-cbl,\n",
      "including\n",
      "erbs,\n",
      "pdgfr,\n",
      "fms,\n",
      "met,\n",
      "c-kit,\n",
      "vegfr,\n",
      "flt-1,\n",
      "ron,\n",
      "fgfr,\n",
      "ir,\n",
      "as\n",
      "well\n",
      "as\n",
      "syk,\n",
      "fyn,\n",
      "lck,\n",
      "fgr,\n",
      "lyn\n",
      "and\n",
      "c-abl\n",
      "[3].\n",
      "in\n",
      "lung\n",
      "cancers\n",
      "the\n",
      "relevant\n",
      "substrates\n",
      "of\n",
      "c-cbl\n",
      "in\n",
      "terms\n",
      "of\n",
      "degradation\n",
      "or\n",
      "signal\n",
      "transduction\n",
      "are\n",
      "yet\n",
      "to\n",
      "be\n",
      "identified.\n",
      "the\n",
      "observation\n",
      "that\n",
      "c-cbl\n",
      "somatic\n",
      "mutations,\n",
      "especially\n",
      "s80n/h94y,\n",
      "q249e\n",
      "and\n",
      "w802*\n",
      "showed\n",
      "increased\n",
      "cell\n",
      "viability\n",
      "and\n",
      "cell\n",
      "motility\n",
      "that\n",
      "is\n",
      "in\n",
      "agreement\n",
      "with\n",
      "the\n",
      "physiological\n",
      "role\n",
      "for\n",
      "d-cbl\n",
      "in\n",
      "the\n",
      "regulation\n",
      "of\n",
      "apoptosis\n",
      "and\n",
      "differentiation\n",
      "identified\n",
      "in\n",
      "drosophila\n",
      "is\n",
      "very\n",
      "significant\n",
      "[33].\n",
      "it\n",
      "has\n",
      "been\n",
      "previously\n",
      "shown\n",
      "that\n",
      "activating\n",
      "c-cbl\n",
      "mutation\n",
      "downregulates\n",
      "egfr\n",
      "signaling\n",
      "and\n",
      "decreases\n",
      "cellular\n",
      "proliferation\n",
      "and\n",
      "migration\n",
      "in\n",
      "breast\n",
      "cancer\n",
      "cell\n",
      "lines\n",
      "[34].\n",
      "although\n",
      "the\n",
      "role\n",
      "of\n",
      "c-cbl\n",
      "in\n",
      "the\n",
      "negative\n",
      "regulation\n",
      "of\n",
      "rtks\n",
      "is\n",
      "well\n",
      "substantiated,\n",
      "thereby\n",
      "suggesting\n",
      "that\n",
      "it\n",
      "is\n",
      "a\n",
      "natural\n",
      "tumor\n",
      "suppressor,\n",
      "studies\n",
      "in\n",
      "cancer\n",
      "cells\n",
      "have\n",
      "revealed\n",
      "both\n",
      "tumor\n",
      "suppressor\n",
      "and\n",
      "tumor\n",
      "promoting\n",
      "activities\n",
      "depending\n",
      "on\n",
      "the\n",
      "type\n",
      "of\n",
      "c-cbl\n",
      "mutation\n",
      "and\n",
      "the\n",
      "number\n",
      "of\n",
      "alleles\n",
      "at\n",
      "the\n",
      "c-cbl\n",
      "locus\n",
      "[24].\n",
      "in\n",
      "agreement\n",
      "with\n",
      "the\n",
      "above,\n",
      "the\n",
      "three\n",
      "c-cbl\n",
      "mutants\n",
      "described\n",
      "here\n",
      "appear\n",
      "to\n",
      "have\n",
      "tumor\n",
      "growth\n",
      "and\n",
      "metastasis\n",
      "promoting\n",
      "properties.\n",
      "although\n",
      "these\n",
      "mutants\n",
      "are\n",
      "outside\n",
      "of\n",
      "the\n",
      "ring\n",
      "finger\n",
      "or\n",
      "the\n",
      "linker\n",
      "region\n",
      "of\n",
      "c-cbl,\n",
      "their\n",
      "downstream\n",
      "effects\n",
      "are\n",
      "significant\n",
      "so\n",
      "as\n",
      "to\n",
      "cause\n",
      "increased\n",
      "proliferation\n",
      "and\n",
      "migration,\n",
      "but\n",
      "the\n",
      "substrate\n",
      "affected\n",
      "by\n",
      "these\n",
      "mutations\n",
      "are\n",
      "not\n",
      "known\n",
      "yet.\n",
      "this\n",
      "raises\n",
      "the\n",
      "possibility\n",
      "that\n",
      "some\n",
      "of\n",
      "the\n",
      "cellular\n",
      "functions\n",
      "of\n",
      "c-cbl\n",
      "are\n",
      "independent\n",
      "of\n",
      "its\n",
      "ubiquitin-ligase\n",
      "activity,\n",
      "an\n",
      "area\n",
      "that\n",
      "we\n",
      "are\n",
      "currently\n",
      "investigating.\n",
      "the\n",
      "oncogenic\n",
      "nature\n",
      "of\n",
      "rtks,\n",
      "addiction\n",
      "of\n",
      "cancers\n",
      "to\n",
      "growth\n",
      "signals\n",
      "and\n",
      "given\n",
      "the\n",
      "clustering\n",
      "of\n",
      "c-cbl,\n",
      "egfr\n",
      "and\n",
      "met\n",
      "mutations,\n",
      "it\n",
      "is\n",
      "possible\n",
      "that\n",
      "the\n",
      "transforming\n",
      "effect\n",
      "of\n",
      "c-cbl\n",
      "mutations\n",
      "is\n",
      "most\n",
      "likely\n",
      "a\n",
      "combinatorial\n",
      "effect\n",
      "of\n",
      "the\n",
      "three.\n",
      "we\n",
      "also\n",
      "show\n",
      "that\n",
      "loh\n",
      "for\n",
      "c-cbl\n",
      "was\n",
      "found\n",
      "in\n",
      "a\n",
      "significant\n",
      "number\n",
      "of\n",
      "samples\n",
      "that\n",
      "harbored\n",
      "met\n",
      "or\n",
      "egfr\n",
      "mutations.\n",
      "the\n",
      "fact\n",
      "that\n",
      "about\n",
      "7%\n",
      "of\n",
      "lung\n",
      "tumor\n",
      "samples\n",
      "are\n",
      "likely\n",
      "to\n",
      "have\n",
      "c-cbl\n",
      "mutations\n",
      "and\n",
      "an\n",
      "additional\n",
      "22%\n",
      "are\n",
      "likely\n",
      "to\n",
      "harbor\n",
      "c-cbl-related\n",
      "loh,\n",
      "makes\n",
      "c-cbl\n",
      "a\n",
      "highly\n",
      "mutated\n",
      "molecule\n",
      "in\n",
      "lung\n",
      "cancer.\n",
      "since\n",
      "loh\n",
      "alone\n",
      "is\n",
      "not\n",
      "enough\n",
      "to\n",
      "cause\n",
      "a\n",
      "transforming\n",
      "event\n",
      "[35],\n",
      "[36],\n",
      "[37],\n",
      "associated\n",
      "mutation\n",
      "in\n",
      "the\n",
      "met\n",
      "or\n",
      "egfr\n",
      "locus\n",
      "or\n",
      "yet\n",
      "another\n",
      "rtk\n",
      "discussed\n",
      "above\n",
      "may\n",
      "play\n",
      "a\n",
      "role\n",
      "in\n",
      "carcinogenesis.\n",
      "we\n",
      "predict\n",
      "that\n",
      "this\n",
      "loh\n",
      "in\n",
      "c-cbl\n",
      "results\n",
      "in\n",
      "haploinsufficiency\n",
      "that\n",
      "downplays\n",
      "rtk\n",
      "ubiquitination\n",
      "leading\n",
      "to\n",
      "hyperactivity\n",
      "of\n",
      "the\n",
      "rtks.\n",
      "however,\n",
      "whether\n",
      "this\n",
      "is\n",
      "sufficient\n",
      "cause\n",
      "for\n",
      "tumorigenesis\n",
      "remains\n",
      "to\n",
      "be\n",
      "determined.\n",
      "consistent\n",
      "with\n",
      "our\n",
      "hypothesis\n",
      "is\n",
      "the\n",
      "fact\n",
      "that\n",
      "c-cbl-/-\n",
      "mice\n",
      "have\n",
      "increased\n",
      "kinase\n",
      "activity\n",
      "in\n",
      "lymphocytes,\n",
      "but\n",
      "is\n",
      "not\n",
      "sufficient\n",
      "for\n",
      "tumor\n",
      "formation\n",
      "[35],\n",
      "[36],\n",
      "[37].\n",
      "c-cbl\n",
      "loh\n",
      "could\n",
      "also\n",
      "lead\n",
      "to\n",
      "increased\n",
      "expression\n",
      "of\n",
      "c-cbl\n",
      "from\n",
      "the\n",
      "other\n",
      "allele\n",
      "to\n",
      "compensate\n",
      "for\n",
      "a\n",
      "loss\n",
      "of\n",
      "an\n",
      "allele.\n",
      "alternately,\n",
      "there\n",
      "could\n",
      "be\n",
      "some\n",
      "form\n",
      "of\n",
      "synergy\n",
      "working\n",
      "with\n",
      "reduced\n",
      "c-cbl\n",
      "levels\n",
      "and\n",
      "mutated\n",
      "receptors\n",
      "that\n",
      "exacerbate\n",
      "the\n",
      "phenotype\n",
      "of\n",
      "each\n",
      "alone.\n",
      "previous\n",
      "studies\n",
      "from\n",
      "our\n",
      "lab\n",
      "and\n",
      "others\n",
      "have\n",
      "shown\n",
      "that\n",
      "east\n",
      "asians\n",
      "with\n",
      "lung\n",
      "cancers\n",
      "have\n",
      "relatively\n",
      "high\n",
      "frequencies\n",
      "of\n",
      "gain-of-function\n",
      "of\n",
      "mutations\n",
      "in\n",
      "rtks\n",
      "such\n",
      "as\n",
      "egfr\n",
      "and\n",
      "met\n",
      "[31].\n",
      "in\n",
      "a\n",
      "cohort\n",
      "of\n",
      "japanese\n",
      "patients\n",
      "an\n",
      "activating\n",
      "met\n",
      "mutation\n",
      "was\n",
      "identified\n",
      "in\n",
      "the\n",
      "splice\n",
      "region\n",
      "that\n",
      "deletes\n",
      "the\n",
      "juxtamembrane\n",
      "domain\n",
      "that\n",
      "is\n",
      "involved\n",
      "e3\n",
      "activity\n",
      "of\n",
      "c-cbl\n",
      "[38].\n",
      "this\n",
      "study\n",
      "also\n",
      "found\n",
      "that\n",
      "activation\n",
      "of\n",
      "met\n",
      "is\n",
      "mutually\n",
      "exclusive\n",
      "of\n",
      "egfr,\n",
      "kras\n",
      "and\n",
      "her2\n",
      "gene\n",
      "mutations\n",
      "[38].\n",
      "we\n",
      "failed\n",
      "to\n",
      "detect\n",
      "such\n",
      "mutations\n",
      "in\n",
      "significant\n",
      "numbers\n",
      "in\n",
      "lung\n",
      "tumor\n",
      "samples\n",
      "obtained\n",
      "from\n",
      "african-americans\n",
      "(n\n",
      "=\n",
      "29)\n",
      "and\n",
      "caucasian\n",
      "(n\n",
      "=\n",
      "50)\n",
      "patients.\n",
      "one\n",
      "met\n",
      "mutation\n",
      "was\n",
      "identified\n",
      "in\n",
      "each\n",
      "of\n",
      "the\n",
      "groups\n",
      "whereas\n",
      "1\n",
      "and\n",
      "3\n",
      "egfr\n",
      "mutations\n",
      "were\n",
      "identified\n",
      "in\n",
      "the\n",
      "african-american\n",
      "and\n",
      "caucasian\n",
      "cohorts\n",
      "respectively.\n",
      "egfr\n",
      "mutations\n",
      "have\n",
      "earlier\n",
      "been\n",
      "identified\n",
      "as\n",
      "one\n",
      "of\n",
      "the\n",
      "key\n",
      "mutations\n",
      "affecting\n",
      "lung\n",
      "adenocarcinoma\n",
      "patients\n",
      "in\n",
      "a\n",
      "comprehensive\n",
      "study\n",
      "of\n",
      "188\n",
      "patients\n",
      "[39].\n",
      "our\n",
      "study\n",
      "encompasses\n",
      "different\n",
      "histologies\n",
      "of\n",
      "nsclc.\n",
      "however,\n",
      "the\n",
      "published\n",
      "series\n",
      "did\n",
      "not\n",
      "find\n",
      "any\n",
      "mutations\n",
      "in\n",
      "c-cbl\n",
      "or\n",
      "met\n",
      "unlike\n",
      "our\n",
      "study\n",
      "that\n",
      "encompassed\n",
      "different\n",
      "subtypes\n",
      "of\n",
      "nsclc.\n",
      "it\n",
      "is\n",
      "important\n",
      "to\n",
      "note,\n",
      "we\n",
      "have\n",
      "recently\n",
      "shown\n",
      "that\n",
      "met\n",
      "mutations\n",
      "in\n",
      "lung\n",
      "cancer\n",
      "are\n",
      "in\n",
      "majority\n",
      "germline\n",
      "[31].\n",
      "we\n",
      "have\n",
      "reported\n",
      "earlier\n",
      "c-cbl\n",
      "mutations\n",
      "in\n",
      "a\n",
      "small\n",
      "cohort\n",
      "of\n",
      "taiwanese\n",
      "lung\n",
      "cancer\n",
      "samples\n",
      "[40].\n",
      "in\n",
      "our\n",
      "efforts\n",
      "to\n",
      "understand\n",
      "the\n",
      "ethnic\n",
      "differences\n",
      "in\n",
      "the\n",
      "lung\n",
      "oncogenome,\n",
      "we\n",
      "also\n",
      "looked\n",
      "at\n",
      "pax\n",
      "transcription\n",
      "factors\n",
      "such\n",
      "as\n",
      "pax5\n",
      "and\n",
      "pax8\n",
      "that\n",
      "are\n",
      "highly\n",
      "expressed\n",
      "in\n",
      "lung\n",
      "cancers;\n",
      "however\n",
      "there\n",
      "was\n",
      "no\n",
      "preferential\n",
      "expression\n",
      "or\n",
      "mutations\n",
      "of\n",
      "the\n",
      "above\n",
      "genes\n",
      "in\n",
      "lung\n",
      "tumor\n",
      "samples\n",
      "of\n",
      "african-americans.\n",
      "in\n",
      "this\n",
      "study,\n",
      "we\n",
      "show\n",
      "relatively\n",
      "high\n",
      "frequency\n",
      "of\n",
      "c-cbl\n",
      "mutations\n",
      "in\n",
      "lung\n",
      "cancers,\n",
      "especially\n",
      "in\n",
      "the\n",
      "large\n",
      "cell\n",
      "type\n",
      "among\n",
      "caucasians\n",
      "and\n",
      "particularly\n",
      "among\n",
      "african-americans.\n",
      "we\n",
      "therefore\n",
      "propose\n",
      "c-cbl\n",
      "as\n",
      "an\n",
      "efficacious\n",
      "target\n",
      "for\n",
      "lung\n",
      "cancers\n",
      "in\n",
      "african-americans\n",
      "that\n",
      "needs\n",
      "to\n",
      "be\n",
      "further\n",
      "substantiated.\n",
      "this\n",
      "is\n",
      "all\n",
      "the\n",
      "more\n",
      "important\n",
      "because\n",
      "the\n",
      "prognosis\n",
      "for\n",
      "african-americans\n",
      "with\n",
      "lung\n",
      "cancer,\n",
      "especially\n",
      "for\n",
      "men\n",
      "is\n",
      "much\n",
      "poorer\n",
      "compared\n",
      "to\n",
      "their\n",
      "caucasian\n",
      "counterparts\n",
      "[41].\n",
      "in\n",
      "conclusion,\n",
      "the\n",
      "results\n",
      "presented\n",
      "in\n",
      "this\n",
      "study\n",
      "demonstrate\n",
      "that\n",
      "c-cbl\n",
      "is\n",
      "frequently\n",
      "mutated\n",
      "or\n",
      "even\n",
      "lost\n",
      "in\n",
      "lung\n",
      "cancers.\n",
      "our\n",
      "results\n",
      "support\n",
      "a\n",
      "role\n",
      "for\n",
      "c-cbl\n",
      "mutants\n",
      "that\n",
      "are\n",
      "independent\n",
      "of\n",
      "its\n",
      "ubiquitination\n",
      "activity.\n",
      "given\n",
      "the\n",
      "relatively\n",
      "high\n",
      "mutation\n",
      "rates\n",
      "in\n",
      "c-cbl\n",
      "as\n",
      "well\n",
      "as\n",
      "rtks\n",
      "such\n",
      "as\n",
      "met\n",
      "and\n",
      "egfr,\n",
      "it\n",
      "is\n",
      "likely\n",
      "that\n",
      "their\n",
      "combined\n",
      "effect\n",
      "could\n",
      "be\n",
      "synergistic\n",
      "in\n",
      "promoting\n",
      "tumorigenesis.\n"
     ]
    }
   ],
   "source": [
    "for i in data_all.loc[1,\"Text\"].lower().split():\n",
    "    print(i)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# some more features engineering"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Feature for the length of the text\n",
    "data_all[\"Text_words\"] = data_all[\"Text\"].map(lambda x: len(str(x).split(\" \")))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_train = data_all.iloc[:len(train_final)]\n",
    "new_test = data_all.iloc[len(train_final):]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "extra columns: {'EIF2B5', 'BAG3', 'WISP3', 'ADGRG1', 'BBS5', 'SF3B2', 'ZFPM2', 'B4GALT7', 'EPHA5', 'APOL1', 'ABCB11', 'SCN9A', 'GNE', 'DNAH5', 'ROCK1', 'NDUFS6', 'DPM1', 'KCNE2', 'MPDU1', 'LITAF', 'RP1', 'AURKC', 'PLA2G6', 'PDE8B', 'AP3B1', 'TRPM1', 'SIX3', 'RGS9', 'SUCLA2', 'IKBKAP', 'RAB27A', 'LRP6', 'SYT6', 'PHOX2B', 'CYP7B1', 'TNFRSF11A', 'SLC17A5', 'SLC7A7', 'SLC33A1', 'MOCS1', 'DNAI1', 'ABCC6', 'KERA', 'SLC22A4', 'CRB1', 'TRPC6', 'CRLF1', 'GALK1', 'BFSP2', 'SLC25A13', 'SLC25A15', 'EPHB2', 'TTK', 'LCT', 'STK19', 'EPHA2', 'WNT4', 'STK33', 'CRNKL1', 'MOCS2', 'GPHN', 'KISS1R', 'XRCC1', 'MYOT', 'SLC27A4', 'LRP4', 'CSF1R', 'RAD54B', 'SPAST', 'TGM5', 'CHST3', 'MCC', 'DYNC2H1', 'TP63', 'SLC25A12', 'FLNB', 'GCM2', 'PNPO', 'PTCH2', 'DNM1L', 'ALOX12B', 'SCN4A', 'ITM2B', 'TSHR', 'RECQL4', 'KCNJ13', 'BMPR1B', 'RPS19', 'DCC', 'OTOF', 'SLC6A5', 'SEPT9', 'AKAP9', 'STAT5B', 'KRIT1', 'RNF6', 'SLC19A2', 'ASS1', 'CILP', 'NDUFS3', 'NEK8', 'COX15', 'SLC7A9', 'KCNMB1', 'CST3', 'GJB3', 'LARGE1', 'SCO1', 'BCS1L', 'PER2', 'SLC4A4', 'SCO2', 'RBBP8', 'KCNQ4', 'PRKRA', 'POLH', 'SLC22A5', 'AGXT', 'ALG10', 'ATP2C1', 'ADAMTS13', 'GRM6', 'GYG1', 'TINF2', 'CLDN16', 'GLE1', 'HABP2', 'LRP5', 'KLF11', 'YARS', 'RPS26', 'EDAR'}\n"
     ]
    }
   ],
   "source": [
    "svd = TruncatedSVD(n_components=25, n_iter=12, random_state=26)\n",
    "\n",
    "one_hot_gene = pd.get_dummies(new_train['Gene'])\n",
    "one_hot_gene_test=pd.get_dummies(new_test[\"Gene\"])\n",
    "one_hot_gene_test=fix_test_columns(one_hot_gene_test,one_hot_gene.columns)\n",
    "truncated_one_hot_gene = svd.fit_transform(one_hot_gene.values)\n",
    "truncated_one_hot_gene_for_test=svd.transform(one_hot_gene_test.values)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes_train=pd.DataFrame(truncated_one_hot_gene,columns=[\"tsvd_gene\"+ str(x) for x in range(0,25)])\n",
    "genes_test=pd.DataFrame(truncated_one_hot_gene_for_test,columns=[\"tsvd_gene\"+ str(x) for x in range(0,25)])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\Franck\\Documents\\Continuum\\Anaconda3\\lib\\site-packages\\ipykernel_launcher.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \"\"\"Entry point for launching an IPython kernel.\n"
     ]
    }
   ],
   "source": [
    "new_test[\"index\"]=range(0,len(new_test))\n",
    "new_train_1=pd.merge(new_train.reset_index(),genes_train.reset_index()).drop(\"index\",axis=1)\n",
    "new_test_1=pd.merge(new_test,genes_test.reset_index()).drop(\"index\",axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_train_1.to_csv(\"checkpoints_databases/nw_working_train.csv\",index=False,encoding=\"utf8\")\n",
    "new_test_1.to_csv(\"checkpoints_databases/nw_working_test.csv\",index=False,encoding=\"utf8\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(3689, 42)"
      ]
     },
     "execution_count": 116,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_train_1.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(986, 42)"
      ]
     },
     "execution_count": 117,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_test_1.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
