{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import string\n",
    "import os\n",
    "from collections import Counter\n",
    "\n",
    "import nltk\n",
    "from nltk.tokenize import sent_tokenize, word_tokenize\n",
    "from nltk.corpus import stopwords\n",
    "from nltk.stem import WordNetLemmatizer\n",
    "pd.options.display.max_colwidth = 50\n",
    "import re\n",
    "\n",
    "from sklearn.feature_extraction.text import TfidfVectorizer\n",
    "from sklearn.decomposition import TruncatedSVD"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Creating base"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "train = pd.read_csv('..//bases/training_variants')\n",
    "test = pd.read_csv('..//bases/test_variants')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "train_texts = pd.read_csv('..//bases/training_text', sep=\"\\|\\|\", engine='python', header=None, skiprows=1, names=[\"ID\",\"Text\"], encoding = \"utf-8\")\n",
    "test_texts = pd.read_csv('..//bases/test_text', sep=\"\\|\\|\", engine='python', header=None, skiprows=1, names=[\"ID\",\"Text\"], encoding = \"utf-8\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "train = pd.merge(train, train_texts, how='left', on='ID')\n",
    "test = pd.merge(test, test_texts, how='left', on='ID')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Transform Amino Acid (AA) Letter to their three-letter abbreviation in order to find them in the text when they appear\n",
    "One_to_Three_AA = {'C': 'Cys', 'D': 'Asp', 'S': 'Ser', 'Q': 'Gln', 'K': 'Lys',\n",
    "         'I': 'Ile', 'P': 'Pro', 'T': 'Thr', 'F': 'Phe', 'N': 'Asn', \n",
    "         'G': 'Gly', 'H': 'His', 'L': 'Leu', 'R': 'Arg', 'W': 'Trp', \n",
    "         'A': 'Ala', 'V': 'Val', 'E': 'Glu', 'Y': 'Tyr', 'M': 'Met'}\n",
    "pattern = re.compile('|'.join(One_to_Three_AA.keys()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#### process the train and test set together\n",
    "data_all = pd.concat((train, test), axis=0, ignore_index=True)\n",
    "data_all_backup = data_all[:] ## We keep backup in case we need to use again\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Adding extra feature(s)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# Feature for the length of the text\n",
    "data_all[\"Text_words\"] = data_all[\"Text\"].map(lambda x: len(str(x).split(\" \")))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Substitutions (subs)\n",
    "## Functions for pre-processing of subs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# find_sub return the substituions that are in text and those that are not\n",
    "def find_sub(data):    \n",
    "    Boolean = [data.Variation[i][:-1] in data.Text[i] or #case 1.\n",
    "               pattern.sub(lambda x: One_to_Three_AA[x.group()], data.Variation[i][:-1]) # case2\n",
    "               in data.Text[i]  for i in data.index] ## because new indexing we use \n",
    "    \n",
    "    sub_in_text = data[Boolean]\n",
    "    not_Boolean = [not i for i in Boolean]  \n",
    "    sub_not_in_text = data[not_Boolean]\n",
    "    \n",
    "    return sub_in_text, sub_not_in_text\n",
    "\n",
    "# find_sub_numberChange searches for other number of a substitution i.e. G12V -> G_V because sometimes mistake in entry\n",
    "# Is currently without One_to_three substitution or Variation[:-1] only the full variation\n",
    "def find_sub_numberChange(data):\n",
    "    Booleans = [] #will contain the different Booleans if found in text\n",
    "    for i in data.index:\n",
    "        split_variation = re.split('(\\d+)', data.Variation[i]) # split based on a number\n",
    "        first_Amino = re.escape(split_variation[0]) #re.escpae uses variable as regex\n",
    "        last_Amino = re.escape(split_variation[-1])\n",
    "        new_regex  = first_Amino + r\"\\d+\" + last_Amino\n",
    "        Boolean = bool(re.search(new_regex, data.Text[i]))\n",
    "        Booleans.append(Boolean)\n",
    "        \n",
    "    sub_number_in_text = data[Booleans]\n",
    "    not_Boolean = [not i for i in Booleans]  \n",
    "    sub_number_no_text = data[not_Boolean]\n",
    "    \n",
    "    return sub_number_in_text, sub_number_no_text\n",
    "\n",
    "# for substitutions that are still not found, use other keywords\n",
    "def find_sub_pattern(data, pattern):    \n",
    "    Boolean = [pattern in data.Text[i] for i in data.index] ## because new indexing we use \n",
    "    \n",
    "    sub_in_text = data[Boolean]\n",
    "    not_Boolean = [not i for i in Boolean]  \n",
    "    sub_not_in_text = data[not_Boolean]\n",
    "    \n",
    "    return sub_in_text, sub_not_in_text\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "##### get_sentences_sub use a window to extract sentences where the subs appear. \n",
    "# If window_left & window_right = 0 => just taking the sentences with subs\n",
    "\n",
    "def get_sentences_sub(data, splitted_sentences, window_left, window_right):\n",
    "    data.index = range(len(data)) #makes sure that index is right\n",
    "    sentences_with_sub = [[] for _ in range(len(data))]\n",
    "    for i in range(len(splitted_sentences)):\n",
    "        sentences = splitted_sentences[i]\n",
    "        one_to_three_variation = pattern.sub(lambda x: One_to_Three_AA[x.group()], data.Variation[i][:-1])\n",
    "        Variation = data.Variation[i][:-1]        \n",
    "        for j in range(len(sentences)):                              \n",
    "            if (Variation in sentences[j]) or (one_to_three_variation in sentences[j]):\n",
    "                new_regex = r\"[\\S]*\" + re.escape(Variation) + r\"[\\S]*\" ###  r\"[\\S]*\" because we look for Variation[:-1] not just Variation\n",
    "                sentences[j] = re.sub(new_regex, 'placeholderMutation', sentences[j]) #case 1\n",
    "                ### We add the space to ' placeholderMutation' because sometimes there are letters in front of it\n",
    "                new_regex = re.escape(one_to_three_variation) + r\"[\\S]*\"\n",
    "                sentences[j] = re.sub(new_regex, ' placeholderMutation', sentences[j]) #case 2\n",
    "                sentences_with_sub[i].extend(sentences[max(j-window_left,0) : min(j+1+window_right, len(sentences)-1)])               \n",
    "    \n",
    "    return sentences_with_sub\n",
    "\n",
    "##### get_sentences_sub_number use a window to extract sentences where the subs appear that have different number i.e. G12V -> G_V\n",
    "\n",
    "def get_sentences_sub_number(data, splitted_sentences, window_left, window_right):\n",
    "    #position_sentences = [[] for _ in range(len(data))]  #### currently not used\n",
    "    data.index = range(len(data))\n",
    "    sentences_with_sub_number = [[] for _ in range(len(data))]\n",
    "    for i in range(len(splitted_sentences)):\n",
    "        sentences = splitted_sentences[i] \n",
    "        for j in range(len(sentences)):\n",
    "            split_variation = re.split('(\\d+)', data.Variation[i]) # split based on a number\n",
    "            first_Amino = re.escape(split_variation[0]) #re.escpae uses variable as regex\n",
    "            last_Amino = re.escape(split_variation[-1])\n",
    "            new_regex  = first_Amino + r\"\\d+\" + last_Amino\n",
    "            Boolean = bool(re.search(new_regex, sentences[j]))            \n",
    "            if Boolean:\n",
    "                sentences[j] = re.sub(new_regex, ' placeholderMutation', sentences[j]) # Again replacing the sentence with placeholder\n",
    "                sentences_with_sub_number[i].extend(sentences[max(j-window_left,0) : min(j+1+window_right,len(sentences)-1)])\n",
    "    \n",
    "    return sentences_with_sub_number\n",
    "\n",
    "# for substitutions that are still not found, use other keywords\n",
    "\n",
    "def get_sentences_pattern(data, splitted_sentences, pattern, window_left, window_right):\n",
    "    data.index = range(len(data)) #makes sure that index is right\n",
    "    sentences_with_sub = [[] for _ in range(len(data))]\n",
    "    for i in range(len(splitted_sentences)):\n",
    "        sentences = splitted_sentences[i] \n",
    "        for j in range(len(sentences)):                              \n",
    "            if (pattern in sentences[j]):\n",
    "                sentences_with_sub[i].extend(sentences[max(j-window_left,0) : min(j+1+window_right, len(sentences)-1)])               \n",
    "    \n",
    "    return sentences_with_sub"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#### Converts list of sentences into one string of sentences for each document => to use for tfidf etc.\n",
    "def sentences_to_string(sentences_list):\n",
    "    sentence_strings = []\n",
    "    for sentences in sentences_list:\n",
    "        sentence_string =  ' '.join(str(sentence) for sentence in sentences)\n",
    "        sentence_strings.append(sentence_string)\n",
    "    \n",
    "    return sentence_strings "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Subs processing of the data set "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "######### First find those that have the format of being a substitution in data\n",
    "data_all['Substitutions_var'] = data_all.Variation.apply(lambda x: bool(re.search('^[A-Z]\\\\d+[A-Z*]$', x))*1) #multiplying by 1 converts True to 1, False to 0 => Maybe modify this later?\n",
    "data_all['Stop_codon_var'] = data_all.Variation.apply(lambda x: bool(re.search('[*]', x))*1) #multiplying by 1 converts True to 1, False to 0\n",
    "data_sub = data_all[data_all['Substitutions_var']==1] ### Now we know the index of where a substitution occurs - the data_sub"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Subs inside the text"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "## First consider the subs that appear in text\n",
    "sub_text, sub_no_text = find_sub(data_sub) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "## use tokenizer to split into sentences of all the subs in text \n",
    "NLTK_sub = [sent_tokenize(sub_text.Text[i]) for i in sub_text.index] "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# extract window for the sub sentences where they appear\n",
    "# !! Use [:] because it makes a copy and doesn't change anything to the original indexes\n",
    "sub_sentences = get_sentences_sub(sub_text[:], NLTK_sub[:], window_left = 0, window_right = 0) # choosing for window 0 as default now\n",
    "sub_sentences_string = sentences_to_string(sub_sentences)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\rafiz\\Anaconda3\\lib\\site-packages\\pandas\\core\\indexing.py:179: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  self._setitem_with_indexer(indexer, value)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "1       Using select c-CBL somatic mutations such as S...\n",
       "2       Using select c-CBL somatic mutations such as S...\n",
       "3       Most of the changes were novel, although 4 cas...\n",
       "4       Finally, the third group constituted mutations...\n",
       "5       Finally, the third group constituted mutations...\n",
       "6       The second group of mutants (M374V, placeholde...\n",
       "8       We investigated the mechanism by which placeho...\n",
       "9       We investigated the mechanism by which CBL-Y37...\n",
       "10      Finally, the third group constituted mutations...\n",
       "11      Purified PCR amplicons (CBLWT, CBLC381A, place...\n",
       "12      Purified PCR amplicons (CBLWT, CBLC381A, CBLK3...\n",
       "13      Purified PCR amplicons (CBLWT, placeholderMuta...\n",
       "14      The second group of mutants (M374V, V430M, pla...\n",
       "15      Purified PCR amplicons (CBLWT, CBLC381A, CBLK3...\n",
       "17      When introduced into Lin- Sca1+ c-Kit+ (LSK) H...\n",
       "18      The second group of mutants placeholderMutatio...\n",
       "19      When introduced into Lin- Sca1+ c-Kit+ (LSK) H...\n",
       "20      Using select c-CBL somatic mutations such as p...\n",
       "21      The first group of damaging mutants included p...\n",
       "22      Finally, the third group constituted mutations...\n",
       "23      Transforming activities of CBL mutants  We det...\n",
       "24      Case UPN 3 had a C>G missense change in exon 9...\n",
       "25      Transforming activities of CBL mutants  We det...\n",
       "26      Twenty-five subjects with a relatively consist...\n",
       "27      The proband in the second family has mutant al...\n",
       "28      Each of these promoter mutations resulted in a...\n",
       "29      In one case the proband is homozygous for a no...\n",
       "30      The proband in the second family has mutant al...\n",
       "32      In one case the proband is homozygous for a no...\n",
       "34      Each of these promoter mutations resulted in a...\n",
       "                              ...                        \n",
       "8959    The of placeholderMutation listed placeholderM...\n",
       "8960    Download placeholderMutation is HeLa five of t...\n",
       "8961    Rodriguez-Viciana also structure non-covalent ...\n",
       "8962    In MRI were bands effort mutants subjects plac...\n",
       "8963    For mutations mutants and Y220C of in placehol...\n",
       "8964    Mutations results placeholderMutation is are t...\n",
       "8965    Ramucirumab cells 3S G12V after expression of ...\n",
       "8966    Stock data placeholderMutation species tumours...\n",
       "8967    The comparison placeholderMutation memory eith...\n",
       "8968    Images placeholderMutation wild-type merlin pl...\n",
       "8969    The placeholderMutation Engl demonstrated anal...\n",
       "8970    HERmark serum larger activation placeholderMut...\n",
       "8971    This approach resulted in the rescue of p53 ca...\n",
       "8972    Genetic variants placeholderMutation patients ...\n",
       "8973    tumor-bearing transitions results is EGFR the ...\n",
       "8974    Therefore represenation of naturally are to bi...\n",
       "8975    After placeholderMutation affected positively ...\n",
       "8976    All patient second protocadherin the successfu...\n",
       "8977    Seven which of carried N676D to normal is are ...\n",
       "8978    RT addition on placeholderMutation harvested t...\n",
       "8979    As of the occurring responders expression Tabl...\n",
       "8980    The proteins extracted records placeholderMuta...\n",
       "8981    Figure practice cells the known placeholderMut...\n",
       "8982    These contrast complexes placeholderMutation s...\n",
       "8983    Figure largeDownload these has placeholderMuta...\n",
       "8984    These disease point both role the was signal f...\n",
       "8985    For group-by-day variants not for KIT placehol...\n",
       "8986    All most placeholderMutation of has with to SM...\n",
       "8987    Figure these are MYC also protein subsequently...\n",
       "8988    In Author mutations response data placeholderM...\n",
       "Name: Text, Length: 7794, dtype: object"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Replace text in data_all\n",
    "data_all.Text.iloc[sub_text.index] = sub_sentences_string #iloc for indexing based on integers\n",
    "data_all.Text.iloc[sub_text.index]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Subs with a different number inside the text\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# now the subs that don't appear in text: one reason is different number in the substitution. f.e. G12V -> G13V \n",
    "sub_number_text, sub_number_no_text = find_sub_numberChange(sub_no_text) # 108 such cases out of 411 = nice improvement"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "## use tokenizer to split into sentences of subs that have different number in text \n",
    "NLTK_sub_number = [sent_tokenize(sub_number_text.Text[i]) for i in sub_number_text.index]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# extract window for the sub sentences where they appear\n",
    "# !! Use [:] because it makes a copy and doesn't change anything to the original indexes\n",
    "sub_number_sentences = get_sentences_sub_number(sub_number_text[:], NLTK_sub_number[:], window_left = 0, window_right = 0)\n",
    "sub_number_string = sentences_to_string(sub_number_sentences)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\rafiz\\Anaconda3\\lib\\site-packages\\pandas\\core\\indexing.py:179: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  self._setitem_with_indexer(indexer, value)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "44      His-tagged versions of the catalytic region of...\n",
       "49      His-tagged versions of the catalytic region of...\n",
       "61      His-tagged versions of the catalytic region of...\n",
       "63      His-tagged versions of the catalytic region of...\n",
       "66      His-tagged versions of the catalytic region of...\n",
       "73      While both were missense mutations, in silico ...\n",
       "208      placeholderMutation and T847I were also found...\n",
       "213     Complex mutation  placeholderMutation and dele...\n",
       "267     As a positive control, we introduced the well-...\n",
       "275     And in the series of Pallis and colleagues (12...\n",
       "306     Recurrent mutations in H3F3A, which encodes th...\n",
       "307     Histone H3-K42me2 may regulate transcription  ...\n",
       "308     Recurrent mutations in H3F3A, which encodes th...\n",
       "521     Both combinations contained mutations of codon...\n",
       "607     In the present study, we have focused on two d...\n",
       "632     Such mutant amino acid specificity was also ap...\n",
       "730     In contrast, ectopic expression of FGFR4  plac...\n",
       "1081    Recently, somatic mutations of Tyr641 (Y641F, ...\n",
       "1083    Recently, somatic mutations of Tyr641 (Y641F, ...\n",
       "1085    Recently, somatic mutations of Tyr641 (Y641F, ...\n",
       "1091    Seven of 12 mutations in RXRA (retinoid × nucl...\n",
       "1092    Seven of 12 mutations in RXRA (retinoid × nucl...\n",
       "1121    Three of the MET mutated variants, Y1248H, L12...\n",
       "1128    Two Met variants tested, M1268T and H1112Y, we...\n",
       "1131    Two Met variants tested, M1268T and  placehold...\n",
       "1136    Here we characterized the response of the MET ...\n",
       "1139    We describe a case of clinical activity of cri...\n",
       "1141    The  placeholderMutation MET mutation segregat...\n",
       "1142    The locations of the RET mutations, E768D, L79...\n",
       "1144    MET  placeholderMutation A new mutation in cod...\n",
       "                              ...                        \n",
       "2773    In contrast, the B-Raf  placeholderMutation mu...\n",
       "2790    A few VUS, such as BRCA2 IVS13+5 G>C and BRCA1...\n",
       "2815    After combination of the multifactorial odds w...\n",
       "2816    Protein modeling predictions were possible for...\n",
       "2837    Variants BRCA1 IVS2-14 C>T,  placeholderMutati...\n",
       "2886    Analysis of potential splicing aberrations wer...\n",
       "2919    Activation of oncogenic alleles of KRAS, BRAF,...\n",
       "3065    In contrast, 11 of the 12 mutated sHDs that we...\n",
       "3067    In contrast, 11 of the 12 mutated sHDs that we...\n",
       "3068    The mutations  placeholderMutation and  placeh...\n",
       "3070    In contrast, 11 of the 12 mutated sHDs that we...\n",
       "3071    The mutations  placeholderMutation and  placeh...\n",
       "3074    In contrast, 11 of the 12 mutated sHDs that we...\n",
       "3075    The mutations  placeholderMutation and  placeh...\n",
       "3076    In contrast, 11 of the 12 mutated sHDs that we...\n",
       "3079    The mutations  placeholderMutation and  placeh...\n",
       "3080    Only the  placeholderMutation mutation, found ...\n",
       "3082    The two mutations that did not generate NTM*, ...\n",
       "3083    In contrast, three mutations (L1601P,  placeho...\n",
       "3090    Of note, L1601P and  placeholderMutation, the ...\n",
       "3092    In contrast, 11 of the 12 mutated sHDs that we...\n",
       "3261    It was reported that substitutions of arginine...\n",
       "3265    In addition, two codon 609 mutations ( placeho...\n",
       "3288    It was reported that substitutions of arginine...\n",
       "3292    In addition, two codon 609 mutations ( placeho...\n",
       "5043    Here we characterized the response of the MET ...\n",
       "5574    However, forced expression of stable MYC ( pla...\n",
       "5808    NRAS (G12A, G12D, Q61K, Q61R, Q61L), BRAF (V60...\n",
       "6895    The two mutations that did not generate NTM*, ...\n",
       "8513    Point mutations (M536I, M538I, I548V,  placeho...\n",
       "Name: Text, Length: 108, dtype: object"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_all.Text.iloc[sub_number_text.index] = sub_number_string #iloc for indexing based on integers\n",
    "data_all.Text.iloc[sub_number_text.index]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Subs that are still not found"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# subs that are still not found, look based on a pattern like the word 'mutat' in text\n",
    "sub_pattern_text, sub_pattern_no_text = find_sub_pattern(sub_number_no_text, 'mutat') # 108 such cases out of 411 = nice improvement"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Class</th>\n",
       "      <th>Gene</th>\n",
       "      <th>ID</th>\n",
       "      <th>Text</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Text_words</th>\n",
       "      <th>Substitutions_var</th>\n",
       "      <th>Stop_codon_var</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>140</th>\n",
       "      <td>5.0</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>140</td>\n",
       "      <td>The accurate determination of perfluoroalkyl s...</td>\n",
       "      <td>I491M</td>\n",
       "      <td>430</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>145</th>\n",
       "      <td>2.0</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>145</td>\n",
       "      <td>The accurate determination of perfluoroalkyl s...</td>\n",
       "      <td>K467T</td>\n",
       "      <td>430</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>259</th>\n",
       "      <td>2.0</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>259</td>\n",
       "      <td>The accurate determination of perfluoroalkyl s...</td>\n",
       "      <td>S464L</td>\n",
       "      <td>430</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>416</th>\n",
       "      <td>4.0</td>\n",
       "      <td>TP53</td>\n",
       "      <td>416</td>\n",
       "      <td>The effects of chlorpromazine on various prope...</td>\n",
       "      <td>P151S</td>\n",
       "      <td>718</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1109</th>\n",
       "      <td>1.0</td>\n",
       "      <td>FANCA</td>\n",
       "      <td>1109</td>\n",
       "      <td>null</td>\n",
       "      <td>S1088F</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1407</th>\n",
       "      <td>6.0</td>\n",
       "      <td>FGFR3</td>\n",
       "      <td>1407</td>\n",
       "      <td>null</td>\n",
       "      <td>K508M</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1613</th>\n",
       "      <td>4.0</td>\n",
       "      <td>VHL</td>\n",
       "      <td>1613</td>\n",
       "      <td>The case of a 40-year-old woman with severe ed...</td>\n",
       "      <td>L158Q</td>\n",
       "      <td>53</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1936</th>\n",
       "      <td>7.0</td>\n",
       "      <td>CARD11</td>\n",
       "      <td>1936</td>\n",
       "      <td>Left ventricular (LV) remodeling is a signific...</td>\n",
       "      <td>G116S</td>\n",
       "      <td>3681</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1942</th>\n",
       "      <td>7.0</td>\n",
       "      <td>CARD11</td>\n",
       "      <td>1942</td>\n",
       "      <td>Left ventricular (LV) remodeling is a signific...</td>\n",
       "      <td>E127G</td>\n",
       "      <td>3681</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2302</th>\n",
       "      <td>7.0</td>\n",
       "      <td>JAK1</td>\n",
       "      <td>2302</td>\n",
       "      <td>Regulatory T (T reg) cells have a major role i...</td>\n",
       "      <td>R724H</td>\n",
       "      <td>8287</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2309</th>\n",
       "      <td>7.0</td>\n",
       "      <td>JAK1</td>\n",
       "      <td>2309</td>\n",
       "      <td>Regulatory T (T reg) cells have a major role i...</td>\n",
       "      <td>R879C</td>\n",
       "      <td>8287</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2504</th>\n",
       "      <td>6.0</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>2504</td>\n",
       "      <td>Identification of protein-protein interaction...</td>\n",
       "      <td>V11A</td>\n",
       "      <td>231</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2755</th>\n",
       "      <td>7.0</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>2755</td>\n",
       "      <td>null</td>\n",
       "      <td>G596C</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2980</th>\n",
       "      <td>2.0</td>\n",
       "      <td>KIT</td>\n",
       "      <td>2980</td>\n",
       "      <td>TO THE EDITOR: I commend Drs. Freeman and Bank...</td>\n",
       "      <td>N655K</td>\n",
       "      <td>362</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Class    Gene    ID                                               Text  \\\n",
       "140     5.0    EGFR   140  The accurate determination of perfluoroalkyl s...   \n",
       "145     2.0    EGFR   145  The accurate determination of perfluoroalkyl s...   \n",
       "259     2.0    EGFR   259  The accurate determination of perfluoroalkyl s...   \n",
       "416     4.0    TP53   416  The effects of chlorpromazine on various prope...   \n",
       "1109    1.0   FANCA  1109                                               null   \n",
       "1407    6.0   FGFR3  1407                                               null   \n",
       "1613    4.0     VHL  1613  The case of a 40-year-old woman with severe ed...   \n",
       "1936    7.0  CARD11  1936  Left ventricular (LV) remodeling is a signific...   \n",
       "1942    7.0  CARD11  1942  Left ventricular (LV) remodeling is a signific...   \n",
       "2302    7.0    JAK1  2302  Regulatory T (T reg) cells have a major role i...   \n",
       "2309    7.0    JAK1  2309  Regulatory T (T reg) cells have a major role i...   \n",
       "2504    6.0   BRCA1  2504   Identification of protein-protein interaction...   \n",
       "2755    7.0    BRAF  2755                                               null   \n",
       "2980    2.0     KIT  2980  TO THE EDITOR: I commend Drs. Freeman and Bank...   \n",
       "\n",
       "     Variation  Text_words  Substitutions_var  Stop_codon_var  \n",
       "140      I491M         430                  1               0  \n",
       "145      K467T         430                  1               0  \n",
       "259      S464L         430                  1               0  \n",
       "416      P151S         718                  1               0  \n",
       "1109    S1088F           1                  1               0  \n",
       "1407     K508M           1                  1               0  \n",
       "1613     L158Q          53                  1               0  \n",
       "1936     G116S        3681                  1               0  \n",
       "1942     E127G        3681                  1               0  \n",
       "2302     R724H        8287                  1               0  \n",
       "2309     R879C        8287                  1               0  \n",
       "2504      V11A         231                  1               0  \n",
       "2755     G596C           1                  1               0  \n",
       "2980     N655K         362                  1               0  "
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sub_pattern_no_text # only 14 texts that include sentences with the word 'mutat', of which three nulls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "## use tokenizer to split into sentences of subs that have keyword 'mutat'\n",
    "NLTK_sub_pattern = [sent_tokenize(sub_pattern_text.Text[i]) for i in sub_pattern_text.index]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# !! Use [:] because it makes a copy and doesn't change anything to the original indexes\n",
    "sub_pattern_sentences = get_sentences_pattern(sub_pattern_text[:], NLTK_sub_pattern[:], 'mutat', window_left = 0, window_right = 0)\n",
    "sub_pattern_string = sentences_to_string(sub_pattern_sentences)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\rafiz\\Anaconda3\\lib\\site-packages\\pandas\\core\\indexing.py:179: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  self._setitem_with_indexer(indexer, value)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "46      Mutations in CBL, encoding an E3 ubiquitin lig...\n",
       "80      This D1 domain contains only two serine residu...\n",
       "102     Identification of a high-risk disease-causing ...\n",
       "104     In this study, we profiled 70 colorectal cance...\n",
       "106     Identification of a high-risk disease-causing ...\n",
       "108     Most Lynch Syndrome-associated CRCs are report...\n",
       "111     Identification of a high-risk disease-causing ...\n",
       "112     Germ-line MSH6 mutations, which are rare in HN...\n",
       "136     In 2011, whole-exome sequencing studies showed...\n",
       "151     We investigated the association between mutati...\n",
       "155     Clinical data and epidermal growth factor rece...\n",
       "161     Clinical data and epidermal growth factor rece...\n",
       "168     Currently available methods of EGFR mutation d...\n",
       "178     Those were analyzed for EGFR and k-ras (Kirste...\n",
       "182     Purpose: Clinical reports about responsiveness...\n",
       "185     ‘Classical' mutations in the EGFR tyrosine kin...\n",
       "215     Purpose: Clinical features of epidermal growth...\n",
       "222     Findings in the last decade showed that mutati...\n",
       "249     ‘Classical' mutations in the EGFR tyrosine kin...\n",
       "344     Direct evidence of E-cadherin mutations trigge...\n",
       "346     Direct evidence of E-cadherin mutations trigge...\n",
       "348     Direct evidence of E-cadherin mutations trigge...\n",
       "371     Inheritance of germ-line mutant alleles of BRC...\n",
       "380     The tumour suppressor gene p53 is frequently m...\n",
       "384     Inheritance of germ-line mutant alleles of BRC...\n",
       "386     Although in the majority of cases only one of ...\n",
       "399     Inheritance of germ-line mutant alleles of BRC...\n",
       "430     Roughly half of all human tumours contain p53 ...\n",
       "441     Here, we report on the global gene expression ...\n",
       "464     In many cases, p53 gene mutations result in hi...\n",
       "                              ...                        \n",
       "3442    The excellent agreement between the structure/...\n",
       "3458    Inheritance of germ-line mutant alleles of BRC...\n",
       "3863    The splicing factor SF3B1 is the most commonly...\n",
       "3866    Recently, mutations in the B-Raf gene have bee...\n",
       "4180    The identification of the causative mutations ...\n",
       "4236    Class 2S: (1) impact on alternative splicing a...\n",
       "4514    FAP is caused by germline mutations in the ade...\n",
       "4548    ‘Classical' mutations in the EGFR tyrosine kin...\n",
       "4820    It occurs in excess among individuals with the...\n",
       "5082    By comparison with cDNA prepared from Ramos ce...\n",
       "5606    Indeed, in the early nineties, gain-of-functio...\n",
       "5704    Using ERCC2 DNA repair gene as an example, we ...\n",
       "6058     Abstract  The BRCA1 gene from individuals at ...\n",
       "6190    Identification of a high-risk disease-causing ...\n",
       "6348    However, PDGFRA mutations in melanomas have no...\n",
       "6447     Abstract  The BRCA1 gene from individuals at ...\n",
       "6579    In humans, MLH1 mutations underlie half of her...\n",
       "6643    Deletion or mutation of these CDK-inhibitors c...\n",
       "6710     Abstract  Background Heterozygous mutations i...\n",
       "6713    The excellent agreement between the structure/...\n",
       "6866    Identification of a high-risk disease-causing ...\n",
       "7341    In humans, MLH1 mutations underlie half of her...\n",
       "7383    We assessed somatic alleles of six receptor ty...\n",
       "7391    We here apply a direct comparative and compreh...\n",
       "7647     Abstract  The BRCA1 gene from individuals at ...\n",
       "7877    We demonstrate that phosphatidylinositol 3-kin...\n",
       "8047    The excellent agreement between the structure/...\n",
       "8113    The excellent agreement between the structure/...\n",
       "8324    Indeed, in the early nineties, gain-of-functio...\n",
       "8921    The systematic characterization of somatic mut...\n",
       "Name: Text, Length: 290, dtype: object"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_all.Text.iloc[sub_pattern_text.index] = sub_pattern_string #iloc for indexing based on integers\n",
    "data_all.Text.iloc[sub_pattern_text.index]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Non-substitutions (no subs)\n",
    "## Functions for pre-processing of no_subs like fusions, deletions, truncations, etc."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "## Gets different non-subs variation types by using regex\n",
    "def variation_regex(data, pattern): \n",
    "    Boolean = [bool(re.search(pattern, i, re.IGNORECASE)) for i in data.Variation]\n",
    "    data_regex = data[Boolean] \n",
    "    not_Boolean = [not i for i in Boolean]  \n",
    "    data_no_regex = data[not_Boolean]\n",
    "    \n",
    "    return (data_regex, data_no_regex)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def custom_find(data):\n",
    "    l=[\"mutat\",\"variat\",\"fusion\",\"deleti\",\"amplific\",\"framesh\",\n",
    "    \"truncat\",\"fs\",\"exon\",\"dupl\"]\n",
    "    Boolean={}\n",
    "    for word in l:\n",
    "        Boolean[word] = [re.search(word,data.Text[i]) for i in data.index] ## because new indexing we use \n",
    "    Bool=[any(tup) for tup in zip(Boolean[\"mutat\"],Boolean[\"variat\"],Boolean[\"fusion\"],\n",
    "        Boolean[\"deleti\"],Boolean[\"amplific\"],Boolean[\"truncat\"],\n",
    "        Boolean[\"fs\"],Boolean[\"framesh\"],Boolean[\"exon\"],Boolean[\"dupl\"])]\n",
    "    no_sub_in_text=data[Bool]\n",
    "    no_Bool=[not i for i in Bool]\n",
    "    no_sub_not_in_text=data[no_Bool]\n",
    "    return no_sub_in_text,no_sub_not_in_text\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def get_sentences_pattern(data, splitted_sentences, pattern, window_left, window_right):\n",
    "    data.index = range(len(data)) #makes sure that index is right\n",
    "    sentences_with_sub = [[] for _ in range(len(data))]\n",
    "    for i in range(len(splitted_sentences)):\n",
    "        sentences = splitted_sentences[i] \n",
    "        for j in range(len(sentences)):                              \n",
    "            if (pattern in sentences[j]):\n",
    "                sentences_with_sub[i].extend(sentences[max(j-window_left,0) : min(j+1+window_right, len(sentences)-1)])               \n",
    "    \n",
    "    return sentences_with_sub"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def get_sentences_nosub(data, splitted_sentences, window_left, window_right):\n",
    "    #position_sentences = [[] for _ in range(len(data))]  #### currently not used\n",
    "    data.index = range(len(data))\n",
    "    sentences_with_sub = [[] for _ in range(len(data))]\n",
    "\n",
    "    for i in range(len(splitted_sentences)):\n",
    "        sentences = splitted_sentences[i]\n",
    "        l=[\"mutat\",\"variat\",\"fusion\",\"deleti\",\"amplific\",\"framesh\",\n",
    "        \"truncat\",\"fs\",\"exon\",\"dupl\"]     \n",
    "        for j in range(len(sentences)):    \n",
    "            for keyword in l:\n",
    "                if keyword in sentences[j]:\n",
    "                    sentences_with_sub[i].extend(sentences[max(j-window_left,0) : min(j+1+window_right, len(sentences)-1)])\n",
    "    return sentences_with_sub"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## No subs processing of the data set "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\rafiz\\Anaconda3\\lib\\site-packages\\pandas\\core\\indexing.py:179: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  self._setitem_with_indexer(indexer, value)\n"
     ]
    }
   ],
   "source": [
    "#### First find those that have the format of being a non-substitutions in the data\n",
    "\n",
    "# Initialize some of the variables already (because we splice them, could be filled with NA's otherwise)\n",
    "# Represents 783 from train and test set\n",
    "data_all['gene_fusion_var'] = 0\n",
    "data_all['Deletion_var'] = 0\n",
    "data_all['del_or_ins_var'] = 0\n",
    "\n",
    "####### Fusions : 'Fusions' ############\n",
    "data_all['Fusion_var'] = data_all.Variation.apply(lambda x: bool(re.search('^fusion', x, re.IGNORECASE))*1) #multiplying by 1 converts True to 1, False to 0\n",
    "_ , new_data_all = variation_regex(data_all, '^fusion')  #37 cases\n",
    "\n",
    "###### Fusions: 'Gene-Gene fusion' ########\n",
    "data_all['gene_fusion_var'].loc[new_data_all.index] = new_data_all.Variation.apply(lambda x: bool(re.search('fusion', x, re.IGNORECASE))*1)\n",
    "_ , new_data_all = variation_regex(new_data_all, 'fusion') #160 cases\n",
    "\n",
    "####### Deletions: 'Deletions' ############\n",
    "data_all['Deletion_var'].loc[new_data_all.index] = new_data_all.Variation.apply(lambda x: bool(re.search('^del', x, re.IGNORECASE))*1)\n",
    "_, new_data_all = variation_regex(new_data_all, '^del') # 88 cases\n",
    "\n",
    "####### Deletions & Insertions wheteher together or seperately (doesn't make a big difference IMO)\n",
    "data_all['del_or_ins_var'].loc[new_data_all.index] = new_data_all.Variation.apply(lambda x: bool(re.search('del|ins', x, re.IGNORECASE))*1)\n",
    "# 196 cases\n",
    "\n",
    "###### Amplifications #########\n",
    "data_all['Amplification_var'] = data_all.Variation.apply(lambda x: bool(re.search('ampl', x, re.IGNORECASE))*1) # 79 cases\n",
    "\n",
    "###### Truncations ########### Don't forget there are 'Truncating mutations'  '_trunc' \n",
    "data_all['Truncation_var'] = data_all.Variation.apply(lambda x: bool(re.search('trunc', x, re.IGNORECASE))*1) # 118 cases\n",
    "\n",
    "####### Exons #########\n",
    "data_all['exon_var'] = data_all.Variation.apply(lambda x: bool(re.search('exon', x, re.IGNORECASE))*1) \n",
    "\n",
    "####### Frameshift mutations ########\n",
    "data_all['frameshift_var'] = data_all.Variation.apply(lambda x: bool(re.search('fs', x, re.IGNORECASE))*1) # 22 cases\n",
    "\n",
    "####### Duplications ##############\n",
    "data_all['dup_var'] = data_all.Variation.apply(lambda x: bool(re.search('dup', x, re.IGNORECASE))*1)\n",
    "\n",
    "####### Hypermethylation ##########\n",
    "data_all['overexpression_var'] = data_all.Variation.apply(lambda x: bool(re.search('Overexpression', x, re.IGNORECASE))*1) \n",
    "\n",
    "####### Splice ##########\n",
    "data_all['splice_var'] = data_all.Variation.apply(lambda x: bool(re.search('splice', x, re.IGNORECASE))*1) \n",
    "\n",
    "####### Hypermethylation ##########\n",
    "data_all['hypermethylation_var'] = data_all.Variation.apply(lambda x: bool(re.search('hypermethylation', x, re.IGNORECASE))*1)\n",
    "\n",
    "####### DNA binding ##########\n",
    "data_all['DNA_binding_var'] = data_all.Variation.apply(lambda x: bool(re.search('DNA', x, re.IGNORECASE))*1)\n",
    "\n",
    "####### Null ##########\n",
    "data_all['null_var'] = data_all.Variation.apply(lambda x: bool(re.search('null', x, re.IGNORECASE))*1)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Class</th>\n",
       "      <th>Gene</th>\n",
       "      <th>ID</th>\n",
       "      <th>Text</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Text_words</th>\n",
       "      <th>Substitutions_var</th>\n",
       "      <th>Stop_codon_var</th>\n",
       "      <th>gene_fusion_var</th>\n",
       "      <th>Deletion_var</th>\n",
       "      <th>...</th>\n",
       "      <th>exon_var</th>\n",
       "      <th>frameshift_var</th>\n",
       "      <th>dup_var</th>\n",
       "      <th>overexpression_var</th>\n",
       "      <th>splice_var</th>\n",
       "      <th>hypermethylation</th>\n",
       "      <th>null</th>\n",
       "      <th>hypermethylation_var</th>\n",
       "      <th>DNA_binding_var</th>\n",
       "      <th>null_var</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>7.0</td>\n",
       "      <td>TERT</td>\n",
       "      <td>31</td>\n",
       "      <td>Sequencing studies have identified many recurr...</td>\n",
       "      <td>Promoter Mutations</td>\n",
       "      <td>21096</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150</th>\n",
       "      <td>7.0</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>150</td>\n",
       "      <td>Alterations ofthe EGFR gene occur frequently i...</td>\n",
       "      <td>EGFRvIII</td>\n",
       "      <td>25595</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152</th>\n",
       "      <td>7.0</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>152</td>\n",
       "      <td>Abstract Glioblastomas with EGFR amplificatio...</td>\n",
       "      <td>EGFRvII</td>\n",
       "      <td>12517</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170</th>\n",
       "      <td>7.0</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>170</td>\n",
       "      <td>Abstract  The epidermal growth factor recepto...</td>\n",
       "      <td>EGFRvV</td>\n",
       "      <td>9121</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>211</th>\n",
       "      <td>7.0</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>211</td>\n",
       "      <td>Oncogenic EGFR mutations are found in 10-35% o...</td>\n",
       "      <td>EGFR-KDD</td>\n",
       "      <td>2724</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>238</th>\n",
       "      <td>7.0</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>238</td>\n",
       "      <td>Tumor cells often subvert normal regulatory me...</td>\n",
       "      <td>EGFRvIV</td>\n",
       "      <td>7536</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>350</th>\n",
       "      <td>1.0</td>\n",
       "      <td>CDH1</td>\n",
       "      <td>350</td>\n",
       "      <td>In diffuse gastric carcinoma, despite common E...</td>\n",
       "      <td>Epigenetic Silencing</td>\n",
       "      <td>3955</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>354</th>\n",
       "      <td>4.0</td>\n",
       "      <td>EP300</td>\n",
       "      <td>354</td>\n",
       "      <td>The transcriptional coactivator p300/CBP (CREB...</td>\n",
       "      <td>R1627</td>\n",
       "      <td>3275</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>355</th>\n",
       "      <td>4.0</td>\n",
       "      <td>EP300</td>\n",
       "      <td>355</td>\n",
       "      <td>The transcriptional coactivator p300/CBP (CREB...</td>\n",
       "      <td>C1385</td>\n",
       "      <td>3275</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>631</th>\n",
       "      <td>4.0</td>\n",
       "      <td>FBXW7</td>\n",
       "      <td>631</td>\n",
       "      <td>This study focused on a cell cycle regulatory ...</td>\n",
       "      <td>Copy Number Loss</td>\n",
       "      <td>29536</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1786</th>\n",
       "      <td>7.0</td>\n",
       "      <td>AR</td>\n",
       "      <td>1786</td>\n",
       "      <td>Androgen receptor (AR) splice variants lacking...</td>\n",
       "      <td>ARv567es</td>\n",
       "      <td>40550</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1801</th>\n",
       "      <td>7.0</td>\n",
       "      <td>AR</td>\n",
       "      <td>1801</td>\n",
       "      <td>Androgen receptor (AR) splice variants lacking...</td>\n",
       "      <td>AR-V7</td>\n",
       "      <td>40550</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1938</th>\n",
       "      <td>2.0</td>\n",
       "      <td>CARD11</td>\n",
       "      <td>1938</td>\n",
       "      <td>Self-tolerance and immunity are actively acqui...</td>\n",
       "      <td>L232LI</td>\n",
       "      <td>5325</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1941</th>\n",
       "      <td>7.0</td>\n",
       "      <td>CARD11</td>\n",
       "      <td>1941</td>\n",
       "      <td>The regulated activation of NF-κB by antigen r...</td>\n",
       "      <td>L225LI</td>\n",
       "      <td>12640</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2054</th>\n",
       "      <td>6.0</td>\n",
       "      <td>MYC</td>\n",
       "      <td>2054</td>\n",
       "      <td>Evasion of apoptosis is critical in Myc-induce...</td>\n",
       "      <td>MYC-nick</td>\n",
       "      <td>13692</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2718</th>\n",
       "      <td>7.0</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>2718</td>\n",
       "      <td>Summary Activated RAS promotes dimerization o...</td>\n",
       "      <td>p61BRAF</td>\n",
       "      <td>2203</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3109</th>\n",
       "      <td>8.0</td>\n",
       "      <td>DNMT3B</td>\n",
       "      <td>3109</td>\n",
       "      <td>Epigenetic changes are among the most common a...</td>\n",
       "      <td>DNMT3B7</td>\n",
       "      <td>16484</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3135</th>\n",
       "      <td>3.0</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>3135</td>\n",
       "      <td>To characterize somatic alterations in colorec...</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>8084</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3254</th>\n",
       "      <td>4.0</td>\n",
       "      <td>CASP8</td>\n",
       "      <td>3254</td>\n",
       "      <td>Caspase-8 (Fas-associating protein with death ...</td>\n",
       "      <td>CASP8L</td>\n",
       "      <td>19171</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4439</th>\n",
       "      <td>NaN</td>\n",
       "      <td>BCL2L11</td>\n",
       "      <td>1118</td>\n",
       "      <td>Abstract Purpose  BIM is essential for the re...</td>\n",
       "      <td>Single Nucleotide Polymorphism</td>\n",
       "      <td>3778</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>20 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Class     Gene    ID                                               Text  \\\n",
       "31      7.0     TERT    31  Sequencing studies have identified many recurr...   \n",
       "150     7.0     EGFR   150  Alterations ofthe EGFR gene occur frequently i...   \n",
       "152     7.0     EGFR   152   Abstract Glioblastomas with EGFR amplificatio...   \n",
       "170     7.0     EGFR   170   Abstract  The epidermal growth factor recepto...   \n",
       "211     7.0     EGFR   211  Oncogenic EGFR mutations are found in 10-35% o...   \n",
       "238     7.0     EGFR   238  Tumor cells often subvert normal regulatory me...   \n",
       "350     1.0     CDH1   350  In diffuse gastric carcinoma, despite common E...   \n",
       "354     4.0    EP300   354  The transcriptional coactivator p300/CBP (CREB...   \n",
       "355     4.0    EP300   355  The transcriptional coactivator p300/CBP (CREB...   \n",
       "631     4.0    FBXW7   631  This study focused on a cell cycle regulatory ...   \n",
       "1786    7.0       AR  1786  Androgen receptor (AR) splice variants lacking...   \n",
       "1801    7.0       AR  1801  Androgen receptor (AR) splice variants lacking...   \n",
       "1938    2.0   CARD11  1938  Self-tolerance and immunity are actively acqui...   \n",
       "1941    7.0   CARD11  1941  The regulated activation of NF-κB by antigen r...   \n",
       "2054    6.0      MYC  2054  Evasion of apoptosis is critical in Myc-induce...   \n",
       "2718    7.0     BRAF  2718   Summary Activated RAS promotes dimerization o...   \n",
       "3109    8.0   DNMT3B  3109  Epigenetic changes are among the most common a...   \n",
       "3135    3.0     KRAS  3135  To characterize somatic alterations in colorec...   \n",
       "3254    4.0    CASP8  3254  Caspase-8 (Fas-associating protein with death ...   \n",
       "4439    NaN  BCL2L11  1118   Abstract Purpose  BIM is essential for the re...   \n",
       "\n",
       "                           Variation  Text_words  Substitutions_var  \\\n",
       "31                Promoter Mutations       21096                  0   \n",
       "150                         EGFRvIII       25595                  0   \n",
       "152                          EGFRvII       12517                  0   \n",
       "170                           EGFRvV        9121                  0   \n",
       "211                         EGFR-KDD        2724                  0   \n",
       "238                          EGFRvIV        7536                  0   \n",
       "350             Epigenetic Silencing        3955                  0   \n",
       "354                            R1627        3275                  0   \n",
       "355                            C1385        3275                  0   \n",
       "631                 Copy Number Loss       29536                  0   \n",
       "1786                        ARv567es       40550                  0   \n",
       "1801                           AR-V7       40550                  0   \n",
       "1938                          L232LI        5325                  0   \n",
       "1941                          L225LI       12640                  0   \n",
       "2054                        MYC-nick       13692                  0   \n",
       "2718                         p61BRAF        2203                  0   \n",
       "3109                         DNMT3B7       16484                  0   \n",
       "3135                        Wildtype        8084                  0   \n",
       "3254                          CASP8L       19171                  0   \n",
       "4439  Single Nucleotide Polymorphism        3778                  0   \n",
       "\n",
       "      Stop_codon_var  gene_fusion_var  Deletion_var    ...     exon_var  \\\n",
       "31                 0                0             0    ...            0   \n",
       "150                0                0             0    ...            0   \n",
       "152                0                0             0    ...            0   \n",
       "170                0                0             0    ...            0   \n",
       "211                0                0             0    ...            0   \n",
       "238                0                0             0    ...            0   \n",
       "350                0                0             0    ...            0   \n",
       "354                0                0             0    ...            0   \n",
       "355                0                0             0    ...            0   \n",
       "631                0                0             0    ...            0   \n",
       "1786               0                0             0    ...            0   \n",
       "1801               0                0             0    ...            0   \n",
       "1938               0                0             0    ...            0   \n",
       "1941               0                0             0    ...            0   \n",
       "2054               0                0             0    ...            0   \n",
       "2718               0                0             0    ...            0   \n",
       "3109               0                0             0    ...            0   \n",
       "3135               0                0             0    ...            0   \n",
       "3254               0                0             0    ...            0   \n",
       "4439               0                0             0    ...            0   \n",
       "\n",
       "      frameshift_var  dup_var  overexpression_var  splice_var  \\\n",
       "31                 0        0                   0           0   \n",
       "150                0        0                   0           0   \n",
       "152                0        0                   0           0   \n",
       "170                0        0                   0           0   \n",
       "211                0        0                   0           0   \n",
       "238                0        0                   0           0   \n",
       "350                0        0                   0           0   \n",
       "354                0        0                   0           0   \n",
       "355                0        0                   0           0   \n",
       "631                0        0                   0           0   \n",
       "1786               0        0                   0           0   \n",
       "1801               0        0                   0           0   \n",
       "1938               0        0                   0           0   \n",
       "1941               0        0                   0           0   \n",
       "2054               0        0                   0           0   \n",
       "2718               0        0                   0           0   \n",
       "3109               0        0                   0           0   \n",
       "3135               0        0                   0           0   \n",
       "3254               0        0                   0           0   \n",
       "4439               0        0                   0           0   \n",
       "\n",
       "      hypermethylation  null  hypermethylation_var  DNA_binding_var  null_var  \n",
       "31                   0     0                     0                0         0  \n",
       "150                  0     0                     0                0         0  \n",
       "152                  0     0                     0                0         0  \n",
       "170                  0     0                     0                0         0  \n",
       "211                  0     0                     0                0         0  \n",
       "238                  0     0                     0                0         0  \n",
       "350                  0     0                     0                0         0  \n",
       "354                  0     0                     0                0         0  \n",
       "355                  0     0                     0                0         0  \n",
       "631                  0     0                     0                0         0  \n",
       "1786                 0     0                     0                0         0  \n",
       "1801                 0     0                     0                0         0  \n",
       "1938                 0     0                     0                0         0  \n",
       "1941                 0     0                     0                0         0  \n",
       "2054                 0     0                     0                0         0  \n",
       "2718                 0     0                     0                0         0  \n",
       "3109                 0     0                     0                0         0  \n",
       "3135                 0     0                     0                0         0  \n",
       "3254                 0     0                     0                0         0  \n",
       "4439                 0     0                     0                0         0  \n",
       "\n",
       "[20 rows x 24 columns]"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Remember there are still unassigned variations that we can assign to for the next run (20 in total)\n",
    "all_variations = data_all.loc[:, \"Substitutions_var\":]\n",
    "data_all[(all_variations.T == 0).all()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# Preprocess all the nosubs together\n",
    "nosub=data_all[data_all['Substitutions_var']==0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Class</th>\n",
       "      <th>Gene</th>\n",
       "      <th>ID</th>\n",
       "      <th>Text</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Text_words</th>\n",
       "      <th>Substitutions_var</th>\n",
       "      <th>Stop_codon_var</th>\n",
       "      <th>gene_fusion_var</th>\n",
       "      <th>Deletion_var</th>\n",
       "      <th>...</th>\n",
       "      <th>exon_var</th>\n",
       "      <th>frameshift_var</th>\n",
       "      <th>dup_var</th>\n",
       "      <th>overexpression_var</th>\n",
       "      <th>splice_var</th>\n",
       "      <th>hypermethylation</th>\n",
       "      <th>null</th>\n",
       "      <th>hypermethylation_var</th>\n",
       "      <th>DNA_binding_var</th>\n",
       "      <th>null_var</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1277</th>\n",
       "      <td>1.0</td>\n",
       "      <td>ARID5B</td>\n",
       "      <td>1277</td>\n",
       "      <td>null</td>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1639</th>\n",
       "      <td>6.0</td>\n",
       "      <td>FLT1</td>\n",
       "      <td>1639</td>\n",
       "      <td>null</td>\n",
       "      <td>Amplification</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1911</th>\n",
       "      <td>1.0</td>\n",
       "      <td>FOXP1</td>\n",
       "      <td>1911</td>\n",
       "      <td>The transcriptional network of the androgen re...</td>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>863</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4944</th>\n",
       "      <td>NaN</td>\n",
       "      <td>AURKB</td>\n",
       "      <td>1623</td>\n",
       "      <td>null</td>\n",
       "      <td>Amplification</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Class    Gene    ID                                               Text  \\\n",
       "1277    1.0  ARID5B  1277                                               null   \n",
       "1639    6.0    FLT1  1639                                               null   \n",
       "1911    1.0   FOXP1  1911  The transcriptional network of the androgen re...   \n",
       "4944    NaN   AURKB  1623                                               null   \n",
       "\n",
       "                 Variation  Text_words  Substitutions_var  Stop_codon_var  \\\n",
       "1277  Truncating Mutations           1                  0               0   \n",
       "1639         Amplification           1                  0               0   \n",
       "1911  Truncating Mutations         863                  0               0   \n",
       "4944         Amplification           1                  0               0   \n",
       "\n",
       "      gene_fusion_var  Deletion_var    ...     exon_var  frameshift_var  \\\n",
       "1277                0             0    ...            0               0   \n",
       "1639                0             0    ...            0               0   \n",
       "1911                0             0    ...            0               0   \n",
       "4944                0             0    ...            0               0   \n",
       "\n",
       "      dup_var  overexpression_var  splice_var  hypermethylation  null  \\\n",
       "1277        0                   0           0                 0     0   \n",
       "1639        0                   0           0                 0     0   \n",
       "1911        0                   0           0                 0     0   \n",
       "4944        0                   0           0                 0     0   \n",
       "\n",
       "      hypermethylation_var  DNA_binding_var  null_var  \n",
       "1277                     0                0         0  \n",
       "1639                     0                0         0  \n",
       "1911                     0                0         0  \n",
       "4944                     0                0         0  \n",
       "\n",
       "[4 rows x 24 columns]"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nosub_text, nosub_no_text = custom_find(nosub)\n",
    "nosub_no_text # only 4 not found, of which three nulls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# find the sentences containing the nosubs\n",
    "NLTK_nosub = [sent_tokenize(nosub_text.Text[i]) for i in nosub_text.index]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "nosub_sentences = get_sentences_nosub(nosub_text[:], NLTK_nosub[:], window_left = 0, window_right = 0) # Retrieves sentences where keyword is included\n",
    "nosub_sentences_string = sentences_to_string(nosub_sentences)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\rafiz\\Anaconda3\\lib\\site-packages\\pandas\\core\\indexing.py:179: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  self._setitem_with_indexer(indexer, value)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0       Cyclin M, an orphan cyclin, is the product of ...\n",
       "7       In this study, we determined the frequency of ...\n",
       "16      Germline and somatic mutations that deregulate...\n",
       "31      Sequencing studies have identified many recurr...\n",
       "33      Studies with comparative genomic hybridization...\n",
       "41      Here, we report that deletion of the miRNA pro...\n",
       "43      Here, we report that deletion of the miRNA pro...\n",
       "48      Experimental Design  Genome-wide DNA methylati...\n",
       "70      Both, human SHQ1 and NAP57 (as an N-terminal f...\n",
       "71      Polyclonal anti-cyclin D2 and anti-cyclin D3 a...\n",
       "72      Here we report the association of three Mre11 ...\n",
       "74      The Mre11 complex specifies ssDNA endonuclease...\n",
       "75      Polyclonal anti-cyclin D2 and anti-cyclin D3 a...\n",
       "86      A Sac I (-1 195)-BamHI (+ 79) fragmnent and a ...\n",
       "87      A Sac I (-1 195)-BamHI (+ 79) fragmnent and a ...\n",
       "88      To determine whether RYBP binding to MDM2 is n...\n",
       "96      Mutations in the known high-risk genes, BRCA1,...\n",
       "100     These alterations can be deletions or mutation...\n",
       "101     Here we identify frequent TGFBR1 and TGFBR2 mu...\n",
       "107     A hypermutated subtype of advanced prostate ca...\n",
       "110     Two distinct functions have been proposed for ...\n",
       "113     Germ-line MSH6 mutations, which are rare in HN...\n",
       "114     EstablishedMSH6-null mice present afrequent oc...\n",
       "115     Kabuki syndrome (KS) is a multiple congenital ...\n",
       "116     Kabuki syndrome (KS) is a multiple congenital ...\n",
       "117     Nonsense mutations surrounding the correspondi...\n",
       "118     Screening for tumor suppressor genes in breast...\n",
       "119     Screening for tumor suppressor genes in breast...\n",
       "120     Background Many cases of familial renal cell c...\n",
       "137     Three cell lines showed overlapping homozygous...\n",
       "                              ...                        \n",
       "7399    A novel alteration in exon 1 of KRAS was detec...\n",
       "7424    We detected missense mutations in 7 of 73 ABC ...\n",
       "7476    A study was conducted to clarify the contribut...\n",
       "7555    In tumors or in drug-resistant cell lines, gen...\n",
       "7822    We conducted a long-term analysis of patients ...\n",
       "7834    Germline deletion of Sox17 led to severe fetal...\n",
       "7849     In contrast to other primary epidermal growth...\n",
       "7886    Gefitinib inhibits the growth of some cancer-d...\n",
       "7928     Abstract Activating mutations of the BRAF gen...\n",
       "7930    An unselected series of 310 colorectal carcino...\n",
       "7952    Common genetic lesions observed in PA, which a...\n",
       "7976    Gefitinib inhibits the growth of some cancer-d...\n",
       "8043    Family phylogeny suggests the duplication of a...\n",
       "8084    A mutational analysis of the tyrosine phosphat...\n",
       "8159    Examples of this include trastuzumab and lapat...\n",
       "8177    We found mutations in the ASXL1 gene in four o...\n",
       "8188     Abstract The discovery of recurrent gene fusi...\n",
       "8217    Inositol polyphosphate 4-phosphatase B (INPP4B...\n",
       "8255    Neurofibromatosis type 2 patients carry a sing...\n",
       "8305    Progression from MDS-refractory anemia with ex...\n",
       "8307     Purpose: Oncogenic fusions consisting of fibr...\n",
       "8423    Glioblastomas frequently carry mutations in th...\n",
       "8534    We provide strong evidence that c-Met is overe...\n",
       "8587    A novel mRNA in-frame fusion between exon 13 o...\n",
       "8637    Terminal deletions of chromosome 10p result in...\n",
       "8662    To estimate the prevalence of MODY3 in Norwegi...\n",
       "8747    An internal tandem duplication (ITD) of the FL...\n",
       "8758    1997), recognition motifs within the target pr...\n",
       "8772    Here we identified BCAM-AKT2 as a cancer-speci...\n",
       "8785    To analyze a multi-institutional series of typ...\n",
       "Name: Text, Length: 779, dtype: object"
      ]
     },
     "execution_count": 95,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Replace text in data_all\n",
    "data_all.Text.iloc[nosub_text.index] = nosub_sentences_string  #iloc for indexing based on integers\n",
    "data_all.Text.iloc[nosub_text.index] "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Not used\n",
    "################ The dummy variables for Gene and Text ##################\n",
    "## TODO: also use dummy for Text? There are 135 shared Genes and 142 shared Text between train and Leaks!  len(set(train.Text) & set(Leaks.Text))\n",
    "#data_all_dummy = data_all_backup[['Gene', 'Text']] # drop those columns we don't need as dummy.\n",
    "#X_dummy = pd.get_dummies(data_all_dummy) # converts categorical variables into dummy variable. From len set => 269 genes + 2090 texts\n",
    "#X_dummy_train = X_dummy[:train.shape[0]]\n",
    "#X_dummy_test = X_dummy[train.shape[0]:]\n",
    "#dummy_names = X_dummy.columns.values #### To remember names if you want to check again what Gene or Text used\n",
    "#X_dummy = X_dummy.values"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Cleaning "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [],
   "source": [
    "stop = set(stopwords.words('english')).update(['et al', 'fig', 'figure', 'acknowledgement' ])\n",
    "exclude = set('!\"#$%&\\'()*+:;<=>?@[\\\\]^_`{|}~') \n",
    "lemma = WordNetLemmatizer()\n",
    "def clean(doc,lemmatiz=False):\n",
    "    stop_free = \" \".join([i for i in doc.lower().split() if i not in stop])\n",
    "    punc_free_0 = [re.sub(\",|\\.|/\",\" \",ch) for ch in stop_free]\n",
    "    if lemmatiz:\n",
    "        punc_free_lem=\"\".join(ch for ch in punc_free_0 if ch not in exclude)\n",
    "        normalized = \" \".join(lemma.lemmatize(word) for word in punc_free_lem.split())\n",
    "        return normalized\n",
    "    else:\n",
    "        punc_free = \"\".join(ch for ch in punc_free_0 if ch not in exclude)\n",
    "        return punc_free"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {},
   "outputs": [],
   "source": [
    "#No lemmatization for the moment, be careful not to lemmatize then w2vec\n",
    "data_all.Text = [clean(doc) for doc in data_all.Text]  \n",
    "#data_all.Background = [clean(doc) for doc in data_all.Background]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
