{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import string\n",
    "import os\n",
    "from collections import Counter\n",
    "\n",
    "import nltk\n",
    "from nltk.tokenize import sent_tokenize, word_tokenize\n",
    "from segtok.segmenter import split_single, split_multi #seems to be the better choice than nltk despite lower speed\n",
    "\n",
    "from nltk.corpus import stopwords\n",
    "from nltk.stem import WordNetLemmatizer\n",
    "pd.options.display.max_colwidth = 50\n",
    "import re\n",
    "\n",
    "from sklearn.feature_extraction.text import TfidfVectorizer\n",
    "from sklearn.decomposition import TruncatedSVD"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Creating base"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "train = pd.read_csv('..//bases/new_training_variants.csv')\n",
    "test = pd.read_csv('..//bases/new_test_variants.csv')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "train_texts = pd.read_csv('..//bases/new_training_text.csv')\n",
    "test_texts = pd.read_csv('..//bases/new_test_text.csv', sep=\"\\|\\|\", engine='python', header=None, skiprows=1, names=[\"ID\",\"Text\"], encoding = \"utf-8\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "train_all = pd.merge(train, train_texts, how='left', on='ID')\n",
    "test_all = pd.merge(test, test_texts, how='left', on='ID')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Class</th>\n",
       "      <th>Gene</th>\n",
       "      <th>ID</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Text</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1.0</td>\n",
       "      <td>FAM58A</td>\n",
       "      <td>0</td>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>Cyclin-dependent kinases (CDKs) regulate a var...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>1</td>\n",
       "      <td>W802*</td>\n",
       "      <td>Abstract Background  Non-small cell lung canc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>2</td>\n",
       "      <td>Q249E</td>\n",
       "      <td>Abstract Background  Non-small cell lung canc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>3</td>\n",
       "      <td>N454D</td>\n",
       "      <td>Recent evidence has demonstrated that acquired...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>4</td>\n",
       "      <td>L399V</td>\n",
       "      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>4.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>5</td>\n",
       "      <td>V391I</td>\n",
       "      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>5.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>6</td>\n",
       "      <td>V430M</td>\n",
       "      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>1.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>7</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>CBL is a negative regulator of activated recep...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>4.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>8</td>\n",
       "      <td>Y371H</td>\n",
       "      <td>Abstract Juvenile myelomonocytic leukemia (JM...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>4.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>9</td>\n",
       "      <td>C384R</td>\n",
       "      <td>Abstract Juvenile myelomonocytic leukemia (JM...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>4.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>10</td>\n",
       "      <td>P395A</td>\n",
       "      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>4.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>11</td>\n",
       "      <td>K382E</td>\n",
       "      <td>Noonan syndrome is an autosomal dominant conge...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>4.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>12</td>\n",
       "      <td>R420Q</td>\n",
       "      <td>Noonan syndrome is an autosomal dominant conge...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>4.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>13</td>\n",
       "      <td>C381A</td>\n",
       "      <td>Noonan syndrome is an autosomal dominant conge...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>5.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>14</td>\n",
       "      <td>P428L</td>\n",
       "      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>4.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>15</td>\n",
       "      <td>D390Y</td>\n",
       "      <td>Noonan syndrome is an autosomal dominant conge...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>1.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>16</td>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>To determine if residual cylindrical refractiv...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>4.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>17</td>\n",
       "      <td>Q367P</td>\n",
       "      <td>Acquired uniparental disomy (aUPD) is a common...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>5.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>18</td>\n",
       "      <td>M374V</td>\n",
       "      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>4.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>19</td>\n",
       "      <td>Y371S</td>\n",
       "      <td>Acquired uniparental disomy (aUPD) is a common...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>6.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>20</td>\n",
       "      <td>H94Y</td>\n",
       "      <td>Abstract Background  Non-small cell lung canc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>4.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>21</td>\n",
       "      <td>C396R</td>\n",
       "      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>4.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>22</td>\n",
       "      <td>G375P</td>\n",
       "      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>4.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>23</td>\n",
       "      <td>S376F</td>\n",
       "      <td>Recent evidence has demonstrated that acquired...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>4.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>24</td>\n",
       "      <td>P417A</td>\n",
       "      <td>Recent evidence has demonstrated that acquired...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>4.0</td>\n",
       "      <td>CBL</td>\n",
       "      <td>25</td>\n",
       "      <td>H398Y</td>\n",
       "      <td>Recent evidence has demonstrated that acquired...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>4.0</td>\n",
       "      <td>SHOC2</td>\n",
       "      <td>26</td>\n",
       "      <td>S2G</td>\n",
       "      <td>Abstract N-myristoylation is a common form of...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>4.0</td>\n",
       "      <td>TERT</td>\n",
       "      <td>27</td>\n",
       "      <td>Y846C</td>\n",
       "      <td>Heterozygous mutations in the telomerase compo...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>7.0</td>\n",
       "      <td>TERT</td>\n",
       "      <td>28</td>\n",
       "      <td>C228T</td>\n",
       "      <td>Sequencing studies have identified many recurr...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>4.0</td>\n",
       "      <td>TERT</td>\n",
       "      <td>29</td>\n",
       "      <td>H412Y</td>\n",
       "      <td>Heterozygous mutations in the telomerase compo...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4645</th>\n",
       "      <td>NaN</td>\n",
       "      <td>PTEN</td>\n",
       "      <td>957</td>\n",
       "      <td>V133I</td>\n",
       "      <td>The tumor suppressor gene PTEN is frequently m...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4646</th>\n",
       "      <td>NaN</td>\n",
       "      <td>ABCC6</td>\n",
       "      <td>958</td>\n",
       "      <td>T1130M</td>\n",
       "      <td>EWS and TLS/FUS genes, which code for RNA bind...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4647</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CARM1</td>\n",
       "      <td>959</td>\n",
       "      <td>S217E</td>\n",
       "      <td>Coactivator-associated arginine methyltransfer...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4648</th>\n",
       "      <td>NaN</td>\n",
       "      <td>EIF2B5</td>\n",
       "      <td>960</td>\n",
       "      <td>T182M</td>\n",
       "      <td>Abstract  p300, which was originally cloned a...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4649</th>\n",
       "      <td>NaN</td>\n",
       "      <td>KRIT1</td>\n",
       "      <td>961</td>\n",
       "      <td>Q201E</td>\n",
       "      <td>Translocation t(12;16)(ql3;pll) is regarded as...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4650</th>\n",
       "      <td>NaN</td>\n",
       "      <td>PDGFRA</td>\n",
       "      <td>962</td>\n",
       "      <td>D842_I843delinsIM</td>\n",
       "      <td>Gastrointestinal stromal tumors (GISTs) common...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4651</th>\n",
       "      <td>NaN</td>\n",
       "      <td>SOX17</td>\n",
       "      <td>963</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>SUMMARY Fetal stem cells differ phenotypicall...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4652</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CHST3</td>\n",
       "      <td>964</td>\n",
       "      <td>F159I</td>\n",
       "      <td>Shp2 you of work difference which that of in t...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4653</th>\n",
       "      <td>NaN</td>\n",
       "      <td>FGFR1</td>\n",
       "      <td>965</td>\n",
       "      <td>V561M</td>\n",
       "      <td>Some protein kinases are known to acquire resi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4654</th>\n",
       "      <td>NaN</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>966</td>\n",
       "      <td>G13R</td>\n",
       "      <td>Colorectal cancer is one of the most common hu...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4655</th>\n",
       "      <td>NaN</td>\n",
       "      <td>TP53</td>\n",
       "      <td>967</td>\n",
       "      <td>K120R</td>\n",
       "      <td>Critical determinants of DNA recognition by p5...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4656</th>\n",
       "      <td>NaN</td>\n",
       "      <td>GLE1</td>\n",
       "      <td>968</td>\n",
       "      <td>I684T</td>\n",
       "      <td>The pathogenesis of most gastrointestinal stro...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4657</th>\n",
       "      <td>NaN</td>\n",
       "      <td>ABCC6</td>\n",
       "      <td>969</td>\n",
       "      <td>D1238H</td>\n",
       "      <td>Characterization of the exome and genome of pr...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4658</th>\n",
       "      <td>NaN</td>\n",
       "      <td>AGXT</td>\n",
       "      <td>970</td>\n",
       "      <td>G82E</td>\n",
       "      <td>Abstract  Cellular signaling mediated by Notc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4659</th>\n",
       "      <td>NaN</td>\n",
       "      <td>TP53</td>\n",
       "      <td>971</td>\n",
       "      <td>S109Y</td>\n",
       "      <td>Among the best-studied therapeutic targets in ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4660</th>\n",
       "      <td>NaN</td>\n",
       "      <td>TP53</td>\n",
       "      <td>972</td>\n",
       "      <td>R248G</td>\n",
       "      <td>The tumor suppressor gene TP53 plays a key rol...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4661</th>\n",
       "      <td>NaN</td>\n",
       "      <td>TP53</td>\n",
       "      <td>973</td>\n",
       "      <td>R280G</td>\n",
       "      <td>Among the best-studied therapeutic targets in ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4662</th>\n",
       "      <td>NaN</td>\n",
       "      <td>LRP5</td>\n",
       "      <td>974</td>\n",
       "      <td>T253I</td>\n",
       "      <td>Introduction  Glioblastoma (GBM) is the most c...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4663</th>\n",
       "      <td>NaN</td>\n",
       "      <td>SEPT9</td>\n",
       "      <td>975</td>\n",
       "      <td>S93F</td>\n",
       "      <td>Gastrointestinal stromal tumor (GIST) is the m...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4664</th>\n",
       "      <td>NaN</td>\n",
       "      <td>SCO2</td>\n",
       "      <td>976</td>\n",
       "      <td>S225F</td>\n",
       "      <td>TGFβ-related growth factors have been implicat...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4665</th>\n",
       "      <td>NaN</td>\n",
       "      <td>GNA11</td>\n",
       "      <td>977</td>\n",
       "      <td>R183C</td>\n",
       "      <td>Uveal melanoma is a neoplasm that arises from ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4666</th>\n",
       "      <td>NaN</td>\n",
       "      <td>RECQL4</td>\n",
       "      <td>978</td>\n",
       "      <td>R1021W</td>\n",
       "      <td>Glioblastomas frequently carry mutations in th...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4667</th>\n",
       "      <td>NaN</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>979</td>\n",
       "      <td>Q61P</td>\n",
       "      <td>Epidermal growth factor receptor (EGFR) has be...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4668</th>\n",
       "      <td>NaN</td>\n",
       "      <td>GNE</td>\n",
       "      <td>980</td>\n",
       "      <td>A557T</td>\n",
       "      <td>To identify a new idiopathic basal ganglia cal...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4669</th>\n",
       "      <td>NaN</td>\n",
       "      <td>ABCC6</td>\n",
       "      <td>981</td>\n",
       "      <td>G1321S</td>\n",
       "      <td>Given known et to cases BRAF cancer or grew ma...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4670</th>\n",
       "      <td>NaN</td>\n",
       "      <td>TP63</td>\n",
       "      <td>982</td>\n",
       "      <td>S580P</td>\n",
       "      <td>IκB kinase β (IKKβ) is involved in tumor devel...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4671</th>\n",
       "      <td>NaN</td>\n",
       "      <td>SCN4A</td>\n",
       "      <td>983</td>\n",
       "      <td>R672G</td>\n",
       "      <td>he identification of subtype-specific transloc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4672</th>\n",
       "      <td>NaN</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>984</td>\n",
       "      <td>N581H</td>\n",
       "      <td>Cardio-facio-cutaneous (CFC) syndrome (MIM 115...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4673</th>\n",
       "      <td>NaN</td>\n",
       "      <td>TSHR</td>\n",
       "      <td>985</td>\n",
       "      <td>S281N</td>\n",
       "      <td>Screening for tumor suppressor genes in breast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4674</th>\n",
       "      <td>NaN</td>\n",
       "      <td>ASS1</td>\n",
       "      <td>986</td>\n",
       "      <td>S180N</td>\n",
       "      <td>Histiocytic neoplasms are clonal, hematopoieti...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4675 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Class    Gene   ID             Variation  \\\n",
       "0       1.0  FAM58A    0  Truncating Mutations   \n",
       "1       2.0     CBL    1                 W802*   \n",
       "2       2.0     CBL    2                 Q249E   \n",
       "3       3.0     CBL    3                 N454D   \n",
       "4       4.0     CBL    4                 L399V   \n",
       "5       4.0     CBL    5                 V391I   \n",
       "6       5.0     CBL    6                 V430M   \n",
       "7       1.0     CBL    7              Deletion   \n",
       "8       4.0     CBL    8                 Y371H   \n",
       "9       4.0     CBL    9                 C384R   \n",
       "10      4.0     CBL   10                 P395A   \n",
       "11      4.0     CBL   11                 K382E   \n",
       "12      4.0     CBL   12                 R420Q   \n",
       "13      4.0     CBL   13                 C381A   \n",
       "14      5.0     CBL   14                 P428L   \n",
       "15      4.0     CBL   15                 D390Y   \n",
       "16      1.0     CBL   16  Truncating Mutations   \n",
       "17      4.0     CBL   17                 Q367P   \n",
       "18      5.0     CBL   18                 M374V   \n",
       "19      4.0     CBL   19                 Y371S   \n",
       "20      6.0     CBL   20                  H94Y   \n",
       "21      4.0     CBL   21                 C396R   \n",
       "22      4.0     CBL   22                 G375P   \n",
       "23      4.0     CBL   23                 S376F   \n",
       "24      4.0     CBL   24                 P417A   \n",
       "25      4.0     CBL   25                 H398Y   \n",
       "26      4.0   SHOC2   26                   S2G   \n",
       "27      4.0    TERT   27                 Y846C   \n",
       "28      7.0    TERT   28                 C228T   \n",
       "29      4.0    TERT   29                 H412Y   \n",
       "...     ...     ...  ...                   ...   \n",
       "4645    NaN    PTEN  957                 V133I   \n",
       "4646    NaN   ABCC6  958                T1130M   \n",
       "4647    NaN   CARM1  959                 S217E   \n",
       "4648    NaN  EIF2B5  960                 T182M   \n",
       "4649    NaN   KRIT1  961                 Q201E   \n",
       "4650    NaN  PDGFRA  962     D842_I843delinsIM   \n",
       "4651    NaN   SOX17  963              Deletion   \n",
       "4652    NaN   CHST3  964                 F159I   \n",
       "4653    NaN   FGFR1  965                 V561M   \n",
       "4654    NaN    KRAS  966                  G13R   \n",
       "4655    NaN    TP53  967                 K120R   \n",
       "4656    NaN    GLE1  968                 I684T   \n",
       "4657    NaN   ABCC6  969                D1238H   \n",
       "4658    NaN    AGXT  970                  G82E   \n",
       "4659    NaN    TP53  971                 S109Y   \n",
       "4660    NaN    TP53  972                 R248G   \n",
       "4661    NaN    TP53  973                 R280G   \n",
       "4662    NaN    LRP5  974                 T253I   \n",
       "4663    NaN   SEPT9  975                  S93F   \n",
       "4664    NaN    SCO2  976                 S225F   \n",
       "4665    NaN   GNA11  977                 R183C   \n",
       "4666    NaN  RECQL4  978                R1021W   \n",
       "4667    NaN    KRAS  979                  Q61P   \n",
       "4668    NaN     GNE  980                 A557T   \n",
       "4669    NaN   ABCC6  981                G1321S   \n",
       "4670    NaN    TP63  982                 S580P   \n",
       "4671    NaN   SCN4A  983                 R672G   \n",
       "4672    NaN    BRAF  984                 N581H   \n",
       "4673    NaN    TSHR  985                 S281N   \n",
       "4674    NaN    ASS1  986                 S180N   \n",
       "\n",
       "                                                   Text  \n",
       "0     Cyclin-dependent kinases (CDKs) regulate a var...  \n",
       "1      Abstract Background  Non-small cell lung canc...  \n",
       "2      Abstract Background  Non-small cell lung canc...  \n",
       "3     Recent evidence has demonstrated that acquired...  \n",
       "4     Oncogenic mutations in the monomeric Casitas B...  \n",
       "5     Oncogenic mutations in the monomeric Casitas B...  \n",
       "6     Oncogenic mutations in the monomeric Casitas B...  \n",
       "7     CBL is a negative regulator of activated recep...  \n",
       "8      Abstract Juvenile myelomonocytic leukemia (JM...  \n",
       "9      Abstract Juvenile myelomonocytic leukemia (JM...  \n",
       "10    Oncogenic mutations in the monomeric Casitas B...  \n",
       "11    Noonan syndrome is an autosomal dominant conge...  \n",
       "12    Noonan syndrome is an autosomal dominant conge...  \n",
       "13    Noonan syndrome is an autosomal dominant conge...  \n",
       "14    Oncogenic mutations in the monomeric Casitas B...  \n",
       "15    Noonan syndrome is an autosomal dominant conge...  \n",
       "16    To determine if residual cylindrical refractiv...  \n",
       "17    Acquired uniparental disomy (aUPD) is a common...  \n",
       "18    Oncogenic mutations in the monomeric Casitas B...  \n",
       "19    Acquired uniparental disomy (aUPD) is a common...  \n",
       "20     Abstract Background  Non-small cell lung canc...  \n",
       "21    Oncogenic mutations in the monomeric Casitas B...  \n",
       "22    Oncogenic mutations in the monomeric Casitas B...  \n",
       "23    Recent evidence has demonstrated that acquired...  \n",
       "24    Recent evidence has demonstrated that acquired...  \n",
       "25    Recent evidence has demonstrated that acquired...  \n",
       "26     Abstract N-myristoylation is a common form of...  \n",
       "27    Heterozygous mutations in the telomerase compo...  \n",
       "28    Sequencing studies have identified many recurr...  \n",
       "29    Heterozygous mutations in the telomerase compo...  \n",
       "...                                                 ...  \n",
       "4645  The tumor suppressor gene PTEN is frequently m...  \n",
       "4646  EWS and TLS/FUS genes, which code for RNA bind...  \n",
       "4647  Coactivator-associated arginine methyltransfer...  \n",
       "4648   Abstract  p300, which was originally cloned a...  \n",
       "4649  Translocation t(12;16)(ql3;pll) is regarded as...  \n",
       "4650  Gastrointestinal stromal tumors (GISTs) common...  \n",
       "4651   SUMMARY Fetal stem cells differ phenotypicall...  \n",
       "4652  Shp2 you of work difference which that of in t...  \n",
       "4653  Some protein kinases are known to acquire resi...  \n",
       "4654  Colorectal cancer is one of the most common hu...  \n",
       "4655  Critical determinants of DNA recognition by p5...  \n",
       "4656  The pathogenesis of most gastrointestinal stro...  \n",
       "4657  Characterization of the exome and genome of pr...  \n",
       "4658   Abstract  Cellular signaling mediated by Notc...  \n",
       "4659  Among the best-studied therapeutic targets in ...  \n",
       "4660  The tumor suppressor gene TP53 plays a key rol...  \n",
       "4661  Among the best-studied therapeutic targets in ...  \n",
       "4662  Introduction  Glioblastoma (GBM) is the most c...  \n",
       "4663  Gastrointestinal stromal tumor (GIST) is the m...  \n",
       "4664  TGFβ-related growth factors have been implicat...  \n",
       "4665  Uveal melanoma is a neoplasm that arises from ...  \n",
       "4666  Glioblastomas frequently carry mutations in th...  \n",
       "4667  Epidermal growth factor receptor (EGFR) has be...  \n",
       "4668  To identify a new idiopathic basal ganglia cal...  \n",
       "4669  Given known et to cases BRAF cancer or grew ma...  \n",
       "4670  IκB kinase β (IKKβ) is involved in tumor devel...  \n",
       "4671  he identification of subtype-specific transloc...  \n",
       "4672  Cardio-facio-cutaneous (CFC) syndrome (MIM 115...  \n",
       "4673  Screening for tumor suppressor genes in breast...  \n",
       "4674  Histiocytic neoplasms are clonal, hematopoieti...  \n",
       "\n",
       "[4675 rows x 5 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#### process the train and test set together\n",
    "data_all = pd.concat((train_all, test_all), axis=0, ignore_index=True)\n",
    "data_all = data_all[['Class', 'Gene', 'ID', 'Variation', 'Text']] # just reordering\n",
    "data_all_backup = data_all[:] ## We keep backup in case we need to use again\n",
    "data_all"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Transform Amino Acid (AA) Letter to their three-letter abbreviation in order to find them in the text when they appear\n",
    "One_to_Three_AA = {'C': 'Cys', 'D': 'Asp', 'S': 'Ser', 'Q': 'Gln', 'K': 'Lys',\n",
    "         'I': 'Ile', 'P': 'Pro', 'T': 'Thr', 'F': 'Phe', 'N': 'Asn', \n",
    "         'G': 'Gly', 'H': 'His', 'L': 'Leu', 'R': 'Arg', 'W': 'Trp', \n",
    "         'A': 'Ala', 'V': 'Val', 'E': 'Glu', 'Y': 'Tyr', 'M': 'Met'}\n",
    "pattern = re.compile('|'.join(One_to_Three_AA.keys()))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Adding extra feature(s)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# Feature for the length of the text\n",
    "data_all[\"Text_words\"] = data_all[\"Text\"].map(lambda x: len(str(x).split(\" \")))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Substitutions (subs)\n",
    "## Functions for pre-processing of subs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# find_sub return the substituions that are in text and those that are not\n",
    "def find_sub(data):    \n",
    "    Boolean = [data.Variation[i][:-1] in data.Text[i] or #case 1.\n",
    "               pattern.sub(lambda x: One_to_Three_AA[x.group()], data.Variation[i][:-1]) # case2\n",
    "               in data.Text[i]  for i in data.index] ## because new indexing we use \n",
    "    \n",
    "    sub_in_text = data[Boolean]\n",
    "    not_Boolean = [not i for i in Boolean]  \n",
    "    sub_not_in_text = data[not_Boolean]\n",
    "    \n",
    "    return sub_in_text, sub_not_in_text\n",
    "\n",
    "# find_sub_numberChange searches for other number of a substitution i.e. G12V -> G_V because sometimes mistake in entry\n",
    "# Is currently without One_to_three substitution or Variation[:-1] only the full variation\n",
    "def find_sub_numberChange(data):\n",
    "    Booleans = [] #will contain the different Booleans if found in text\n",
    "    for i in data.index:\n",
    "        split_variation = re.split('(\\d+)', data.Variation[i]) # split based on a number\n",
    "        first_Amino = re.escape(split_variation[0]) #re.escpae uses variable as regex\n",
    "        last_Amino = re.escape(split_variation[-1])\n",
    "        new_regex  = first_Amino + r\"\\d+\" + last_Amino\n",
    "        Boolean = bool(re.search(new_regex, data.Text[i]))\n",
    "        Booleans.append(Boolean)\n",
    "        \n",
    "    sub_number_in_text = data[Booleans]\n",
    "    not_Boolean = [not i for i in Booleans]  \n",
    "    sub_number_no_text = data[not_Boolean]\n",
    "    \n",
    "    return sub_number_in_text, sub_number_no_text\n",
    "\n",
    "# for substitutions that are still not found, use other keywords\n",
    "def find_sub_pattern(data, pattern):    \n",
    "    Boolean = [pattern in data.Text[i] for i in data.index] ## because new indexing we use \n",
    "    \n",
    "    sub_in_text = data[Boolean]\n",
    "    not_Boolean = [not i for i in Boolean]  \n",
    "    sub_not_in_text = data[not_Boolean]\n",
    "    \n",
    "    return sub_in_text, sub_not_in_text\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "##### get_sentences_sub use a window to extract sentences where the subs appear. \n",
    "# If window_left & window_right = 0 => just taking the sentences with subs\n",
    "\n",
    "def get_sentences_sub(data, splitted_sentences, window_left, window_right):\n",
    "    data.index = range(len(data)) #makes sure that index is right\n",
    "    sentences_with_sub = [[] for _ in range(len(data))]\n",
    "    for i in range(len(splitted_sentences)):\n",
    "        sentences = splitted_sentences[i]\n",
    "        one_to_three_variation = pattern.sub(lambda x: One_to_Three_AA[x.group()], data.Variation[i][:-1])\n",
    "        Variation = data.Variation[i][:-1]        \n",
    "        for j in range(len(sentences)):                              \n",
    "            if (Variation in sentences[j]) or (one_to_three_variation in sentences[j]):\n",
    "                new_regex = r\"[\\S]*\" + re.escape(Variation) + r\"[\\S]*\" ###  r\"[\\S]*\" because we look for Variation[:-1] not just Variation\n",
    "                sentences[j] = re.sub(new_regex, 'placeholderMutation', sentences[j]) #case 1\n",
    "                ### We add the space to ' placeholderMutation' because sometimes there are letters in front of it\n",
    "                new_regex = re.escape(one_to_three_variation) + r\"[\\S]*\"\n",
    "                sentences[j] = re.sub(new_regex, ' placeholderMutation', sentences[j]) #case 2\n",
    "                sentences_with_sub[i].extend(sentences[max(j-window_left,0) : min(j+1+window_right, len(sentences)-1)])               \n",
    "    \n",
    "    return sentences_with_sub\n",
    "\n",
    "##### get_sentences_sub_number use a window to extract sentences where the subs appear that have different number i.e. G12V -> G_V\n",
    "\n",
    "def get_sentences_sub_number(data, splitted_sentences, window_left, window_right):\n",
    "    #position_sentences = [[] for _ in range(len(data))]  #### currently not used\n",
    "    data.index = range(len(data))\n",
    "    sentences_with_sub_number = [[] for _ in range(len(data))]\n",
    "    for i in range(len(splitted_sentences)):\n",
    "        sentences = splitted_sentences[i] \n",
    "        for j in range(len(sentences)):\n",
    "            split_variation = re.split('(\\d+)', data.Variation[i]) # split based on a number\n",
    "            first_Amino = re.escape(split_variation[0]) #re.escpae uses variable as regex\n",
    "            last_Amino = re.escape(split_variation[-1])\n",
    "            new_regex  = first_Amino + r\"\\d+\" + last_Amino\n",
    "            Boolean = bool(re.search(new_regex, sentences[j]))            \n",
    "            if Boolean:\n",
    "                sentences[j] = re.sub(new_regex, ' placeholderMutation', sentences[j]) # Again replacing the sentence with placeholder\n",
    "                sentences_with_sub_number[i].extend(sentences[max(j-window_left,0) : min(j+1+window_right,len(sentences)-1)])\n",
    "    \n",
    "    return sentences_with_sub_number\n",
    "\n",
    "# for substitutions that are still not found, use other keywords\n",
    "\n",
    "def get_sentences_pattern(data, splitted_sentences, pattern, window_left, window_right):\n",
    "    data.index = range(len(data)) #makes sure that index is right\n",
    "    sentences_with_sub = [[] for _ in range(len(data))]\n",
    "    for i in range(len(splitted_sentences)):\n",
    "        sentences = splitted_sentences[i] \n",
    "        for j in range(len(sentences)):                              \n",
    "            if (pattern in sentences[j]):\n",
    "                sentences_with_sub[i].extend(sentences[max(j-window_left,0) : min(j+1+window_right, len(sentences)-1)])               \n",
    "    \n",
    "    return sentences_with_sub"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "###### LENGTH of sentences <= 10 : we run split again with segtok instead of NLTK\n",
    "###### LENGTH < 5: window of 2,2   5 <= LENGTH <= 10: window of 1,1 \n",
    "\n",
    "def get_window_sub(data, splitted_sentences):\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#### Converts list of sentences into one string of sentences for each document => to use for tfidf etc.\n",
    "def sentences_to_string(sentences_list):\n",
    "    sentence_strings = []\n",
    "    for sentences in sentences_list:\n",
    "        sentence_string =  ' '.join(str(sentence) for sentence in sentences)\n",
    "        sentence_strings.append(sentence_string)\n",
    "    \n",
    "    return sentence_strings "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Subs processing of the data set "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Length of total subs: 3820\n"
     ]
    }
   ],
   "source": [
    "######### First find those that have the format of being a substitution in data\n",
    "data_all['Substitutions_var'] = data_all.Variation.apply(lambda x: bool(re.search('^[A-Z]\\\\d+[A-Z*]$', x))*1) #multiplying by 1 converts True to 1, False to 0 => Maybe modify this later?\n",
    "data_all['Stop_codon_var'] = data_all.Variation.apply(lambda x: bool(re.search('[*]', x))*1) #multiplying by 1 converts True to 1, False to 0\n",
    "data_sub = data_all[data_all['Substitutions_var']==1] ### Now we know the index of where a substitution occurs - the data_sub\n",
    "print(\"Length of total subs: %i\" %len(data_sub)) # other ways to process it like finding the word 'mutation'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Subs inside the text"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "## First consider the subs that appear in text\n",
    "sub_text, sub_no_text = find_sub(data_sub) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "## use tokenizer to split into sentences of all the subs in text \n",
    "NLTK_sub = [sent_tokenize(sub_text.Text[i]) for i in sub_text.index] "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 396,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# extract window for the sub sentences where they appear\n",
    "# !! Use [:] because it makes a copy and doesn't change anything to the original indexes\n",
    "sub_sentences = get_sentences_sub(sub_text[:], NLTK_sub[:], window_left = 0, window_right = 0) # choosing for window 0 as default now\n",
    "sub_sentences_string = sentences_to_string(sub_sentences)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 345,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\rafiz\\Anaconda3\\lib\\site-packages\\pandas\\core\\indexing.py:179: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  self._setitem_with_indexer(indexer, value)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "1       Using select c-CBL somatic mutations such as S...\n",
       "2       Using select c-CBL somatic mutations such as S...\n",
       "3       Most of the changes were novel, although 4 cas...\n",
       "4       Finally, the third group constituted mutations...\n",
       "5       Finally, the third group constituted mutations...\n",
       "6       The second group of mutants (M374V, placeholde...\n",
       "8       We investigated the mechanism by which placeho...\n",
       "9       We investigated the mechanism by which CBL-Y37...\n",
       "10      Finally, the third group constituted mutations...\n",
       "11      Purified PCR amplicons (CBLWT, CBLC381A, place...\n",
       "12      Purified PCR amplicons (CBLWT, CBLC381A, CBLK3...\n",
       "13      Purified PCR amplicons (CBLWT, placeholderMuta...\n",
       "14      The second group of mutants (M374V, V430M, pla...\n",
       "15      Purified PCR amplicons (CBLWT, CBLC381A, CBLK3...\n",
       "17      When introduced into Lin- Sca1+ c-Kit+ (LSK) H...\n",
       "18      The second group of mutants placeholderMutatio...\n",
       "19      When introduced into Lin- Sca1+ c-Kit+ (LSK) H...\n",
       "20      Using select c-CBL somatic mutations such as p...\n",
       "21      The first group of damaging mutants included p...\n",
       "22      Finally, the third group constituted mutations...\n",
       "23      Transforming activities of CBL mutants  We det...\n",
       "24      Case UPN 3 had a C>G missense change in exon 9...\n",
       "25      Transforming activities of CBL mutants  We det...\n",
       "26      Twenty-five subjects with a relatively consist...\n",
       "27      The proband in the second family has mutant al...\n",
       "28      Each of these promoter mutations resulted in a...\n",
       "29      In one case the proband is homozygous for a no...\n",
       "30      The proband in the second family has mutant al...\n",
       "32      In one case the proband is homozygous for a no...\n",
       "34      Each of these promoter mutations resulted in a...\n",
       "                              ...                        \n",
       "4641    Several recurrent mutations were found, includ...\n",
       "4642    Yeast displays were cells with are was residue...\n",
       "4643    2A), that would result in a  placeholderMutati...\n",
       "4644    These residues are conserved in mammalian PDGF...\n",
       "4645    PTEN cDNA with missense mutations (S10N, Y16C,...\n",
       "4646    The however the co-workers not proteins cells ...\n",
       "4647    In this study, we report that CARM1 function i...\n",
       "4648    This the variant issue confirmed placeholderMu...\n",
       "4649    After the no cloning variants mutant dye in 12...\n",
       "4652    Shp2 you of work difference which that of in t...\n",
       "4653    The cellular wild-type FGFR1 activity was pote...\n",
       "4654    Interestingly, none of the 3 of 43 cancer cell...\n",
       "4655    Mutants R283Q, placeholderMutation and both C2...\n",
       "4656    For mTORC1 placeholderMutation 100 information...\n",
       "4657    Modulation of were the out were 19 has cancer ...\n",
       "4658    TGF-β analysis evolutionary homozygous T-cell–...\n",
       "4660    Frequent substitutions at ‘‘hotspot’’ codons i...\n",
       "4662    Thus blood methods was highlights arise to car...\n",
       "4663    The were EMBL human DNMT3A McCoy dose placehol...\n",
       "4664    Procedures placeholderMutation results placeho...\n",
       "4665    and Table 2TABLE 2 Frequency of Mutations in p...\n",
       "4666    Top placeholderMutation crystallography-based ...\n",
       "4667    Interestingly, none of the 3 of 43 cancer cell...\n",
       "4668    In 18 anti-androgen by Ras p53N239S dimers by ...\n",
       "4669    Given known et to cases BRAF cancer or grew ma...\n",
       "4670    In we panel further RB-SWI through gammaR1 eac...\n",
       "4671    SEs mutations maintains of of BACH1 in SSA QIA...\n",
       "4672    BRAF\\t6\\tc.721A→C\\tp.T241P\\t+\\t2\\t1\\t− BRAF\\t6...\n",
       "4673    Recent 1977 lipid for placeholderMutation neph...\n",
       "4674    PIPKIIα antibodies SU11274 15 placeholderMutat...\n",
       "Name: Text, Length: 3294, dtype: object"
      ]
     },
     "execution_count": 345,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Replace text in data_all\n",
    "data_all.Text.iloc[sub_text.index] = sub_sentences_string #iloc for indexing based on integers\n",
    "data_all.Text.iloc[sub_text.index]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'sub_sentences_string' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-16-da2e2da9a1c3>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m()\u001b[0m\n\u001b[0;32m      1\u001b[0m \u001b[1;31m#### Now use a window for the smaller size sentences (with LENGTH <= 10 sentences)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m----> 2\u001b[1;33m \u001b[0mlen\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0msub_sentences_string\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[1;31mNameError\u001b[0m: name 'sub_sentences_string' is not defined"
     ]
    }
   ],
   "source": [
    "#### Now use a window for the smaller size sentences (with LENGTH <= 10 sentences)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Subs with a different number inside the text\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 346,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# now the subs that don't appear in text: one reason is different number in the substitution. f.e. G12V -> G13V \n",
    "sub_number_text, sub_number_no_text = find_sub_numberChange(sub_no_text) # 108 such cases out of 411 = nice improvement"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 347,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "## use tokenizer to split into sentences of subs that have different number in text \n",
    "NLTK_sub_number = [sent_tokenize(sub_number_text.Text[i]) for i in sub_number_text.index]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 348,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# extract window for the sub sentences where they appear\n",
    "# !! Use [:] because it makes a copy and doesn't change anything to the original indexes\n",
    "sub_number_sentences = get_sentences_sub_number(sub_number_text[:], NLTK_sub_number[:], window_left = 0, window_right = 0)\n",
    "sub_number_string = sentences_to_string(sub_number_sentences)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 349,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\rafiz\\Anaconda3\\lib\\site-packages\\pandas\\core\\indexing.py:179: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  self._setitem_with_indexer(indexer, value)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "44      His-tagged versions of the catalytic region of...\n",
       "49      His-tagged versions of the catalytic region of...\n",
       "61      His-tagged versions of the catalytic region of...\n",
       "63      His-tagged versions of the catalytic region of...\n",
       "66      His-tagged versions of the catalytic region of...\n",
       "73      While both were missense mutations, in silico ...\n",
       "208      placeholderMutation and T847I were also found...\n",
       "213     Complex mutation  placeholderMutation and dele...\n",
       "267     As a positive control, we introduced the well-...\n",
       "275     And in the series of Pallis and colleagues (12...\n",
       "306     Recurrent mutations in H3F3A, which encodes th...\n",
       "307     Histone H3-K42me2 may regulate transcription  ...\n",
       "308     Recurrent mutations in H3F3A, which encodes th...\n",
       "521     Both combinations contained mutations of codon...\n",
       "607     In the present study, we have focused on two d...\n",
       "632     Such mutant amino acid specificity was also ap...\n",
       "730     In contrast, ectopic expression of FGFR4  plac...\n",
       "1081    Recently, somatic mutations of Tyr641 (Y641F, ...\n",
       "1083    Recently, somatic mutations of Tyr641 (Y641F, ...\n",
       "1085    Recently, somatic mutations of Tyr641 (Y641F, ...\n",
       "1091    Seven of 12 mutations in RXRA (retinoid × nucl...\n",
       "1092    Seven of 12 mutations in RXRA (retinoid × nucl...\n",
       "1121    Three of the MET mutated variants, Y1248H, L12...\n",
       "1128    Two Met variants tested, M1268T and H1112Y, we...\n",
       "1131    Two Met variants tested, M1268T and  placehold...\n",
       "1136    Here we characterized the response of the MET ...\n",
       "1139    We describe a case of clinical activity of cri...\n",
       "1141    The  placeholderMutation MET mutation segregat...\n",
       "1142    The locations of the RET mutations, E768D, L79...\n",
       "1144    MET  placeholderMutation A new mutation in cod...\n",
       "                              ...                        \n",
       "3288    It was reported that substitutions of arginine...\n",
       "3292    In addition, two codon 609 mutations ( placeho...\n",
       "3432    Here we characterized the response of the MET ...\n",
       "3466    However, forced expression of stable MYC ( pla...\n",
       "3487    NRAS (G12A, G12D, Q61K, Q61R, Q61L), BRAF (V60...\n",
       "3566    The two mutations that did not generate NTM*, ...\n",
       "3661    Point mutations (M536I, M538I, I548V,  placeho...\n",
       "3703    The limit of detection was determined by mixin...\n",
       "3707    Mutations at these residues are also found in ...\n",
       "3714    Here we characterized the response of the MET ...\n",
       "3731    Among KRAS G12 mutations, the abundance of  pl...\n",
       "3735    Age/ sex TNM stage Primary tumor (site) Associ...\n",
       "3772    The two mutations that did not generate NTM*, ...\n",
       "3820    Gastric cancers, on the other hand, had the fe...\n",
       "3875     placeholderMutation, P780-Y781insertionGSP (a...\n",
       "3879    However, forced expression of stable MYC ( pla...\n",
       "3896    ERBB2 followed a similar pattern, where extrac...\n",
       "3966    Notably, patient number 5, with a  placeholder...\n",
       "4036    Point mutations (M536I, M538I, I548V,  placeho...\n",
       "4089    This mutation resulted in a glycine-to-cystein...\n",
       "4119    Finally, the position of an acidic substitutio...\n",
       "4153    ERBB2 followed a similar pattern, where extrac...\n",
       "4198    Although mutations of R140 in IDH2 exclusively...\n",
       "4207    We also identified a novel RAC1  placeholderMu...\n",
       "4290    Whereas activating KRAS  placeholderMutation m...\n",
       "4398    Such mutant amino acid specificity was also ap...\n",
       "4431    NRAS (G12A, G12D, Q61K, Q61R, Q61L), BRAF (V60...\n",
       "4520    Additional reports of extracellular domain mut...\n",
       "4540    Its various functions include the conversion o...\n",
       "4616    Moreover mutations L755S, L755P,  placeholderM...\n",
       "Name: Text, Length: 131, dtype: object"
      ]
     },
     "execution_count": 349,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_all.Text.iloc[sub_number_text.index] = sub_number_string #iloc for indexing based on integers\n",
    "data_all.Text.iloc[sub_number_text.index]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Subs that are still not found"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 350,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# subs that are still not found, look based on a pattern like the word 'mutat' in text\n",
    "sub_pattern_text, sub_pattern_no_text = find_sub_pattern(sub_number_no_text, 'mutat') # 108 such cases out of 411 = nice improvement"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 352,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Class</th>\n",
       "      <th>Gene</th>\n",
       "      <th>ID</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Text</th>\n",
       "      <th>Substitutions_var</th>\n",
       "      <th>Stop_codon_var</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>140</th>\n",
       "      <td>5.0</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>140</td>\n",
       "      <td>I491M</td>\n",
       "      <td>The accurate determination of perfluoroalkyl s...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>145</th>\n",
       "      <td>2.0</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>145</td>\n",
       "      <td>K467T</td>\n",
       "      <td>The accurate determination of perfluoroalkyl s...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>259</th>\n",
       "      <td>2.0</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>259</td>\n",
       "      <td>S464L</td>\n",
       "      <td>The accurate determination of perfluoroalkyl s...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>416</th>\n",
       "      <td>4.0</td>\n",
       "      <td>TP53</td>\n",
       "      <td>416</td>\n",
       "      <td>P151S</td>\n",
       "      <td>The effects of chlorpromazine on various prope...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1109</th>\n",
       "      <td>1.0</td>\n",
       "      <td>FANCA</td>\n",
       "      <td>1109</td>\n",
       "      <td>S1088F</td>\n",
       "      <td>null</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1407</th>\n",
       "      <td>6.0</td>\n",
       "      <td>FGFR3</td>\n",
       "      <td>1407</td>\n",
       "      <td>K508M</td>\n",
       "      <td>null</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1613</th>\n",
       "      <td>4.0</td>\n",
       "      <td>VHL</td>\n",
       "      <td>1613</td>\n",
       "      <td>L158Q</td>\n",
       "      <td>The case of a 40-year-old woman with severe ed...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1936</th>\n",
       "      <td>7.0</td>\n",
       "      <td>CARD11</td>\n",
       "      <td>1936</td>\n",
       "      <td>G116S</td>\n",
       "      <td>Left ventricular (LV) remodeling is a signific...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1942</th>\n",
       "      <td>7.0</td>\n",
       "      <td>CARD11</td>\n",
       "      <td>1942</td>\n",
       "      <td>E127G</td>\n",
       "      <td>Left ventricular (LV) remodeling is a signific...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2302</th>\n",
       "      <td>7.0</td>\n",
       "      <td>JAK1</td>\n",
       "      <td>2302</td>\n",
       "      <td>R724H</td>\n",
       "      <td>Regulatory T (T reg) cells have a major role i...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2309</th>\n",
       "      <td>7.0</td>\n",
       "      <td>JAK1</td>\n",
       "      <td>2309</td>\n",
       "      <td>R879C</td>\n",
       "      <td>Regulatory T (T reg) cells have a major role i...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2504</th>\n",
       "      <td>6.0</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>2504</td>\n",
       "      <td>V11A</td>\n",
       "      <td>Identification of protein-protein interaction...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2755</th>\n",
       "      <td>7.0</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>2755</td>\n",
       "      <td>G596C</td>\n",
       "      <td>null</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2980</th>\n",
       "      <td>2.0</td>\n",
       "      <td>KIT</td>\n",
       "      <td>2980</td>\n",
       "      <td>N655K</td>\n",
       "      <td>TO THE EDITOR: I commend Drs. Freeman and Bank...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Class    Gene    ID Variation  \\\n",
       "140     5.0    EGFR   140     I491M   \n",
       "145     2.0    EGFR   145     K467T   \n",
       "259     2.0    EGFR   259     S464L   \n",
       "416     4.0    TP53   416     P151S   \n",
       "1109    1.0   FANCA  1109    S1088F   \n",
       "1407    6.0   FGFR3  1407     K508M   \n",
       "1613    4.0     VHL  1613     L158Q   \n",
       "1936    7.0  CARD11  1936     G116S   \n",
       "1942    7.0  CARD11  1942     E127G   \n",
       "2302    7.0    JAK1  2302     R724H   \n",
       "2309    7.0    JAK1  2309     R879C   \n",
       "2504    6.0   BRCA1  2504      V11A   \n",
       "2755    7.0    BRAF  2755     G596C   \n",
       "2980    2.0     KIT  2980     N655K   \n",
       "\n",
       "                                                   Text  Substitutions_var  \\\n",
       "140   The accurate determination of perfluoroalkyl s...                  1   \n",
       "145   The accurate determination of perfluoroalkyl s...                  1   \n",
       "259   The accurate determination of perfluoroalkyl s...                  1   \n",
       "416   The effects of chlorpromazine on various prope...                  1   \n",
       "1109                                               null                  1   \n",
       "1407                                               null                  1   \n",
       "1613  The case of a 40-year-old woman with severe ed...                  1   \n",
       "1936  Left ventricular (LV) remodeling is a signific...                  1   \n",
       "1942  Left ventricular (LV) remodeling is a signific...                  1   \n",
       "2302  Regulatory T (T reg) cells have a major role i...                  1   \n",
       "2309  Regulatory T (T reg) cells have a major role i...                  1   \n",
       "2504   Identification of protein-protein interaction...                  1   \n",
       "2755                                               null                  1   \n",
       "2980  TO THE EDITOR: I commend Drs. Freeman and Bank...                  1   \n",
       "\n",
       "      Stop_codon_var  \n",
       "140                0  \n",
       "145                0  \n",
       "259                0  \n",
       "416                0  \n",
       "1109               0  \n",
       "1407               0  \n",
       "1613               0  \n",
       "1936               0  \n",
       "1942               0  \n",
       "2302               0  \n",
       "2309               0  \n",
       "2504               0  \n",
       "2755               0  \n",
       "2980               0  "
      ]
     },
     "execution_count": 352,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sub_pattern_no_text # only 14 texts that still don't include sentences with the word 'mutat', of which three nulls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 353,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "## use tokenizer to split into sentences of subs that have keyword 'mutat'\n",
    "NLTK_sub_pattern = [sent_tokenize(sub_pattern_text.Text[i]) for i in sub_pattern_text.index]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 354,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# !! Use [:] because it makes a copy and doesn't change anything to the original indexes\n",
    "sub_pattern_sentences = get_sentences_pattern(sub_pattern_text[:], NLTK_sub_pattern[:], 'mutat', window_left = 0, window_right = 0)\n",
    "sub_pattern_string = sentences_to_string(sub_pattern_sentences)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 355,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\rafiz\\Anaconda3\\lib\\site-packages\\pandas\\core\\indexing.py:179: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  self._setitem_with_indexer(indexer, value)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "46      Mutations in CBL, encoding an E3 ubiquitin lig...\n",
       "80      This D1 domain contains only two serine residu...\n",
       "102     Identification of a high-risk disease-causing ...\n",
       "104     In this study, we profiled 70 colorectal cance...\n",
       "106     Identification of a high-risk disease-causing ...\n",
       "108     Most Lynch Syndrome-associated CRCs are report...\n",
       "111     Identification of a high-risk disease-causing ...\n",
       "112     Germ-line MSH6 mutations, which are rare in HN...\n",
       "136     In 2011, whole-exome sequencing studies showed...\n",
       "151     We investigated the association between mutati...\n",
       "155     Clinical data and epidermal growth factor rece...\n",
       "161     Clinical data and epidermal growth factor rece...\n",
       "168     Currently available methods of EGFR mutation d...\n",
       "178     Those were analyzed for EGFR and k-ras (Kirste...\n",
       "182     Purpose: Clinical reports about responsiveness...\n",
       "185     ‘Classical' mutations in the EGFR tyrosine kin...\n",
       "215     Purpose: Clinical features of epidermal growth...\n",
       "222     Findings in the last decade showed that mutati...\n",
       "249     ‘Classical' mutations in the EGFR tyrosine kin...\n",
       "344     Direct evidence of E-cadherin mutations trigge...\n",
       "346     Direct evidence of E-cadherin mutations trigge...\n",
       "348     Direct evidence of E-cadherin mutations trigge...\n",
       "371     Inheritance of germ-line mutant alleles of BRC...\n",
       "380     The tumour suppressor gene p53 is frequently m...\n",
       "384     Inheritance of germ-line mutant alleles of BRC...\n",
       "386     Although in the majority of cases only one of ...\n",
       "399     Inheritance of germ-line mutant alleles of BRC...\n",
       "430     Roughly half of all human tumours contain p53 ...\n",
       "441     Here, we report on the global gene expression ...\n",
       "464     In many cases, p53 gene mutations result in hi...\n",
       "                              ...                        \n",
       "4408    These include LOH3 of p53 (2â€”4),Rb (5), APCI...\n",
       "4409    Germline mutation in serine/threonine kinase 1...\n",
       "4440    The excellent agreement between the structure/...\n",
       "4448    The point mutation in the N-terminal part of t...\n",
       "4458    This ‘double hit’ w x is often accomplished by...\n",
       "4461    Inheritance of germ-line mutant alleles of BRC...\n",
       "4463    The excellent agreement between the structure/...\n",
       "4464    Among the best-studied therapeutic targets in ...\n",
       "4485    Among the best-studied therapeutic targets in ...\n",
       "4486    Among the best-studied therapeutic targets in ...\n",
       "4488    We enrolled patients with any cancer diagnosis...\n",
       "4489    Among the best-studied therapeutic targets in ...\n",
       "4504    Bioinformatics annotation, filtering of varian...\n",
       "4511    Indeed, in the early nineties, gain-of-functio...\n",
       "4514    Recently, mutations in the B-Raf gene have bee...\n",
       "4527    The systematic characterization of somatic mut...\n",
       "4539    In humans, MLH1 mutations underlie half of her...\n",
       "4549    Among the best-studied therapeutic targets in ...\n",
       "4550    Among the best-studied therapeutic targets in ...\n",
       "4566    We enrolled patients with any cancer diagnosis...\n",
       "4575     Abstract  The BRCA1 gene from individuals at ...\n",
       "4577    Among the best-studied therapeutic targets in ...\n",
       "4584    ras is mutated in approximately 20% of all hum...\n",
       "4590    This class included truncating mutations, with...\n",
       "4591    The excellent agreement between the structure/...\n",
       "4606    Deletion or mutation of these CDK-inhibitors c...\n",
       "4608    Sequence analysis indicated that 8 of 24 (33%)...\n",
       "4634    It has recently been reported that somatic mut...\n",
       "4659    Among the best-studied therapeutic targets in ...\n",
       "4661    Among the best-studied therapeutic targets in ...\n",
       "Name: Text, Length: 381, dtype: object"
      ]
     },
     "execution_count": 355,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_all.Text.iloc[sub_pattern_text.index] = sub_pattern_string #iloc for indexing based on integers\n",
    "data_all.Text.iloc[sub_pattern_text.index]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Non-substitutions (no subs)\n",
    "## Functions for pre-processing of no_subs like fusions, deletions, truncations, etc."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 358,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "## Gets different non-subs variation types by using regex\n",
    "def variation_regex(data, pattern): \n",
    "    Boolean = [bool(re.search(pattern, i, re.IGNORECASE)) for i in data.Variation]\n",
    "    data_regex = data[Boolean] \n",
    "    not_Boolean = [not i for i in Boolean]  \n",
    "    data_no_regex = data[not_Boolean]\n",
    "    \n",
    "    return (data_regex, data_no_regex)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 359,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def custom_find(data):\n",
    "    l=[\"mutat\",\"variat\",\"fusion\",\"deleti\",\"amplific\",\"framesh\",\n",
    "    \"truncat\",\"fs\",\"exon\",\"dupl\"]\n",
    "    Boolean={}\n",
    "    for word in l:\n",
    "        Boolean[word] = [re.search(word,data.Text[i]) for i in data.index] ## because new indexing we use \n",
    "    Bool=[any(tup) for tup in zip(Boolean[\"mutat\"],Boolean[\"variat\"],Boolean[\"fusion\"],\n",
    "        Boolean[\"deleti\"],Boolean[\"amplific\"],Boolean[\"truncat\"],\n",
    "        Boolean[\"fs\"],Boolean[\"framesh\"],Boolean[\"exon\"],Boolean[\"dupl\"])]\n",
    "    no_sub_in_text=data[Bool]\n",
    "    no_Bool=[not i for i in Bool]\n",
    "    no_sub_not_in_text=data[no_Bool]\n",
    "    return no_sub_in_text,no_sub_not_in_text\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 360,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def get_sentences_pattern(data, splitted_sentences, pattern, window_left, window_right):\n",
    "    data.index = range(len(data)) #makes sure that index is right\n",
    "    sentences_with_sub = [[] for _ in range(len(data))]\n",
    "    for i in range(len(splitted_sentences)):\n",
    "        sentences = splitted_sentences[i] \n",
    "        for j in range(len(sentences)):                              \n",
    "            if (pattern in sentences[j]):\n",
    "                sentences_with_sub[i].extend(sentences[max(j-window_left,0) : min(j+1+window_right, len(sentences)-1)])               \n",
    "    \n",
    "    return sentences_with_sub"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 361,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def get_sentences_nosub(data, splitted_sentences, window_left, window_right):\n",
    "    #position_sentences = [[] for _ in range(len(data))]  #### currently not used\n",
    "    data.index = range(len(data))\n",
    "    sentences_with_sub = [[] for _ in range(len(data))]\n",
    "\n",
    "    for i in range(len(splitted_sentences)):\n",
    "        sentences = splitted_sentences[i]\n",
    "        l=[\"mutat\",\"variat\",\"fusion\",\"deleti\",\"amplific\",\"framesh\",\n",
    "        \"truncat\",\"fs\",\"exon\",\"dupl\"]     \n",
    "        for j in range(len(sentences)):    \n",
    "            for keyword in l:\n",
    "                if keyword in sentences[j]:\n",
    "                    sentences_with_sub[i].extend(sentences[max(j-window_left,0) : min(j+1+window_right, len(sentences)-1)])\n",
    "    return sentences_with_sub"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## No subs processing of the data set "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 362,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\rafiz\\Anaconda3\\lib\\site-packages\\pandas\\core\\indexing.py:179: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  self._setitem_with_indexer(indexer, value)\n"
     ]
    }
   ],
   "source": [
    "#### First find those that have the format of being a non-substitutions in the data\n",
    "\n",
    "# Initialize some of the variables already (because we splice them, could be filled with NA's otherwise)\n",
    "# Represents 783 from train and test set\n",
    "data_all['gene_fusion_var'] = 0\n",
    "data_all['Deletion_var'] = 0\n",
    "data_all['del_or_ins_var'] = 0\n",
    "\n",
    "####### Fusions : 'Fusions' ############\n",
    "data_all['Fusion_var'] = data_all.Variation.apply(lambda x: bool(re.search('^fusion', x, re.IGNORECASE))*1) #multiplying by 1 converts True to 1, False to 0\n",
    "_ , new_data_all = variation_regex(data_all, '^fusion')  #37 cases\n",
    "\n",
    "###### Fusions: 'Gene-Gene fusion' ########\n",
    "data_all['gene_fusion_var'].loc[new_data_all.index] = new_data_all.Variation.apply(lambda x: bool(re.search('fusion', x, re.IGNORECASE))*1)\n",
    "_ , new_data_all = variation_regex(new_data_all, 'fusion') #160 cases\n",
    "\n",
    "####### Deletions: 'Deletions' ############\n",
    "data_all['Deletion_var'].loc[new_data_all.index] = new_data_all.Variation.apply(lambda x: bool(re.search('^del', x, re.IGNORECASE))*1)\n",
    "_, new_data_all = variation_regex(new_data_all, '^del') # 88 cases\n",
    "\n",
    "####### Deletions & Insertions wheteher together or seperately (doesn't make a big difference IMO)\n",
    "data_all['del_or_ins_var'].loc[new_data_all.index] = new_data_all.Variation.apply(lambda x: bool(re.search('del|ins', x, re.IGNORECASE))*1)\n",
    "# 196 cases\n",
    "\n",
    "###### Amplifications #########\n",
    "data_all['Amplification_var'] = data_all.Variation.apply(lambda x: bool(re.search('ampl', x, re.IGNORECASE))*1) # 79 cases\n",
    "\n",
    "###### Truncations ########### Don't forget there are 'Truncating mutations'  '_trunc' \n",
    "data_all['Truncation_var'] = data_all.Variation.apply(lambda x: bool(re.search('trunc', x, re.IGNORECASE))*1) # 118 cases\n",
    "\n",
    "####### Exons #########\n",
    "data_all['exon_var'] = data_all.Variation.apply(lambda x: bool(re.search('exon', x, re.IGNORECASE))*1) \n",
    "\n",
    "####### Frameshift mutations ########\n",
    "data_all['frameshift_var'] = data_all.Variation.apply(lambda x: bool(re.search('fs', x, re.IGNORECASE))*1) # 22 cases\n",
    "\n",
    "####### Duplications ##############\n",
    "data_all['dup_var'] = data_all.Variation.apply(lambda x: bool(re.search('dup', x, re.IGNORECASE))*1)\n",
    "\n",
    "####### Hypermethylation ##########\n",
    "data_all['overexpression_var'] = data_all.Variation.apply(lambda x: bool(re.search('Overexpression', x, re.IGNORECASE))*1) \n",
    "\n",
    "####### Splice ##########\n",
    "data_all['splice_var'] = data_all.Variation.apply(lambda x: bool(re.search('splice', x, re.IGNORECASE))*1) \n",
    "\n",
    "####### Hypermethylation ##########\n",
    "data_all['hypermethylation_var'] = data_all.Variation.apply(lambda x: bool(re.search('hypermethylation', x, re.IGNORECASE))*1)\n",
    "\n",
    "####### DNA binding ##########\n",
    "data_all['DNA_binding_var'] = data_all.Variation.apply(lambda x: bool(re.search('DNA', x, re.IGNORECASE))*1)\n",
    "\n",
    "####### Null ##########\n",
    "data_all['null_var'] = data_all.Variation.apply(lambda x: bool(re.search('null', x, re.IGNORECASE))*1)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 363,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Class</th>\n",
       "      <th>Gene</th>\n",
       "      <th>ID</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Text</th>\n",
       "      <th>Substitutions_var</th>\n",
       "      <th>Stop_codon_var</th>\n",
       "      <th>gene_fusion_var</th>\n",
       "      <th>Deletion_var</th>\n",
       "      <th>del_or_ins_var</th>\n",
       "      <th>...</th>\n",
       "      <th>Amplification_var</th>\n",
       "      <th>Truncation_var</th>\n",
       "      <th>exon_var</th>\n",
       "      <th>frameshift_var</th>\n",
       "      <th>dup_var</th>\n",
       "      <th>overexpression_var</th>\n",
       "      <th>splice_var</th>\n",
       "      <th>hypermethylation_var</th>\n",
       "      <th>DNA_binding_var</th>\n",
       "      <th>null_var</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>7.0</td>\n",
       "      <td>TERT</td>\n",
       "      <td>31</td>\n",
       "      <td>Promoter Mutations</td>\n",
       "      <td>Sequencing studies have identified many recurr...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150</th>\n",
       "      <td>7.0</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>150</td>\n",
       "      <td>EGFRvIII</td>\n",
       "      <td>Alterations ofthe EGFR gene occur frequently i...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152</th>\n",
       "      <td>7.0</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>152</td>\n",
       "      <td>EGFRvII</td>\n",
       "      <td>Abstract Glioblastomas with EGFR amplificatio...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170</th>\n",
       "      <td>7.0</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>170</td>\n",
       "      <td>EGFRvV</td>\n",
       "      <td>Abstract  The epidermal growth factor recepto...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>211</th>\n",
       "      <td>7.0</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>211</td>\n",
       "      <td>EGFR-KDD</td>\n",
       "      <td>Oncogenic EGFR mutations are found in 10-35% o...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>238</th>\n",
       "      <td>7.0</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>238</td>\n",
       "      <td>EGFRvIV</td>\n",
       "      <td>Tumor cells often subvert normal regulatory me...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>350</th>\n",
       "      <td>1.0</td>\n",
       "      <td>CDH1</td>\n",
       "      <td>350</td>\n",
       "      <td>Epigenetic Silencing</td>\n",
       "      <td>In diffuse gastric carcinoma, despite common E...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>354</th>\n",
       "      <td>4.0</td>\n",
       "      <td>EP300</td>\n",
       "      <td>354</td>\n",
       "      <td>R1627</td>\n",
       "      <td>The transcriptional coactivator p300/CBP (CREB...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>355</th>\n",
       "      <td>4.0</td>\n",
       "      <td>EP300</td>\n",
       "      <td>355</td>\n",
       "      <td>C1385</td>\n",
       "      <td>The transcriptional coactivator p300/CBP (CREB...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>631</th>\n",
       "      <td>4.0</td>\n",
       "      <td>FBXW7</td>\n",
       "      <td>631</td>\n",
       "      <td>Copy Number Loss</td>\n",
       "      <td>This study focused on a cell cycle regulatory ...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1786</th>\n",
       "      <td>7.0</td>\n",
       "      <td>AR</td>\n",
       "      <td>1786</td>\n",
       "      <td>ARv567es</td>\n",
       "      <td>Androgen receptor (AR) splice variants lacking...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1801</th>\n",
       "      <td>7.0</td>\n",
       "      <td>AR</td>\n",
       "      <td>1801</td>\n",
       "      <td>AR-V7</td>\n",
       "      <td>Androgen receptor (AR) splice variants lacking...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1938</th>\n",
       "      <td>2.0</td>\n",
       "      <td>CARD11</td>\n",
       "      <td>1938</td>\n",
       "      <td>L232LI</td>\n",
       "      <td>Self-tolerance and immunity are actively acqui...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1941</th>\n",
       "      <td>7.0</td>\n",
       "      <td>CARD11</td>\n",
       "      <td>1941</td>\n",
       "      <td>L225LI</td>\n",
       "      <td>The regulated activation of NF-κB by antigen r...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2054</th>\n",
       "      <td>6.0</td>\n",
       "      <td>MYC</td>\n",
       "      <td>2054</td>\n",
       "      <td>MYC-nick</td>\n",
       "      <td>Evasion of apoptosis is critical in Myc-induce...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2718</th>\n",
       "      <td>7.0</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>2718</td>\n",
       "      <td>p61BRAF</td>\n",
       "      <td>Summary Activated RAS promotes dimerization o...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3109</th>\n",
       "      <td>8.0</td>\n",
       "      <td>DNMT3B</td>\n",
       "      <td>3109</td>\n",
       "      <td>DNMT3B7</td>\n",
       "      <td>Epigenetic changes are among the most common a...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3135</th>\n",
       "      <td>3.0</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>3135</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>To characterize somatic alterations in colorec...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3254</th>\n",
       "      <td>4.0</td>\n",
       "      <td>CASP8</td>\n",
       "      <td>3254</td>\n",
       "      <td>CASP8L</td>\n",
       "      <td>Caspase-8 (Fas-associating protein with death ...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3389</th>\n",
       "      <td>4.0</td>\n",
       "      <td>BCL2L11</td>\n",
       "      <td>3389</td>\n",
       "      <td>Single Nucleotide Polymorphism</td>\n",
       "      <td>Abstract Purpose  BIM is essential for the re...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3947</th>\n",
       "      <td>NaN</td>\n",
       "      <td>BCL2L11</td>\n",
       "      <td>259</td>\n",
       "      <td>Single Nucleotide Polymorphism</td>\n",
       "      <td>Abstract Purpose  BIM is essential for the re...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>21 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Class     Gene    ID                       Variation  \\\n",
       "31      7.0     TERT    31              Promoter Mutations   \n",
       "150     7.0     EGFR   150                        EGFRvIII   \n",
       "152     7.0     EGFR   152                         EGFRvII   \n",
       "170     7.0     EGFR   170                          EGFRvV   \n",
       "211     7.0     EGFR   211                        EGFR-KDD   \n",
       "238     7.0     EGFR   238                         EGFRvIV   \n",
       "350     1.0     CDH1   350            Epigenetic Silencing   \n",
       "354     4.0    EP300   354                           R1627   \n",
       "355     4.0    EP300   355                           C1385   \n",
       "631     4.0    FBXW7   631                Copy Number Loss   \n",
       "1786    7.0       AR  1786                        ARv567es   \n",
       "1801    7.0       AR  1801                           AR-V7   \n",
       "1938    2.0   CARD11  1938                          L232LI   \n",
       "1941    7.0   CARD11  1941                          L225LI   \n",
       "2054    6.0      MYC  2054                        MYC-nick   \n",
       "2718    7.0     BRAF  2718                         p61BRAF   \n",
       "3109    8.0   DNMT3B  3109                         DNMT3B7   \n",
       "3135    3.0     KRAS  3135                        Wildtype   \n",
       "3254    4.0    CASP8  3254                          CASP8L   \n",
       "3389    4.0  BCL2L11  3389  Single Nucleotide Polymorphism   \n",
       "3947    NaN  BCL2L11   259  Single Nucleotide Polymorphism   \n",
       "\n",
       "                                                   Text  Substitutions_var  \\\n",
       "31    Sequencing studies have identified many recurr...                  0   \n",
       "150   Alterations ofthe EGFR gene occur frequently i...                  0   \n",
       "152    Abstract Glioblastomas with EGFR amplificatio...                  0   \n",
       "170    Abstract  The epidermal growth factor recepto...                  0   \n",
       "211   Oncogenic EGFR mutations are found in 10-35% o...                  0   \n",
       "238   Tumor cells often subvert normal regulatory me...                  0   \n",
       "350   In diffuse gastric carcinoma, despite common E...                  0   \n",
       "354   The transcriptional coactivator p300/CBP (CREB...                  0   \n",
       "355   The transcriptional coactivator p300/CBP (CREB...                  0   \n",
       "631   This study focused on a cell cycle regulatory ...                  0   \n",
       "1786  Androgen receptor (AR) splice variants lacking...                  0   \n",
       "1801  Androgen receptor (AR) splice variants lacking...                  0   \n",
       "1938  Self-tolerance and immunity are actively acqui...                  0   \n",
       "1941  The regulated activation of NF-κB by antigen r...                  0   \n",
       "2054  Evasion of apoptosis is critical in Myc-induce...                  0   \n",
       "2718   Summary Activated RAS promotes dimerization o...                  0   \n",
       "3109  Epigenetic changes are among the most common a...                  0   \n",
       "3135  To characterize somatic alterations in colorec...                  0   \n",
       "3254  Caspase-8 (Fas-associating protein with death ...                  0   \n",
       "3389   Abstract Purpose  BIM is essential for the re...                  0   \n",
       "3947   Abstract Purpose  BIM is essential for the re...                  0   \n",
       "\n",
       "      Stop_codon_var  gene_fusion_var  Deletion_var  del_or_ins_var    ...     \\\n",
       "31                 0                0             0               0    ...      \n",
       "150                0                0             0               0    ...      \n",
       "152                0                0             0               0    ...      \n",
       "170                0                0             0               0    ...      \n",
       "211                0                0             0               0    ...      \n",
       "238                0                0             0               0    ...      \n",
       "350                0                0             0               0    ...      \n",
       "354                0                0             0               0    ...      \n",
       "355                0                0             0               0    ...      \n",
       "631                0                0             0               0    ...      \n",
       "1786               0                0             0               0    ...      \n",
       "1801               0                0             0               0    ...      \n",
       "1938               0                0             0               0    ...      \n",
       "1941               0                0             0               0    ...      \n",
       "2054               0                0             0               0    ...      \n",
       "2718               0                0             0               0    ...      \n",
       "3109               0                0             0               0    ...      \n",
       "3135               0                0             0               0    ...      \n",
       "3254               0                0             0               0    ...      \n",
       "3389               0                0             0               0    ...      \n",
       "3947               0                0             0               0    ...      \n",
       "\n",
       "      Amplification_var  Truncation_var  exon_var  frameshift_var  dup_var  \\\n",
       "31                    0               0         0               0        0   \n",
       "150                   0               0         0               0        0   \n",
       "152                   0               0         0               0        0   \n",
       "170                   0               0         0               0        0   \n",
       "211                   0               0         0               0        0   \n",
       "238                   0               0         0               0        0   \n",
       "350                   0               0         0               0        0   \n",
       "354                   0               0         0               0        0   \n",
       "355                   0               0         0               0        0   \n",
       "631                   0               0         0               0        0   \n",
       "1786                  0               0         0               0        0   \n",
       "1801                  0               0         0               0        0   \n",
       "1938                  0               0         0               0        0   \n",
       "1941                  0               0         0               0        0   \n",
       "2054                  0               0         0               0        0   \n",
       "2718                  0               0         0               0        0   \n",
       "3109                  0               0         0               0        0   \n",
       "3135                  0               0         0               0        0   \n",
       "3254                  0               0         0               0        0   \n",
       "3389                  0               0         0               0        0   \n",
       "3947                  0               0         0               0        0   \n",
       "\n",
       "      overexpression_var  splice_var  hypermethylation_var  DNA_binding_var  \\\n",
       "31                     0           0                     0                0   \n",
       "150                    0           0                     0                0   \n",
       "152                    0           0                     0                0   \n",
       "170                    0           0                     0                0   \n",
       "211                    0           0                     0                0   \n",
       "238                    0           0                     0                0   \n",
       "350                    0           0                     0                0   \n",
       "354                    0           0                     0                0   \n",
       "355                    0           0                     0                0   \n",
       "631                    0           0                     0                0   \n",
       "1786                   0           0                     0                0   \n",
       "1801                   0           0                     0                0   \n",
       "1938                   0           0                     0                0   \n",
       "1941                   0           0                     0                0   \n",
       "2054                   0           0                     0                0   \n",
       "2718                   0           0                     0                0   \n",
       "3109                   0           0                     0                0   \n",
       "3135                   0           0                     0                0   \n",
       "3254                   0           0                     0                0   \n",
       "3389                   0           0                     0                0   \n",
       "3947                   0           0                     0                0   \n",
       "\n",
       "      null_var  \n",
       "31           0  \n",
       "150          0  \n",
       "152          0  \n",
       "170          0  \n",
       "211          0  \n",
       "238          0  \n",
       "350          0  \n",
       "354          0  \n",
       "355          0  \n",
       "631          0  \n",
       "1786         0  \n",
       "1801         0  \n",
       "1938         0  \n",
       "1941         0  \n",
       "2054         0  \n",
       "2718         0  \n",
       "3109         0  \n",
       "3135         0  \n",
       "3254         0  \n",
       "3389         0  \n",
       "3947         0  \n",
       "\n",
       "[21 rows x 21 columns]"
      ]
     },
     "execution_count": 363,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Remember there are still unassigned variations that we can assign to for the next run (21 in total)\n",
    "all_variations = data_all.loc[:, \"Substitutions_var\":]\n",
    "data_all[(all_variations.T == 0).all()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 366,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# Preprocess all the nosubs together\n",
    "nosub=data_all[data_all['Substitutions_var']==0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 367,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Class</th>\n",
       "      <th>Gene</th>\n",
       "      <th>ID</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Text</th>\n",
       "      <th>Substitutions_var</th>\n",
       "      <th>Stop_codon_var</th>\n",
       "      <th>gene_fusion_var</th>\n",
       "      <th>Deletion_var</th>\n",
       "      <th>del_or_ins_var</th>\n",
       "      <th>...</th>\n",
       "      <th>Amplification_var</th>\n",
       "      <th>Truncation_var</th>\n",
       "      <th>exon_var</th>\n",
       "      <th>frameshift_var</th>\n",
       "      <th>dup_var</th>\n",
       "      <th>overexpression_var</th>\n",
       "      <th>splice_var</th>\n",
       "      <th>hypermethylation_var</th>\n",
       "      <th>DNA_binding_var</th>\n",
       "      <th>null_var</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1277</th>\n",
       "      <td>1.0</td>\n",
       "      <td>ARID5B</td>\n",
       "      <td>1277</td>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>null</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1639</th>\n",
       "      <td>6.0</td>\n",
       "      <td>FLT1</td>\n",
       "      <td>1639</td>\n",
       "      <td>Amplification</td>\n",
       "      <td>null</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1911</th>\n",
       "      <td>1.0</td>\n",
       "      <td>FOXP1</td>\n",
       "      <td>1911</td>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>The transcriptional network of the androgen re...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3426</th>\n",
       "      <td>2.0</td>\n",
       "      <td>AURKB</td>\n",
       "      <td>3426</td>\n",
       "      <td>Amplification</td>\n",
       "      <td>null</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Class    Gene    ID             Variation  \\\n",
       "1277    1.0  ARID5B  1277  Truncating Mutations   \n",
       "1639    6.0    FLT1  1639         Amplification   \n",
       "1911    1.0   FOXP1  1911  Truncating Mutations   \n",
       "3426    2.0   AURKB  3426         Amplification   \n",
       "\n",
       "                                                   Text  Substitutions_var  \\\n",
       "1277                                               null                  0   \n",
       "1639                                               null                  0   \n",
       "1911  The transcriptional network of the androgen re...                  0   \n",
       "3426                                               null                  0   \n",
       "\n",
       "      Stop_codon_var  gene_fusion_var  Deletion_var  del_or_ins_var    ...     \\\n",
       "1277               0                0             0               0    ...      \n",
       "1639               0                0             0               0    ...      \n",
       "1911               0                0             0               0    ...      \n",
       "3426               0                0             0               0    ...      \n",
       "\n",
       "      Amplification_var  Truncation_var  exon_var  frameshift_var  dup_var  \\\n",
       "1277                  0               1         0               0        0   \n",
       "1639                  1               0         0               0        0   \n",
       "1911                  0               1         0               0        0   \n",
       "3426                  1               0         0               0        0   \n",
       "\n",
       "      overexpression_var  splice_var  hypermethylation_var  DNA_binding_var  \\\n",
       "1277                   0           0                     0                0   \n",
       "1639                   0           0                     0                0   \n",
       "1911                   0           0                     0                0   \n",
       "3426                   0           0                     0                0   \n",
       "\n",
       "      null_var  \n",
       "1277         0  \n",
       "1639         0  \n",
       "1911         0  \n",
       "3426         0  \n",
       "\n",
       "[4 rows x 21 columns]"
      ]
     },
     "execution_count": 367,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nosub_text, nosub_no_text = custom_find(nosub)\n",
    "nosub_no_text # only 4 not found, of which three nulls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 368,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# find the sentences containing the nosubs\n",
    "NLTK_nosub = [sent_tokenize(nosub_text.Text[i]) for i in nosub_text.index]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 188,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "nosub_sentences = get_sentences_nosub(nosub_text[:], NLTK_nosub[:], window_left = 0, window_right = 0) # Retrieves sentences where keyword is included\n",
    "nosub_sentences_string = sentences_to_string(nosub_sentences)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 369,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\rafiz\\Anaconda3\\lib\\site-packages\\pandas\\core\\indexing.py:179: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  self._setitem_with_indexer(indexer, value)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0       Cyclin M, an orphan cyclin, is the product of ...\n",
       "7       In this study, we determined the frequency of ...\n",
       "16      Germline and somatic mutations that deregulate...\n",
       "31      Sequencing studies have identified many recurr...\n",
       "33      Studies with comparative genomic hybridization...\n",
       "41      Here, we report that deletion of the miRNA pro...\n",
       "43      Here, we report that deletion of the miRNA pro...\n",
       "48      Experimental Design  Genome-wide DNA methylati...\n",
       "70      Both, human SHQ1 and NAP57 (as an N-terminal f...\n",
       "71      Polyclonal anti-cyclin D2 and anti-cyclin D3 a...\n",
       "72      Here we report the association of three Mre11 ...\n",
       "74      The Mre11 complex specifies ssDNA endonuclease...\n",
       "75      Polyclonal anti-cyclin D2 and anti-cyclin D3 a...\n",
       "86      A Sac I (-1 195)-BamHI (+ 79) fragmnent and a ...\n",
       "87      A Sac I (-1 195)-BamHI (+ 79) fragmnent and a ...\n",
       "88      To determine whether RYBP binding to MDM2 is n...\n",
       "96      Mutations in the known high-risk genes, BRCA1,...\n",
       "100     These alterations can be deletions or mutation...\n",
       "101     Here we identify frequent TGFBR1 and TGFBR2 mu...\n",
       "107     A hypermutated subtype of advanced prostate ca...\n",
       "110     Two distinct functions have been proposed for ...\n",
       "113     Germ-line MSH6 mutations, which are rare in HN...\n",
       "114     EstablishedMSH6-null mice present afrequent oc...\n",
       "115     Kabuki syndrome (KS) is a multiple congenital ...\n",
       "116     Kabuki syndrome (KS) is a multiple congenital ...\n",
       "117     Nonsense mutations surrounding the correspondi...\n",
       "118     Screening for tumor suppressor genes in breast...\n",
       "119     Screening for tumor suppressor genes in breast...\n",
       "120     Background Many cases of familial renal cell c...\n",
       "137     Three cell lines showed overlapping homozygous...\n",
       "                              ...                        \n",
       "4308    Somatic mutations in the tyrosine kinase (TK) ...\n",
       "4311    Mice with targeted deletions of EGFR display d...\n",
       "4313    Somatic mutations introduced into the epiderma...\n",
       "4318    Somatic mutations introduced into the epiderma...\n",
       "4323     In contrast to other primary epidermal growth...\n",
       "4342    Gefitinib inhibits the growth of some cancer-d...\n",
       "4365     Abstract  Malignant pleural effusions (MPEs) ...\n",
       "4405    Somatic mutations in the tyrosine kinase (TK) ...\n",
       "4422     In contrast to other primary epidermal growth...\n",
       "4430    Epidermal growth factor receptor (EGFR) gene m...\n",
       "4453     Abstract  Malignant pleural effusions (MPEs) ...\n",
       "4456     In contrast to other primary epidermal growth...\n",
       "4465    Studies of the mechanism of c-Src regulation h...\n",
       "4496    Currently available methods of EGFR mutation d...\n",
       "4497    We identify several novel mutated genes as wel...\n",
       "4506    Somatic mutations introduced into the epiderma...\n",
       "4508    Oncogenic EGFR mutations are found in 10-35% o...\n",
       "4525    Currently available methods of EGFR mutation d...\n",
       "4535    Epidermal growth factor receptor (EGFR) gene m...\n",
       "4537    Gefitinib inhibits the growth of some cancer-d...\n",
       "4558     In contrast to other primary epidermal growth...\n",
       "4559    Purpose: Clinical features of epidermal growth...\n",
       "4569    Epidermal growth factor receptor (EGFR) gene m...\n",
       "4580    Gefitinib inhibits the growth of some cancer-d...\n",
       "4583    Gefitinib inhibits the growth of some cancer-d...\n",
       "4602    Gefitinib inhibits the growth of some cancer-d...\n",
       "4623     In contrast to other primary epidermal growth...\n",
       "4624    This information, along with remarkable sensit...\n",
       "4650     In contrast to other primary epidermal growth...\n",
       "4651     Abstract  Malignant pleural effusions (MPEs) ...\n",
       "Name: Text, Length: 851, dtype: object"
      ]
     },
     "execution_count": 369,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "### Replacement inside the Text of data_all\n",
    "data_all.Text.iloc[nosub_text.index] = nosub_sentences_string #iloc for indexing based on integers\n",
    "data_all.Text.iloc[nosub_text.index]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 189,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Not used\n",
    "################ The dummy variables for Gene and Text ##################\n",
    "## TODO: also use dummy for Text? There are 135 shared Genes and 142 shared Text between train and Leaks!  len(set(train.Text) & set(Leaks.Text))\n",
    "#data_all_dummy = data_all_backup[['Gene', 'Text']] # drop those columns we don't need as dummy.\n",
    "#X_dummy = pd.get_dummies(data_all_dummy) # converts categorical variables into dummy variable. From len set => 269 genes + 2090 texts\n",
    "#X_dummy_train = X_dummy[:train.shape[0]]\n",
    "#X_dummy_test = X_dummy[train.shape[0]:]\n",
    "#dummy_names = X_dummy.columns.values #### To remember names if you want to check again what Gene or Text used\n",
    "#X_dummy = X_dummy.values"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Cleaning "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 190,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "stop = set(stopwords.words('english'))\n",
    "stop.update(('et al', 'fig', 'figure', 'acknowledgement' )) #maybe add some more\n",
    "\n",
    "exclude = set('!\"#$%&\\'()*+:;<=>?@[\\\\]^_`{|}~') \n",
    "lemma = WordNetLemmatizer()\n",
    "def clean(doc,lemmatiz=False):\n",
    "    stop_free = \" \".join([i for i in doc.lower().split() if i not in stop])\n",
    "    punc_free_0 = [re.sub(\",|\\.|/\",\" \",ch) for ch in stop_free]\n",
    "    if lemmatiz:\n",
    "        punc_free_lem=\"\".join(ch for ch in punc_free_0 if ch not in exclude)\n",
    "        normalized = \" \".join(lemma.lemmatize(word) for word in punc_free_lem.split())\n",
    "        return normalized\n",
    "    else:\n",
    "        punc_free = \"\".join(ch for ch in punc_free_0 if ch not in exclude)\n",
    "        return punc_free"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 191,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#No lemmatization for the moment, be careful not to lemmatize then w2vec\n",
    "data_all.Text = [clean(doc) for doc in data_all[\"Text\"]]  \n",
    "#data_all.Background = [clean(doc) for doc in data_all.Background]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Adding extra feature(s)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 247,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# Feature for the length of the text\n",
    "data_all[\"Text_words\"] = data_all[\"Text\"].map(lambda x: len(str(x).split(\" \")))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 269,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "new_train = data_all.iloc[:len(train_final)]\n",
    "new_test = data_all.iloc[len(train_final):]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 270,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "svd = TruncatedSVD(n_components=25, n_iter=12, random_state=26)\n",
    "\n",
    "one_hot_gene = pd.get_dummies(new_train['Gene'])\n",
    "one_hot_gene_test=pd.get_dummies(new_test[\"Gene\"])\n",
    "one_hot_gene_test=fix_test_columns(one_hot_gene_test,one_hot_gene.columns)\n",
    "truncated_one_hot_gene = svd.fit_transform(one_hot_gene.values)\n",
    "truncated_one_hot_gene_for_test=svd.transform(one_hot_gene_test.values)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 271,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "genes_train=pd.DataFrame(truncated_one_hot_gene,columns=[\"tsvd_gene\"+ str(x) for x in range(0,25)])\n",
    "genes_test=pd.DataFrame(truncated_one_hot_gene_for_test,columns=[\"tsvd_gene\"+ str(x) for x in range(0,25)])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 272,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\Franck\\Documents\\Continuum\\Anaconda3\\lib\\site-packages\\ipykernel_launcher.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \"\"\"Entry point for launching an IPython kernel.\n"
     ]
    }
   ],
   "source": [
    "new_test[\"index\"]=range(0,len(new_test))\n",
    "new_train_1=pd.merge(new_train.reset_index(),genes_train.reset_index()).drop(\"index\",axis=1)\n",
    "new_test_1=pd.merge(new_test,genes_test.reset_index()).drop(\"index\",axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 273,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "new_train_1.to_csv(\"checkpoints_databases/w_working_train.csv\",index=False,encoding=\"utf8\")\n",
    "new_test_1.to_csv(\"checkpoints_databases/w_working_test.csv\",index=False,encoding=\"utf8\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 265,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(986, 47)"
      ]
     },
     "execution_count": 265,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_test_1.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
